<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>DLPFC</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">KODAMA-Analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="Installation.html">Installation</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Single-cell
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="singlecell_datamatrix.html">Data matrix</a>
    </li>
    <li>
      <a href="singlecell_seurat.html">Seurat</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Spatial
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="MERFISH.html">MERFISH (Moffitt et al., 2018)</a>
    </li>
    <li>
      <a href="DLPFC.html">Visium DLPFC (Maynard et al., 2021)</a>
    </li>
    <li>
      <a href="Prostate.html">Visium (10X database - Prostate)</a>
    </li>
    <li>
      <a href="VisiumHD.html">Visium HD (10X database - Colorectal cancer)</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Wrapper
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Seurat.html">Seurat</a>
    </li>
    <li>
      <a href="Giotto.html">Giotto</a>
    </li>
    <li>
      <a href="SpatialExperiment.html">SpatialExperiment</a>
    </li>
  </ul>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/tkcaccia/KODAMA-Analysis">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">DLPFC</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<<<<<<< HEAD
<strong>Last updated:</strong> 2025-04-13
=======
<strong>Last updated:</strong> 2025-03-05
>>>>>>> d89baaa7bfc3bffd762b0db8bac45eeb05443b97
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>KODAMA-Analysis/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240618code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240618)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240618code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240618)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<<<<<<< HEAD
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomtkcacciaKODAMAAnalysistree5f5ac634a8274a7b7c4cb315454f9c9aa7a12f86targetblank5f5ac63a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/tree/5f5ac634a8274a7b7c4cb315454f9c9aa7a12f86" target="_blank">5f5ac63</a>
=======
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomtkcacciaKODAMAAnalysistree7de0c7444fdebf22d8827606a2b840417e344179targetblank7de0c74a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/tree/7de0c7444fdebf22d8827606a2b840417e344179" target="_blank">7de0c74</a>
>>>>>>> d89baaa7bfc3bffd762b0db8bac45eeb05443b97
</a>
</p>
</div>
<div
<<<<<<< HEAD
id="strongRepositoryversionstrongahrefhttpsgithubcomtkcacciaKODAMAAnalysistree5f5ac634a8274a7b7c4cb315454f9c9aa7a12f86targetblank5f5ac63a"
=======
id="strongRepositoryversionstrongahrefhttpsgithubcomtkcacciaKODAMAAnalysistree7de0c7444fdebf22d8827606a2b840417e344179targetblank7de0c74a"
>>>>>>> d89baaa7bfc3bffd762b0db8bac45eeb05443b97
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<<<<<<< HEAD
<a href="https://github.com/tkcaccia/KODAMA-Analysis/tree/5f5ac634a8274a7b7c4cb315454f9c9aa7a12f86" target="_blank">5f5ac63</a>.
=======
<a href="https://github.com/tkcaccia/KODAMA-Analysis/tree/7de0c7444fdebf22d8827606a2b840417e344179" target="_blank">7de0c74</a>.
>>>>>>> d89baaa7bfc3bffd762b0db8bac45eeb05443b97
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  KODAMA.svg
<<<<<<< HEAD
    Untracked:  analysis/singlecell_datamatrix.Rmd
    Untracked:  analysis/singlecell_seurat.Rmd
=======
    Untracked:  analysis/old_DLPFC.Rmd
>>>>>>> d89baaa7bfc3bffd762b0db8bac45eeb05443b97
    Untracked:  code/Acinar_Cell_Carcinoma.ipynb
    Untracked:  code/Adenocarcinoma.ipynb
    Untracked:  code/Adjacent_normal_section.ipynb
    Untracked:  code/DLFPC_preprocessing.R
    Untracked:  code/DLPFC - BANKSY.R
    Untracked:  code/DLPFC - BASS.R
    Untracked:  code/DLPFC - BAYESPACE.R
    Untracked:  code/DLPFC - Nonspatial.R
    Untracked:  code/DLPFC - PRECAST.R
    Untracked:  code/DLPFC_comparison.R
    Untracked:  code/DLPFC_results_analysis.R
    Untracked:  code/MERFISH - BANKSY.R
    Untracked:  code/MERFISH - BASS.R
    Untracked:  code/MERFISH - BAYESPACE.R
    Untracked:  code/MERFISH - Nonspatial.R
    Untracked:  code/MERFISH - PRECAST.R
    Untracked:  code/MERFISH_comparison.R
    Untracked:  code/MERFISH_results_analysis.R
    Untracked:  code/VisiumHD-CRC.ipynb
<<<<<<< HEAD
    Untracked:  code/VisiumHDassignment.py
=======
>>>>>>> d89baaa7bfc3bffd762b0db8bac45eeb05443b97
    Untracked:  code/deep learning code DLPFC.R
    Untracked:  code/save tiles.py
    Untracked:  data/Adenocarcinoma.csv
    Untracked:  data/Annotations/
    Untracked:  data/DLFPC-Br5292-input.RData
    Untracked:  data/DLFPC-Br5595-input.RData
    Untracked:  data/DLFPC-Br8100-input.RData
    Untracked:  data/DLPFC-general.RData
    Untracked:  data/MERFISH-input.RData
    Untracked:  data/spots_classification_ALL.csv
    Untracked:  data/spots_classification_Acinar_Cell_Carcinoma.csv
    Untracked:  data/spots_classification_IF.csv
    Untracked:  data/spots_classification_Normal_prostate.csv
    Untracked:  data/trajectories.RData
    Untracked:  data/trajectories_VISIUMHD.RData
    Untracked:  output/BANSKY-results.RData
    Untracked:  output/BASS-results.RData
    Untracked:  output/BayesSpace-results.RData
    Untracked:  output/CRC-image.RData
    Untracked:  output/CRC-image2.RData
    Untracked:  output/CRC.png
    Untracked:  output/CRC2.png
    Untracked:  output/CRC7.png
    Untracked:  output/CRC8.png
    Untracked:  output/CRC_boxplot.png
    Untracked:  output/CRC_boxplot.svg
<<<<<<< HEAD
    Untracked:  output/CRC_boxplot2.svg
=======
>>>>>>> d89baaa7bfc3bffd762b0db8bac45eeb05443b97
    Untracked:  output/CRC_linee.svg
    Untracked:  output/DL.RData
    Untracked:  output/DLFPC-All-2.RData
    Untracked:  output/DLFPC-All.RData
    Untracked:  output/DLFPC-Br5292.RData
    Untracked:  output/DLFPC-Br5595.RData
    Untracked:  output/DLFPC-Br8100.RData
<<<<<<< HEAD
    Untracked:  output/DLFPC-variablesXdeeplearning.RData
=======
>>>>>>> d89baaa7bfc3bffd762b0db8bac45eeb05443b97
    Untracked:  output/DLPFC-BANSKY-results.RData
    Untracked:  output/DLPFC-BASS-results.RData
    Untracked:  output/DLPFC-BayesSpace-results.RData
    Untracked:  output/DLPFC-Nonspatial-results.RData
    Untracked:  output/DLPFC-PRECAST-results.RData
<<<<<<< HEAD
    Untracked:  output/DLPFC_all_cluster.svg
    Untracked:  output/DLPFCpathway.RData
=======
    Untracked:  output/DLPFC1.svg
    Untracked:  output/DLPFC_all_cluster.svg
>>>>>>> d89baaa7bfc3bffd762b0db8bac45eeb05443b97
    Untracked:  output/Figure 1 - boxplot.pdf
    Untracked:  output/Figure 2 - DLPFC 10.pdf
    Untracked:  output/Figures/
    Untracked:  output/KODAMA-results.RData
    Untracked:  output/KODAMA_DLPFC_All_original.svg
    Untracked:  output/KODAMA_DLPFC_Br5595.svg
    Untracked:  output/KODAMA_DLPFC_Br5595_slide.svg
<<<<<<< HEAD
    Untracked:  output/Loupe.csv
=======
>>>>>>> d89baaa7bfc3bffd762b0db8bac45eeb05443b97
    Untracked:  output/MERFISH-BANSKY-results.RData
    Untracked:  output/MERFISH-BASS-results.RData
    Untracked:  output/MERFISH-BayesSpace-results.RData
    Untracked:  output/MERFISH-KODAMA-results.RData
<<<<<<< HEAD
    Untracked:  output/MERFISH-Nonspatial-results.RData
=======
    Untracked:  output/MERFISH-NONSPATIAL-results.RData
>>>>>>> d89baaa7bfc3bffd762b0db8bac45eeb05443b97
    Untracked:  output/MERFISH-PRECAST-results.RData
    Untracked:  output/MERFISH.RData
    Untracked:  output/Nonspatial-results.RData
    Untracked:  output/Prostate.RData
<<<<<<< HEAD
    Untracked:  output/VisiumHD-RNA.RData
    Untracked:  output/VisiumHD-genes.pdf
    Untracked:  output/VisiumHD.RData
    Untracked:  output/boh.svg
    Untracked:  output/desmoplastic_distance_carcinoma.csv
=======
    Untracked:  output/RGB_DLPFC.png
    Untracked:  output/VisiumHD.RData
    Untracked:  output/VisiumHD3.RData
>>>>>>> d89baaa7bfc3bffd762b0db8bac45eeb05443b97
    Untracked:  output/image.RData
    Untracked:  output/pp.RData
    Untracked:  output/pp2.RData
    Untracked:  output/pp3.RData
    Untracked:  output/pp4.RData
    Untracked:  output/pp5.RData
    Untracked:  output/prostate1.svg
    Untracked:  output/prostate2.svg
    Untracked:  output/prostate3.svg
    Untracked:  output/prostate4.svg
    Untracked:  output/prostate5.svg
    Untracked:  output/prostate6.svg
    Untracked:  output/prostate7.svg
<<<<<<< HEAD
    Untracked:  output/subclusters1.csv
    Untracked:  output/subclusters2.csv
    Untracked:  output/subclusters3.csv
    Untracked:  output/tight_boundary.geojson
    Untracked:  output/trajectory.csv
=======
    Untracked:  output/tight_boundary.geojson
>>>>>>> d89baaa7bfc3bffd762b0db8bac45eeb05443b97

Unstaged changes:
    Deleted:    analysis/D1.Rmd
    Deleted:    analysis/DLPFC-12.Rmd
    Deleted:    analysis/DLPFC-4.Rmd
    Modified:   analysis/DLPFC.Rmd
    Deleted:    analysis/DLPFC1.Rmd
    Deleted:    analysis/DLPFC10.Rmd
    Deleted:    analysis/DLPFC2.Rmd
    Deleted:    analysis/DLPFC3.Rmd
    Deleted:    analysis/DLPFC4.Rmd
    Deleted:    analysis/DLPFC5.Rmd
    Deleted:    analysis/DLPFC6.Rmd
    Deleted:    analysis/DLPFC7.Rmd
    Deleted:    analysis/DLPFC8.Rmd
    Deleted:    analysis/DLPFC9.Rmd
    Deleted:    analysis/Du1.Rmd
    Deleted:    analysis/Du10.Rmd
    Deleted:    analysis/Du11.Rmd
    Deleted:    analysis/Du12.Rmd
    Deleted:    analysis/Du13.Rmd
    Deleted:    analysis/Du14.Rmd
    Deleted:    analysis/Du15.Rmd
    Deleted:    analysis/Du16.Rmd
    Deleted:    analysis/Du17.Rmd
    Deleted:    analysis/Du18.Rmd
    Deleted:    analysis/Du19.Rmd
    Deleted:    analysis/Du2.Rmd
    Deleted:    analysis/Du20.Rmd
    Deleted:    analysis/Du3.Rmd
    Deleted:    analysis/Du4.Rmd
    Deleted:    analysis/Du5.Rmd
    Deleted:    analysis/Du6.Rmd
    Deleted:    analysis/Du7.Rmd
    Deleted:    analysis/Du8.Rmd
    Deleted:    analysis/Du9.Rmd
    Modified:   analysis/Giotto.Rmd
    Modified:   analysis/MERFISH.Rmd
    Deleted:    analysis/MERFISH1a (copy).Rmd
    Deleted:    analysis/MERFISH1a.Rmd
    Deleted:    analysis/MERFISH1b (copy).Rmd
    Deleted:    analysis/MERFISH1b.Rmd
    Deleted:    analysis/MERFISH2a (copy).Rmd
    Deleted:    analysis/MERFISH2a.Rmd
    Deleted:    analysis/MERFISH2b (copy).Rmd
    Deleted:    analysis/MERFISH2b.Rmd
    Deleted:    analysis/MERFISH3a (copy).Rmd
    Deleted:    analysis/MERFISH3a.Rmd
    Deleted:    analysis/MERFISH3b (copy).Rmd
    Deleted:    analysis/MERFISH3b.Rmd
    Deleted:    analysis/MERFISH4a (copy).Rmd
    Deleted:    analysis/MERFISH4a.Rmd
    Deleted:    analysis/MERFISH4b (copy).Rmd
    Deleted:    analysis/MERFISH4b.Rmd
    Modified:   analysis/Prostate.Rmd
    Deleted:    analysis/STARmap.Rmd
<<<<<<< HEAD
    Modified:   analysis/Seurat.Rmd
    Deleted:    analysis/Simulation.Rmd
    Deleted:    analysis/Single-cell.Rmd
    Modified:   analysis/SpatialExperiment.Rmd
=======
>>>>>>> d89baaa7bfc3bffd762b0db8bac45eeb05443b97
    Modified:   analysis/VisiumHD.Rmd
    Modified:   code/VisiumHD_CRC_download.sh
    Deleted:    data/Pathology.csv

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/DLPFC.Rmd</code>) and HTML
(<code>docs/DLPFC.html</code>) files. If you’ve configured a remote Git
repository (see <code>?wflow_git_remote</code>), click on the hyperlinks
in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/3305d55e512f9647a5a090bf36911e444b9e361e/docs/DLPFC.html" target="_blank">3305d55</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-12-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/16e454f9af3e4ba9c31e8d733a21f0fc4b770feb/analysis/DLPFC.Rmd" target="_blank">16e454f</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-12-20
</td>
<td>
Start my new project
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/7b2cb8c2c66c61a5abd5703a20bb718daeae4ac5/docs/DLPFC.html" target="_blank">7b2cb8c</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-12-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/8a5f6c814dca158afc33cfad9e738cc30b449074/analysis/DLPFC.Rmd" target="_blank">8a5f6c8</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-12-16
</td>
<td>
Start my new project
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/374d5f014b2a22cb41b1c5950c364ae56235250d/docs/DLPFC.html" target="_blank">374d5f0</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-12-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/6efcd94e9db0755a88ab2711f51ad94b088ec2e1/analysis/DLPFC.Rmd" target="_blank">6efcd94</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-12-14
</td>
<td>
Start my new project
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/f6bab12992a50883783f079203211492efa5647d/docs/DLPFC.html" target="_blank">f6bab12</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/3490a1170290b0e69afacd4f37310c4e18d627cb/analysis/DLPFC.Rmd" target="_blank">3490a11</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-19
</td>
<td>
Start my new project
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/43ad12f5138e9e9d2f128946a6fcf3d511bb4fd1/analysis/DLPFC.Rmd" target="_blank">43ad12f</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-18
</td>
<td>
Start my new project
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/2c73e5e098bf69904d0b42b83bdd001386d372a5/analysis/DLPFC.Rmd" target="_blank">2c73e5e</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-18
</td>
<td>
Start my new project
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/d75807ed0cab32318b488050becfbc666c63603a/analysis/DLPFC.Rmd" target="_blank">d75807e</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-17
</td>
<td>
Start my new project
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/ede33d24ef309bb21827f046b9fb02418d1f1692/docs/DLPFC.html" target="_blank">ede33d2</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/1ec4751f221ecef8a9baa226d64ca9a529c318f8/analysis/DLPFC.Rmd" target="_blank">1ec4751</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-17
</td>
<td>
Start my new project
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/773f9efd21c5247e87d6a4487fdfef012d376794/docs/DLPFC.html" target="_blank">773f9ef</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/df9888188a540922da484770f06fd4fd4b80c2c7/analysis/DLPFC.Rmd" target="_blank">df98881</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-16
</td>
<td>
Start my new project
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/47b016303e8748b5297be43db18b2a9f8c3569c5/analysis/DLPFC.Rmd" target="_blank">47b0163</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-15
</td>
<td>
Start my new project
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/fd8d0925cad5881c332a822c5d7e5befd27ef73d/docs/DLPFC.html" target="_blank">fd8d092</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/e31e3e86a247626d539d237a3145860abb73c612/analysis/DLPFC.Rmd" target="_blank">e31e3e8</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-15
</td>
<td>
Start my new project
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/1edc32b6806afe1639b806a6a7a2bdc873ec6a24/docs/DLPFC.html" target="_blank">1edc32b</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-11
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/eaad1a02a20fd111e43c1888bb87e1181c208687/analysis/DLPFC.Rmd" target="_blank">eaad1a0</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-11
</td>
<td>
Start my new project
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/c9d54eec14966990195df616e6151496bfb3cc5d/docs/DLPFC.html" target="_blank">c9d54ee</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-11
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/fa049dea1943554383690dd66e78cf06e3ff912c/analysis/DLPFC.Rmd" target="_blank">fa049de</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-11
</td>
<td>
Start my new project
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/454b8fc544e210b86b7c3fe74998caef36c23d7c/analysis/DLPFC.Rmd" target="_blank">454b8fc</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-11
</td>
<td>
Start my new project
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/1352d911e77a96a9c1216f1473ee5d13bc6bb3b4/docs/DLPFC.html" target="_blank">1352d91</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/1b119a0510fc5d6ad48183c7d7ce8455f9813d33/analysis/DLPFC.Rmd" target="_blank">1b119a0</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-10
</td>
<td>
Start my new project
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/6038af1ae7d982283a26fe00a2ca6547cfe2ad13/docs/DLPFC.html" target="_blank">6038af1</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/d1416285bd90b9ebb07152a6d73396f800e25d1d/analysis/DLPFC.Rmd" target="_blank">d141628</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-09
</td>
<td>
Start my new project
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/d1192e95a8ed7985fb6d9678ce0a6d850f882066/docs/DLPFC.html" target="_blank">d1192e9</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-08-12
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/3374e66ce76ec3d26fcb912a9f6eba2f7341ed8f/docs/DLPFC.html" target="_blank">3374e66</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-08-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/35ce733e4e0012d79f696f9c7dcdbf67e1623e80/docs/DLPFC.html" target="_blank">35ce733</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-08-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/06f7055948b77da17ff4bf8e475037218baab49e/analysis/DLPFC.Rmd" target="_blank">06f7055</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-08-02
</td>
<td>
Start my new project
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/7be8f59b0c4d82e877c5dfc8cfbfb7936ceceeb6/analysis/DLPFC.Rmd" target="_blank">7be8f59</a>
</td>
<td>
tkcaccia
</td>
<td>
2024-07-15
</td>
<td>
updates
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/f8ca54aee4428df50d2cd0f41cbf2a0ceae01d7d/analysis/DLPFC.Rmd" target="_blank">f8ca54a</a>
</td>
<td>
tkcaccia
</td>
<td>
2024-07-14
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/f8ca54aee4428df50d2cd0f41cbf2a0ceae01d7d/docs/DLPFC.html" target="_blank">f8ca54a</a>
</td>
<td>
tkcaccia
</td>
<td>
2024-07-14
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/3ea09a699f0476bb6ca620f6aa6d7e88b7d4a1b5/docs/DLPFC.html" target="_blank">3ea09a6</a>
</td>
<td>
GitHub
</td>
<td>
2024-07-08
</td>
<td>
Update DLPFC.html
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/93915d8a77bf92f678416206af1f8613dd2aabe4/docs/DLPFC.html" target="_blank">93915d8</a>
</td>
<td>
GitHub
</td>
<td>
2024-07-04
</td>
<td>
Update DLPFC.html
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/ee4ee17ff874a26eaf8c1e76e4a85ce3b9de867e/docs/DLPFC.html" target="_blank">ee4ee17</a>
</td>
<td>
GitHub
</td>
<td>
2024-06-19
</td>
<td>
Add files via upload
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/615fc05e05b60aed64683d1204afddc6b8c32f36/analysis/DLPFC.Rmd" target="_blank">615fc05</a>
</td>
<td>
GitHub
</td>
<td>
2024-06-19
</td>
<td>
Add files via upload
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>Here, we apply KODAMA to analyze the human dorsolateral prefrontal
cortex (DLPFC) data by 10x Visium from <a
href="https://www.nature.com/articles/s41593-020-00787-0">Maynard et
al., 2021</a>. The links to download the raw data and H&amp;E full
resolution images can be found in the <a
href="https://github.com/LieberInstitute/spatialLIBD">LieberInstitute/spatialLIBD</a>
github page.</p>
</div>
<div id="loading-the-required-libraries" class="section level1">
<h1>Loading the required libraries</h1>
<pre class="r"><code>library(&quot;nnSVG&quot;)
library(&quot;scater&quot;)
library(&quot;scran&quot;)
library(&quot;scry&quot;)
library(&quot;SPARK&quot;)
library(&quot;harmony&quot;)
library(&quot;Seurat&quot;)
library(&quot;spatialLIBD&quot;)
library(&quot;KODAMAextra&quot;)
library(&quot;mclust&quot;)
library(&quot;slingshot&quot;)
library(&quot;irlba&quot;)
library(&quot;Rnanoflann&quot;)
library(&quot;ggpubr&quot;)</code></pre>
</div>
<div id="download-the-dataset" class="section level1">
<h1>Download the dataset</h1>
<pre class="r"><code>spe &lt;- fetch_data(type = &#39;spe&#39;)</code></pre>
<div id="extract-the-metadata-information" class="section level2">
<h2>Extract the metadata information</h2>
<pre class="r"><code>n.cores=40
splitting = 100
spatial.resolution = 0.3
aa_noise=3
gene_number=2000
graph = 20
seed=543210


set.seed(seed)
ID=unlist(lapply(strsplit(rownames(colData(spe)),&quot;-&quot;),function(x) x[1]))
samples=colData(spe)$sample_id
rownames(colData(spe))=paste(ID,samples,sep=&quot;-&quot;)

txtfile=paste(splitting,spatial.resolution,aa_noise,2,gene_number,sep=&quot;_&quot;)

sample_names=c(&quot;151507&quot;,
               &quot;151508&quot;,
               &quot;151509&quot;,
               &quot;151510&quot;,
               &quot;151669&quot;,
               &quot;151670&quot;,
               &quot;151671&quot;,
               &quot;151672&quot;,
               &quot;151673&quot;,
               &quot;151674&quot;,
               &quot;151675&quot;,
               &quot;151676&quot;)
subject_names= c(&quot;Br5292&quot;,&quot;Br5595&quot;, &quot;Br8100&quot;)
metaData = SingleCellExperiment::colData(spe)
expr = SingleCellExperiment::counts(spe)
sample_names &lt;- paste0(&quot;sample_&quot;, unique(colData(spe)$sample_id))
sample_names &lt;-  unique(colData(spe)$sample_id)
dim(spe)</code></pre>
<pre><code>[1] 33538 47681</code></pre>
<pre class="r"><code># identify mitochondrial genes
is_mito &lt;- grepl(&quot;(^MT-)|(^mt-)&quot;, rowData(spe)$gene_name)
table(is_mito)</code></pre>
<pre><code>is_mito
FALSE  TRUE 
33525    13 </code></pre>
<pre class="r"><code># calculate per-spot QC metrics
spe &lt;- addPerCellQC(spe, subsets = list(mito = is_mito))

# select QC thresholds
qc_lib_size &lt;- colData(spe)$sum &lt; 500
qc_detected &lt;- colData(spe)$detected &lt; 250
qc_mito &lt;- colData(spe)$subsets_mito_percent &gt; 30
qc_cell_count &lt;- colData(spe)$cell_count &gt; 12

# spots to discard
discard &lt;- qc_lib_size | qc_detected | qc_mito | qc_cell_count
table(discard)</code></pre>
<pre><code>discard
FALSE  TRUE 
46653  1028 </code></pre>
<pre class="r"><code>colData(spe)$discard &lt;- discard
# filter low-quality spots
spe &lt;- spe[, !colData(spe)$discard]
dim(spe)</code></pre>
<pre><code>[1] 33538 46653</code></pre>
<pre class="r"><code>spe &lt;- filter_genes(
  spe,
  filter_genes_ncounts = 2,   #ncounts
  filter_genes_pcspots = 0.5,
  filter_mito = TRUE
)

dim(spe)</code></pre>
<pre><code>[1]  6623 46653</code></pre>
<pre class="r"><code>sel= !is.na(colData(spe)$layer_guess_reordered)
spe = spe[,sel]
dim(spe)</code></pre>
<pre><code>[1]  6623 46318</code></pre>
<pre class="r"><code>spe &lt;- computeLibraryFactors(spe)
spe &lt;- logNormCounts(spe)

 subjects=colData(spe)$subject
 labels=as.factor(colData(spe)$layer_guess_reordered)
 xy=as.matrix(spatialCoords(spe))
 samples=colData(spe)$sample_id
 
 cols_cluster &lt;- c(&quot;#0000b6&quot;, &quot;#81b29a&quot;, &quot;#f2cc8f&quot;,&quot;#e07a5f&quot;,
          &quot;#cc00b6&quot;, &quot;#81ccff&quot;, &quot;#33b233&quot;)

<<<<<<< HEAD
  plot_slide(xy,samples,labels,col=cols_cluster,size.dot = 1)</code></pre>
=======
  plot_slide(xy,samples,labels,col=cols_cluster)</code></pre>
>>>>>>> d89baaa7bfc3bffd762b0db8bac45eeb05443b97
<p><img src="figure/DLPFC.Rmd/unnamed-chunk-4-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/374d5f014b2a22cb41b1c5950c364ae56235250d/docs/figure/DLPFC.Rmd/unnamed-chunk-4-1.png" target="_blank">374d5f0</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-12-14
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/f6bab12992a50883783f079203211492efa5647d/docs/figure/DLPFC.Rmd/unnamed-chunk-4-1.png" target="_blank">f6bab12</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>png 
  2 </code></pre>
</div>
<div id="gene-selection" class="section level2">
<h2>Gene selection</h2>
<p>The identification of genes that display spatial expression patterns
is performed using the SPARKX method (<a
href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-021-02404-0">Zhu
et al. (2021)</a>). The genes are ranked based on the median value of
the logarithm value of the p-value obtained in each slide
individually.</p>
<pre class="r"><code>top=multi_SPARKX(spe,n.cores=n.cores)</code></pre>
<pre><code>Warning in asMethod(object): sparse-&gt;dense coercion: allocating vector of size
2.3 GiB</code></pre>
<pre class="r"><code>data=as.matrix(t(logcounts(spe)[top[1:gene_number],]))

genes=spe@rowRanges@elementMetadata$gene_name
names(genes)=spe@rowRanges@elementMetadata$gene_id

samples=colData(spe)$sample_id
labels=as.factor(colData(spe)$layer_guess_reordered)
names(labels)=rownames(colData(spe))
subjects=colData(spe)$subject
genes=rowData(spe)[,&quot;gene_name&quot;]
names(genes)=rowData(spe)$gene_id
genes_top=genes[top[1:gene_number]]</code></pre>
</div>
</div>
<div id="patient-br5595" class="section level1">
<h1>Patient Br5595</h1>
<pre class="r"><code>subject_names=&quot;Br5595&quot;
nclusters=5

spe_sub &lt;- spe[, colData(spe)$subject ==  subject_names]
 # subjects=colData(spe_sub)$subject
dim(spe_sub)</code></pre>
<pre><code>[1]  6623 14646</code></pre>
<pre class="r"><code>#  spe_sub &lt;- runPCA(spe_sub, 50,subset_row = top[1:gene_number], scale=TRUE)
#pca=reducedDim(spe_sub,type = &quot;PCA&quot;)[,1:50]
  
spe_sub &lt;- spe[, colData(spe)$subject ==  subject_names]
sel= subjects ==  subject_names
        
data_Br5595=data[sel,top[1:gene_number]]
        
RNA.scaled=scale(data_Br5595)
pca_results &lt;- irlba(A = RNA.scaled, nv = 50)
pca_Br5595 &lt;- pca_results$u %*% diag(pca_results$d)[,1:50]
rownames(pca_Br5595)=rownames(data_Br5595)
colnames(pca_Br5595)=paste(&quot;PC&quot;,1:50,sep=&quot;&quot;)
labels=as.factor(colData(spe_sub)$layer_guess_reordered)
names(labels)=rownames(colData(spe_sub))
xy=as.matrix(spatialCoords(spe_sub))
rownames(xy)=rownames(colData(spe_sub))
samples=colData(spe_sub)$sample_id

        subject_names_Br5595=colData(spe_sub)$subject

plot(pca_Br5595, pch=20,col=as.factor(colData(spe_sub)$sample_id))</code></pre>
<p><img src="figure/DLPFC.Rmd/unnamed-chunk-7-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/374d5f014b2a22cb41b1c5950c364ae56235250d/docs/figure/DLPFC.Rmd/unnamed-chunk-7-1.png" target="_blank">374d5f0</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-12-14
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/fd8d0925cad5881c332a822c5d7e5befd27ef73d/docs/figure/DLPFC.Rmd/unnamed-chunk-7-1.png" target="_blank">fd8d092</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/1352d911e77a96a9c1216f1473ee5d13bc6bb3b4/docs/figure/DLPFC.Rmd/unnamed-chunk-7-1.png" target="_blank">1352d91</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/6038af1ae7d982283a26fe00a2ca6547cfe2ad13/docs/figure/DLPFC.Rmd/unnamed-chunk-7-1.png" target="_blank">6038af1</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="kodama-analysis" class="section level2">
<h2>KODAMA analysis</h2>
<pre class="r"><code>set.seed(seed)
kk=KODAMA.matrix.parallel(pca_Br5595,
                          spatial = xy,
                          samples=samples,
<<<<<<< HEAD
=======
                          FUN= &quot;fastpls&quot; ,
>>>>>>> d89baaa7bfc3bffd762b0db8bac45eeb05443b97
                          landmarks = 100000,
                          splitting = splitting,
                          ncomp = 50,
                          spatial.resolution = spatial.resolution,
                          n.cores=n.cores,
                          seed = seed)</code></pre>
<pre><code>Calculating Network

Calculating Network spatial
socket cluster with 40 nodes on host &#39;localhost&#39;
================================================================================
Finished parallel computation

[1] &quot;Calculation of dissimilarity matrix...&quot;
================================================================================</code></pre>
<pre class="r"><code>print(&quot;KODAMA finished&quot;)</code></pre>
<pre><code>[1] &quot;KODAMA finished&quot;</code></pre>
<pre class="r"><code>config=umap.defaults
config$n_threads = n.cores
config$n_sgd_threads = &quot;auto&quot;

kk_UMAP=KODAMA.visualization(kk,method=&quot;UMAP&quot;,config=config)
plot(kk_UMAP,pch=20,col=cols_cluster[labels])</code></pre>
<p><img src="figure/DLPFC.Rmd/unnamed-chunk-8-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-1">
Past versions of unnamed-chunk-8-1.png
</button>
</p>
<div id="fig-unnamed-chunk-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/374d5f014b2a22cb41b1c5950c364ae56235250d/docs/figure/DLPFC.Rmd/unnamed-chunk-8-1.png" target="_blank">374d5f0</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-12-14
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/fd8d0925cad5881c332a822c5d7e5befd27ef73d/docs/figure/DLPFC.Rmd/unnamed-chunk-8-1.png" target="_blank">fd8d092</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/1352d911e77a96a9c1216f1473ee5d13bc6bb3b4/docs/figure/DLPFC.Rmd/unnamed-chunk-8-1.png" target="_blank">1352d91</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/6038af1ae7d982283a26fe00a2ca6547cfe2ad13/docs/figure/DLPFC.Rmd/unnamed-chunk-8-1.png" target="_blank">6038af1</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>png 
  2 </code></pre>
</div>
<div id="graph-based-clustering" class="section level2">
<h2>Graph-based clustering</h2>
<pre class="r"><code>    # Graph-based clustering

g &lt;- bluster::makeSNNGraph(as.matrix(kk_UMAP), k = 20)
g_walk &lt;- igraph::cluster_walktrap(g)
clu &lt;- as.character(igraph::cut_at(g_walk, no = 2))
plot(kk_UMAP,pch=20,col=cols_cluster[as.factor(clu)])</code></pre>
<p><img src="figure/DLPFC.Rmd/unnamed-chunk-10-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-1">
Past versions of unnamed-chunk-10-1.png
</button>
</p>
<div id="fig-unnamed-chunk-10-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/7b2cb8c2c66c61a5abd5703a20bb718daeae4ac5/docs/figure/DLPFC.Rmd/unnamed-chunk-10-1.png" target="_blank">7b2cb8c</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/374d5f014b2a22cb41b1c5950c364ae56235250d/docs/figure/DLPFC.Rmd/unnamed-chunk-10-1.png" target="_blank">374d5f0</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-12-14
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/fd8d0925cad5881c332a822c5d7e5befd27ef73d/docs/figure/DLPFC.Rmd/unnamed-chunk-10-1.png" target="_blank">fd8d092</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/c9d54eec14966990195df616e6151496bfb3cc5d/docs/figure/DLPFC.Rmd/unnamed-chunk-10-1.png" target="_blank">c9d54ee</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-11
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/1352d911e77a96a9c1216f1473ee5d13bc6bb3b4/docs/figure/DLPFC.Rmd/unnamed-chunk-10-1.png" target="_blank">1352d91</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/6038af1ae7d982283a26fe00a2ca6547cfe2ad13/docs/figure/DLPFC.Rmd/unnamed-chunk-10-1.png" target="_blank">6038af1</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/35ce733e4e0012d79f696f9c7dcdbf67e1623e80/docs/figure/DLPFC.Rmd/unnamed-chunk-10-1.png" target="_blank">35ce733</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-08-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/f8ca54aee4428df50d2cd0f41cbf2a0ceae01d7d/docs/figure/DLPFC.Rmd/unnamed-chunk-10-1.png" target="_blank">f8ca54a</a>
</td>
<td>
tkcaccia
</td>
<td>
2024-07-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
<<<<<<< HEAD
<pre class="r"><code>  plot_slide(xy,as.factor(samples),clu,col=cols_cluster,size.dot = 1)</code></pre>
=======
<pre class="r"><code>  plot_slide(xy,as.factor(samples),clu,col=cols_cluster)</code></pre>
>>>>>>> d89baaa7bfc3bffd762b0db8bac45eeb05443b97
<p><img src="figure/DLPFC.Rmd/unnamed-chunk-10-2.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-2">
Past versions of unnamed-chunk-10-2.png
</button>
</p>
<div id="fig-unnamed-chunk-10-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/7b2cb8c2c66c61a5abd5703a20bb718daeae4ac5/docs/figure/DLPFC.Rmd/unnamed-chunk-10-2.png" target="_blank">7b2cb8c</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-12-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>png 
  2 </code></pre>
<pre class="r"><code>FB=names(which.min(table(clu)))
selFB=clu!=FB

 # kk_UMAP=kk_UMAP[selFB,]
#  labels=labels[selFB]
#  samples=samples[selFB]
#  xy=xy[selFB,]
    
      
g &lt;- bluster::makeSNNGraph(as.matrix(kk_UMAP[selFB,]), k = graph)
g_walk &lt;- igraph::cluster_walktrap(g)
clu &lt;- as.character(igraph::cut_at(g_walk, no = nclusters))
plot(kk_UMAP[selFB,],pch=20,col=as.factor(clu))</code></pre>
<p><img src="figure/DLPFC.Rmd/unnamed-chunk-12-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-1">
Past versions of unnamed-chunk-12-1.png
</button>
</p>
<div id="fig-unnamed-chunk-12-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/7b2cb8c2c66c61a5abd5703a20bb718daeae4ac5/docs/figure/DLPFC.Rmd/unnamed-chunk-12-1.png" target="_blank">7b2cb8c</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/374d5f014b2a22cb41b1c5950c364ae56235250d/docs/figure/DLPFC.Rmd/unnamed-chunk-12-1.png" target="_blank">374d5f0</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-12-14
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/fd8d0925cad5881c332a822c5d7e5befd27ef73d/docs/figure/DLPFC.Rmd/unnamed-chunk-12-1.png" target="_blank">fd8d092</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/1edc32b6806afe1639b806a6a7a2bdc873ec6a24/docs/figure/DLPFC.Rmd/unnamed-chunk-12-1.png" target="_blank">1edc32b</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-11
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/c9d54eec14966990195df616e6151496bfb3cc5d/docs/figure/DLPFC.Rmd/unnamed-chunk-12-1.png" target="_blank">c9d54ee</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-11
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/1352d911e77a96a9c1216f1473ee5d13bc6bb3b4/docs/figure/DLPFC.Rmd/unnamed-chunk-12-1.png" target="_blank">1352d91</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/6038af1ae7d982283a26fe00a2ca6547cfe2ad13/docs/figure/DLPFC.Rmd/unnamed-chunk-12-1.png" target="_blank">6038af1</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/35ce733e4e0012d79f696f9c7dcdbf67e1623e80/docs/figure/DLPFC.Rmd/unnamed-chunk-12-1.png" target="_blank">35ce733</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-08-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/f8ca54aee4428df50d2cd0f41cbf2a0ceae01d7d/docs/figure/DLPFC.Rmd/unnamed-chunk-12-1.png" target="_blank">f8ca54a</a>
</td>
<td>
tkcaccia
</td>
<td>
2024-07-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ref=refine_SVM(xy[selFB,],clu,samples[selFB],cost=100)</code></pre>
<pre><code>[1] &quot;151669&quot;
[1] &quot;151670&quot;
[1] &quot;151671&quot;
[1] &quot;151672&quot;</code></pre>
<pre class="r"><code>u=unique(samples[selFB])
for(j in u){
  sel=samples[selFB]==j
  print(mclust::adjustedRandIndex(labels[selFB][sel],ref[sel]))
}</code></pre>
<pre><code>[1] 0.7433529
[1] 0.7503761
[1] 0.8065421
[1] 0.745585</code></pre>
<<<<<<< HEAD
<pre class="r"><code>plot_slide(xy,samples,labels,col=cols_cluster,size.dot = 1)</code></pre>
=======
<pre class="r"><code>plot_slide(xy,samples,labels,col=cols_cluster)</code></pre>
>>>>>>> d89baaa7bfc3bffd762b0db8bac45eeb05443b97
<p><img src="figure/DLPFC.Rmd/unnamed-chunk-12-2.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-2">
Past versions of unnamed-chunk-12-2.png
</button>
</p>
<div id="fig-unnamed-chunk-12-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/7b2cb8c2c66c61a5abd5703a20bb718daeae4ac5/docs/figure/DLPFC.Rmd/unnamed-chunk-12-2.png" target="_blank">7b2cb8c</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/374d5f014b2a22cb41b1c5950c364ae56235250d/docs/figure/DLPFC.Rmd/unnamed-chunk-12-2.png" target="_blank">374d5f0</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-12-14
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/fd8d0925cad5881c332a822c5d7e5befd27ef73d/docs/figure/DLPFC.Rmd/unnamed-chunk-12-2.png" target="_blank">fd8d092</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/1edc32b6806afe1639b806a6a7a2bdc873ec6a24/docs/figure/DLPFC.Rmd/unnamed-chunk-12-2.png" target="_blank">1edc32b</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-11
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/c9d54eec14966990195df616e6151496bfb3cc5d/docs/figure/DLPFC.Rmd/unnamed-chunk-12-2.png" target="_blank">c9d54ee</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-11
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/1352d911e77a96a9c1216f1473ee5d13bc6bb3b4/docs/figure/DLPFC.Rmd/unnamed-chunk-12-2.png" target="_blank">1352d91</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/35ce733e4e0012d79f696f9c7dcdbf67e1623e80/docs/figure/DLPFC.Rmd/unnamed-chunk-12-2.png" target="_blank">35ce733</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-08-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/f8ca54aee4428df50d2cd0f41cbf2a0ceae01d7d/docs/figure/DLPFC.Rmd/unnamed-chunk-12-2.png" target="_blank">f8ca54a</a>
</td>
<td>
tkcaccia
</td>
<td>
2024-07-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
<<<<<<< HEAD
<pre class="r"><code>plot_slide(xy[selFB,],samples[selFB],ref,col=cols_cluster,size.dot = 1)</code></pre>
=======
<pre class="r"><code>plot_slide(xy[selFB,],samples[selFB],ref,col=cols_cluster)</code></pre>
>>>>>>> d89baaa7bfc3bffd762b0db8bac45eeb05443b97
<p><img src="figure/DLPFC.Rmd/unnamed-chunk-12-3.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-3">
Past versions of unnamed-chunk-12-3.png
</button>
</p>
<div id="fig-unnamed-chunk-12-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/7b2cb8c2c66c61a5abd5703a20bb718daeae4ac5/docs/figure/DLPFC.Rmd/unnamed-chunk-12-3.png" target="_blank">7b2cb8c</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/374d5f014b2a22cb41b1c5950c364ae56235250d/docs/figure/DLPFC.Rmd/unnamed-chunk-12-3.png" target="_blank">374d5f0</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-12-14
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/35ce733e4e0012d79f696f9c7dcdbf67e1623e80/docs/figure/DLPFC.Rmd/unnamed-chunk-12-3.png" target="_blank">35ce733</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-08-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/f8ca54aee4428df50d2cd0f41cbf2a0ceae01d7d/docs/figure/DLPFC.Rmd/unnamed-chunk-12-3.png" target="_blank">f8ca54a</a>
</td>
<td>
tkcaccia
</td>
<td>
2024-07-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>kk_UMAP_Br5595=kk_UMAP
samples_Br5595=samples
xy_Br5595=xy
labels_Br5595=labels
ref_Br5595=ref
clu_Br5595=clu

save(top,kk_UMAP_Br5595,samples_Br5595,xy_Br5595,labels_Br5595,subject_names_Br5595,ref_Br5595,clu_Br5595,selFB,file=&quot;output/DLFPC-Br5595.RData&quot;)

save(top,data_Br5595,pca_Br5595,samples_Br5595,xy_Br5595,labels_Br5595,subject_names_Br5595,selFB,file=&quot;data/DLFPC-Br5595-input.RData&quot;)</code></pre>
</div>
</div>
<div id="patient-br5292" class="section level1">
<h1>Patient Br5292</h1>
<pre class="r"><code>subject_names=&quot;Br5292&quot;
nclusters=7

<<<<<<< HEAD
spe_sub &lt;- spe[, colData(spe)$subject ==  subject_names]
dim(spe_sub)</code></pre>
<pre><code>[1]  6623 17734</code></pre>
<pre class="r"><code>spe_sub &lt;- spe[, colData(spe)$subject ==  subject_names]
sel= subjects ==  subject_names
data_Br5292=data[sel,top[1:gene_number]]
RNA.scaled=scale(data_Br5292)
pca_results &lt;- irlba(A = RNA.scaled, nv = 50)
pca_Br5292 &lt;- pca_results$u %*% diag(pca_results$d)[,1:50]
rownames(pca_Br5292)=rownames(data_Br5292)
colnames(pca_Br5292)=paste(&quot;PC&quot;,1:50,sep=&quot;&quot;)
labels=as.factor(colData(spe_sub)$layer_guess_reordered)
names(labels)=rownames(colData(spe_sub))
xy=as.matrix(spatialCoords(spe_sub))
rownames(xy)=rownames(colData(spe_sub))
samples=colData(spe_sub)$sample_id
subject_names_Br5292=colData(spe_sub)$subject
plot(pca_Br5292, pch=20,col=as.factor(colData(spe_sub)$sample_id))</code></pre>
=======
  spe_sub &lt;- spe[, colData(spe)$subject ==  subject_names]
  dim(spe_sub)</code></pre>
<pre><code>[1]  6623 17734</code></pre>
<pre class="r"><code>#  spe_sub &lt;- runPCA(spe_sub, 50,subset_row = top[1:gene_number], scale=TRUE)

  #pca=reducedDim(spe_sub,type = &quot;PCA&quot;)[,1:50]
  
  
  
    
        
        spe_sub &lt;- spe[, colData(spe)$subject ==  subject_names]
        sel= subjects ==  subject_names
        

        data_Br5292=data[sel,top[1:gene_number]]
        
        RNA.scaled=scale(data_Br5292)
        pca_results &lt;- irlba(A = RNA.scaled, nv = 50)
        pca_Br5292 &lt;- pca_results$u %*% diag(pca_results$d)[,1:50]
        rownames(pca_Br5292)=rownames(data_Br5292)
        colnames(pca_Br5292)=paste(&quot;PC&quot;,1:50,sep=&quot;&quot;)
        labels=as.factor(colData(spe_sub)$layer_guess_reordered)
        names(labels)=rownames(colData(spe_sub))
        xy=as.matrix(spatialCoords(spe_sub))
        rownames(xy)=rownames(colData(spe_sub))
        samples=colData(spe_sub)$sample_id
        subject_names_Br5292=colData(spe_sub)$subject
  plot(pca_Br5292, pch=20,col=as.factor(colData(spe_sub)$sample_id))</code></pre>
>>>>>>> d89baaa7bfc3bffd762b0db8bac45eeb05443b97
<p><img src="figure/DLPFC.Rmd/unnamed-chunk-13-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-13-1">
Past versions of unnamed-chunk-13-1.png
</button>
</p>
<div id="fig-unnamed-chunk-13-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/7b2cb8c2c66c61a5abd5703a20bb718daeae4ac5/docs/figure/DLPFC.Rmd/unnamed-chunk-13-1.png" target="_blank">7b2cb8c</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/374d5f014b2a22cb41b1c5950c364ae56235250d/docs/figure/DLPFC.Rmd/unnamed-chunk-13-1.png" target="_blank">374d5f0</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-12-14
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/fd8d0925cad5881c332a822c5d7e5befd27ef73d/docs/figure/DLPFC.Rmd/unnamed-chunk-13-1.png" target="_blank">fd8d092</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/1352d911e77a96a9c1216f1473ee5d13bc6bb3b4/docs/figure/DLPFC.Rmd/unnamed-chunk-13-1.png" target="_blank">1352d91</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/6038af1ae7d982283a26fe00a2ca6547cfe2ad13/docs/figure/DLPFC.Rmd/unnamed-chunk-13-1.png" target="_blank">6038af1</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/35ce733e4e0012d79f696f9c7dcdbf67e1623e80/docs/figure/DLPFC.Rmd/unnamed-chunk-13-1.png" target="_blank">35ce733</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-08-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/f8ca54aee4428df50d2cd0f41cbf2a0ceae01d7d/docs/figure/DLPFC.Rmd/unnamed-chunk-13-1.png" target="_blank">f8ca54a</a>
</td>
<td>
tkcaccia
</td>
<td>
2024-07-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="kodama-analysis-1" class="section level2">
<h2>KODAMA analysis</h2>
<pre class="r"><code>set.seed(seed)
kk=KODAMA.matrix.parallel(pca_Br5292,
                          
                          spatial = xy,
                          samples=samples,
<<<<<<< HEAD
=======
                          FUN= &quot;fastpls&quot; ,
>>>>>>> d89baaa7bfc3bffd762b0db8bac45eeb05443b97
                          landmarks = 100000,
                          splitting = splitting,
                          ncomp = 50,
                          spatial.resolution = spatial.resolution,
                          n.cores=n.cores,
                          seed = seed)</code></pre>
<pre><code>Calculating Network

Calculating Network spatial
socket cluster with 40 nodes on host &#39;localhost&#39;
================================================================================
Finished parallel computation

[1] &quot;Calculation of dissimilarity matrix...&quot;
================================================================================</code></pre>
<pre class="r"><code>  print(&quot;KODAMA finished&quot;)</code></pre>
<pre><code>[1] &quot;KODAMA finished&quot;</code></pre>
<pre class="r"><code>     config=umap.defaults
     config$n_threads = n.cores
     config$n_sgd_threads = &quot;auto&quot;
     kk_UMAP=KODAMA.visualization(kk,method=&quot;UMAP&quot;,config=config)

     plot(kk_UMAP,pch=20,col=as.factor(labels))</code></pre>
<p><img src="figure/DLPFC.Rmd/unnamed-chunk-14-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-1">
Past versions of unnamed-chunk-14-1.png
</button>
</p>
<div id="fig-unnamed-chunk-14-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/7b2cb8c2c66c61a5abd5703a20bb718daeae4ac5/docs/figure/DLPFC.Rmd/unnamed-chunk-14-1.png" target="_blank">7b2cb8c</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/374d5f014b2a22cb41b1c5950c364ae56235250d/docs/figure/DLPFC.Rmd/unnamed-chunk-14-1.png" target="_blank">374d5f0</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-12-14
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/fd8d0925cad5881c332a822c5d7e5befd27ef73d/docs/figure/DLPFC.Rmd/unnamed-chunk-14-1.png" target="_blank">fd8d092</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/1352d911e77a96a9c1216f1473ee5d13bc6bb3b4/docs/figure/DLPFC.Rmd/unnamed-chunk-14-1.png" target="_blank">1352d91</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/6038af1ae7d982283a26fe00a2ca6547cfe2ad13/docs/figure/DLPFC.Rmd/unnamed-chunk-14-1.png" target="_blank">6038af1</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/35ce733e4e0012d79f696f9c7dcdbf67e1623e80/docs/figure/DLPFC.Rmd/unnamed-chunk-14-1.png" target="_blank">35ce733</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-08-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="graph-based-clustering-1" class="section level2">
<h2>Graph-based clustering</h2>
<pre class="r"><code>    # Graph-based clustering

  
        
        g &lt;- bluster::makeSNNGraph(as.matrix(kk_UMAP), k = graph)
        
        g_walk &lt;- igraph::cluster_walktrap(g)
        clu &lt;- as.character(igraph::cut_at(g_walk, no = nclusters))
        
    plot(kk_UMAP,pch=20,col=as.factor(clu))</code></pre>
<p><img src="figure/DLPFC.Rmd/unnamed-chunk-15-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-15-1">
Past versions of unnamed-chunk-15-1.png
</button>
</p>
<div id="fig-unnamed-chunk-15-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/7b2cb8c2c66c61a5abd5703a20bb718daeae4ac5/docs/figure/DLPFC.Rmd/unnamed-chunk-15-1.png" target="_blank">7b2cb8c</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/374d5f014b2a22cb41b1c5950c364ae56235250d/docs/figure/DLPFC.Rmd/unnamed-chunk-15-1.png" target="_blank">374d5f0</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-12-14
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/fd8d0925cad5881c332a822c5d7e5befd27ef73d/docs/figure/DLPFC.Rmd/unnamed-chunk-15-1.png" target="_blank">fd8d092</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/1352d911e77a96a9c1216f1473ee5d13bc6bb3b4/docs/figure/DLPFC.Rmd/unnamed-chunk-15-1.png" target="_blank">1352d91</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/6038af1ae7d982283a26fe00a2ca6547cfe2ad13/docs/figure/DLPFC.Rmd/unnamed-chunk-15-1.png" target="_blank">6038af1</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/35ce733e4e0012d79f696f9c7dcdbf67e1623e80/docs/figure/DLPFC.Rmd/unnamed-chunk-15-1.png" target="_blank">35ce733</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-08-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>        ref=refine_SVM(xy,clu,samples,cost=100)</code></pre>
<pre><code>[1] &quot;151507&quot;
[1] &quot;151508&quot;
[1] &quot;151509&quot;
[1] &quot;151510&quot;</code></pre>
<pre class="r"><code>        u=unique(samples)
        for(j in u){
          sel=samples==j
          
            print(mclust::adjustedRandIndex(labels[sel],ref[sel]))
        }</code></pre>
<pre><code>[1] 0.4462599
[1] 0.4843513
[1] 0.4496646
[1] 0.4064282</code></pre>
<pre class="r"><code>          g &lt;- bluster::makeSNNGraph(as.matrix(kk_UMAP), k = graph)
          
          g_walk &lt;- igraph::cluster_walktrap(g)
          clu &lt;- as.character(igraph::cut_at(g_walk, no = nclusters))
          ref=refine_SVM(xy,clu,samples,cost=100)</code></pre>
<pre><code>[1] &quot;151507&quot;
[1] &quot;151508&quot;
[1] &quot;151509&quot;
[1] &quot;151510&quot;</code></pre>
<pre class="r"><code>          u=unique(samples)
          for(j in u){
            sel=samples==j
            print(mclust::adjustedRandIndex(labels[sel],ref[sel]))
          }</code></pre>
<pre><code>[1] 0.4462599
[1] 0.4843513
[1] 0.4496646
[1] 0.4064282</code></pre>
<<<<<<< HEAD
<pre class="r"><code>plot_slide(xy,samples,labels,col=cols_cluster,size.dot = 1)</code></pre>
=======
<pre class="r"><code>plot_slide(xy,samples,labels,col=cols_cluster)</code></pre>
>>>>>>> d89baaa7bfc3bffd762b0db8bac45eeb05443b97
<p><img src="figure/DLPFC.Rmd/unnamed-chunk-15-2.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-15-2">
Past versions of unnamed-chunk-15-2.png
</button>
</p>
<div id="fig-unnamed-chunk-15-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/7b2cb8c2c66c61a5abd5703a20bb718daeae4ac5/docs/figure/DLPFC.Rmd/unnamed-chunk-15-2.png" target="_blank">7b2cb8c</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/374d5f014b2a22cb41b1c5950c364ae56235250d/docs/figure/DLPFC.Rmd/unnamed-chunk-15-2.png" target="_blank">374d5f0</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-12-14
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/fd8d0925cad5881c332a822c5d7e5befd27ef73d/docs/figure/DLPFC.Rmd/unnamed-chunk-15-2.png" target="_blank">fd8d092</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/1352d911e77a96a9c1216f1473ee5d13bc6bb3b4/docs/figure/DLPFC.Rmd/unnamed-chunk-15-2.png" target="_blank">1352d91</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<<<<<<< HEAD
<pre class="r"><code>    plot_slide(xy,samples,ref,col=cols_cluster,size.dot = 1)</code></pre>
=======
<pre class="r"><code>    plot_slide(xy,samples,ref,col=cols_cluster)</code></pre>
>>>>>>> d89baaa7bfc3bffd762b0db8bac45eeb05443b97
<p><img src="figure/DLPFC.Rmd/unnamed-chunk-15-3.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-15-3">
Past versions of unnamed-chunk-15-3.png
</button>
</p>
<div id="fig-unnamed-chunk-15-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/7b2cb8c2c66c61a5abd5703a20bb718daeae4ac5/docs/figure/DLPFC.Rmd/unnamed-chunk-15-3.png" target="_blank">7b2cb8c</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/374d5f014b2a22cb41b1c5950c364ae56235250d/docs/figure/DLPFC.Rmd/unnamed-chunk-15-3.png" target="_blank">374d5f0</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-12-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>kk_UMAP_Br5292=kk_UMAP
samples_Br5292=samples
xy_Br5292=xy
labels_Br5292=labels
ref_Br5292=ref
clu_Br5292=clu
save(top,kk_UMAP_Br5292,pca_Br5292,samples_Br5292,xy_Br5292,subject_names_Br5292,labels_Br5292,ref_Br5292,clu_Br5292,file=&quot;output/DLFPC-Br5292.RData&quot;)

save(top,data_Br5292,pca_Br5292,samples_Br5292,xy_Br5292,labels_Br5292,subject_names_Br5292,file=&quot;data/DLFPC-Br5292-input.RData&quot;)</code></pre>
</div>
</div>
<div id="patient-br8100" class="section level1">
<h1>Patient Br8100</h1>
<pre class="r"><code>subject_names=&quot;Br8100&quot;

nclusters=7

  spe_sub &lt;- spe[, colData(spe)$subject ==  subject_names]
  dim(spe_sub)</code></pre>
<pre><code>[1]  6623 13938</code></pre>
<pre class="r"><code>#  spe_sub &lt;- runPCA(spe_sub, 50,subset_row = top[1:gene_number], scale=TRUE)

  #pca=reducedDim(spe_sub,type = &quot;PCA&quot;)[,1:50]
  
  
  
    
        
        spe_sub &lt;- spe[, colData(spe)$subject ==  subject_names]
        sel= subjects ==  subject_names
        

        data_Br8100=data[sel,top[1:gene_number]]
        
        RNA.scaled=scale(data_Br8100)
        pca_results &lt;- irlba(A = RNA.scaled, nv = 50)
        pca_Br8100 &lt;- pca_results$u %*% diag(pca_results$d)[,1:50]
        rownames(pca_Br8100)=rownames(data_Br8100)
        colnames(pca_Br8100)=paste(&quot;PC&quot;,1:50,sep=&quot;&quot;)
        labels=as.factor(colData(spe_sub)$layer_guess_reordered)
        names(labels)=rownames(colData(spe_sub))
        xy=as.matrix(spatialCoords(spe_sub))
        rownames(xy)=rownames(colData(spe_sub))
        samples=colData(spe_sub)$sample_id
        subject_names_Br8100=colData(spe_sub)$subject

  plot(pca_Br8100, pch=20,col=as.factor(colData(spe_sub)$sample_id))</code></pre>
<p><img src="figure/DLPFC.Rmd/unnamed-chunk-16-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-16-1">
Past versions of unnamed-chunk-16-1.png
</button>
</p>
<div id="fig-unnamed-chunk-16-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/7b2cb8c2c66c61a5abd5703a20bb718daeae4ac5/docs/figure/DLPFC.Rmd/unnamed-chunk-16-1.png" target="_blank">7b2cb8c</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/fd8d0925cad5881c332a822c5d7e5befd27ef73d/docs/figure/DLPFC.Rmd/unnamed-chunk-16-1.png" target="_blank">fd8d092</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/1352d911e77a96a9c1216f1473ee5d13bc6bb3b4/docs/figure/DLPFC.Rmd/unnamed-chunk-16-1.png" target="_blank">1352d91</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/6038af1ae7d982283a26fe00a2ca6547cfe2ad13/docs/figure/DLPFC.Rmd/unnamed-chunk-16-1.png" target="_blank">6038af1</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/35ce733e4e0012d79f696f9c7dcdbf67e1623e80/docs/figure/DLPFC.Rmd/unnamed-chunk-16-1.png" target="_blank">35ce733</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-08-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="kodama-analysis-2" class="section level2">
<h2>KODAMA analysis</h2>
<pre class="r"><code>set.seed(seed)
kk=KODAMA.matrix.parallel(pca_Br8100,
                          spatial = xy,
                          samples=samples,
<<<<<<< HEAD
=======
                          FUN= &quot;fastpls&quot; ,
>>>>>>> d89baaa7bfc3bffd762b0db8bac45eeb05443b97
                          landmarks = 100000,
                          splitting = splitting,
                          ncomp = 50,
                          spatial.resolution = spatial.resolution,
                          n.cores=n.cores,
                          seed = seed)</code></pre>
<pre><code>Calculating Network

Calculating Network spatial
socket cluster with 40 nodes on host &#39;localhost&#39;
================================================================================
Finished parallel computation

[1] &quot;Calculation of dissimilarity matrix...&quot;
================================================================================</code></pre>
<pre class="r"><code>  print(&quot;KODAMA finished&quot;)</code></pre>
<pre><code>[1] &quot;KODAMA finished&quot;</code></pre>
<pre class="r"><code>     config=umap.defaults
     config$n_threads = n.cores
     config$n_sgd_threads = &quot;auto&quot;
     kk_UMAP=KODAMA.visualization(kk,method=&quot;UMAP&quot;,config=config)

     plot(kk_UMAP,pch=20,col=as.factor(labels))</code></pre>
<p><img src="figure/DLPFC.Rmd/unnamed-chunk-17-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-17-1">
Past versions of unnamed-chunk-17-1.png
</button>
</p>
<div id="fig-unnamed-chunk-17-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/7b2cb8c2c66c61a5abd5703a20bb718daeae4ac5/docs/figure/DLPFC.Rmd/unnamed-chunk-17-1.png" target="_blank">7b2cb8c</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/374d5f014b2a22cb41b1c5950c364ae56235250d/docs/figure/DLPFC.Rmd/unnamed-chunk-17-1.png" target="_blank">374d5f0</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-12-14
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/fd8d0925cad5881c332a822c5d7e5befd27ef73d/docs/figure/DLPFC.Rmd/unnamed-chunk-17-1.png" target="_blank">fd8d092</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/1352d911e77a96a9c1216f1473ee5d13bc6bb3b4/docs/figure/DLPFC.Rmd/unnamed-chunk-17-1.png" target="_blank">1352d91</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="graph-based-clustering-2" class="section level2">
<h2>Graph-based clustering</h2>
<pre class="r"><code>    # Graph-based clustering

  
        
        g &lt;- bluster::makeSNNGraph(as.matrix(kk_UMAP), k = graph)
        
        g_walk &lt;- igraph::cluster_walktrap(g)
        clu &lt;- as.character(igraph::cut_at(g_walk, no = nclusters))
        ref=refine_SVM(xy,clu,samples,cost=100)</code></pre>
<pre><code>[1] &quot;151673&quot;
[1] &quot;151674&quot;
[1] &quot;151675&quot;
[1] &quot;151676&quot;</code></pre>
<pre class="r"><code>        u=unique(samples)
        for(j in u){
          sel=samples==j
          
            print(mclust::adjustedRandIndex(labels[sel],ref[sel]))
        }</code></pre>
<pre><code>[1] 0.599265
[1] 0.658843
[1] 0.6486061
[1] 0.5928946</code></pre>
<pre class="r"><code>          g &lt;- bluster::makeSNNGraph(as.matrix(kk_UMAP), k = graph)
          
          g_walk &lt;- igraph::cluster_walktrap(g)
          clu &lt;- as.character(igraph::cut_at(g_walk, no = nclusters))
          
           plot(kk_UMAP,pch=20,col=as.factor(clu))</code></pre>
<p><img src="figure/DLPFC.Rmd/unnamed-chunk-18-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-1">
Past versions of unnamed-chunk-18-1.png
</button>
</p>
<div id="fig-unnamed-chunk-18-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/7b2cb8c2c66c61a5abd5703a20bb718daeae4ac5/docs/figure/DLPFC.Rmd/unnamed-chunk-18-1.png" target="_blank">7b2cb8c</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/374d5f014b2a22cb41b1c5950c364ae56235250d/docs/figure/DLPFC.Rmd/unnamed-chunk-18-1.png" target="_blank">374d5f0</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-12-14
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/fd8d0925cad5881c332a822c5d7e5befd27ef73d/docs/figure/DLPFC.Rmd/unnamed-chunk-18-1.png" target="_blank">fd8d092</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/1352d911e77a96a9c1216f1473ee5d13bc6bb3b4/docs/figure/DLPFC.Rmd/unnamed-chunk-18-1.png" target="_blank">1352d91</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/6038af1ae7d982283a26fe00a2ca6547cfe2ad13/docs/figure/DLPFC.Rmd/unnamed-chunk-18-1.png" target="_blank">6038af1</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/35ce733e4e0012d79f696f9c7dcdbf67e1623e80/docs/figure/DLPFC.Rmd/unnamed-chunk-18-1.png" target="_blank">35ce733</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-08-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>          ref=refine_SVM(xy,clu,samples,cost=100)</code></pre>
<pre><code>[1] &quot;151673&quot;
[1] &quot;151674&quot;
[1] &quot;151675&quot;
[1] &quot;151676&quot;</code></pre>
<pre class="r"><code>          u=unique(samples)
          for(j in u){
            sel=samples==j
            print(mclust::adjustedRandIndex(labels[sel],ref[sel]))
          }</code></pre>
<pre><code>[1] 0.599265
[1] 0.658843
[1] 0.6486061
[1] 0.5928946</code></pre>
<<<<<<< HEAD
<pre class="r"><code> plot_slide(xy,samples,labels,col=cols_cluster,size.dot = 1)</code></pre>
=======
<pre class="r"><code> plot_slide(xy,samples,labels,col=cols_cluster)</code></pre>
>>>>>>> d89baaa7bfc3bffd762b0db8bac45eeb05443b97
<p><img src="figure/DLPFC.Rmd/unnamed-chunk-18-2.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-2">
Past versions of unnamed-chunk-18-2.png
</button>
</p>
<div id="fig-unnamed-chunk-18-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/7b2cb8c2c66c61a5abd5703a20bb718daeae4ac5/docs/figure/DLPFC.Rmd/unnamed-chunk-18-2.png" target="_blank">7b2cb8c</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/1352d911e77a96a9c1216f1473ee5d13bc6bb3b4/docs/figure/DLPFC.Rmd/unnamed-chunk-18-2.png" target="_blank">1352d91</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/6038af1ae7d982283a26fe00a2ca6547cfe2ad13/docs/figure/DLPFC.Rmd/unnamed-chunk-18-2.png" target="_blank">6038af1</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/35ce733e4e0012d79f696f9c7dcdbf67e1623e80/docs/figure/DLPFC.Rmd/unnamed-chunk-18-2.png" target="_blank">35ce733</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-08-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<<<<<<< HEAD
<pre class="r"><code>    plot_slide(xy,samples,ref,col=cols_cluster,size.dot = 1)</code></pre>
=======
<pre class="r"><code>    plot_slide(xy,samples,ref,col=cols_cluster)</code></pre>
>>>>>>> d89baaa7bfc3bffd762b0db8bac45eeb05443b97
<p><img src="figure/DLPFC.Rmd/unnamed-chunk-18-3.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-3">
Past versions of unnamed-chunk-18-3.png
</button>
</p>
<div id="fig-unnamed-chunk-18-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/7b2cb8c2c66c61a5abd5703a20bb718daeae4ac5/docs/figure/DLPFC.Rmd/unnamed-chunk-18-3.png" target="_blank">7b2cb8c</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-12-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>kk_UMAP_Br8100=kk_UMAP
samples_Br8100=samples
xy_Br8100=xy
labels_Br8100=labels

ref_Br8100=ref
clu_Br8100=clu 
save(top,kk_UMAP_Br8100,pca_Br8100,samples_Br8100,xy_Br8100,subject_names_Br8100,labels_Br8100,ref_Br8100,clu_Br8100,file=&quot;output/DLFPC-Br8100.RData&quot;)


save(top,data_Br8100,pca_Br8100,samples_Br8100,xy_Br8100,labels_Br8100,subject_names_Br8100,file=&quot;data/DLFPC-Br8100-input.RData&quot;)</code></pre>
</div>
</div>
<div id="saving-the-results" class="section level1">
<h1>Saving the results</h1>
<<<<<<< HEAD
=======
<pre class="r"><code>result_KODAMA &lt;- list(clusters=list(),
                      tissue_segments=list(),
                      feature_extraction=list(),
                      xy=list(),
                      subjects=list(),
                      samples=list())

result_KODAMA$clusters=c(result_KODAMA$clusters,tapply(ref_Br5292,samples_Br5292,function(x) x))
result_KODAMA$clusters=c(result_KODAMA$clusters,tapply(ref_Br5595,samples_Br5595[selFB],function(x) x))
result_KODAMA$clusters=c(result_KODAMA$clusters,tapply(ref_Br8100,samples_Br8100,function(x) x))

result_KODAMA$tissue_segments=c(result_KODAMA$tissue_segments,tapply(labels_Br5292,samples_Br5292,function(x) x))
result_KODAMA$tissue_segments=c(result_KODAMA$tissue_segments,tapply(labels_Br5595[selFB],samples_Br5595[selFB],function(x) x))
result_KODAMA$tissue_segments=c(result_KODAMA$tissue_segments,tapply(labels_Br8100,samples_Br8100,function(x) x))

result_KODAMA$feature_extraction=c(result_KODAMA$feature_extraction,by(kk_UMAP_Br5292,samples_Br5292,function(x) x))
result_KODAMA$feature_extraction=c(result_KODAMA$feature_extraction,by(kk_UMAP_Br5595[selFB,],samples_Br5595[selFB],function(x) x))
result_KODAMA$feature_extraction=c(result_KODAMA$feature_extraction,by(kk_UMAP_Br8100,samples_Br8100,function(x) x))

result_KODAMA$xy=c(result_KODAMA$xy,by(xy_Br5292,samples_Br5292,function(x) x))
result_KODAMA$xy=c(result_KODAMA$xy,by(xy_Br5595[selFB,],samples_Br5595[selFB],function(x) x))
result_KODAMA$xy=c(result_KODAMA$xy,by(xy_Br8100,samples_Br8100,function(x) x))

result_KODAMA$samples=c(result_KODAMA$samples,tapply(samples_Br5292,samples_Br5292,function(x) x))
result_KODAMA$samples=c(result_KODAMA$samples,tapply(samples_Br5595[selFB],samples_Br5595[selFB],function(x) x))
result_KODAMA$samples=c(result_KODAMA$samples,tapply(samples_Br8100,samples_Br8100,function(x) x))

result_KODAMA$subjects=c(result_KODAMA$subjects,tapply(subject_names_Br5292,samples_Br5292,function(x) x))
result_KODAMA$subjects=c(result_KODAMA$subjects,tapply(subject_names_Br5595[selFB],samples_Br5595[selFB],function(x) x))
result_KODAMA$subjects=c(result_KODAMA$subjects,tapply(subject_names_Br8100,samples_Br8100,function(x) x))


ground_true &lt;- list(clusters=list(),
                    tissue_segments=list(),
                    xy=list(),
                    subjects=list(),
                    samples=list())

ground_true$tissue_segments=c(ground_true$tissue_segments,tapply(labels_Br5292,samples_Br5292,function(x) x))
ground_true$tissue_segments=c(ground_true$tissue_segments,tapply(labels_Br5595,samples_Br5595,function(x) x))
ground_true$tissue_segments=c(ground_true$tissue_segments,tapply(labels_Br8100,samples_Br8100,function(x) x))

ground_true$xy=c(ground_true$xy,by(xy_Br5292,samples_Br5292,function(x) x))
ground_true$xy=c(ground_true$xy,by(xy_Br5595,samples_Br5595,function(x) x))
ground_true$xy=c(ground_true$xy,by(xy_Br8100,samples_Br8100,function(x) x))

ground_true$clusters=ground_true$tissue_segments



ground_true$samples=c(ground_true$samples,tapply(samples_Br5292,samples_Br5292,function(x) x))
ground_true$samples=c(ground_true$samples,tapply(samples_Br5595,samples_Br5595,function(x) x))
ground_true$samples=c(ground_true$samples,tapply(samples_Br8100,samples_Br8100,function(x) x))

ground_true$subjects=c(ground_true$subjects,tapply(subject_names_Br5292,samples_Br5292,function(x) x))
ground_true$subjects=c(ground_true$subjects,tapply(subject_names_Br5595,samples_Br5595,function(x) x))
ground_true$subjects=c(ground_true$subjects,tapply(subject_names_Br8100,samples_Br8100,function(x) x))


save(result_KODAMA,ground_true,file=&quot;output/KODAMA-results.RData&quot;)</code></pre>
<p>#seurat list preprocessing</p>
<pre class="r"><code>source(&quot;code/DLFPC_preprocessing.R&quot;)</code></pre>
>>>>>>> d89baaa7bfc3bffd762b0db8bac45eeb05443b97
<pre><code>[1] 1
[1] 2
[1] 3
[1] 4
[1] 5
[1] 6
[1] 7
[1] 8
[1] 9
[1] 10
[1] 11
[1] 12</code></pre>
</div>
<div id="slides" class="section level1">
<h1>12 Slides</h1>
<p>PCA and HARMONY</p>
<<<<<<< HEAD
=======
<pre class="r"><code> dim(spe_sub)</code></pre>
<pre><code>[1]  6623 13938</code></pre>
>>>>>>> d89baaa7bfc3bffd762b0db8bac45eeb05443b97
<pre class="r"><code> spe &lt;- runPCA(spe, 50,subset_row = top[1:gene_number], scale=TRUE)


 subjects=colData(spe)$subject
 labels=as.factor(colData(spe)$layer_guess_reordered)
 xy=as.matrix(spatialCoords(spe))
 samples=colData(spe)$sample_id
 
  
 

 spe &lt;- RunHarmony(spe, &quot;subject&quot;,lambda=NULL)
 pca=reducedDim(spe,type = &quot;HARMONY&quot;)[,1:50]
 
 plot(pca, pch=20,col=as.factor(colData(spe_sub)$sample_id))</code></pre>
<<<<<<< HEAD
<p><img src="figure/DLPFC.Rmd/unnamed-chunk-20-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-20-1">
Past versions of unnamed-chunk-20-1.png
</button>
</p>
<div id="fig-unnamed-chunk-20-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/374d5f014b2a22cb41b1c5950c364ae56235250d/docs/figure/DLPFC.Rmd/unnamed-chunk-20-1.png" target="_blank">374d5f0</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-12-14
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/fd8d0925cad5881c332a822c5d7e5befd27ef73d/docs/figure/DLPFC.Rmd/unnamed-chunk-20-1.png" target="_blank">fd8d092</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/1352d911e77a96a9c1216f1473ee5d13bc6bb3b4/docs/figure/DLPFC.Rmd/unnamed-chunk-20-1.png" target="_blank">1352d91</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/6038af1ae7d982283a26fe00a2ca6547cfe2ad13/docs/figure/DLPFC.Rmd/unnamed-chunk-20-1.png" target="_blank">6038af1</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/35ce733e4e0012d79f696f9c7dcdbf67e1623e80/docs/figure/DLPFC.Rmd/unnamed-chunk-20-1.png" target="_blank">35ce733</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-08-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>KODAMA</p>
<pre class="r"><code>set.seed(seed)
kk=KODAMA.matrix.parallel(pca,
                          spatial = xy,
                          samples=samples,
                          landmarks = 100000,
                          splitting = splitting,
                          ncomp = 50,
                          spatial.resolution = spatial.resolution,
                          n.cores=n.cores,
                          seed = seed)</code></pre>
<pre><code>Calculating Network

Calculating Network spatial
socket cluster with 40 nodes on host &#39;localhost&#39;
================================================================================
Finished parallel computation

[1] &quot;Calculation of dissimilarity matrix...&quot;
================================================================================</code></pre>
<pre class="r"><code>print(&quot;KODAMA finished&quot;)</code></pre>
<pre><code>[1] &quot;KODAMA finished&quot;</code></pre>
<pre class="r"><code>config=umap.defaults
config$n_threads = n.cores
config$n_sgd_threads = &quot;auto&quot;

kk_UMAP=KODAMA.visualization(kk,method=&quot;UMAP&quot;,config=config)
plot(kk_UMAP,pch=20,col=as.factor(labels))</code></pre>
=======
>>>>>>> d89baaa7bfc3bffd762b0db8bac45eeb05443b97
<p><img src="figure/DLPFC.Rmd/unnamed-chunk-21-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-21-1">
Past versions of unnamed-chunk-21-1.png
</button>
</p>
<div id="fig-unnamed-chunk-21-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/7b2cb8c2c66c61a5abd5703a20bb718daeae4ac5/docs/figure/DLPFC.Rmd/unnamed-chunk-21-1.png" target="_blank">7b2cb8c</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/374d5f014b2a22cb41b1c5950c364ae56235250d/docs/figure/DLPFC.Rmd/unnamed-chunk-21-1.png" target="_blank">374d5f0</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-12-14
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/fd8d0925cad5881c332a822c5d7e5befd27ef73d/docs/figure/DLPFC.Rmd/unnamed-chunk-21-1.png" target="_blank">fd8d092</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/1352d911e77a96a9c1216f1473ee5d13bc6bb3b4/docs/figure/DLPFC.Rmd/unnamed-chunk-21-1.png" target="_blank">1352d91</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/6038af1ae7d982283a26fe00a2ca6547cfe2ad13/docs/figure/DLPFC.Rmd/unnamed-chunk-21-1.png" target="_blank">6038af1</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<<<<<<< HEAD
<pre class="r"><code>df &lt;- data.frame(kk_UMAP[,1:2], tissue=labels,check.names = FALSE)
 
plot1 = ggplot(df, aes(`Dimension 1`, `Dimension 2`, color = tissue)) +labs(title=&quot;KODAMA&quot;) +
     geom_point(size = 1) +
     theme_bw() + theme(legend.position = &quot;bottom&quot;,
                        legend.text = element_text(size = 20),
                        legend.title = element_text(size = 20),
                        axis.title = element_text(size = 22),       # x and y axis labels
                        axis.text = element_text(size = 16),        # tick labels
                        plot.title = element_text(size = 26, face = &quot;bold&quot;, hjust = 0) )+
     scale_color_manual(&quot;Domain&quot;, values = cols_cluster) +
     
     guides(color = guide_legend(nrow = 1,override.aes = list(size = 10)))
plot1</code></pre>
<p><img src="figure/DLPFC.Rmd/unnamed-chunk-21-2.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-21-2">
Past versions of unnamed-chunk-21-2.png
</button>
</p>
<div id="fig-unnamed-chunk-21-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/7b2cb8c2c66c61a5abd5703a20bb718daeae4ac5/docs/figure/DLPFC.Rmd/unnamed-chunk-21-2.png" target="_blank">7b2cb8c</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/374d5f014b2a22cb41b1c5950c364ae56235250d/docs/figure/DLPFC.Rmd/unnamed-chunk-21-2.png" target="_blank">374d5f0</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-12-14
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/fd8d0925cad5881c332a822c5d7e5befd27ef73d/docs/figure/DLPFC.Rmd/unnamed-chunk-21-2.png" target="_blank">fd8d092</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/1352d911e77a96a9c1216f1473ee5d13bc6bb3b4/docs/figure/DLPFC.Rmd/unnamed-chunk-21-2.png" target="_blank">1352d91</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/6038af1ae7d982283a26fe00a2ca6547cfe2ad13/docs/figure/DLPFC.Rmd/unnamed-chunk-21-2.png" target="_blank">6038af1</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>png 
  2 </code></pre>
<p>CLUSTER</p>
<pre class="r"><code>  g &lt;- bluster::makeSNNGraph(as.matrix(kk_UMAP), k = graph)
  g_walk &lt;- igraph::cluster_walktrap(g)
  clu &lt;- as.character(igraph::cut_at(g_walk, no = 7))
  plot(kk_UMAP,pch=20,col=cols_cluster[as.factor(clu)]) </code></pre>
<p><img src="figure/DLPFC.Rmd/unnamed-chunk-23-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-1">
Past versions of unnamed-chunk-23-1.png
</button>
</p>
<div id="fig-unnamed-chunk-23-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/7b2cb8c2c66c61a5abd5703a20bb718daeae4ac5/docs/figure/DLPFC.Rmd/unnamed-chunk-23-1.png" target="_blank">7b2cb8c</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/374d5f014b2a22cb41b1c5950c364ae56235250d/docs/figure/DLPFC.Rmd/unnamed-chunk-23-1.png" target="_blank">374d5f0</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-12-14
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/fd8d0925cad5881c332a822c5d7e5befd27ef73d/docs/figure/DLPFC.Rmd/unnamed-chunk-23-1.png" target="_blank">fd8d092</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/6038af1ae7d982283a26fe00a2ca6547cfe2ad13/docs/figure/DLPFC.Rmd/unnamed-chunk-23-1.png" target="_blank">6038af1</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/35ce733e4e0012d79f696f9c7dcdbf67e1623e80/docs/figure/DLPFC.Rmd/unnamed-chunk-23-1.png" target="_blank">35ce733</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-08-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>  plot_slide(xy,samples,clu,col=cols_cluster,size.dot = 1)</code></pre>
<p><img src="figure/DLPFC.Rmd/unnamed-chunk-23-2.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-2">
Past versions of unnamed-chunk-23-2.png
</button>
</p>
<div id="fig-unnamed-chunk-23-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/7b2cb8c2c66c61a5abd5703a20bb718daeae4ac5/docs/figure/DLPFC.Rmd/unnamed-chunk-23-2.png" target="_blank">7b2cb8c</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/374d5f014b2a22cb41b1c5950c364ae56235250d/docs/figure/DLPFC.Rmd/unnamed-chunk-23-2.png" target="_blank">374d5f0</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-12-14
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/fd8d0925cad5881c332a822c5d7e5befd27ef73d/docs/figure/DLPFC.Rmd/unnamed-chunk-23-2.png" target="_blank">fd8d092</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>  mito=colData(spe)$subsets_mito_percent
  sel_local=(labels==&quot;Layer3&quot; | labels==&quot;Layer4&quot;) &amp; 
    (samples==&quot;151669&quot; | samples==&quot;151670&quot; | samples==&quot;151671&quot; | samples==&quot;151672&quot;) &amp;
    (clu==1 | clu==6)

  
   
   
library(ggpubr)
library(ggplot2)

df=data.frame(mito=mito[sel_local],labels=as.character(clu[sel_local]))

my_comparisons=list(c(&quot;1&quot;,&quot;6&quot;))
Nplot1=ggboxplot(df, x = &quot;labels&quot;, y = &quot;mito&quot;, width = 0.8,palette = cols_cluster[c(1,6)] ,las=2,
                 fill=&quot;labels&quot;,
                 shape=21)+  
    ylab(&quot;Mitochondrial gene percentage&quot;)+ 
    xlab(&quot;&quot;)+
    
    stat_compare_means(comparisons = my_comparisons,method=&quot;wilcox.test&quot;)+
    theme(legend.position = &quot;none&quot;,plot.margin = unit(c(2,1,1,1), &quot;cm&quot;))

Nplot1</code></pre>
<p><img src="figure/DLPFC.Rmd/unnamed-chunk-23-3.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-3">
Past versions of unnamed-chunk-23-3.png
</button>
</p>
<div id="fig-unnamed-chunk-23-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/7b2cb8c2c66c61a5abd5703a20bb718daeae4ac5/docs/figure/DLPFC.Rmd/unnamed-chunk-23-3.png" target="_blank">7b2cb8c</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/374d5f014b2a22cb41b1c5950c364ae56235250d/docs/figure/DLPFC.Rmd/unnamed-chunk-23-3.png" target="_blank">374d5f0</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-12-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>png 
  2 </code></pre>
<pre><code>png 
  2 </code></pre>
<p>CLUSTER</p>
<p>We removed the spots with uncertain quality level.</p>
<pre class="r"><code>  sel_rem=which((clu %in% names(sort(table(clu)))[1:2]) |
                 rownames(kk_UMAP) %in% rownames(xy_Br5595[!selFB,]))

  kk_UMAP_clear=kk_UMAP[-sel_rem,]
  labels_clear=labels[-sel_rem]
  samples_clear=samples[-sel_rem]
  xy_clear=xy[-sel_rem,]
  data_clear=data[-sel_rem,]
  subjects_clear=subjects[-sel_rem]
  
  clu_clear=kmeans(kk_UMAP_clear,7,nstart = 100)$cluster
  plot(kk_UMAP_clear,col=cols_cluster[labels_clear],pch=20)</code></pre>
<p><img src="figure/DLPFC.Rmd/unnamed-chunk-25-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>png 
  2 </code></pre>
<p>KODAMA plot colored by cluster.</p>
<pre class="r"><code>  plot(kk_UMAP_clear,col=cols_cluster[clu_clear],pch=20)</code></pre>
<p><img src="figure/DLPFC.Rmd/unnamed-chunk-27-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-27-1">
Past versions of unnamed-chunk-27-1.png
</button>
</p>
<div id="fig-unnamed-chunk-27-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/3305d55e512f9647a5a090bf36911e444b9e361e/docs/figure/DLPFC.Rmd/unnamed-chunk-27-1.png" target="_blank">3305d55</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-12-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/7b2cb8c2c66c61a5abd5703a20bb718daeae4ac5/docs/figure/DLPFC.Rmd/unnamed-chunk-27-1.png" target="_blank">7b2cb8c</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/6038af1ae7d982283a26fe00a2ca6547cfe2ad13/docs/figure/DLPFC.Rmd/unnamed-chunk-27-1.png" target="_blank">6038af1</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/35ce733e4e0012d79f696f9c7dcdbf67e1623e80/docs/figure/DLPFC.Rmd/unnamed-chunk-27-1.png" target="_blank">35ce733</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-08-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>u=unique(samples)
for(i in 1:length(u)){
  sel=samples==u[i]
  print(adjustedRandIndex(labels_clear[sel],clu_clear[sel]))
}</code></pre>
<pre><code>[1] 0.5451047
[1] 0.4831456
[1] 0.4813363
[1] 0.4645803
[1] 0.3600907
[1] 0.3385299
[1] 0.4023525
[1] 0.4332784
[1] 0.5425434
[1] 0.5684745
[1] 0.5398805
[1] 0.5276553</code></pre>
<pre class="r"><code>ref=refine_SVM(xy_clear,clu_clear,samples_clear,cost=100)</code></pre>
<pre><code>[1] &quot;151507&quot;
[1] &quot;151508&quot;
[1] &quot;151509&quot;
[1] &quot;151510&quot;
[1] &quot;151669&quot;
[1] &quot;151670&quot;
[1] &quot;151671&quot;
[1] &quot;151672&quot;
[1] &quot;151673&quot;
[1] &quot;151674&quot;
[1] &quot;151675&quot;
[1] &quot;151676&quot;</code></pre>
<pre class="r"><code>names(ref)=rownames(data_clear)
names(labels_clear)=rownames(data_clear)

u=unique(samples)
for(i in 1:length(u)){
  sel=samples[-sel_rem]==u[i]
  print(adjustedRandIndex(labels_clear[sel],ref[sel]))
}</code></pre>
<pre><code>[1] 0.5757955
[1] 0.5134738
[1] 0.4944026
[1] 0.4885008
[1] 0.4065839
[1] 0.3925779
[1] 0.4671617
[1] 0.5502752
[1] 0.5714412
[1] 0.6139012
[1] 0.6011734
[1] 0.5864472</code></pre>
</div>
<div id="trajectory-color" class="section level1">
<h1>Trajectory color</h1>
<pre class="r"><code>d &lt;- slingshot(kk_UMAP_clear, clusterLabels = clu_clear)
trajectory=d@metadata$curves$Lineage1$s
k=Rnanoflann::nn(trajectory,kk_UMAP_clear,1)
map_color=rainbow(nrow(trajectory))[k$indices]
plot(kk_UMAP_clear,pch=20,col=map_color)
points(trajectory,pch=21,bg=&quot;white&quot;,col=2,lwd=2)</code></pre>
<p><img src="figure/DLPFC.Rmd/unnamed-chunk-29-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot_slide(xy_clear,samples_clear,k$indices,col=rainbow(nrow(trajectory)),size.dot = 1)</code></pre>
<p><img src="figure/DLPFC.Rmd/unnamed-chunk-29-2.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>png 
  2 </code></pre>
<p>The cluster number is reorder based on the trajectory</p>
<pre class="r"><code>oo=order(tapply(k$indices,ref,mean))
tra=1:7
names(tra)=oo
ref_ordered=tra[as.character(ref)]
   
plot_slide(xy_clear,samples_clear,ref_ordered,col=cols_cluster,size.dot = 1)</code></pre>
<p><img src="figure/DLPFC.Rmd/unnamed-chunk-31-1.png" width="960" style="display: block; margin: auto;" /></p>
=======
<p>KODAMA</p>
<pre class="r"><code>set.seed(seed)
kk=KODAMA.matrix.parallel(pca,
                          spatial = xy,
                          samples=samples,
                          FUN= &quot;fastpls&quot; ,
                          landmarks = 100000,
                          splitting = splitting,
                          ncomp = 50,
                          spatial.resolution = spatial.resolution,
                          n.cores=n.cores,
                          seed = seed)</code></pre>
<pre><code>Calculating Network

Calculating Network spatial
socket cluster with 40 nodes on host &#39;localhost&#39;
================================================================================
Finished parallel computation

[1] &quot;Calculation of dissimilarity matrix...&quot;
================================================================================</code></pre>
<pre class="r"><code>print(&quot;KODAMA finished&quot;)</code></pre>
<pre><code>[1] &quot;KODAMA finished&quot;</code></pre>
<pre class="r"><code>config=umap.defaults
config$n_threads = n.cores
config$n_sgd_threads = &quot;auto&quot;

kk_UMAP=KODAMA.visualization(kk,method=&quot;UMAP&quot;,config=config)
plot(kk_UMAP,pch=20,col=as.factor(labels))</code></pre>
<p><img src="figure/DLPFC.Rmd/unnamed-chunk-22-1.png" width="960" style="display: block; margin: auto;" /></p>
>>>>>>> d89baaa7bfc3bffd762b0db8bac45eeb05443b97
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-31-1">
Past versions of unnamed-chunk-31-1.png
</button>
</p>
<div id="fig-unnamed-chunk-31-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<<<<<<< HEAD
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/3305d55e512f9647a5a090bf36911e444b9e361e/docs/figure/DLPFC.Rmd/unnamed-chunk-31-1.png" target="_blank">3305d55</a>
=======
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/7b2cb8c2c66c61a5abd5703a20bb718daeae4ac5/docs/figure/DLPFC.Rmd/unnamed-chunk-22-1.png" target="_blank">7b2cb8c</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/374d5f014b2a22cb41b1c5950c364ae56235250d/docs/figure/DLPFC.Rmd/unnamed-chunk-22-1.png" target="_blank">374d5f0</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-12-14
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/fd8d0925cad5881c332a822c5d7e5befd27ef73d/docs/figure/DLPFC.Rmd/unnamed-chunk-22-1.png" target="_blank">fd8d092</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/1352d911e77a96a9c1216f1473ee5d13bc6bb3b4/docs/figure/DLPFC.Rmd/unnamed-chunk-22-1.png" target="_blank">1352d91</a>
>>>>>>> d89baaa7bfc3bffd762b0db8bac45eeb05443b97
</td>
<td>
Stefano Cacciatore
</td>
<td>
<<<<<<< HEAD
2024-12-20
=======
2024-10-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/6038af1ae7d982283a26fe00a2ca6547cfe2ad13/docs/figure/DLPFC.Rmd/unnamed-chunk-22-1.png" target="_blank">6038af1</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/35ce733e4e0012d79f696f9c7dcdbf67e1623e80/docs/figure/DLPFC.Rmd/unnamed-chunk-22-1.png" target="_blank">35ce733</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-08-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>CLUSTER</p>
<pre class="r"><code>        g &lt;- bluster::makeSNNGraph(as.matrix(kk_UMAP), k = graph)
        g_walk &lt;- igraph::cluster_walktrap(g)
        clu &lt;- as.character(igraph::cut_at(g_walk, no = 7))
        plot(kk_UMAP,pch=20,col=as.factor(clu)) </code></pre>
<p><img src="figure/DLPFC.Rmd/unnamed-chunk-24-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-24-1">
Past versions of unnamed-chunk-24-1.png
</button>
</p>
<div id="fig-unnamed-chunk-24-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/7b2cb8c2c66c61a5abd5703a20bb718daeae4ac5/docs/figure/DLPFC.Rmd/unnamed-chunk-24-1.png" target="_blank">7b2cb8c</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/374d5f014b2a22cb41b1c5950c364ae56235250d/docs/figure/DLPFC.Rmd/unnamed-chunk-24-1.png" target="_blank">374d5f0</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-12-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>    plot_slide(xy,samples,clu,col=cols_cluster)</code></pre>
<p><img src="figure/DLPFC.Rmd/unnamed-chunk-24-2.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-24-2">
Past versions of unnamed-chunk-24-2.png
</button>
</p>
<div id="fig-unnamed-chunk-24-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/7b2cb8c2c66c61a5abd5703a20bb718daeae4ac5/docs/figure/DLPFC.Rmd/unnamed-chunk-24-2.png" target="_blank">7b2cb8c</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/374d5f014b2a22cb41b1c5950c364ae56235250d/docs/figure/DLPFC.Rmd/unnamed-chunk-24-2.png" target="_blank">374d5f0</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-12-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>    mito=colData(spe)$subsets_mito_percent
    sel_local=(labels==&quot;Layer3&quot; | labels==&quot;Layer4&quot;) &amp; 
      (samples==&quot;151669&quot; | samples==&quot;151670&quot; | samples==&quot;151671&quot; | samples==&quot;151672&quot;) &amp;
      (clu==1 | clu==6)
  boxplot(mito[sel_local]~clu[sel_local],col=cols_cluster[c(1,6)] ,ylab=&quot;mito percent&quot;)</code></pre>
<p><img src="figure/DLPFC.Rmd/unnamed-chunk-24-3.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-24-3">
Past versions of unnamed-chunk-24-3.png
</button>
</p>
<div id="fig-unnamed-chunk-24-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/3305d55e512f9647a5a090bf36911e444b9e361e/docs/figure/DLPFC.Rmd/unnamed-chunk-24-3.png" target="_blank">3305d55</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-12-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/7b2cb8c2c66c61a5abd5703a20bb718daeae4ac5/docs/figure/DLPFC.Rmd/unnamed-chunk-24-3.png" target="_blank">7b2cb8c</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-12-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>   wilcox.test(mito[sel_local]~clu[sel_local])</code></pre>
<pre><code>
    Wilcoxon rank sum test with continuity correction

data:  mito[sel_local] by clu[sel_local]
W = 323194, p-value = 0.01216
alternative hypothesis: true location shift is not equal to 0</code></pre>
<p>CLUSTER</p>
<pre class="r"><code>  sel_rem=which(clu %in% names(sort(table(clu)))[1:2])

  kk_UMAP_clear=kk_UMAP[-sel_rem,]
  labels_clear=labels[-sel_rem]
  samples_clear=samples[-sel_rem]
  xy_clear=xy[-sel_rem,]
  data_clear=data[-sel_rem,]
  subjects_clear=subjects[-sel_rem]
  
  clu_clear=kmeans(kk_UMAP_clear,7,nstart = 100)$cluster
  plot(kk_UMAP_clear,col=cols_cluster[labels_clear],pch=20)</code></pre>
<p><img src="figure/DLPFC.Rmd/unnamed-chunk-26-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-26-1">
Past versions of unnamed-chunk-26-1.png
</button>
</p>
<div id="fig-unnamed-chunk-26-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/3305d55e512f9647a5a090bf36911e444b9e361e/docs/figure/DLPFC.Rmd/unnamed-chunk-26-1.png" target="_blank">3305d55</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-12-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/7b2cb8c2c66c61a5abd5703a20bb718daeae4ac5/docs/figure/DLPFC.Rmd/unnamed-chunk-26-1.png" target="_blank">7b2cb8c</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/6038af1ae7d982283a26fe00a2ca6547cfe2ad13/docs/figure/DLPFC.Rmd/unnamed-chunk-26-1.png" target="_blank">6038af1</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-10-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/35ce733e4e0012d79f696f9c7dcdbf67e1623e80/docs/figure/DLPFC.Rmd/unnamed-chunk-26-1.png" target="_blank">35ce733</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-08-03
>>>>>>> d89baaa7bfc3bffd762b0db8bac45eeb05443b97
</td>
</tr>
</tbody>
</table>
</div>
</div>
<<<<<<< HEAD
<pre><code>png 
  2 </code></pre>
<pre><code>png 
  2 </code></pre>
<pre><code>png 
  2 </code></pre>
<p>The variable to be used into the deep learning approach</p>
<pre class="r"><code>save(ref_ordered,samples_clear,subjects_clear,labels_clear,
     file=&quot;output/DLFPC-variablesXdeeplearning.RData&quot;)</code></pre>
<pre class="r"><code>library(ggalluvial)


plot_sankey_subject &lt;- function(subject_id, labels_clear, ref_ordered, subjects_clear, cols_cluster) {
  sel_sub &lt;- subjects_clear == subject_id
  al &lt;- data.frame(
    expand.grid(list(levels(labels_clear), 1:7)),
    freq = as.numeric(table(labels_clear[sel_sub], ref_ordered[sel_sub]))
  )
  al$Var2 &lt;- as.factor(al$Var2)
  
  ggplot(data = al, aes(axis1 = Var1, axis2 = Var2, y = freq)) +
    geom_alluvium(aes(fill = Var2)) +
    geom_stratum() +
    scale_fill_manual(values = cols_cluster) +
    geom_text(stat = &quot;stratum&quot;, aes(label = after_stat(stratum))) +
    theme_void()
}

Sankey_Br5595 &lt;- plot_sankey_subject(&quot;Br5595&quot;, labels_clear, ref_ordered, subjects_clear, cols_cluster)
Sankey_Br5595</code></pre>
<p><img src="figure/DLPFC.Rmd/unnamed-chunk-35-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Sankey_Br5292 &lt;- plot_sankey_subject(&quot;Br5292&quot;, labels_clear, ref_ordered, subjects_clear, cols_cluster)
Sankey_Br5292</code></pre>
<p><img src="figure/DLPFC.Rmd/unnamed-chunk-35-2.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Sankey_Br8100 &lt;- plot_sankey_subject(&quot;Br8100&quot;, labels_clear, ref_ordered, subjects_clear, cols_cluster)
Sankey_Br8100</code></pre>
<p><img src="figure/DLPFC.Rmd/unnamed-chunk-35-3.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>png 
  2 </code></pre>
<pre class="r"><code>subclusters=sort(names(sort(table(ref_ordered,labels_clear)[,&quot;Layer3&quot;],decreasing = TRUE)[1:3]))
print(subclusters)




sa_sel=(ref_ordered %in% subclusters) &amp; (samples_clear %in% c(&quot;151669&quot;,&quot;151670&quot;,&quot;151671&quot;,&quot;151672&quot;)) &amp; c(labels_clear %in% &quot;Layer3&quot;)


plot_slide(xy_clear[sa_sel,],samples_clear[sa_sel],ref_ordered[sa_sel],col=cols_cluster,size.dot = 1)</code></pre>
<p><img src="figure/DLPFC.Rmd/unnamed-chunk-37-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>s1=rownames(data_clear)[which(ref_ordered==subclusters[1] &amp; sa_sel)]
s2=rownames(data_clear)[which(ref_ordered==subclusters[2] &amp; sa_sel)]
s3=rownames(data_clear)[which(ref_ordered==subclusters[3] &amp; sa_sel)]



data_clear_N=data_clear[c(s1,s2,s3),]
data_clear_N=10*t(t(data_clear_N)/colMaxs(data_clear_N))
colnames(data_clear_N)=genes[colnames(data_clear_N)]

lab_S123 &lt;- factor(rep(c(&quot;Group1&quot;, &quot;Group2&quot;, &quot;Group3&quot;), times = c(length(s1), length(s2), length(s3))),levels=c(&quot;Group1&quot;,&quot;Group2&quot;,&quot;Group3&quot;))
da123=multi_analysis(data_clear_N[c(s1,s2,s3),],lab_S123,FUN=&quot;continuous.test&quot;,range=&quot;95%CI&quot;)


lab_S12 &lt;- factor(rep(c(&quot;Group1&quot;, &quot;Group2&quot;), times = c(length(s1), length(s2))))
lab_S13 &lt;- factor(rep(c(&quot;Group1&quot;, &quot;Group3&quot;), times = c(length(s1), length(s3))))
da12=multi_analysis(data_clear_N[c(s1,s2),],lab_S12,FUN=&quot;continuous.test&quot;,alternative = &quot;greater&quot;)
da13=multi_analysis(data_clear_N[c(s1,s3),],lab_S13,FUN=&quot;continuous.test&quot;,alternative = &quot;greater&quot;)

rank=order(-log(1+as.numeric(da12$`p-value`))-log(1+as.numeric(da13$`p-value`)),decreasing = TRUE)[1:50]

df=data.frame(da123[,1],
           sublayer1=da123[,2],
           sublayer3=da123[,3],
           sublayer5=da123[,4],
           L1vsL3_pvalue=da12[,4],
           L1vsL3_FDR=da12[,5],
           L1vsL5_pvalue=da13[,4],
           L1vsL5_FDR=da13[,5])[rank,]


write.csv(df,&quot;output/subclusters1.csv&quot;)
####

lab_S21 &lt;- factor(rep(c(&quot;Group2&quot;, &quot;Group1&quot;), times = c(length(s2), length(s1))),levels=c(&quot;Group2&quot;,&quot;Group1&quot;))
lab_S23 &lt;- factor(rep(c(&quot;Group2&quot;, &quot;Group3&quot;), times = c(length(s2), length(s3))),levels=c(&quot;Group2&quot;,&quot;Group3&quot;))
da21=multi_analysis(data_clear_N[c(s2,s1),],lab_S21,FUN=&quot;continuous.test&quot;,alternative = &quot;greater&quot;,range=&quot;95%CI&quot;)
da23=multi_analysis(data_clear_N[c(s2,s3),],lab_S23,FUN=&quot;continuous.test&quot;,alternative = &quot;greater&quot;,range=&quot;95%CI&quot;)

rank=order(-log(1+as.numeric(da21$`p-value`))-log(1+as.numeric(da23$`p-value`)),decreasing = TRUE)[1:50]

df=data.frame(da123[,1],
           sublayer1=da123[,2],
           sublayer3=da123[,3],
           sublayer5=da123[,4],
           L3vsL1_pvalue=da21[,4],
           L3vsL1_FDR=da21[,5],
           L3vsL5_pvalue=da23[,4],
           L3vsL5_FDR=da23[,5])[rank,]


write.csv(df,&quot;output/subclusters2.csv&quot;)

####


lab_S31 &lt;- factor(rep(c(&quot;Group3&quot;, &quot;Group1&quot;), times = c(length(s3), length(s1))),levels=c(&quot;Group3&quot;,&quot;Group1&quot;))
lab_S32 &lt;- factor(rep(c(&quot;Group3&quot;, &quot;Group2&quot;), times = c(length(s3), length(s2))),levels=c(&quot;Group3&quot;,&quot;Group2&quot;))
da31=multi_analysis(data_clear_N[c(s3,s1),],lab_S31,FUN=&quot;continuous.test&quot;,alternative = &quot;greater&quot;)
da32=multi_analysis(data_clear_N[c(s3,s2),],lab_S32,FUN=&quot;continuous.test&quot;,alternative = &quot;greater&quot;)

rank=order(-log(1+as.numeric(da31$`p-value`))-log(1+as.numeric(da32$`p-value`)),decreasing = TRUE)[1:50]

df=data.frame(da123[,1],
           sublayer1=da123[,2],
           sublayer3=da123[,3],
           sublayer5=da123[,4],
           L5vsL1_pvalue=da31[,4],
           L5vsL1_FDR=da31[,5],
           L5vsL3_pvalue=da32[,4],
           L5vsL3_FDR=da32[,5])[rank,]


write.csv(df,&quot;output/subclusters3.csv&quot;)</code></pre>
=======
<pre class="r"><code>  svg(&quot;output/KODAMA_DLPFC_All_original.svg&quot;)
  plot(kk_UMAP_clear,col=cols_cluster[labels_clear],pch=20)
dev.off()</code></pre>
<pre><code>png 
  2 </code></pre>
<pre class="r"><code>  plot(kk_UMAP_clear,col=cols_cluster[clu_clear],pch=20)</code></pre>
<p><img src="figure/DLPFC.Rmd/unnamed-chunk-28-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-28-1">
Past versions of unnamed-chunk-28-1.png
</button>
</p>
<div id="fig-unnamed-chunk-28-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/3305d55e512f9647a5a090bf36911e444b9e361e/docs/figure/DLPFC.Rmd/unnamed-chunk-28-1.png" target="_blank">3305d55</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-12-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>u=unique(samples)
for(i in 1:length(u)){
  sel=samples==u[i]
  print(adjustedRandIndex(labels_clear[sel],clu_clear[sel]))
}</code></pre>
<pre><code>[1] 0.5538318
[1] 0.5015062
[1] 0.4896804
[1] 0.4760243
[1] 0.3739001
[1] 0.3609056
[1] 0.4066322
[1] 0.4383128
[1] 0.548363
[1] 0.572728
[1] 0.536385
[1] 0.5275014</code></pre>
<pre class="r"><code>ref=refine_SVM(xy_clear,clu_clear,samples_clear,cost=100)</code></pre>
<pre><code>[1] &quot;151507&quot;
[1] &quot;151508&quot;
[1] &quot;151509&quot;
[1] &quot;151510&quot;
[1] &quot;151669&quot;
[1] &quot;151670&quot;
[1] &quot;151671&quot;
[1] &quot;151672&quot;
[1] &quot;151673&quot;
[1] &quot;151674&quot;
[1] &quot;151675&quot;
[1] &quot;151676&quot;</code></pre>
<pre class="r"><code>names(ref)=rownames(data_clear)
names(labels_clear)=rownames(data_clear)
save(labels_clear,ref,subjects_clear,samples_clear,file=&quot;output/DL.RData&quot;)

u=unique(samples)
for(i in 1:length(u)){
  sel=samples[-sel_rem]==u[i]
  print(adjustedRandIndex(labels_clear[sel],ref[sel]))
}</code></pre>
<pre><code>[1] 0.5874289
[1] 0.5343093
[1] 0.507712
[1] 0.5028689
[1] 0.4241738
[1] 0.426007
[1] 0.4838957
[1] 0.5635095
[1] 0.5759575
[1] 0.6173067
[1] 0.6233622
[1] 0.5917995</code></pre>
<pre class="r"><code>plot_slide(xy_clear,samples_clear,ref,col=cols_cluster)</code></pre>
<p><img src="figure/DLPFC.Rmd/unnamed-chunk-28-2.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-28-2">
Past versions of unnamed-chunk-28-2.png
</button>
</p>
<div id="fig-unnamed-chunk-28-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/3305d55e512f9647a5a090bf36911e444b9e361e/docs/figure/DLPFC.Rmd/unnamed-chunk-28-2.png" target="_blank">3305d55</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-12-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>png 
  2 </code></pre>
<pre class="r"><code>subclusters=sort(names(sort(table(ref,labels_clear)[,&quot;Layer3&quot;],decreasing = TRUE)[1:3]))

sa_sel=(ref %in% subclusters) &amp; (samples_clear %in% c(&quot;151669&quot;,&quot;151670&quot;,&quot;151671&quot;,&quot;151672&quot;)) &amp; c(labels_clear %in% &quot;Layer3&quot;)


plot_slide(xy_clear[sa_sel,],samples_clear[sa_sel],ref[sa_sel],col=cols_cluster)</code></pre>
<p><img src="figure/DLPFC.Rmd/unnamed-chunk-31-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-31-1">
Past versions of unnamed-chunk-31-1.png
</button>
</p>
<div id="fig-unnamed-chunk-31-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/3305d55e512f9647a5a090bf36911e444b9e361e/docs/figure/DLPFC.Rmd/unnamed-chunk-31-1.png" target="_blank">3305d55</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-12-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>s1=which(ref==subclusters[1] &amp; sa_sel)
s2=which(ref==subclusters[2] &amp; sa_sel)
s3=which(ref==subclusters[3] &amp; sa_sel)
lab_S12 &lt;- factor(rep(c(&quot;Group1&quot;, &quot;Group2&quot;), times = c(length(s1), length(s2))))
lab_S13 &lt;- factor(rep(c(&quot;Group1&quot;, &quot;Group3&quot;), times = c(length(s1), length(s3))))
da12=multi_analysis(data_clear[c(s1,s2),],lab_S12,FUN=&quot;continuous.test&quot;,alternative = &quot;greater&quot;)
da13=multi_analysis(data_clear[c(s1,s3),],lab_S13,FUN=&quot;continuous.test&quot;,alternative = &quot;greater&quot;)

rank=order(as.numeric(da12$`p-value`)+as.numeric(da13$`p-value`))[1:50]

data.frame(da12[,1],
           sublayer1=da12[,2],
           sublayer3=da12[,3],
           sublayer5=da13[,3],
           L1vsL3_pvalue=da12[,4],
           L1vsL3_FDR=da12[,5],
           L1vsL5_pvalue=da13[,4],
           L1vsL5_FDR=da13[,5])[rank,]</code></pre>
<pre><code>                         da12...1.           sublayer1           sublayer3
213  ENSG00000115756, median [IQR]     1.009 [0 1.588]         0 [0 0.745]
105  ENSG00000162545, median [IQR]  2.639 [2.131 3.05] 2.128 [1.665 2.502]
45   ENSG00000251562, median [IQR] 1.697 [1.072 2.425]     1.152 [0 1.713]
272  ENSG00000128989, median [IQR] 2.035 [1.433 2.521] 1.609 [1.082 2.085]
281  ENSG00000156508, median [IQR]     2.9 [2.427 3.3]  2.66 [2.252 2.992]
42   ENSG00000170323, median [IQR]         0 [0 1.085]             0 [0 0]
22   ENSG00000173432, median [IQR]         0 [0 1.286]             0 [0 0]
227  ENSG00000138814, median [IQR]     1.485 [0 2.038] 1.168 [0.603 1.616]
1704 ENSG00000162734, median [IQR]     1.294 [0 1.837]     1.014 [0 1.485]
329  ENSG00000135919, median [IQR]         0 [0 1.315]         0 [0 0.858]
13   ENSG00000171617, median [IQR] 2.601 [2.022 3.074]  2.079 [1.54 2.531]
195  ENSG00000150625, median [IQR] 2.512 [1.981 2.953] 2.299 [1.848 2.654]
540  ENSG00000144119, median [IQR]             0 [0 0]             0 [0 0]
244  ENSG00000123689, median [IQR]             0 [0 0]             0 [0 0]
77   ENSG00000164326, median [IQR]         0 [0 1.359]             0 [0 0]
775  ENSG00000174444, median [IQR] 1.864 [1.294 2.363] 1.668 [1.163 2.115]
963  ENSG00000099308, median [IQR]      1.01 [0 1.547]      0.73 [0 1.181]
537  ENSG00000198944, median [IQR]         0 [0 0.953]             0 [0 0]
468  ENSG00000250067, median [IQR]         0 [0 1.384]          0 [0 0.99]
180  ENSG00000221890, median [IQR]     1.012 [0 1.566]     0.817 [0 1.226]
255  ENSG00000197766, median [IQR]             0 [0 0]             0 [0 0]
569  ENSG00000179388, median [IQR]             0 [0 0]             0 [0 0]
383  ENSG00000088832, median [IQR]     1.101 [0 1.625]      0.812 [0 1.24]
381  ENSG00000172247, median [IQR]     1.072 [0 1.589]     0.887 [0 1.259]
288  ENSG00000214548, median [IQR]     0.952 [0 1.546]      0.75 [0 1.204]
333  ENSG00000196616, median [IQR]             0 [0 0]             0 [0 0]
1186 ENSG00000117983, median [IQR]             0 [0 0]             0 [0 0]
1072 ENSG00000120049, median [IQR]             0 [0 0]             0 [0 0]
632  ENSG00000115310, median [IQR]  2.15 [1.568 2.604] 1.813 [1.283 2.211]
344  ENSG00000166819, median [IQR]             0 [0 0]             0 [0 0]
553  ENSG00000181418, median [IQR]     0.846 [0 1.442]     0.622 [0 1.064]
134  ENSG00000111341, median [IQR]          0 [0 1.45]         0 [0 1.027]
1479 ENSG00000126583, median [IQR]         0 [0 0.942]             0 [0 0]
267  ENSG00000189241, median [IQR]  1.497 [0.83 2.045] 1.312 [0.871 1.781]
319  ENSG00000066248, median [IQR]     1.065 [0 1.576]     0.859 [0 1.301]
378  ENSG00000137825, median [IQR]          0 [0 1.07]          0 [0 0.65]
65   ENSG00000171476, median [IQR] 1.547 [0.903 2.118] 1.225 [0.847 1.726]
853  ENSG00000157005, median [IQR]             0 [0 0]             0 [0 0]
1246 ENSG00000135447, median [IQR]         0 [0 0.818]             0 [0 0]
591  ENSG00000125869, median [IQR]             0 [0 0]             0 [0 0]
353  ENSG00000059915, median [IQR]     1.037 [0 1.536]     0.824 [0 1.231]
1152 ENSG00000125534, median [IQR]     1.248 [0 1.757]     1.064 [0 1.558]
1995 ENSG00000163682, median [IQR] 1.904 [1.351 2.375] 1.789 [1.266 2.202]
1095 ENSG00000141404, median [IQR]             0 [0 0]             0 [0 0]
161  ENSG00000127585, median [IQR] 1.546 [0.994 2.095] 1.335 [0.907 1.815]
1452 ENSG00000165443, median [IQR]         0 [0 1.161]          0 [0 0.92]
465  ENSG00000120251, median [IQR]         0 [0 1.375]         0 [0 1.034]
220  ENSG00000105290, median [IQR]     1.463 [0 2.008] 1.309 [0.865 1.772]
852  ENSG00000145920, median [IQR]     1.029 [0 1.558]     0.864 [0 1.284]
74   ENSG00000198668, median [IQR]  3.163 [2.72 3.523] 3.025 [2.669 3.332]
               sublayer5 L1vsL3_pvalue L1vsL3_FDR L1vsL5_pvalue L1vsL5_FDR
213           0 [0 0.98]      1.23e-88   1.23e-85      8.88e-87   1.78e-83
105  2.346 [1.852 2.745]      2.13e-89   4.26e-86      4.13e-57   4.13e-54
45   1.332 [0.738 1.924]      1.56e-57   7.81e-55      2.61e-45   1.74e-42
272    1.8 [1.244 2.256]      8.12e-45   2.71e-42       5.1e-27   2.55e-24
281  2.701 [2.261 3.078]      2.69e-27   5.38e-25      8.51e-27   3.40e-24
42               0 [0 0]      4.78e-33   1.06e-30      1.32e-23   4.41e-21
22           0 [0 0.745]      9.57e-43   2.73e-40       3.8e-22   1.09e-19
227      1.293 [0 1.787]      2.12e-22   3.53e-20      6.23e-16   1.13e-13
1704     1.082 [0 1.546]      6.11e-15   6.79e-13      6.32e-19   1.58e-16
329          0 [0 1.014]      8.43e-23   1.53e-20      1.15e-14   1.92e-12
13   2.444 [1.924 2.866]      4.52e-69   3.01e-66       4.5e-14   6.92e-12
195  2.372 [1.913 2.752]      6.63e-19   9.47e-17      9.47e-13   1.26e-10
540              0 [0 0]      9.15e-12   7.32e-10      1.49e-16   2.99e-14
244              0 [0 0]      8.42e-13   7.32e-11      1.95e-11   2.26e-09
77           0 [0 1.082]      7.48e-56   2.99e-53      6.62e-11   6.97e-09
775  1.685 [1.161 2.139]      8.35e-11   5.96e-09      1.61e-17   3.57e-15
963      0.831 [0 1.332]      6.46e-13   5.87e-11      2.68e-10   2.55e-08
537          0 [0 0.347]      2.63e-17   3.51e-15       4.7e-10   4.27e-08
468           0 [0 1.14]      2.96e-15   3.48e-13      9.65e-10   8.39e-08
180      0.802 [0 1.325]       1.1e-09   6.69e-08      4.46e-12   5.58e-10
255              0 [0 0]      1.48e-09   8.46e-08      1.12e-10   1.12e-08
569              0 [0 0]      2.51e-10   1.62e-08      1.48e-09   1.18e-07
383      0.928 [0 1.412]      4.61e-16   5.76e-14      5.66e-09   4.19e-07
381       0.897 [0 1.34]       6.7e-09   3.27e-07      2.04e-11   2.26e-09
288      0.799 [0 1.313]      4.94e-10   3.08e-08      8.04e-09   5.75e-07
333              0 [0 0]      3.92e-09   2.01e-07      9.89e-09   6.82e-07
1186             0 [0 0]       3.9e-09   2.01e-07      1.44e-08   9.57e-07
1072             0 [0 0]       2.4e-08   1.07e-06       6.7e-14   9.57e-12
632  2.024 [1.439 2.486]       2.8e-34   7.01e-32      2.79e-08   1.75e-06
344              0 [0 0]       3.5e-08   1.46e-06      1.05e-09   8.78e-08
553      0.687 [0 1.249]      6.87e-12   5.72e-10       1.6e-07   9.42e-06
134          0 [0 1.178]      1.23e-07   4.82e-06      4.53e-08   2.75e-06
1479         0 [0 0.616]      6.51e-11   4.82e-09      3.37e-07   1.76e-05
267   1.375 [0.79 1.884]      4.55e-09   2.28e-07      3.44e-07   1.76e-05
319      0.924 [0 1.374]      2.77e-08   1.18e-06      1.05e-06   4.90e-05
378          0 [0 0.833]      4.18e-11   3.22e-09      1.13e-06   4.98e-05
65   1.435 [0.901 1.953]      1.85e-19   2.85e-17      1.17e-06   4.98e-05
853              0 [0 0]      1.34e-06   4.95e-05      2.59e-09   1.99e-07
1246             0 [0 0]      1.36e-06   4.95e-05      1.97e-07   1.06e-05
591              0 [0 0]      3.94e-06   1.29e-04      1.05e-06   4.90e-05
353      0.894 [0 1.371]      7.68e-09   3.57e-07      8.97e-06   3.26e-04
1152     1.109 [0 1.594]      9.12e-06   2.76e-04      4.25e-07   2.12e-05
1995 1.795 [1.306 2.252]      8.84e-06   2.72e-04      6.03e-06   2.28e-04
1095             0 [0 0]      5.77e-06   1.83e-04      1.29e-05   4.52e-04
161   1.454 [0.91 1.977]      2.17e-10   1.45e-08      2.17e-05   7.36e-04
1452         0 [0 0.973]      2.17e-05   6.21e-04      4.07e-06   1.66e-04
465          0 [0 1.157]      3.36e-09   1.82e-07      2.83e-05   9.29e-04
220  1.357 [0.802 1.851]      2.51e-05   7.06e-04      6.55e-06   2.42e-04
852      0.888 [0 1.376]      2.94e-05   8.16e-04      6.01e-06   2.28e-04
74   3.093 [2.716 3.423]       9.2e-14   9.20e-12      3.77e-05   1.22e-03</code></pre>
<pre class="r"><code>a1=apply(data_clear,2,function(x) wilcox.test(x[ref==subclusters[2] &amp; sa_sel],x[(ref==subclusters[3]) &amp; sa_sel],alternative = &quot;greater&quot;)$p.value)
  a2=apply(data_clear,2,function(x) wilcox.test(x[ref==subclusters[2] &amp; sa_sel],x[(ref==subclusters[1] ) &amp; sa_sel],alternative = &quot;greater&quot;)$p.value)
  
aa=a1+a2

box_sel=(samples_clear==&quot;151669&quot; | samples_clear==&quot;151670&quot; | samples_clear==&quot;151671&quot; | samples_clear==&quot;151672&quot;) &amp; (ref==subclusters[1] | ref==subclusters[2] | ref==subclusters[3])
tapply(data_clear[box_sel,names(sort(aa)[1])],as.numeric(as.vector(ref[box_sel])),mean)</code></pre>
<pre><code>        1         3         5 
0.1664618 0.5226190 0.3352899 </code></pre>
<pre class="r"><code>df=data.frame(variable=data_clear[box_sel,names(sort(aa)[1])],labels=as.numeric(as.vector(ref[box_sel])))

my_comparisons=list()
my_comparisons[[1]]=c(1,3)
my_comparisons[[2]]=c(1,2)

Nplot1=ggboxplot(df, x = &quot;labels&quot;, y = &quot;variable&quot;,fill=&quot;labels&quot;,
                 width = 0.8,
                 palette=cols_cluster)+  
  ylab(names(sort(aa)[1]))+ xlab(&quot;&quot;)+
  stat_compare_means(comparisons = my_comparisons,method=&quot;wilcox.test&quot;)

Nplot1</code></pre>
<p><img src="figure/DLPFC.Rmd/unnamed-chunk-31-2.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-31-2">
Past versions of unnamed-chunk-31-2.png
</button>
</p>
<div id="fig-unnamed-chunk-31-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/3305d55e512f9647a5a090bf36911e444b9e361e/docs/figure/DLPFC.Rmd/unnamed-chunk-31-2.png" target="_blank">3305d55</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-12-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>sort(aa)[1:10]</code></pre>
<pre><code>ENSG00000100285 ENSG00000197971 ENSG00000171345 ENSG00000123416 ENSG00000105711 
   5.243017e-23    3.316729e-21    2.544932e-18    4.923129e-15    2.818593e-14 
ENSG00000159763 ENSG00000153002 ENSG00000151552 ENSG00000164919 ENSG00000246705 
   3.448144e-13    2.162544e-11    1.290837e-10    2.811478e-10    5.038213e-10 </code></pre>
<pre class="r"><code>a_choice=names(sort(aa)[1])
tapply(data_clear[box_sel,a_choice],as.numeric(as.vector(ref[box_sel])),mean)</code></pre>
<pre><code>        1         3         5 
0.1664618 0.5226190 0.3352899 </code></pre>
<pre class="r"><code>df=data.frame(variable=data_clear[box_sel,a_choice],labels=as.numeric(as.vector(ref[box_sel])))

my_comparisons=list()
my_comparisons[[1]]=c(2,3)
my_comparisons[[2]]=c(1,2)

Nplot2=ggboxplot(df, x = &quot;labels&quot;, y = &quot;variable&quot;,fill=&quot;labels&quot;,
                 width = 0.8,
                 palette=cols_cluster)+  
  ylab(a_choice)+ xlab(&quot;&quot;)+
  stat_compare_means(comparisons = my_comparisons,method=&quot;wilcox.test&quot;)

Nplot2</code></pre>
<p><img src="figure/DLPFC.Rmd/unnamed-chunk-31-3.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-31-3">
Past versions of unnamed-chunk-31-3.png
</button>
</p>
<div id="fig-unnamed-chunk-31-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/3305d55e512f9647a5a090bf36911e444b9e361e/docs/figure/DLPFC.Rmd/unnamed-chunk-31-3.png" target="_blank">3305d55</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-12-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>aa=apply(data_clear,2,function(x) wilcox.test(x[ref==subclusters[3] &amp; sa_sel],x[(ref==subclusters[1] ) &amp; sa_sel],alternative = &quot;greater&quot;)$p.value) +
  apply(data_clear,2,function(x) wilcox.test(x[ref==subclusters[3] &amp; sa_sel],x[(ref==subclusters[2]) &amp; sa_sel],alternative = &quot;greater&quot;)$p.value)
sort(aa)[1:10]</code></pre>
<pre><code>ENSG00000128245 ENSG00000143933 ENSG00000163032 ENSG00000117152 ENSG00000144959 
   1.282461e-05    8.868432e-04    1.928869e-03    5.248686e-03    7.369777e-03 
ENSG00000162706 ENSG00000084234 ENSG00000171858 ENSG00000182492 ENSG00000187094 
   1.419858e-02    1.496842e-02    1.766745e-02    1.807298e-02    1.933249e-02 </code></pre>
<pre class="r"><code>tapply(data_clear[box_sel,names(sort(aa)[1])],as.numeric(as.vector(ref[box_sel])),mean)</code></pre>
<pre><code>       1        3        5 
2.533871 2.515540 2.641904 </code></pre>
<pre class="r"><code>df=data.frame(variable=data_clear[box_sel,names(sort(aa)[1])],labels=as.numeric(as.vector(ref[box_sel])))

my_comparisons=list()
my_comparisons[[1]]=c(2,3)
my_comparisons[[2]]=c(1,3)

Nplot3=ggboxplot(df, x = &quot;labels&quot;, y = &quot;variable&quot;,fill=&quot;labels&quot;,
                 width = 0.8,
                 palette=cols_cluster)+  
  ylab(names(sort(aa)[1]))+ xlab(&quot;&quot;)+
  stat_compare_means(comparisons = my_comparisons,method=&quot;wilcox.test&quot;)

Nplot3</code></pre>
<p><img src="figure/DLPFC.Rmd/unnamed-chunk-31-4.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-31-4">
Past versions of unnamed-chunk-31-4.png
</button>
</p>
<div id="fig-unnamed-chunk-31-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/3305d55e512f9647a5a090bf36911e444b9e361e/docs/figure/DLPFC.Rmd/unnamed-chunk-31-4.png" target="_blank">3305d55</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-12-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>library(ggalluvial)

sel_sub=subjects_clear==&quot;Br5292&quot;
al=data.frame(expand.grid(list(levels(labels_clear),1:7)),freq=as.numeric(table(labels_clear[sel_sub],ref[sel_sub])))
al$Var2=as.factor(al$Var2)


ggplot(data = al,
       aes(axis1 = Var1, axis2 = Var2, y = freq)) +
  geom_alluvium(aes(fill = Var2)) +
  geom_stratum() +
  geom_text(stat = &quot;stratum&quot;,
            aes(label = after_stat(stratum))) +
  theme_void()</code></pre>
<p><img src="figure/DLPFC.Rmd/unnamed-chunk-31-5.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-31-5">
Past versions of unnamed-chunk-31-5.png
</button>
</p>
<div id="fig-unnamed-chunk-31-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/3305d55e512f9647a5a090bf36911e444b9e361e/docs/figure/DLPFC.Rmd/unnamed-chunk-31-5.png" target="_blank">3305d55</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-12-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>sel_sub=subjects_clear==&quot;Br5595&quot;
al=data.frame(expand.grid(list(levels(labels_clear),1:7)),freq=as.numeric(table(labels_clear[sel_sub],ref[sel_sub])))
al$Var2=as.factor(al$Var2)


ggplot(data = al,
       aes(axis1 = Var1, axis2 = Var2, y = freq)) +
  geom_alluvium(aes(fill = Var2)) +
  geom_stratum() +
  geom_text(stat = &quot;stratum&quot;,
            aes(label = after_stat(stratum))) +
  theme_void()</code></pre>
<p><img src="figure/DLPFC.Rmd/unnamed-chunk-31-6.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-31-6">
Past versions of unnamed-chunk-31-6.png
</button>
</p>
<div id="fig-unnamed-chunk-31-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/3305d55e512f9647a5a090bf36911e444b9e361e/docs/figure/DLPFC.Rmd/unnamed-chunk-31-6.png" target="_blank">3305d55</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-12-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>sel_sub=subjects_clear==&quot;Br8100&quot;
al=data.frame(expand.grid(list(levels(labels_clear),1:7)),freq=as.numeric(table(labels_clear[sel_sub],ref[sel_sub])))
al$Var2=as.factor(al$Var2)


ggplot(data = al,
       aes(axis1 = Var1, axis2 = Var2, y = freq)) +
  geom_alluvium(aes(fill = Var2)) +
  geom_stratum() +
  geom_text(stat = &quot;stratum&quot;,
            aes(label = after_stat(stratum))) +
  theme_void()</code></pre>
<p><img src="figure/DLPFC.Rmd/unnamed-chunk-31-7.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-31-7">
Past versions of unnamed-chunk-31-7.png
</button>
</p>
<div id="fig-unnamed-chunk-31-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/3305d55e512f9647a5a090bf36911e444b9e361e/docs/figure/DLPFC.Rmd/unnamed-chunk-31-7.png" target="_blank">3305d55</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-12-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code> library(&quot;GSVA&quot;)
 library(&quot;GSA&quot;)
 library(&quot;VAM&quot;)
 geneset=GSA.read.gmt(&quot;../Genesets/msigdb_v2023.2.Hs_GMTs/h.all.v2023.2.Hs.symbols.gmt&quot;)</code></pre>
<pre><code>12345678910111213141516171819202122232425262728293031323334353637383940414243444546474849501
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49</code></pre>
<pre class="r"><code> names(geneset$genesets)=geneset$geneset.names
 genesets=geneset$genesets

countdata &lt;- as.matrix(t(logcounts(spe)))

# library(&quot;gprofiler2&quot;)
# genes=gconvert(rownames(spe),organism=&quot;hsapiens&quot;,target=&quot;GENECARDS&quot;,filter_na = F)$target

spot_name=colnames(spe)
colnames(countdata)=genes



li=lapply(genesets,function(x) which(genes %in% x))

VAM=vamForCollection(gene.expr=countdata, gene.set.collection=li)
pathway=VAM$distance.sq
  pathway_clear=pathway[-sel_rem,]





aa=apply(pathway_clear,2,function(x) wilcox.test(x[ref==subclusters[1] &amp; sa_sel],x[(ref==subclusters[2] ) &amp; sa_sel],alternative = &quot;greater&quot;)$p.value) +
  apply(pathway_clear,2,function(x) wilcox.test(x[ref==subclusters[1] &amp; sa_sel],x[(ref==subclusters[3]) &amp; sa_sel],alternative = &quot;greater&quot;)$p.value)
sort(aa)[1:10]</code></pre>
<pre><code>           HALLMARK_MYC_TARGETS_V1           HALLMARK_MITOTIC_SPINDLE 
                      4.774718e-42                       8.750945e-31 
  HALLMARK_TNFA_SIGNALING_VIA_NFKB            HALLMARK_G2M_CHECKPOINT 
                      3.301382e-30                       6.218125e-29 
              HALLMARK_P53_PATHWAY            HALLMARK_UV_RESPONSE_DN 
                      9.492812e-28                       2.873370e-26 
          HALLMARK_APICAL_JUNCTION       HALLMARK_ALLOGRAFT_REJECTION 
                      2.810605e-24                       5.699346e-24 
HALLMARK_UNFOLDED_PROTEIN_RESPONSE   HALLMARK_ESTROGEN_RESPONSE_EARLY 
                      5.519530e-23                       2.293869e-22 </code></pre>
<pre class="r"><code>box_sel=(samples_clear==&quot;151669&quot; | samples_clear==&quot;151670&quot; | samples_clear==&quot;151671&quot; | samples_clear==&quot;151672&quot;) &amp; (ref==subclusters[1] | ref==subclusters[2] | ref==subclusters[3])
tapply(pathway_clear[box_sel,names(sort(aa)[1])],as.numeric(as.vector(ref[box_sel])),mean)</code></pre>
<pre><code>       1        3        5 
284.1904 261.3498 267.3446 </code></pre>
<pre class="r"><code>df=data.frame(variable=log(1+pathway_clear[box_sel,names(sort(aa)[1])]),labels=as.numeric(as.vector(ref[box_sel])))

my_comparisons=list()
my_comparisons[[1]]=c(1,3)
my_comparisons[[2]]=c(1,2)

Nplot1=ggboxplot(df, x = &quot;labels&quot;, y = &quot;variable&quot;,fill=&quot;labels&quot;,
                 width = 0.8,
                 palette=cols_cluster)+  
  ylab(names(sort(aa)[1]))+ xlab(&quot;&quot;)+
  stat_compare_means(comparisons = my_comparisons,method=&quot;wilcox.test&quot;)

Nplot1</code></pre>
<p><img src="figure/DLPFC.Rmd/unnamed-chunk-31-8.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-31-8">
Past versions of unnamed-chunk-31-8.png
</button>
</p>
<div id="fig-unnamed-chunk-31-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/3305d55e512f9647a5a090bf36911e444b9e361e/docs/figure/DLPFC.Rmd/unnamed-chunk-31-8.png" target="_blank">3305d55</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-12-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>a1=apply(pathway_clear,2,function(x) wilcox.test(x[ref==subclusters[2] &amp; sa_sel],x[(ref==subclusters[3]) &amp; sa_sel],alternative = &quot;greater&quot;)$p.value)
  a2=apply(pathway_clear,2,function(x) wilcox.test(x[ref==subclusters[2] &amp; sa_sel],x[(ref==subclusters[1] ) &amp; sa_sel],alternative = &quot;greater&quot;)$p.value)
  
aa=a1+a2
  
sort(aa)[1:10]</code></pre>
<pre><code>      HALLMARK_INTERFERON_ALPHA_RESPONSE 
                               0.1483417 
HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY 
                               0.2129212 
      HALLMARK_OXIDATIVE_PHOSPHORYLATION 
                               0.3477291 
        HALLMARK_CHOLESTEROL_HOMEOSTASIS 
                               0.4482731 
                   HALLMARK_ADIPOGENESIS 
                               0.9777236 
            HALLMARK_PANCREAS_BETA_CELLS 
                               1.2410666 
                 HALLMARK_MYC_TARGETS_V2 
                               1.2924749 
      HALLMARK_INTERFERON_GAMMA_RESPONSE 
                               1.4447239 
                     HALLMARK_GLYCOLYSIS 
                               1.5469754 
                 HALLMARK_APICAL_SURFACE 
                               1.6554406 </code></pre>
<pre class="r"><code>a_choice=names(sort(aa)[1])
tapply(pathway_clear[box_sel,a_choice],as.numeric(as.vector(ref[box_sel])),mean)</code></pre>
<pre><code>       1        3        5 
30.91670 29.33472 29.43140 </code></pre>
<pre class="r"><code>df=data.frame(variable=pathway_clear[box_sel,a_choice],labels=as.numeric(as.vector(ref[box_sel])))

my_comparisons=list()
my_comparisons[[1]]=c(2,3)
my_comparisons[[2]]=c(1,2)

Nplot2=ggboxplot(df, x = &quot;labels&quot;, y = &quot;variable&quot;,fill=&quot;labels&quot;,
                 width = 0.8,
                 palette=cols_cluster)+  
  ylab(a_choice)+ xlab(&quot;&quot;)+
  stat_compare_means(comparisons = my_comparisons,method=&quot;wilcox.test&quot;)

Nplot2</code></pre>
<p><img src="figure/DLPFC.Rmd/unnamed-chunk-31-9.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-31-9">
Past versions of unnamed-chunk-31-9.png
</button>
</p>
<div id="fig-unnamed-chunk-31-9" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/3305d55e512f9647a5a090bf36911e444b9e361e/docs/figure/DLPFC.Rmd/unnamed-chunk-31-9.png" target="_blank">3305d55</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-12-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>aa=apply(pathway_clear,2,function(x) wilcox.test(x[ref==subclusters[3] &amp; sa_sel],x[(ref==subclusters[1] ) &amp; sa_sel],alternative = &quot;greater&quot;)$p.value) +
  apply(pathway_clear,2,function(x) wilcox.test(x[ref==subclusters[3] &amp; sa_sel],x[(ref==subclusters[2]) &amp; sa_sel],alternative = &quot;greater&quot;)$p.value)
sort(aa)[1:10]</code></pre>
<pre><code>                   HALLMARK_ADIPOGENESIS 
                               0.5028839 
      HALLMARK_OXIDATIVE_PHOSPHORYLATION 
                               0.6522859 
HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY 
                               0.7874708 
          HALLMARK_FATTY_ACID_METABOLISM 
                               0.9371399 
                 HALLMARK_APICAL_SURFACE 
                               0.9807717 
             HALLMARK_HEDGEHOG_SIGNALING 
                               0.9839734 
                   HALLMARK_ANGIOGENESIS 
                               0.9859129 
                     HALLMARK_COMPLEMENT 
                               1.0000000 
                HALLMARK_APICAL_JUNCTION 
                               1.0000000 
                HALLMARK_MITOTIC_SPINDLE 
                               1.0000000 </code></pre>
<pre class="r"><code>tapply(pathway_clear[box_sel,names(sort(aa)[1])],as.numeric(as.vector(ref[box_sel])),mean)</code></pre>
<pre><code>       1        3        5 
229.6962 225.1243 229.4126 </code></pre>
<pre class="r"><code>df=data.frame(variable=pathway_clear[box_sel,names(sort(aa)[1])],labels=as.numeric(as.vector(ref[box_sel])))

my_comparisons=list()
my_comparisons[[1]]=c(2,3)
my_comparisons[[2]]=c(1,3)

Nplot3=ggboxplot(df, x = &quot;labels&quot;, y = &quot;variable&quot;,fill=&quot;labels&quot;,
                 width = 0.8,
                 palette=cols_cluster)+  
  ylab(names(sort(aa)[1]))+ xlab(&quot;&quot;)+
  stat_compare_means(comparisons = my_comparisons,method=&quot;wilcox.test&quot;)

Nplot3</code></pre>
<p><img src="figure/DLPFC.Rmd/unnamed-chunk-31-10.png" width="960" style="display: block; margin: auto;" /></p>
<p>TRAJECTORY</p>
<pre class="r"><code>d &lt;- slingshot(kk_UMAP_clear, clusterLabels = clu_clear)
trajectory=d@metadata$curves$Lineage1$s
k=Rnanoflann::nn(trajectory,kk_UMAP_clear,1)
map_color=rainbow(nrow(trajectory))[k$indices]
plot(kk_UMAP_clear,pch=20,col=map_color)
points(trajectory,pch=21,bg=&quot;white&quot;,col=2,lwd=2)</code></pre>
<p><img src="figure/DLPFC.Rmd/unnamed-chunk-32-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-32-1">
Past versions of unnamed-chunk-32-1.png
</button>
</p>
<div id="fig-unnamed-chunk-32-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/3305d55e512f9647a5a090bf36911e444b9e361e/docs/figure/DLPFC.Rmd/unnamed-chunk-32-1.png" target="_blank">3305d55</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-12-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot_slide(xy_clear,samples_clear,k$indices,col=rainbow(nrow(trajectory)))</code></pre>
<p><img src="figure/DLPFC.Rmd/unnamed-chunk-32-2.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-32-2">
Past versions of unnamed-chunk-32-2.png
</button>
</p>
<div id="fig-unnamed-chunk-32-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/3305d55e512f9647a5a090bf36911e444b9e361e/docs/figure/DLPFC.Rmd/unnamed-chunk-32-2.png" target="_blank">3305d55</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-12-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>png 
  2 </code></pre>
>>>>>>> d89baaa7bfc3bffd762b0db8bac45eeb05443b97
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<<<<<<< HEAD
<pre><code>R version 4.4.3 (2025-02-28)
=======
<pre><code>R version 4.4.2 (2024-10-31)
>>>>>>> d89baaa7bfc3bffd762b0db8bac45eeb05443b97
Platform: x86_64-pc-linux-gnu
Running under: Ubuntu 20.04.6 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0 
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

time zone: Etc/UTC
tzcode source: system (glibc)

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
<<<<<<< HEAD
 [1] patchwork_1.3.0             ggalluvial_0.12.5          
 [3] ggpubr_0.6.0                Rnanoflann_0.0.3           
 [5] irlba_2.3.5.1               slingshot_2.12.0           
 [7] TrajectoryUtils_1.12.0      princurve_2.1.6            
 [9] mclust_6.1.1                KODAMAextra_1.2            
[11] e1071_1.7-16                doParallel_1.0.17          
[13] iterators_1.0.14            foreach_1.5.2              
[15] KODAMA_3.0                  Matrix_1.7-3               
[17] umap_0.2.10.0               Rtsne_0.17                 
[19] minerva_1.5.10              spatialLIBD_1.16.2         
[21] SpatialExperiment_1.14.0    Seurat_5.2.1               
[23] SeuratObject_5.0.2          sp_2.2-0                   
[25] harmony_1.2.3               Rcpp_1.0.14                
[27] SPARK_1.1.1                 scry_1.16.0                
[29] scran_1.32.0                scater_1.32.1              
[31] ggplot2_3.5.1               scuttle_1.14.0             
[33] SingleCellExperiment_1.26.0 SummarizedExperiment_1.34.0
[35] Biobase_2.64.0              GenomicRanges_1.56.2       
[37] GenomeInfoDb_1.40.1         IRanges_2.38.1             
[39] S4Vectors_0.42.1            BiocGenerics_0.50.0        
[41] MatrixGenerics_1.16.0       matrixStats_1.5.0          
[43] nnSVG_1.8.0                 workflowr_1.7.1            

loaded via a namespace (and not attached):
  [1] goftest_1.2-3             DT_0.33                  
  [3] Biostrings_2.72.1         vctrs_0.6.5              
  [5] spatstat.random_3.3-3     digest_0.6.37            
  [7] png_0.1-8                 proxy_0.4-27             
  [9] git2r_0.33.0              ggrepel_0.9.6            
 [11] deldir_2.0-4              parallelly_1.43.0        
 [13] magick_2.8.6              MASS_7.3-65              
 [15] reshape2_1.4.4            httpuv_1.6.15            
 [17] withr_3.0.2               xfun_0.51                
 [19] survival_3.8-3            memoise_2.0.1            
 [21] benchmarkme_1.0.8         ggbeeswarm_0.7.2         
 [23] zoo_1.8-13                pbapply_1.7-2            
 [25] Formula_1.2-5             rematch2_2.1.2           
 [27] KEGGREST_1.44.1           promises_1.3.2           
 [29] httr_1.4.7                rstatix_0.7.2            
 [31] restfulr_0.0.15           globals_0.16.3           
 [33] fitdistrplus_1.2-2        ps_1.9.0                 
 [35] rstudioapi_0.17.1         UCSC.utils_1.0.0         
 [37] miniUI_0.1.1.1            generics_0.1.3           
 [39] processx_3.8.6            curl_6.2.2               
 [41] fields_16.3.1             zlibbioc_1.50.0          
 [43] ScaledMatrix_1.12.0       polyclip_1.10-7          
 [45] doSNOW_1.0.20             GenomeInfoDbData_1.2.12  
 [47] ExperimentHub_2.12.0      SparseArray_1.4.8        
 [49] golem_0.5.1               xtable_1.8-4             
 [51] stringr_1.5.1             pracma_2.4.4             
 [53] evaluate_1.0.3            S4Arrays_1.4.1           
 [55] BiocFileCache_2.12.0      colorspace_2.1-1         
 [57] filelock_1.0.3            ROCR_1.0-11              
 [59] reticulate_1.42.0         spatstat.data_3.1-6      
 [61] shinyWidgets_0.9.0        magrittr_2.0.3           
 [63] lmtest_0.9-40             later_1.4.1              
 [65] viridis_0.6.5             lattice_0.22-7           
 [67] misc3d_0.9-1              spatstat.geom_3.3-6      
 [69] future.apply_1.11.3       getPass_0.2-4            
 [71] scattermore_1.2           XML_3.99-0.18            
 [73] cowplot_1.1.3             RcppAnnoy_0.0.22         
 [75] class_7.3-23              pillar_1.10.1            
 [77] nlme_3.1-168              compiler_4.4.3           
 [79] beachmat_2.20.0           RSpectra_0.16-2          
 [81] stringi_1.8.7             tensor_1.5               
 [83] GenomicAlignments_1.40.0  plyr_1.8.9               
 [85] crayon_1.5.3              abind_1.4-8              
 [87] BiocIO_1.14.0             locfit_1.5-9.12          
 [89] bit_4.6.0                 dplyr_1.1.4              
 [91] whisker_0.4.1             codetools_0.2-20         
 [93] BiocSingular_1.20.0       openssl_2.3.2            
 [95] bslib_0.9.0               paletteer_1.6.0          
 [97] plotly_4.10.4             mime_0.13                
 [99] splines_4.4.3             fastDummies_1.7.5        
[101] dbplyr_2.5.0              sparseMatrixStats_1.16.0 
[103] attempt_0.3.1             knitr_1.50               
[105] blob_1.2.4                BiocVersion_3.19.1       
[107] fs_1.6.5                  listenv_0.9.1            
[109] DelayedMatrixStats_1.26.0 rdist_0.0.5              
[111] ggsignif_0.6.4            tibble_3.2.1             
[113] callr_3.7.6               statmod_1.5.0            
[115] pkgconfig_2.0.3           tools_4.4.3              
[117] BRISC_1.0.6               cachem_1.1.0             
[119] RhpcBLASctl_0.23-42       RSQLite_2.3.9            
[121] viridisLite_0.4.2         DBI_1.2.3                
[123] fastmap_1.2.0             rmarkdown_2.29           
[125] scales_1.3.0              grid_4.4.3               
[127] ica_1.0-3                 Rsamtools_2.20.0         
[129] broom_1.0.8               AnnotationHub_3.12.0     
[131] sass_0.4.9                BiocManager_1.30.25      
[133] dotCall64_1.2             carData_3.0-5            
[135] RANN_2.6.2                snow_0.4-4               
[137] farver_2.1.2              yaml_2.3.10              
[139] rtracklayer_1.64.0        cli_3.6.4                
[141] purrr_1.0.4               lifecycle_1.0.4          
[143] askpass_1.2.1             uwot_0.2.3               
[145] backports_1.5.0           bluster_1.14.0           
[147] sessioninfo_1.2.3         BiocParallel_1.38.0      
[149] gtable_0.3.6              rjson_0.2.23             
[151] ggridges_0.5.6            progressr_0.15.1         
[153] limma_3.60.6              jsonlite_2.0.0           
[155] edgeR_4.2.2               RcppHNSW_0.6.0           
[157] bitops_1.0-9              benchmarkmeData_1.0.4    
[159] bit64_4.6.0-1             spatstat.utils_3.1-3     
[161] BiocNeighbors_1.22.0      matlab_1.0.4.1           
[163] jquerylib_0.1.4           metapod_1.12.0           
[165] config_0.3.2              dqrng_0.4.1              
[167] spatstat.univar_3.1-2     lazyeval_0.2.2           
[169] shiny_1.10.0              htmltools_0.5.8.1        
[171] sctransform_0.4.1         rappdirs_0.3.3           
[173] glue_1.8.0                tcltk_4.4.3              
[175] spam_2.11-1               XVector_0.44.0           
[177] RCurl_1.98-1.17           rprojroot_2.0.4          
[179] gridExtra_2.3             igraph_2.1.4             
[181] R6_2.6.1                  tidyr_1.3.1              
[183] labeling_0.4.3            CompQuadForm_1.4.3       
[185] cluster_2.1.8.1           DelayedArray_0.30.1      
[187] tidyselect_1.2.1          vipor_0.4.7              
[189] maps_3.4.2.1              car_3.1-3                
[191] AnnotationDbi_1.66.0      future_1.34.0            
[193] rsvd_1.0.5                munsell_0.5.1            
[195] KernSmooth_2.23-26        data.table_1.17.0        
[197] htmlwidgets_1.6.4         RColorBrewer_1.1-3       
[199] rlang_1.1.5               spatstat.sparse_3.1-0    
[201] spatstat.explore_3.4-2    beeswarm_0.4.0           </code></pre>
=======
 [1] VAM_1.1.0                   MASS_7.3-61                
 [3] GSA_1.03.3                  GSVA_1.52.3                
 [5] ggalluvial_0.12.5           ggpubr_0.6.0               
 [7] Rnanoflann_0.0.3            irlba_2.3.5.1              
 [9] slingshot_2.12.0            TrajectoryUtils_1.12.0     
[11] princurve_2.1.6             mclust_6.1.1               
[13] KODAMAextra_1.2             e1071_1.7-16               
[15] doParallel_1.0.17           iterators_1.0.14           
[17] foreach_1.5.2               KODAMA_3.0                 
[19] Matrix_1.7-1                umap_0.2.10.0              
[21] Rtsne_0.17                  minerva_1.5.10             
[23] spatialLIBD_1.16.2          SpatialExperiment_1.14.0   
[25] Seurat_5.1.0                SeuratObject_5.0.2         
[27] sp_2.1-4                    harmony_1.2.3              
[29] Rcpp_1.0.13-1               SPARK_1.1.1                
[31] scry_1.16.0                 scran_1.32.0               
[33] scater_1.32.1               ggplot2_3.5.1              
[35] scuttle_1.14.0              SingleCellExperiment_1.26.0
[37] SummarizedExperiment_1.34.0 Biobase_2.64.0             
[39] GenomicRanges_1.56.2        GenomeInfoDb_1.40.1        
[41] IRanges_2.38.1              S4Vectors_0.42.1           
[43] BiocGenerics_0.50.0         MatrixGenerics_1.16.0      
[45] matrixStats_1.4.1           nnSVG_1.8.0                
[47] workflowr_1.7.1            

loaded via a namespace (and not attached):
  [1] GSEABase_1.66.0           goftest_1.2-3            
  [3] DT_0.33                   HDF5Array_1.32.1         
  [5] Biostrings_2.72.1         vctrs_0.6.5              
  [7] spatstat.random_3.3-2     digest_0.6.37            
  [9] png_0.1-8                 proxy_0.4-27             
 [11] git2r_0.33.0              ggrepel_0.9.6            
 [13] deldir_2.0-4              parallelly_1.41.0        
 [15] magick_2.8.4              reshape2_1.4.4           
 [17] httpuv_1.6.15             withr_3.0.1              
 [19] xfun_0.49                 survival_3.8-3           
 [21] memoise_2.0.1             benchmarkme_1.0.8        
 [23] ggbeeswarm_0.7.2          zoo_1.8-12               
 [25] pbapply_1.7-2             Formula_1.2-5            
 [27] rematch2_2.1.2            KEGGREST_1.44.1          
 [29] promises_1.3.2            httr_1.4.7               
 [31] rstatix_0.7.2             restfulr_0.0.15          
 [33] rhdf5filters_1.16.0       globals_0.16.3           
 [35] fitdistrplus_1.2-2        rhdf5_2.48.0             
 [37] ps_1.8.1                  rstudioapi_0.17.1        
 [39] UCSC.utils_1.0.0          miniUI_0.1.1.1           
 [41] generics_0.1.3            processx_3.8.5           
 [43] curl_6.0.1                fields_16.3              
 [45] zlibbioc_1.50.0           ScaledMatrix_1.12.0      
 [47] polyclip_1.10-7           doSNOW_1.0.20            
 [49] GenomeInfoDbData_1.2.12   ExperimentHub_2.12.0     
 [51] SparseArray_1.4.8         golem_0.5.1              
 [53] xtable_1.8-4              stringr_1.5.1            
 [55] pracma_2.4.4              evaluate_1.0.1           
 [57] S4Arrays_1.4.1            BiocFileCache_2.12.0     
 [59] colorspace_2.1-1          filelock_1.0.3           
 [61] ROCR_1.0-11               reticulate_1.38.0        
 [63] spatstat.data_3.1-4       shinyWidgets_0.8.7       
 [65] magrittr_2.0.3            lmtest_0.9-40            
 [67] later_1.4.0               viridis_0.6.5            
 [69] lattice_0.22-6            misc3d_0.9-1             
 [71] spatstat.geom_3.3-4       future.apply_1.11.3      
 [73] getPass_0.2-4             scattermore_1.2          
 [75] XML_3.99-0.18             cowplot_1.1.3            
 [77] RcppAnnoy_0.0.22          class_7.3-23             
 [79] pillar_1.9.0              nlme_3.1-166             
 [81] compiler_4.4.2            beachmat_2.20.0          
 [83] RSpectra_0.16-2           stringi_1.8.4            
 [85] tensor_1.5                GenomicAlignments_1.40.0 
 [87] plyr_1.8.9                crayon_1.5.3             
 [89] abind_1.4-8               BiocIO_1.14.0            
 [91] locfit_1.5-9.10           bit_4.5.0.1              
 [93] dplyr_1.1.4               whisker_0.4.1            
 [95] codetools_0.2-20          BiocSingular_1.20.0      
 [97] openssl_2.2.2             bslib_0.8.0              
 [99] paletteer_1.6.0           plotly_4.10.4            
[101] mime_0.12                 splines_4.4.2            
[103] fastDummies_1.7.4         dbplyr_2.5.0             
[105] sparseMatrixStats_1.16.0  attempt_0.3.1            
[107] knitr_1.49                blob_1.2.4               
[109] utf8_1.2.4                BiocVersion_3.19.1       
[111] fs_1.6.5                  listenv_0.9.1            
[113] DelayedMatrixStats_1.26.0 rdist_0.0.5              
[115] ggsignif_0.6.4            tibble_3.2.1             
[117] callr_3.7.6               statmod_1.5.0            
[119] pkgconfig_2.0.3           tools_4.4.2              
[121] BRISC_1.0.6               cachem_1.1.0             
[123] RhpcBLASctl_0.23-42       RSQLite_2.3.9            
[125] viridisLite_0.4.2         DBI_1.2.3                
[127] fastmap_1.2.0             rmarkdown_2.29           
[129] scales_1.3.0              grid_4.4.2               
[131] ica_1.0-3                 Rsamtools_2.20.0         
[133] broom_1.0.7               AnnotationHub_3.12.0     
[135] sass_0.4.9                patchwork_1.3.0          
[137] BiocManager_1.30.25       dotCall64_1.2            
[139] graph_1.82.0              carData_3.0-5            
[141] RANN_2.6.2                snow_0.4-4               
[143] farver_2.1.2              yaml_2.3.10              
[145] rtracklayer_1.64.0        cli_3.6.3                
[147] purrr_1.0.2               leiden_0.4.3.1           
[149] lifecycle_1.0.4           askpass_1.2.1            
[151] uwot_0.2.2                backports_1.5.0          
[153] bluster_1.14.0            sessioninfo_1.2.2        
[155] annotate_1.82.0           BiocParallel_1.38.0      
[157] gtable_0.3.5              rjson_0.2.23             
[159] ggridges_0.5.6            progressr_0.14.0         
[161] limma_3.60.3              jsonlite_1.8.9           
[163] edgeR_4.2.2               RcppHNSW_0.6.0           
[165] bitops_1.0-9              benchmarkmeData_1.0.4    
[167] bit64_4.6.0-1             spatstat.utils_3.1-2     
[169] BiocNeighbors_1.22.0      matlab_1.0.4.1           
[171] jquerylib_0.1.4           metapod_1.12.0           
[173] config_0.3.2              dqrng_0.4.1              
[175] spatstat.univar_3.1-1     lazyeval_0.2.2           
[177] shiny_1.10.0              htmltools_0.5.8.1        
[179] sctransform_0.4.1         rappdirs_0.3.3           
[181] glue_1.8.0                tcltk_4.4.2              
[183] spam_2.11-0               XVector_0.44.0           
[185] RCurl_1.98-1.16           rprojroot_2.0.4          
[187] gridExtra_2.3             igraph_2.0.3             
[189] R6_2.5.1                  tidyr_1.3.1              
[191] labeling_0.4.3            CompQuadForm_1.4.3       
[193] cluster_2.1.8             Rhdf5lib_1.26.0          
[195] DelayedArray_0.30.1       tidyselect_1.2.1         
[197] vipor_0.4.7               maps_3.4.2.1             
[199] car_3.1-3                 AnnotationDbi_1.66.0     
[201] future_1.34.0             rsvd_1.0.5               
[203] munsell_0.5.1             KernSmooth_2.23-26       
[205] data.table_1.15.4         htmlwidgets_1.6.4        
[207] RColorBrewer_1.1-3        rlang_1.1.4              
[209] spatstat.sparse_3.1-0     spatstat.explore_3.3-4   
[211] fansi_1.0.6               beeswarm_0.4.0           </code></pre>
>>>>>>> d89baaa7bfc3bffd762b0db8bac45eeb05443b97
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
