<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>DLPFC</title>

<script src="site_libs/header-attrs-2.27/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">KODAMA-Analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="Installation.html">Installation</a>
</li>
<li>
  <a href="Single-cell.html">Single-cell</a>
</li>
<li>
  <a href="Simulation.html">Simulation</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Real data analyses
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="DLPFC-4.html">DLPFC (4 slides)</a>
    </li>
    <li>
      <a href="DLPFC-12.html">DLPFC (12 slides)</a>
    </li>
    <li>
      <a href="Prostate.html">Prostate</a>
    </li>
    <li>
      <a href="STARmap.html">STARmap (Wang et al., 2018)</a>
    </li>
    <li>
      <a href="MERFISH.html">MERFISH (Moffitt et al., 2018)</a>
    </li>
    <li>
      <a href="VisiumHD.html">Visium HD</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Wrapper
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Seurat.html">Seurat</a>
    </li>
    <li>
      <a href="Giotto.html">Giotto</a>
    </li>
    <li>
      <a href="SpatialExperiment.html">SpatialExperiment</a>
    </li>
  </ul>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/tkcaccia/KODAMA-Analysis">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">DLPFC</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-07-19
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>KODAMA-Analysis/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240618code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240618)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240618code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240618)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomtkcacciaKODAMAAnalysistree3f7aad62bc5856b2bdec967ad08fe8dfcb44b2adtargetblank3f7aad6a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/tree/3f7aad62bc5856b2bdec967ad08fe8dfcb44b2ad" target="_blank">3f7aad6</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomtkcacciaKODAMAAnalysistree3f7aad62bc5856b2bdec967ad08fe8dfcb44b2adtargetblank3f7aad6a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/tkcaccia/KODAMA-Analysis/tree/3f7aad62bc5856b2bdec967ad08fe8dfcb44b2ad" target="_blank">3f7aad6</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Unstaged changes:
    Deleted:    analysis/figure/DLPFC-12.Rmd/unnamed-chunk-10-1.png
    Modified:   analysis/index.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/DLPFC-4.Rmd</code>) and HTML
(<code>docs/DLPFC-4.html</code>) files. If you’ve configured a remote
Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/7719e6fa061de87bfbb218191112b56918b21913/analysis/DLPFC-4.Rmd" target="_blank">7719e6f</a>
</td>
<td>
GitHub
</td>
<td>
2024-07-16
</td>
<td>
Update DLPFC-4.Rmd
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/7be8f59b0c4d82e877c5dfc8cfbfb7936ceceeb6/analysis/DLPFC-4.Rmd" target="_blank">7be8f59</a>
</td>
<td>
tkcaccia
</td>
<td>
2024-07-15
</td>
<td>
updates
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/7be8f59b0c4d82e877c5dfc8cfbfb7936ceceeb6/docs/DLPFC-4.html" target="_blank">7be8f59</a>
</td>
<td>
tkcaccia
</td>
<td>
2024-07-15
</td>
<td>
updates
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>We will analyze all 4 slides</p>
<pre class="r"><code>library(&quot;nnSVG&quot;)
library(&quot;scater&quot;)
library(&quot;scran&quot;)
library(&quot;scry&quot;)
library(&quot;SPARK&quot;)
library(&quot;harmony&quot;)
library(&quot;Seurat&quot;)
library(&quot;spatialLIBD&quot;)
library(&quot;KODAMA&quot;)
library(&quot;KODAMAextra&quot;)


sample_names=c(&quot;151507&quot;,
               &quot;151508&quot;,
               &quot;151509&quot;,
               &quot;151510&quot;,
               &quot;151669&quot;,
               &quot;151670&quot;,
               &quot;151671&quot;,
               &quot;151672&quot;,
               &quot;151673&quot;,
               &quot;151674&quot;,
               &quot;151675&quot;,
               &quot;151676&quot;)

subject_names= c(&quot;Br5292&quot;,&quot;Br5595&quot;, &quot;Br8100&quot;)

#spe &lt;- fetch_data(type = &#39;spe&#39;,destdir=&quot;../Temporary&quot;)
load(&quot;../DLFPC.RData&quot;)

metaData = SingleCellExperiment::colData(spe)
expr = SingleCellExperiment::counts(spe)
sample_names &lt;- paste0(&quot;sample_&quot;, unique(colData(spe)$sample_id))
sample_names &lt;-  unique(colData(spe)$sample_id)
dim(spe)</code></pre>
<pre><code>[1] 33538 47681</code></pre>
<p>identify mitochondrial genes</p>
<pre class="r"><code>is_mito &lt;- grepl(&quot;(^MT-)|(^mt-)&quot;, rowData(spe)$gene_name)
table(is_mito)</code></pre>
<pre><code>is_mito
FALSE  TRUE 
33525    13 </code></pre>
<p>calculate per-spot QC metrics and select the QC threshoulds</p>
<pre class="r"><code>spe &lt;- addPerCellQC(spe, subsets = list(mito = is_mito))

# select QC thresholds
qc_lib_size &lt;- colData(spe)$sum &lt; 500
qc_detected &lt;- colData(spe)$detected &lt; 250
qc_mito &lt;- colData(spe)$subsets_mito_percent &gt; 30
qc_cell_count &lt;- colData(spe)$cell_count &gt; 12</code></pre>
<p>Spot to discard and filter low-quality</p>
<pre class="r"><code>discard &lt;- qc_lib_size | qc_detected | qc_mito | qc_cell_count
table(discard)</code></pre>
<pre><code>discard
FALSE  TRUE 
46653  1028 </code></pre>
<pre class="r"><code>colData(spe)$discard &lt;- discard

spe &lt;- spe[, !colData(spe)$discard]
dim(spe)</code></pre>
<pre><code>[1] 33538 46653</code></pre>
<p>Horizontalization</p>
<pre class="r"><code># readjust xy
xy=spatialCoords(spe)
samples=unique(colData(spe)$sample_id)
for(j in 1:length(samples)){
  sel=samples[j]==colData(spe)$sample_id
  xy[sel,1]=spatialCoords(spe)[sel,1]+12000*(j-1)
}
spatialCoords(spe)=xy</code></pre>
<p>Gene filtering</p>
<pre class="r"><code>spe &lt;- filter_genes(
  spe,
  filter_genes_ncounts = 2,   #ncounts
  filter_genes_pcspots = 0.5,
  filter_mito = TRUE
)

dim(spe)</code></pre>
<pre><code>[1]  6623 46653</code></pre>
<p>REmove the spots that have not been manually assigned</p>
<pre class="r"><code>sel= !is.na(colData(spe)$layer_guess_reordered)
spe = spe[,sel]
dim(spe)</code></pre>
<pre><code>[1]  6623 46318</code></pre>
<p>normalization</p>
<pre class="r"><code>spe &lt;- computeLibraryFactors(spe)
spe &lt;- logNormCounts(spe)</code></pre>
<p>Identification of the gene spatially relevvant</p>
<pre class="r"><code>gene_i=NULL
pvalue_i=NULL
pvalue_mat=matrix(nrow=nrow(spe),ncol=length(sample_names))
for(i in 1:length(sample_names)){
  sel=colData(spe)$sample_id==sample_names[i]
  spe_sub= spe[,sel]

  sparkX &lt;- sparkx(logcounts(spe_sub),spatialCoords(spe_sub),numCores=1,option=&quot;mixture&quot;)

  gene_i=c(gene_i,rowData(spe)$gene_id)
  pvalue_i=c(pvalue_i,sparkX$res_mtest$combinedPval)
  pvalue_mat[,i]=sparkX$res_mtest$combinedPval
  print(sample_names[i])
}</code></pre>
<pre><code>## ===== SPARK-X INPUT INFORMATION ==== 
## number of total samples: 4170 
## number of total genes: 6623 
## Running with single core, may take some time 
## Testing With Projection Kernel
## Testing With Gaussian Kernel 1
## Testing With Gaussian Kernel 2
## Testing With Gaussian Kernel 3
## Testing With Gaussian Kernel 4
## Testing With Gaussian Kernel 5
## Testing With Cosine Kernel 1
## Testing With Cosine Kernel 2
## Testing With Cosine Kernel 3
## Testing With Cosine Kernel 4
## Testing With Cosine Kernel 5
[1] &quot;151507&quot;
## ===== SPARK-X INPUT INFORMATION ==== 
## number of total samples: 4285 
## number of total genes: 6623 
## Running with single core, may take some time 
## Testing With Projection Kernel
## Testing With Gaussian Kernel 1
## Testing With Gaussian Kernel 2
## Testing With Gaussian Kernel 3
## Testing With Gaussian Kernel 4
## Testing With Gaussian Kernel 5
## Testing With Cosine Kernel 1
## Testing With Cosine Kernel 2
## Testing With Cosine Kernel 3
## Testing With Cosine Kernel 4
## Testing With Cosine Kernel 5</code></pre>
<pre><code>Warning in FUN(newX[, i], ...): There are p-values that are exactly 1!
Warning in FUN(newX[, i], ...): There are p-values that are exactly 1!</code></pre>
<pre><code>[1] &quot;151508&quot;
## ===== SPARK-X INPUT INFORMATION ==== 
## number of total samples: 4708 
## number of total genes: 6623 
## Running with single core, may take some time 
## Testing With Projection Kernel
## Testing With Gaussian Kernel 1
## Testing With Gaussian Kernel 2
## Testing With Gaussian Kernel 3
## Testing With Gaussian Kernel 4
## Testing With Gaussian Kernel 5
## Testing With Cosine Kernel 1
## Testing With Cosine Kernel 2
## Testing With Cosine Kernel 3
## Testing With Cosine Kernel 4
## Testing With Cosine Kernel 5
[1] &quot;151509&quot;
## ===== SPARK-X INPUT INFORMATION ==== 
## number of total samples: 4571 
## number of total genes: 6623 
## Running with single core, may take some time 
## Testing With Projection Kernel
## Testing With Gaussian Kernel 1
## Testing With Gaussian Kernel 2
## Testing With Gaussian Kernel 3
## Testing With Gaussian Kernel 4
## Testing With Gaussian Kernel 5
## Testing With Cosine Kernel 1
## Testing With Cosine Kernel 2
## Testing With Cosine Kernel 3
## Testing With Cosine Kernel 4
## Testing With Cosine Kernel 5</code></pre>
<pre><code>Warning in FUN(newX[, i], ...): There are p-values that are exactly 1!
Warning in FUN(newX[, i], ...): There are p-values that are exactly 1!</code></pre>
<pre><code>[1] &quot;151510&quot;
## ===== SPARK-X INPUT INFORMATION ==== 
## number of total samples: 3587 
## number of total genes: 6623 
## Running with single core, may take some time 
## Testing With Projection Kernel
## Testing With Gaussian Kernel 1
## Testing With Gaussian Kernel 2
## Testing With Gaussian Kernel 3
## Testing With Gaussian Kernel 4
## Testing With Gaussian Kernel 5
## Testing With Cosine Kernel 1
## Testing With Cosine Kernel 2
## Testing With Cosine Kernel 3
## Testing With Cosine Kernel 4
## Testing With Cosine Kernel 5</code></pre>
<pre><code>Warning in FUN(newX[, i], ...): There are p-values that are exactly 1!</code></pre>
<pre><code>[1] &quot;151669&quot;
## ===== SPARK-X INPUT INFORMATION ==== 
## number of total samples: 3274 
## number of total genes: 6623 
## Running with single core, may take some time 
## Testing With Projection Kernel
## Testing With Gaussian Kernel 1
## Testing With Gaussian Kernel 2
## Testing With Gaussian Kernel 3
## Testing With Gaussian Kernel 4
## Testing With Gaussian Kernel 5
## Testing With Cosine Kernel 1
## Testing With Cosine Kernel 2
## Testing With Cosine Kernel 3
## Testing With Cosine Kernel 4
## Testing With Cosine Kernel 5</code></pre>
<pre><code>Warning in FUN(newX[, i], ...): There are p-values that are exactly 1!</code></pre>
<pre><code>[1] &quot;151670&quot;
## ===== SPARK-X INPUT INFORMATION ==== 
## number of total samples: 4013 
## number of total genes: 6623 
## Running with single core, may take some time 
## Testing With Projection Kernel
## Testing With Gaussian Kernel 1
## Testing With Gaussian Kernel 2
## Testing With Gaussian Kernel 3
## Testing With Gaussian Kernel 4
## Testing With Gaussian Kernel 5
## Testing With Cosine Kernel 1
## Testing With Cosine Kernel 2
## Testing With Cosine Kernel 3
## Testing With Cosine Kernel 4
## Testing With Cosine Kernel 5
[1] &quot;151671&quot;
## ===== SPARK-X INPUT INFORMATION ==== 
## number of total samples: 3772 
## number of total genes: 6623 
## Running with single core, may take some time 
## Testing With Projection Kernel
## Testing With Gaussian Kernel 1
## Testing With Gaussian Kernel 2
## Testing With Gaussian Kernel 3
## Testing With Gaussian Kernel 4
## Testing With Gaussian Kernel 5
## Testing With Cosine Kernel 1
## Testing With Cosine Kernel 2
## Testing With Cosine Kernel 3
## Testing With Cosine Kernel 4
## Testing With Cosine Kernel 5</code></pre>
<pre><code>Warning in FUN(newX[, i], ...): There are p-values that are exactly 1!</code></pre>
<pre><code>[1] &quot;151672&quot;
## ===== SPARK-X INPUT INFORMATION ==== 
## number of total samples: 3568 
## number of total genes: 6623 
## Running with single core, may take some time 
## Testing With Projection Kernel
## Testing With Gaussian Kernel 1
## Testing With Gaussian Kernel 2
## Testing With Gaussian Kernel 3
## Testing With Gaussian Kernel 4
## Testing With Gaussian Kernel 5
## Testing With Cosine Kernel 1
## Testing With Cosine Kernel 2
## Testing With Cosine Kernel 3
## Testing With Cosine Kernel 4
## Testing With Cosine Kernel 5
[1] &quot;151673&quot;
## ===== SPARK-X INPUT INFORMATION ==== 
## number of total samples: 3576 
## number of total genes: 6623 
## Running with single core, may take some time 
## Testing With Projection Kernel
## Testing With Gaussian Kernel 1
## Testing With Gaussian Kernel 2
## Testing With Gaussian Kernel 3
## Testing With Gaussian Kernel 4
## Testing With Gaussian Kernel 5
## Testing With Cosine Kernel 1
## Testing With Cosine Kernel 2
## Testing With Cosine Kernel 3
## Testing With Cosine Kernel 4
## Testing With Cosine Kernel 5
[1] &quot;151674&quot;
## ===== SPARK-X INPUT INFORMATION ==== 
## number of total samples: 3468 
## number of total genes: 6623 
## Running with single core, may take some time 
## Testing With Projection Kernel
## Testing With Gaussian Kernel 1
## Testing With Gaussian Kernel 2
## Testing With Gaussian Kernel 3
## Testing With Gaussian Kernel 4
## Testing With Gaussian Kernel 5
## Testing With Cosine Kernel 1
## Testing With Cosine Kernel 2
## Testing With Cosine Kernel 3
## Testing With Cosine Kernel 4
## Testing With Cosine Kernel 5</code></pre>
<pre><code>Warning in FUN(newX[, i], ...): There are p-values that are exactly 1!</code></pre>
<pre><code>[1] &quot;151675&quot;
## ===== SPARK-X INPUT INFORMATION ==== 
## number of total samples: 3326 
## number of total genes: 6623 
## Running with single core, may take some time 
## Testing With Projection Kernel
## Testing With Gaussian Kernel 1
## Testing With Gaussian Kernel 2
## Testing With Gaussian Kernel 3
## Testing With Gaussian Kernel 4
## Testing With Gaussian Kernel 5
## Testing With Cosine Kernel 1
## Testing With Cosine Kernel 2
## Testing With Cosine Kernel 3
## Testing With Cosine Kernel 4
## Testing With Cosine Kernel 5</code></pre>
<pre><code>Warning in FUN(newX[, i], ...): There are p-values that are exactly 1!</code></pre>
<pre><code>[1] &quot;151676&quot;</code></pre>
<pre class="r"><code>oo=order(pvalue_i)
top_genes=gene_i[oo]
n=ave(1:length(top_genes), top_genes, FUN = seq_along)
top_genes=top_genes[n==1]

oo=order(apply(pvalue_mat,1,function(x) median(-log(x))),decreasing = TRUE)
top=gene_i[oo]</code></pre>
<p>Selection of the patient Br5595</p>
<pre class="r"><code>subject_names=&quot;Br5595&quot;

  spe_sub &lt;- spe[, colData(spe)$subject ==  subject_names]
  subjects=colData(spe_sub)$subject
  dim(spe_sub)</code></pre>
<pre><code>[1]  6623 14646</code></pre>
<pre class="r"><code>  spe_sub &lt;- runPCA(spe_sub, 50,subset_row = top[1:2000], scale=TRUE)

  pca=reducedDim(spe_sub,type = &quot;PCA&quot;)[,1:50]
  labels=as.factor(colData(spe_sub)$layer_guess_reordered)
  xy=as.matrix(spatialCoords(spe_sub))
  samples=colData(spe_sub)$sample_id
  data=t(logcounts(spe_sub)[top[1:2000],])
  
  plot(pca, col=as.factor(colData(spe_sub)$sample_id))</code></pre>
<p><img src="figure/DLPFC-4.Rmd/unnamed-chunk-10-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-1">
Past versions of unnamed-chunk-10-1.png
</button>
</p>
<div id="fig-unnamed-chunk-10-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/7be8f59b0c4d82e877c5dfc8cfbfb7936ceceeb6/docs/figure/DLPFC-4.Rmd/unnamed-chunk-10-1.png" target="_blank">7be8f59</a>
</td>
<td>
tkcaccia
</td>
<td>
2024-07-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>KODAMA</p>
<pre class="r"><code>kk=KODAMA.matrix.parallel(pca,
                            spatial = xy,
                            FUN= &quot;PLS&quot; ,
                            landmarks = 100000,
                            splitting = 100,
                            f.par.pls = 50,
                            spatial.resolution = 0.4,
                            n.cores=4)</code></pre>
<pre><code>socket cluster with 4 nodes on host &#39;localhost&#39;
================================================================================[1] &quot;Finished parallel computation&quot;

[1] &quot;Calculation of dissimilarity matrix...&quot;
================================================================================</code></pre>
<pre class="r"><code>  print(&quot;KODAMA finished&quot;)</code></pre>
<pre><code>[1] &quot;KODAMA finished&quot;</code></pre>
<pre class="r"><code>     config=umap.defaults
     config$n_threads = 4
     config$n_sgd_threads = &quot;auto&quot;
     kk_UMAP=KODAMA.visualization(kk,method=&quot;UMAP&quot;,config=config)

     plot(kk_UMAP,col=as.factor(labels))</code></pre>
<p><img src="figure/DLPFC-4.Rmd/unnamed-chunk-11-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-1">
Past versions of unnamed-chunk-11-1.png
</button>
</p>
<div id="fig-unnamed-chunk-11-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/7be8f59b0c4d82e877c5dfc8cfbfb7936ceceeb6/docs/figure/DLPFC-4.Rmd/unnamed-chunk-11-1.png" target="_blank">7be8f59</a>
</td>
<td>
tkcaccia
</td>
<td>
2024-07-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>CLUSTER</p>
<pre class="r"><code>library(&quot;mclust&quot;)

  clu=kmeans(kk_UMAP,7,nstart = 100)$cluster
  plot(kk_UMAP,col=labels,pch=20)</code></pre>
<p><img src="figure/DLPFC-4.Rmd/unnamed-chunk-12-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-1">
Past versions of unnamed-chunk-12-1.png
</button>
</p>
<div id="fig-unnamed-chunk-12-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/7be8f59b0c4d82e877c5dfc8cfbfb7936ceceeb6/docs/figure/DLPFC-4.Rmd/unnamed-chunk-12-1.png" target="_blank">7be8f59</a>
</td>
<td>
tkcaccia
</td>
<td>
2024-07-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>  plot(kk_UMAP,col=clu,pch=20)</code></pre>
<p><img src="figure/DLPFC-4.Rmd/unnamed-chunk-12-2.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-2">
Past versions of unnamed-chunk-12-2.png
</button>
</p>
<div id="fig-unnamed-chunk-12-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/7be8f59b0c4d82e877c5dfc8cfbfb7936ceceeb6/docs/figure/DLPFC-4.Rmd/unnamed-chunk-12-2.png" target="_blank">7be8f59</a>
</td>
<td>
tkcaccia
</td>
<td>
2024-07-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot(xy,col=clu,pch=20)</code></pre>
<p><img src="figure/DLPFC-4.Rmd/unnamed-chunk-12-3.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-3">
Past versions of unnamed-chunk-12-3.png
</button>
</p>
<div id="fig-unnamed-chunk-12-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/7be8f59b0c4d82e877c5dfc8cfbfb7936ceceeb6/docs/figure/DLPFC-4.Rmd/unnamed-chunk-12-3.png" target="_blank">7be8f59</a>
</td>
<td>
tkcaccia
</td>
<td>
2024-07-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>u=unique(samples)
for(i in 1:length(u)){
  sel=samples==u[i]
  print(adjustedRandIndex(labels[sel],clu[sel]))
}</code></pre>
<pre><code>[1] 0.3935705
[1] 0.3864863
[1] 0.4110915
[1] 0.4308478</code></pre>
<pre class="r"><code>ref=refine_SVM(xy,clu,samples,cost=100)</code></pre>
<pre><code>[1] &quot;151669&quot;
[1] &quot;151670&quot;
[1] &quot;151671&quot;
[1] &quot;151672&quot;</code></pre>
<pre class="r"><code>u=unique(samples)
for(i in 1:length(u)){
  sel=samples==u[i]
  print(adjustedRandIndex(labels[sel],ref[sel]))
}</code></pre>
<pre><code>[1] 0.409757
[1] 0.3940192
[1] 0.4264205
[1] 0.4558965</code></pre>
<pre class="r"><code>plot(xy,col=ref,pch=20)</code></pre>
<p><img src="figure/DLPFC-4.Rmd/unnamed-chunk-12-4.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-4">
Past versions of unnamed-chunk-12-4.png
</button>
</p>
<div id="fig-unnamed-chunk-12-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/7be8f59b0c4d82e877c5dfc8cfbfb7936ceceeb6/docs/figure/DLPFC-4.Rmd/unnamed-chunk-12-4.png" target="_blank">7be8f59</a>
</td>
<td>
tkcaccia
</td>
<td>
2024-07-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.1 (2024-06-14)
Platform: x86_64-pc-linux-gnu
Running under: Ubuntu 20.04.6 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0 
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

time zone: Etc/UTC
tzcode source: system (glibc)

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] mclust_6.1.1                KODAMAextra_1.0            
 [3] e1071_1.7-14                doParallel_1.0.17          
 [5] iterators_1.0.14            foreach_1.5.2              
 [7] KODAMA_3.1                  umap_0.2.10.0              
 [9] Rtsne_0.17                  minerva_1.5.10             
[11] spatialLIBD_1.16.2          SpatialExperiment_1.14.0   
[13] Seurat_5.1.0                SeuratObject_5.0.2         
[15] sp_2.1-4                    harmony_1.2.0              
[17] Rcpp_1.0.12                 SPARK_1.1.1                
[19] scry_1.16.0                 scran_1.32.0               
[21] scater_1.32.0               ggplot2_3.5.1              
[23] scuttle_1.14.0              SingleCellExperiment_1.26.0
[25] SummarizedExperiment_1.34.0 Biobase_2.64.0             
[27] GenomicRanges_1.56.1        GenomeInfoDb_1.40.1        
[29] IRanges_2.38.1              S4Vectors_0.42.1           
[31] BiocGenerics_0.50.0         MatrixGenerics_1.16.0      
[33] matrixStats_1.3.0           nnSVG_1.8.0                
[35] workflowr_1.7.1            

loaded via a namespace (and not attached):
  [1] goftest_1.2-3             DT_0.33                  
  [3] Biostrings_2.72.1         vctrs_0.6.5              
  [5] spatstat.random_3.3-1     digest_0.6.36            
  [7] png_0.1-8                 proxy_0.4-27             
  [9] git2r_0.33.0              ggrepel_0.9.5            
 [11] deldir_2.0-4              parallelly_1.37.1        
 [13] magick_2.8.4              MASS_7.3-61              
 [15] reshape2_1.4.4            httpuv_1.6.15            
 [17] withr_3.0.0               xfun_0.45                
 [19] survival_3.7-0            memoise_2.0.1            
 [21] benchmarkme_1.0.8         ggbeeswarm_0.7.2         
 [23] zoo_1.8-12                pbapply_1.7-2            
 [25] rematch2_2.1.2            KEGGREST_1.44.1          
 [27] promises_1.3.0            httr_1.4.7               
 [29] restfulr_0.0.15           globals_0.16.3           
 [31] fitdistrplus_1.2-1        ps_1.7.7                 
 [33] rstudioapi_0.16.0         UCSC.utils_1.0.0         
 [35] miniUI_0.1.1.1            generics_0.1.3           
 [37] processx_3.8.4            curl_5.2.1               
 [39] fields_16.2               zlibbioc_1.50.0          
 [41] ScaledMatrix_1.12.0       polyclip_1.10-6          
 [43] doSNOW_1.0.20             GenomeInfoDbData_1.2.12  
 [45] ExperimentHub_2.12.0      SparseArray_1.4.8        
 [47] golem_0.4.1               xtable_1.8-4             
 [49] stringr_1.5.1             pracma_2.4.4             
 [51] evaluate_0.24.0           S4Arrays_1.4.1           
 [53] BiocFileCache_2.12.0      irlba_2.3.5.1            
 [55] colorspace_2.1-0          filelock_1.0.3           
 [57] ROCR_1.0-11               reticulate_1.38.0        
 [59] spatstat.data_3.1-2       shinyWidgets_0.8.6       
 [61] magrittr_2.0.3            lmtest_0.9-40            
 [63] later_1.3.2               viridis_0.6.5            
 [65] lattice_0.22-6            spatstat.geom_3.3-2      
 [67] future.apply_1.11.2       getPass_0.2-4            
 [69] scattermore_1.2           XML_3.99-0.17            
 [71] cowplot_1.1.3             RcppAnnoy_0.0.22         
 [73] class_7.3-22              pillar_1.9.0             
 [75] nlme_3.1-165              compiler_4.4.1           
 [77] beachmat_2.20.0           RSpectra_0.16-1          
 [79] stringi_1.8.4             tensor_1.5               
 [81] GenomicAlignments_1.40.0  plyr_1.8.9               
 [83] crayon_1.5.3              abind_1.4-5              
 [85] BiocIO_1.14.0             locfit_1.5-9.10          
 [87] bit_4.0.5                 dplyr_1.1.4              
 [89] whisker_0.4.1             codetools_0.2-20         
 [91] BiocSingular_1.20.0       openssl_2.2.0            
 [93] bslib_0.7.0               paletteer_1.6.0          
 [95] plotly_4.10.4             mime_0.12                
 [97] splines_4.4.1             fastDummies_1.7.3        
 [99] dbplyr_2.5.0              sparseMatrixStats_1.16.0 
[101] attempt_0.3.1             knitr_1.48               
[103] blob_1.2.4                utf8_1.2.4               
[105] BiocVersion_3.19.1        fs_1.6.4                 
[107] listenv_0.9.1             DelayedMatrixStats_1.26.0
[109] rdist_0.0.5               tibble_3.2.1             
[111] Matrix_1.7-0              callr_3.7.6              
[113] statmod_1.5.0             pkgconfig_2.0.3          
[115] tools_4.4.1               BRISC_1.0.5              
[117] cachem_1.1.0              RSQLite_2.3.7            
[119] viridisLite_0.4.2         DBI_1.2.3                
[121] fastmap_1.2.0             rmarkdown_2.27           
[123] scales_1.3.0              grid_4.4.1               
[125] ica_1.0-3                 Rsamtools_2.20.0         
[127] AnnotationHub_3.12.0      sass_0.4.9               
[129] patchwork_1.2.0           BiocManager_1.30.23      
[131] dotCall64_1.1-1           RANN_2.6.1               
[133] snow_0.4-4                yaml_2.3.9               
[135] rtracklayer_1.64.0        cli_3.6.3                
[137] purrr_1.0.2               leiden_0.4.3.1           
[139] lifecycle_1.0.4           askpass_1.2.0            
[141] uwot_0.2.2                bluster_1.14.0           
[143] sessioninfo_1.2.2         BiocParallel_1.38.0      
[145] gtable_0.3.5              rjson_0.2.21             
[147] ggridges_0.5.6            progressr_0.14.0         
[149] limma_3.60.3              jsonlite_1.8.8           
[151] edgeR_4.2.1               RcppHNSW_0.6.0           
[153] bitops_1.0-7              benchmarkmeData_1.0.4    
[155] bit64_4.0.5               spatstat.utils_3.0-5     
[157] BiocNeighbors_1.22.0      matlab_1.0.4.1           
[159] jquerylib_0.1.4           highr_0.11               
[161] metapod_1.12.0            config_0.3.2             
[163] dqrng_0.4.1               spatstat.univar_3.0-0    
[165] lazyeval_0.2.2            shiny_1.8.1.1            
[167] htmltools_0.5.8.1         sctransform_0.4.1        
[169] rappdirs_0.3.3            glue_1.7.0               
[171] spam_2.10-0               XVector_0.44.0           
[173] RCurl_1.98-1.16           rprojroot_2.0.4          
[175] gridExtra_2.3             igraph_2.0.3             
[177] R6_2.5.1                  tidyr_1.3.1              
[179] CompQuadForm_1.4.3        cluster_2.1.6            
[181] DelayedArray_0.30.1       tidyselect_1.2.1         
[183] vipor_0.4.7               maps_3.4.2               
[185] AnnotationDbi_1.66.0      future_1.33.2            
[187] rsvd_1.0.5                munsell_0.5.1            
[189] KernSmooth_2.23-24        data.table_1.15.4        
[191] htmlwidgets_1.6.4         RColorBrewer_1.1-3       
[193] rlang_1.1.4               spatstat.sparse_3.1-0    
[195] spatstat.explore_3.3-1    fansi_1.0.6              
[197] beeswarm_0.4.0           </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
