<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>About</title>

<script src="site_libs/header-attrs-2.27/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">KODAMA-Analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="Installation.html">Installation</a>
</li>
<li>
  <a href="single-cell.html">Single-cell</a>
</li>
<li>
  <a href="simulation.html">Simulation</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Real data analyses
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="DLPFC-4.html">DLPFC (4 slides)</a>
    </li>
    <li>
      <a href="DLPFC-12.html">DLPFC (12 slides)</a>
    </li>
    <li>
      <a href="Prostate.html">Prostate</a>
    </li>
    <li>
      <a href="STARmap.html">STARmap (Wang et al., 2018)</a>
    </li>
    <li>
      <a href="MERFISH.html">MERFISH (Moffitt et al., 2018)</a>
    </li>
    <li>
      <a href="VisiumHD.html">Visium HD</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Wrapper
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Seurat.html">Seurat</a>
    </li>
    <li>
      <a href="Giotto.html">Giotto</a>
    </li>
  </ul>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/tkcaccia/KODAMA-Analysis">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">About</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-07-16
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>KODAMA-Analysis/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240618code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240618)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240618code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240618)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomtkcacciaKODAMAAnalysistree8cc2284afda92440fdbf51d8a3774323c2910506targetblank8cc2284a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/tree/8cc2284afda92440fdbf51d8a3774323c2910506" target="_blank">8cc2284</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomtkcacciaKODAMAAnalysistree8cc2284afda92440fdbf51d8a3774323c2910506targetblank8cc2284a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/tkcaccia/KODAMA-Analysis/tree/8cc2284afda92440fdbf51d8a3774323c2910506" target="_blank">8cc2284</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  .gitignore

Unstaged changes:
    Deleted:    analysis/figure/DLPFC-12.Rmd/unnamed-chunk-10-1.png

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/Giotto.Rmd</code>) and HTML
(<code>docs/Giotto.html</code>) files. If you’ve configured a remote Git
repository (see <code>?wflow_git_remote</code>), click on the hyperlinks
in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/8cc2284afda92440fdbf51d8a3774323c2910506/analysis/Giotto.Rmd" target="_blank">8cc2284</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-07-16
</td>
<td>
Start my new project
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/6494c7a779855737e0e200c15955c637a9854946/docs/Giotto.html" target="_blank">6494c7a</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-07-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/481cada9973bff246b74b928d7be680ee7468db7/analysis/Giotto.Rmd" target="_blank">481cada</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-07-16
</td>
<td>
Start my new project
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/a4f0a75f037ee66fe7e4d5f6b2554370e04e7a1a/docs/Giotto.html" target="_blank">a4f0a75</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-07-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/6a64ca2784eb29fa525807e1e5ce6cc86f6e7c03/analysis/Giotto.Rmd" target="_blank">6a64ca2</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-07-16
</td>
<td>
Start my new project
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/ca0df735e620512fc41a44a5bc241c44df6a2a3d/docs/Giotto.html" target="_blank">ca0df73</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-07-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/77fe01163e4fcb1b0fbe6a7b0431a152b92a65cc/analysis/Giotto.Rmd" target="_blank">77fe011</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-07-16
</td>
<td>
Start my new project
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/7be8f59b0c4d82e877c5dfc8cfbfb7936ceceeb6/docs/Giotto.html" target="_blank">7be8f59</a>
</td>
<td>
tkcaccia
</td>
<td>
2024-07-15
</td>
<td>
updates
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/f8ca54aee4428df50d2cd0f41cbf2a0ceae01d7d/analysis/Giotto.Rmd" target="_blank">f8ca54a</a>
</td>
<td>
tkcaccia
</td>
<td>
2024-07-14
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/f8ca54aee4428df50d2cd0f41cbf2a0ceae01d7d/docs/Giotto.html" target="_blank">f8ca54a</a>
</td>
<td>
tkcaccia
</td>
<td>
2024-07-14
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/89a11c19aabfd67ffe5ad726376a5a2ee4b20539/analysis/Giotto.Rmd" target="_blank">89a11c1</a>
</td>
<td>
GitHub
</td>
<td>
2024-07-08
</td>
<td>
Add files via upload
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/2b5aad71a70086d1a4ae5ae75e37bcc58131fb7e/docs/Giotto.html" target="_blank">2b5aad7</a>
</td>
<td>
GitHub
</td>
<td>
2024-07-08
</td>
<td>
Add files via upload
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="giotto-visium-brain" class="section level1">
<h1>Giotto Visium Brain</h1>
<div id="introduction-to-giotto-visium-platform" class="section level2">
<h2>Introduction to Giotto Visium Platform</h2>
<div id="overview" class="section level3">
<h3>Overview</h3>
<p>10X Genomics recently launched the Giotto Visium platform, a
cutting-edge tool for acquiring spatial gene expression data using
Visium Spatial Gene Expression slides. This innovative technology allows
researchers to analyze gene expression patterns while preserving spatial
information, which is crucial for understanding the intricate
organization and functionality of tissues, especially in complex
structures like the brain.</p>
</div>
<div id="dataset-description" class="section level3">
<h3>Dataset Description</h3>
<p>The data used in this tutorial originates from a Visium Spatial Gene
Expression slide of the adult mouse. This dataset is available on the
10X Genomics support site and can be downloaded from the following link:
<a
href="https://support.10xgenomics.com/spatial-gene-expression/datasets/1.1.0/V1_Adult_Mouse_Brain">V1_Adult_Mouse_Brain</a>.
Ensure that the necessary files are properly organized in the
<code>data_path</code> directory.</p>
</div>
</div>
<div id="part-1-setup-and-installation" class="section level2">
<h2>Part 1: Setup and Installation</h2>
<div id="package-installation" class="section level3">
<h3>Package Installation</h3>
<p>Before proceeding, ensure the required R packages are installed:</p>
<pre class="r"><code># Ensure Giotto Suite is installed
if (!&quot;Giotto&quot; %in% installed.packages()) {
  devtools::install_github(&quot;drieslab/Giotto@suite&quot;)
}

# Ensure GiottoData is installed
if (!&quot;GiottoData&quot; %in% installed.packages()) {
  devtools::install_github(&quot;drieslab/GiottoData&quot;)
}

# Check and install Python environment for Giotto
library(Giotto)

genv_exists &lt;- checkGiottoEnvironment()
if (!genv_exists) {
  installGiottoEnvironment()  # Run this command once to install Giotto environment
}

# Install additional required packages
if (!&quot;KODAMA&quot; %in% installed.packages()) {
  devtools::install_github(&quot;tkcaccia/KODAMA&quot;)
}
if (!&quot;KODAMAextra&quot; %in% installed.packages()) {
  devtools::install_github(&quot;tkcaccia/KODAMAextra&quot;)
}</code></pre>
</div>
</div>
<div id="part-2-setup-and-installation" class="section level2">
<h2>Part 2: Setup and Installation</h2>
<div id="package-installation-1" class="section level3">
<h3>Package Installation</h3>
<p>Before proceeding, ensure the required R packages are installed:</p>
<pre class="r"><code># Ensure Giotto Suite is installed
if (!&quot;Giotto&quot; %in% installed.packages()) {
  devtools::install_github(&quot;drieslab/Giotto@suite&quot;)
}

# Ensure GiottoData is installed
if (!&quot;GiottoData&quot; %in% installed.packages()) {
  devtools::install_github(&quot;drieslab/GiottoData&quot;)
}

# Check and install Python environment for Giotto
library(Giotto)

genv_exists &lt;- checkGiottoEnvironment()
if (!genv_exists) {
  installGiottoEnvironment()  # Run this command once to install Giotto environment
}

# Install additional required packages
if (!&quot;KODAMA&quot; %in% installed.packages()) {
  devtools::install_github(&quot;tkcaccia/KODAMA&quot;)
}
if (!&quot;KODAMAextra&quot; %in% installed.packages()) {
  devtools::install_github(&quot;tkcaccia/KODAMAextra&quot;)
}

library(KODAMA)
library(KODAMAextra)
library(Giotto)
library(GiottoData)</code></pre>
</div>
</div>
<div id="part-3-data-preparation-and-analysis" class="section level2">
<h2>Part 3: Data Preparation and Analysis</h2>
<div id="setting-up-the-environment" class="section level3">
<h3>Setting Up the Environment</h3>
<p>Set Working Directory and Giotto Instructions</p>
<pre class="r"><code># Set working directory
results_folder &lt;- &#39;../result&#39;

# Create Giotto instructions
my_python_path &lt;- NULL  # Use the default Python environment for Giotto
instrs &lt;- createGiottoInstructions(save_dir = results_folder,
                                   save_plot = TRUE,
                                   show_plot = FALSE,
                                   python_path = my_python_path)</code></pre>
<p>###Loading and Preparing Data Create Giotto Visium Object</p>
<pre class="r"><code># Provide path to Visium data folder
data_path &lt;- &#39;../DATA/data_path&#39;

# Create Giotto Visium object
visium_brain &lt;- createGiottoVisiumObject(visium_dir = data_path,
                                         expr_data = &#39;raw&#39;,
                                         png_name = &#39;tissue_lowres_image.png&#39;,
                                         gene_column_index = 2,
                                         instructions = instrs)</code></pre>
</div>
</div>
<div id="part-4-exploratory-data-analysis" class="section level2">
<h2>Part 4: Exploratory Data Analysis</h2>
<div id="visualizing-data-and-quality-control" class="section level4">
<h4>Visualizing Data and Quality Control</h4>
<pre class="r"><code># Show associated images with Giotto object
showGiottoImageNames(visium_brain)</code></pre>
<pre><code>Image type: largeImage 

--&gt; Name: image </code></pre>
<pre class="r"><code># Check metadata
pDataDT(visium_brain)</code></pre>
<pre><code>                 cell_ID in_tissue array_row array_col
                  &lt;char&gt;     &lt;int&gt;     &lt;int&gt;     &lt;int&gt;
   1: AAACAACGAATAGTTC-1         0         0        16
   2: AAACAAGTATCTCCCA-1         1        50       102
   3: AAACAATCTACTAGCA-1         1         3        43
   4: AAACACCAATAACTGC-1         1        59        19
   5: AAACAGAGCGACTCCT-1         1        14        94
  ---                                                 
4988: TTGTTTCACATCCAGG-1         1        58        42
4989: TTGTTTCATTAGTCTA-1         1        60        30
4990: TTGTTTCCATACAACT-1         1        45        27
4991: TTGTTTGTATTACACG-1         0        73        41
4992: TTGTTTGTGTAAATTC-1         1         7        51</code></pre>
<pre class="r"><code># Visualize spatial plot
spatPlot2D(gobject = visium_brain, cell_color = &#39;in_tissue&#39;, point_size = 2,
           cell_color_code = c(&#39;0&#39; = &#39;lightgrey&#39;, &#39;1&#39; = &#39;blue&#39;),
           show_image = TRUE, image_name = &#39;image&#39;, show_plot= TRUE, return_plot = FALSE, save_plot = TRUE)</code></pre>
<p><img src="figure/Giotto.Rmd/unnamed-chunk-5-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/ca0df735e620512fc41a44a5bc241c44df6a2a3d/docs/figure/Giotto.Rmd/unnamed-chunk-5-1.png" target="_blank">ca0df73</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-07-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="part-5-data-processing-and-analysis" class="section level2">
<h2>Part 5: Data Processing and Analysis</h2>
<div id="data-filtering-and-normalization" class="section level3">
<h3>Data Filtering and Normalization</h3>
<pre class="r"><code># Subset spots covered by tissue
metadata &lt;- pDataDT(visium_brain)
in_tissue_barcodes &lt;- metadata[in_tissue == 1]$cell_ID
visium_brain &lt;- subsetGiotto(visium_brain, cell_ids = in_tissue_barcodes)

# Filter data
visium_brain &lt;- filterGiotto(gobject = visium_brain,
                             expression_threshold = 1,
                             feat_det_in_min_cells = 50,
                             min_det_feats_per_cell = 1000,
                             expression_values = c(&#39;raw&#39;),
                             verbose = TRUE)</code></pre>
<pre><code>
Feature type:  rna 
Number of cells removed:  4  out of  2702 
Number of feats removed:  7311  out of  22125 </code></pre>
<pre class="r"><code># Normalize data
visium_brain &lt;- normalizeGiotto(gobject = visium_brain, scalefactor = 6000, verbose = TRUE)

## add gene &amp; cell statistics
visium_brain &lt;- addStatistics(gobject = visium_brain)

## visualize
spatPlot2D(gobject = visium_brain, show_image = T, point_alpha = 0.7,
           cell_color = &#39;nr_feats&#39;, color_as_factor = F,save_plot = TRUE, show_plot= TRUE, return_plot = FALSE)</code></pre>
<p><img src="figure/Giotto.Rmd/unnamed-chunk-6-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/ca0df735e620512fc41a44a5bc241c44df6a2a3d/docs/figure/Giotto.Rmd/unnamed-chunk-6-1.png" target="_blank">ca0df73</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-07-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>visium_brain &lt;- calculateHVF(gobject = visium_brain, save_plot = TRUE, show_plot= TRUE, return_plot = FALSE)</code></pre>
<p><img src="figure/Giotto.Rmd/unnamed-chunk-6-2.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-2">
Past versions of unnamed-chunk-6-2.png
</button>
</p>
<div id="fig-unnamed-chunk-6-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/ca0df735e620512fc41a44a5bc241c44df6a2a3d/docs/figure/Giotto.Rmd/unnamed-chunk-6-2.png" target="_blank">ca0df73</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-07-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="part-6-advanced-analysis-and-visualization"
class="section level2">
<h2>Part 6: Advanced Analysis and Visualization</h2>
<div id="dimensionality-reduction-and-clustering"
class="section level3">
<h3>Dimensionality Reduction and Clustering</h3>
<pre class="r"><code># Perform PCA
gene_metadata &lt;- fDataDT(visium_brain)
featgenes &lt;- gene_metadata[hvf == &#39;yes&#39; &amp; perc_cells &gt; 3 &amp; mean_expr_det &gt; 0.4]$feat_ID
visium_brain &lt;- runPCA(gobject = visium_brain,
                       feats_to_use = featgenes)

# Perform KODAMA analysis and visualization
visium_brain &lt;- RunKODAMAmatrix(visium_brain, f.par.pls = 50, FUN = &quot;PLS&quot;, n.cores = 4)</code></pre>
<pre><code>socket cluster with 4 nodes on host &#39;localhost&#39;
================================================================================[1] &quot;Finished parallel computation&quot;

[1] &quot;Calculation of dissimilarity matrix...&quot;
================================================================================</code></pre>
<pre class="r"><code>visium_brain &lt;- RunKODAMAvisualization(visium_brain, method = &quot;UMAP&quot;)

visium_brain &lt;- createNearestNetwork(gobject = visium_brain,dim_reduction_to_use = &quot;KODAMA&quot;, dim_reduction_name=&quot;KODAMA&quot;,dimensions_to_use = 1:2, k = 15)


# Perform clustering and visualize results
visium_brain &lt;- doLeidenCluster(gobject = visium_brain, resolution = 0.6, n_iterations = 1000,network_name = &quot;sNN.KODAMA&quot;)


spatDimPlot(gobject = visium_brain, dim_reduction_to_use =&quot;KODAMA&quot;, dim_reduction_name=&quot;KODAMA&quot;,cell_color = &#39;leiden_clus&#39;,
            dim_point_size = 2, spat_point_size = 2.5, show_plot= TRUE, return_plot = FALSE, save_plot = TRUE)</code></pre>
<p><img src="figure/Giotto.Rmd/unnamed-chunk-7-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/ca0df735e620512fc41a44a5bc241c44df6a2a3d/docs/figure/Giotto.Rmd/unnamed-chunk-7-1.png" target="_blank">ca0df73</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-07-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>This tutorial provides a comprehensive guide to using the Giotto
Visium platform for spatial gene expression analysis, covering setup,
data handling, advanced analysis techniques, and visualization. Each
section aims to help you effectively leverage this powerful tool for
your research endeavors.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.3 (2024-02-29 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19045)

Matrix products: default


locale:
[1] LC_COLLATE=English_United States.utf8 
[2] LC_CTYPE=English_United States.utf8   
[3] LC_MONETARY=English_United States.utf8
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.utf8    

time zone: Africa/Johannesburg
tzcode source: internal

attached base packages:
[1] parallel  stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] GiottoData_0.2.13 KODAMAextra_1.0   e1071_1.7-14      doParallel_1.0.17
 [5] iterators_1.0.14  foreach_1.5.2     KODAMA_3.1        umap_0.2.10.0    
 [9] Rtsne_0.17        minerva_1.5.10    Giotto_4.0.8      GiottoClass_0.3.1

loaded via a namespace (and not attached):
  [1] RColorBrewer_1.1-3          rstudioapi_0.16.0          
  [3] jsonlite_1.8.8              magrittr_2.0.3             
  [5] magick_2.8.3                farver_2.1.2               
  [7] rmarkdown_2.27              fs_1.6.4                   
  [9] zlibbioc_1.48.2             ragg_1.3.2                 
 [11] vctrs_0.6.5                 GiottoUtils_0.1.8          
 [13] RCurl_1.98-1.14             askpass_1.2.0              
 [15] terra_1.7-78                htmltools_0.5.8.1          
 [17] S4Arrays_1.2.1              SparseArray_1.2.4          
 [19] parallelly_1.37.1           sass_0.4.9                 
 [21] bslib_0.7.0                 htmlwidgets_1.6.4          
 [23] plyr_1.8.9                  plotly_4.10.4              
 [25] cachem_1.1.0                whisker_0.4.1              
 [27] igraph_2.0.3                lifecycle_1.0.4            
 [29] pkgconfig_2.0.3             rsvd_1.0.5                 
 [31] Matrix_1.6-5                R6_2.5.1                   
 [33] fastmap_1.2.0               future_1.33.2              
 [35] GenomeInfoDbData_1.2.11     MatrixGenerics_1.14.0      
 [37] digest_0.6.36               colorspace_2.1-0           
 [39] S4Vectors_0.40.2            rprojroot_2.0.4            
 [41] irlba_2.3.5.1               RSpectra_0.16-1            
 [43] textshaping_0.4.0           GenomicRanges_1.54.1       
 [45] beachmat_2.18.1             labeling_0.4.3             
 [47] fansi_1.0.6                 httr_1.4.7                 
 [49] abind_1.4-5                 compiler_4.3.3             
 [51] proxy_0.4-27                withr_3.0.0                
 [53] backports_1.5.0             BiocParallel_1.36.0        
 [55] highr_0.11                  R.utils_2.12.3             
 [57] openssl_2.2.0               rappdirs_0.3.3             
 [59] DelayedArray_0.28.0         rjson_0.2.21               
 [61] gtools_3.9.5                GiottoVisuals_0.2.3        
 [63] tools_4.3.3                 httpuv_1.6.15              
 [65] future.apply_1.11.2         R.oo_1.26.0                
 [67] glue_1.7.0                  dbscan_1.2-0               
 [69] promises_1.3.0              grid_4.3.3                 
 [71] checkmate_2.3.1             reshape2_1.4.4             
 [73] snow_0.4-4                  generics_0.1.3             
 [75] gtable_0.3.5                R.methodsS3_1.8.2          
 [77] class_7.3-22                tidyr_1.3.1                
 [79] data.table_1.15.4           ScaledMatrix_1.10.0        
 [81] BiocSingular_1.18.0         sp_2.1-4                   
 [83] utf8_1.2.4                  XVector_0.42.0             
 [85] BiocGenerics_0.48.1         ggrepel_0.9.5              
 [87] pillar_1.9.0                stringr_1.5.1              
 [89] later_1.3.2                 dplyr_1.1.4                
 [91] lattice_0.22-6              deldir_2.0-4               
 [93] tidyselect_1.2.1            SingleCellExperiment_1.24.0
 [95] knitr_1.48                  git2r_0.33.0               
 [97] IRanges_2.36.0              SummarizedExperiment_1.32.0
 [99] scattermore_1.2             stats4_4.3.3               
[101] xfun_0.45                   Biobase_2.62.0             
[103] matrixStats_1.3.0           stringi_1.8.4              
[105] workflowr_1.7.1             lazyeval_0.2.2             
[107] yaml_2.3.8                  evaluate_0.24.0            
[109] codetools_0.2-20            tibble_3.2.1               
[111] colorRamp2_0.1.0            cli_3.6.2                  
[113] reticulate_1.38.0           systemfonts_1.1.0          
[115] munsell_0.5.1               jquerylib_0.1.4            
[117] Rcpp_1.0.12                 GenomeInfoDb_1.38.8        
[119] doSNOW_1.0.20               globals_0.16.3             
[121] png_0.1-8                   ggplot2_3.5.1              
[123] bitops_1.0-7                listenv_0.9.1              
[125] SpatialExperiment_1.12.0    viridisLite_0.4.2          
[127] scales_1.3.0                purrr_1.0.2                
[129] crayon_1.5.3                rlang_1.1.4                
[131] cowplot_1.1.3              </code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
