<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>About</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">KODAMA-Analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="Installation.html">Installation</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Spatial
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Simulation.html">Simulation</a>
    </li>
    <li>
      <a href="MERFISH.html">MERFISH (Moffitt et al., 2018)</a>
    </li>
    <li>
      <a href="DLPFC.html">Visium DLPFC (Maynard et al., 2021)</a>
    </li>
    <li>
      <a href="Prostate.html">Visium (10X database - Prostate)</a>
    </li>
    <li>
      <a href="VisiumHD.html">Visium HD (10X database - Colorectal cancer)</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Wrapper
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Seurat.html">Seurat</a>
    </li>
    <li>
      <a href="Giotto.html">Giotto</a>
    </li>
    <li>
      <a href="SpatialExperiment.html">SpatialExperiment</a>
    </li>
  </ul>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/tkcaccia/KODAMA-Analysis">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">About</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-01-14
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>KODAMA-Analysis/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240618code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240618)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240618code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240618)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomtkcacciaKODAMAAnalysistree7de0c7444fdebf22d8827606a2b840417e344179targetblank7de0c74a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/tree/7de0c7444fdebf22d8827606a2b840417e344179" target="_blank">7de0c74</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomtkcacciaKODAMAAnalysistree7de0c7444fdebf22d8827606a2b840417e344179targetblank7de0c74a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/tkcaccia/KODAMA-Analysis/tree/7de0c7444fdebf22d8827606a2b840417e344179" target="_blank">7de0c74</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/figure/

Untracked files:
    Untracked:  KODAMA.svg
    Untracked:  code/Acinar_Cell_Carcinoma.ipynb
    Untracked:  code/Adenocarcinoma.ipynb
    Untracked:  code/Adjacent_normal_section.ipynb
    Untracked:  code/DLFPC_preprocessing.R
    Untracked:  code/DLPFC - BANKSY.R
    Untracked:  code/DLPFC - BASS.R
    Untracked:  code/DLPFC - BAYESPACE.R
    Untracked:  code/DLPFC - Nonspatial.R
    Untracked:  code/DLPFC - PRECAST.R
    Untracked:  code/DLPFC_comparison.R
    Untracked:  code/DLPFC_results_analysis.R
    Untracked:  code/MERFISH - BANKSY.R
    Untracked:  code/MERFISH - BASS.R
    Untracked:  code/MERFISH - BAYESPACE.R
    Untracked:  code/MERFISH - Nonspatial.R
    Untracked:  code/MERFISH - PRECAST.R
    Untracked:  code/MERFISH_comparison.R
    Untracked:  code/MERFISH_results_analysis.R
    Untracked:  code/VisiumHD-CRC.ipynb
    Untracked:  code/deep learning code DLPFC.R
    Untracked:  code/save tiles.py
    Untracked:  data/Adenocarcinoma.csv
    Untracked:  data/Annotations/
    Untracked:  data/DLFPC-Br5292-input.RData
    Untracked:  data/DLFPC-Br5595-input.RData
    Untracked:  data/DLFPC-Br8100-input.RData
    Untracked:  data/DLPFC-general.RData
    Untracked:  data/MERFISH-input.RData
    Untracked:  data/spots_classification_ALL.csv
    Untracked:  data/spots_classification_Acinar_Cell_Carcinoma.csv
    Untracked:  data/spots_classification_IF.csv
    Untracked:  data/spots_classification_Normal_prostate.csv
    Untracked:  data/trajectories.RData
    Untracked:  data/trajectories_VISIUMHD.RData
    Untracked:  output/BANSKY-results.RData
    Untracked:  output/BASS-results.RData
    Untracked:  output/BayesSpace-results.RData
    Untracked:  output/CRC-image.RData
    Untracked:  output/CRC-image2.RData
    Untracked:  output/CRC.png
    Untracked:  output/CRC2.png
    Untracked:  output/CRC7.png
    Untracked:  output/CRC8.png
    Untracked:  output/DL.RData
    Untracked:  output/DLFPC-All-2.RData
    Untracked:  output/DLFPC-All.RData
    Untracked:  output/DLFPC-Br5292.RData
    Untracked:  output/DLFPC-Br5595.RData
    Untracked:  output/DLFPC-Br8100.RData
    Untracked:  output/DLPFC1.svg
    Untracked:  output/DLPFC_all_cluster.svg
    Untracked:  output/Figure 1 - boxplot.pdf
    Untracked:  output/Figure 2 - DLPFC 10.pdf
    Untracked:  output/Figures/
    Untracked:  output/KODAMA-results.RData
    Untracked:  output/KODAMA_DLPFC_All_original.svg
    Untracked:  output/KODAMA_DLPFC_Br5595.svg
    Untracked:  output/KODAMA_DLPFC_Br5595_slide.svg
    Untracked:  output/MERFISH-BANSKY-results.RData
    Untracked:  output/MERFISH-BASS-results.RData
    Untracked:  output/MERFISH-BayesSpace-results.RData
    Untracked:  output/MERFISH-KODAMA-results.RData
    Untracked:  output/MERFISH-NONSPATIAL-results.RData
    Untracked:  output/MERFISH-PRECAST-results.RData
    Untracked:  output/MERFISH.RData
    Untracked:  output/Nonspatial-results.RData
    Untracked:  output/PRECAST-results.RData
    Untracked:  output/Prostate.RData
    Untracked:  output/VisiumHD3.RData
    Untracked:  output/a1.RData
    Untracked:  output/a2.RData
    Untracked:  output/a3.RData
    Untracked:  output/a4.RData
    Untracked:  output/a5.RData
    Untracked:  output/a6.RData
    Untracked:  output/a7.RData
    Untracked:  output/a8.RData
    Untracked:  output/image.RData
    Untracked:  output/pp.RData
    Untracked:  output/pp2.RData
    Untracked:  output/pp3.RData
    Untracked:  output/pp4.RData
    Untracked:  output/pp5.RData
    Untracked:  output/prostate1.svg
    Untracked:  output/prostate2.svg
    Untracked:  output/prostate3.svg
    Untracked:  output/prostate4.svg
    Untracked:  output/prostate5.svg
    Untracked:  output/prostate6.svg
    Untracked:  output/prostate7.svg
    Untracked:  output/tight_boundary.geojson

Unstaged changes:
    Deleted:    analysis/D1.Rmd
    Deleted:    analysis/DLPFC-12.Rmd
    Deleted:    analysis/DLPFC-4.Rmd
    Deleted:    analysis/DLPFC1.Rmd
    Deleted:    analysis/DLPFC10.Rmd
    Deleted:    analysis/DLPFC2.Rmd
    Deleted:    analysis/DLPFC3.Rmd
    Deleted:    analysis/DLPFC4.Rmd
    Deleted:    analysis/DLPFC5.Rmd
    Deleted:    analysis/DLPFC6.Rmd
    Deleted:    analysis/DLPFC7.Rmd
    Deleted:    analysis/DLPFC8.Rmd
    Deleted:    analysis/DLPFC9.Rmd
    Deleted:    analysis/Du1.Rmd
    Deleted:    analysis/Du10.Rmd
    Deleted:    analysis/Du11.Rmd
    Deleted:    analysis/Du12.Rmd
    Deleted:    analysis/Du13.Rmd
    Deleted:    analysis/Du14.Rmd
    Deleted:    analysis/Du15.Rmd
    Deleted:    analysis/Du16.Rmd
    Deleted:    analysis/Du17.Rmd
    Deleted:    analysis/Du18.Rmd
    Deleted:    analysis/Du19.Rmd
    Deleted:    analysis/Du2.Rmd
    Deleted:    analysis/Du20.Rmd
    Deleted:    analysis/Du3.Rmd
    Deleted:    analysis/Du4.Rmd
    Deleted:    analysis/Du5.Rmd
    Deleted:    analysis/Du6.Rmd
    Deleted:    analysis/Du7.Rmd
    Deleted:    analysis/Du8.Rmd
    Deleted:    analysis/Du9.Rmd
    Modified:   analysis/Giotto.Rmd
    Modified:   analysis/MERFISH.Rmd
    Deleted:    analysis/MERFISH1a (copy).Rmd
    Deleted:    analysis/MERFISH1a.Rmd
    Deleted:    analysis/MERFISH1b (copy).Rmd
    Deleted:    analysis/MERFISH1b.Rmd
    Deleted:    analysis/MERFISH2a (copy).Rmd
    Deleted:    analysis/MERFISH2a.Rmd
    Deleted:    analysis/MERFISH2b (copy).Rmd
    Deleted:    analysis/MERFISH2b.Rmd
    Deleted:    analysis/MERFISH3a (copy).Rmd
    Deleted:    analysis/MERFISH3a.Rmd
    Deleted:    analysis/MERFISH3b (copy).Rmd
    Deleted:    analysis/MERFISH3b.Rmd
    Deleted:    analysis/MERFISH4a (copy).Rmd
    Deleted:    analysis/MERFISH4a.Rmd
    Deleted:    analysis/MERFISH4b (copy).Rmd
    Deleted:    analysis/MERFISH4b.Rmd
    Modified:   analysis/Prostate.Rmd
    Deleted:    analysis/STARmap.Rmd
    Modified:   analysis/VisiumHD.Rmd
    Modified:   code/VisiumHD_CRC_download.sh
    Deleted:    data/Pathology.csv

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/MERFISH.Rmd</code>) and HTML
(<code>docs/MERFISH.html</code>) files. If you’ve configured a remote
Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/74c3d1ebf8344239ed74609acf5e78c813641d71/analysis/MERFISH.Rmd" target="_blank">74c3d1e</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2025-01-10
</td>
<td>
Start my new project
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/432fc49eabe29b73dbed6a076d99a15956b907b6/docs/MERFISH.html" target="_blank">432fc49</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2025-01-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/52e75ec2f094e5f858a29a576e63a5d15e01a040/analysis/MERFISH.Rmd" target="_blank">52e75ec</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2025-01-10
</td>
<td>
Start my new project
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/2c6734f3f4c95e91bbd44f6be8cd3e45509ec5e0/docs/MERFISH.html" target="_blank">2c6734f</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-09-05
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/6565005e5ece6769c20aeca9591a9574317acb73/analysis/MERFISH.Rmd" target="_blank">6565005</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-09-05
</td>
<td>
Start my new project
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/d1192e95a8ed7985fb6d9678ce0a6d850f882066/docs/MERFISH.html" target="_blank">d1192e9</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-08-12
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/3374e66ce76ec3d26fcb912a9f6eba2f7341ed8f/docs/MERFISH.html" target="_blank">3374e66</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-08-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/d5e7c3c43e23a44595f76a257442e4efb4cfab91/analysis/MERFISH.Rmd" target="_blank">d5e7c3c</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-08-05
</td>
<td>
Start my new project
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/35ce733e4e0012d79f696f9c7dcdbf67e1623e80/docs/MERFISH.html" target="_blank">35ce733</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-08-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/82fe167363197a5889eb59cb7e39511dfde2f22e/docs/MERFISH.html" target="_blank">82fe167</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-07-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/6f7daaccbcc7b968bc736664a45f9ebcaf73fba3/docs/MERFISH.html" target="_blank">6f7daac</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-07-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/7be8f59b0c4d82e877c5dfc8cfbfb7936ceceeb6/docs/MERFISH.html" target="_blank">7be8f59</a>
</td>
<td>
tkcaccia
</td>
<td>
2024-07-15
</td>
<td>
updates
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/f8ca54aee4428df50d2cd0f41cbf2a0ceae01d7d/analysis/MERFISH.Rmd" target="_blank">f8ca54a</a>
</td>
<td>
tkcaccia
</td>
<td>
2024-07-14
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/f8ca54aee4428df50d2cd0f41cbf2a0ceae01d7d/docs/MERFISH.html" target="_blank">f8ca54a</a>
</td>
<td>
tkcaccia
</td>
<td>
2024-07-14
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/0638f9320233629c838d11f98cc4c9ef4b9137ad/analysis/MERFISH.Rmd" target="_blank">0638f93</a>
</td>
<td>
GitHub
</td>
<td>
2024-07-08
</td>
<td>
Update MERFISH.Rmd
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/d3de9e1bf63e1e35f3d967eaa80f3c8735ee32bf/docs/MERFISH.html" target="_blank">d3de9e1</a>
</td>
<td>
GitHub
</td>
<td>
2024-07-08
</td>
<td>
Update MERFISH.html
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/32f206ec6291cf1653fd45eb12008b97207ec96c/docs/MERFISH.html" target="_blank">32f206e</a>
</td>
<td>
GitHub
</td>
<td>
2024-07-08
</td>
<td>
Update MERFISH.html
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/940ec1385b5fac0e259842572da1962def1b36f1/docs/MERFISH.html" target="_blank">940ec13</a>
</td>
<td>
GitHub
</td>
<td>
2024-07-04
</td>
<td>
Update MERFISH.html
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/9ee53df126bf1758843f7581538a33d39bd97884/analysis/MERFISH.Rmd" target="_blank">9ee53df</a>
</td>
<td>
GitHub
</td>
<td>
2024-06-26
</td>
<td>
Update MERFISH.Rmd
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/35e8f25827e79ca29d8b44970ada451bee48482e/docs/MERFISH.html" target="_blank">35e8f25</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-06-26
</td>
<td>
Update docs with MERFISH.html
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/c3eee75f5d792665d77cce1691993c8025604892/analysis/MERFISH.Rmd" target="_blank">c3eee75</a>
</td>
<td>
GitHub
</td>
<td>
2024-06-26
</td>
<td>
Update MERFISH.Rmd
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/e7ddec66f16b8c7bfac6f4175eb87249e0fd2c28/analysis/MERFISH.Rmd" target="_blank">e7ddec6</a>
</td>
<td>
GitHub
</td>
<td>
2024-06-26
</td>
<td>
Update MERFISH.Rmd
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/eb29180b902b2329560807513555cbfa36849d1e/analysis/MERFISH.Rmd" target="_blank">eb29180</a>
</td>
<td>
GitHub
</td>
<td>
2024-06-26
</td>
<td>
Update MERFISH.Rmd
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/bd1988e34ddc2bb2ef6e4826bd15a1342e7d2f00/analysis/MERFISH.Rmd" target="_blank">bd1988e</a>
</td>
<td>
GitHub
</td>
<td>
2024-06-26
</td>
<td>
Update MERFISH.Rmd
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/911564c3294bcefc453a7097a348901b86544728/analysis/MERFISH.Rmd" target="_blank">911564c</a>
</td>
<td>
GitHub
</td>
<td>
2024-06-26
</td>
<td>
Update MERFISH.Rmd
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/246f0bdb380bae90145007978345f50c5020d4ee/analysis/MERFISH.Rmd" target="_blank">246f0bd</a>
</td>
<td>
GitHub
</td>
<td>
2024-06-25
</td>
<td>
Update MERFISH.Rmd
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/20a6dacd3051961f479a262b51ceb97224ff1d61/docs/MERFISH.html" target="_blank">20a6dac</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-06-25
</td>
<td>
Update docs with MERFISH.html
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/e56c4ff46a93e0e36aa1b2ca686bc55e06f1bdff/analysis/MERFISH.Rmd" target="_blank">e56c4ff</a>
</td>
<td>
GitHub
</td>
<td>
2024-06-20
</td>
<td>
Update MERFISH.Rmd
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/9df203d58ecad69cfd1e59c1dd8e1c571546628d/analysis/MERFISH.Rmd" target="_blank">9df203d</a>
</td>
<td>
GitHub
</td>
<td>
2024-06-20
</td>
<td>
Update MERFISH.Rmd
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/be3cdeb69c835b7428b656238860fc3a01504afc/analysis/MERFISH.Rmd" target="_blank">be3cdeb</a>
</td>
<td>
GitHub
</td>
<td>
2024-06-20
</td>
<td>
Update MERFISH.Rmd
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/2e7fa5ea49a52d62b390d184c49f12113aa2f189/analysis/MERFISH.Rmd" target="_blank">2e7fa5e</a>
</td>
<td>
GitHub
</td>
<td>
2024-06-20
</td>
<td>
Update MERFISH.Rmd
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/a21b9033dd2c224cd5dfe2f99aef2531cf55fd89/analysis/MERFISH.Rmd" target="_blank">a21b903</a>
</td>
<td>
GitHub
</td>
<td>
2024-06-20
</td>
<td>
Update MERFISH.Rmd
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/ee4ee17ff874a26eaf8c1e76e4a85ce3b9de867e/docs/MERFISH.html" target="_blank">ee4ee17</a>
</td>
<td>
GitHub
</td>
<td>
2024-06-19
</td>
<td>
Add files via upload
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/615fc05e05b60aed64683d1204afddc6b8c32f36/analysis/MERFISH.Rmd" target="_blank">615fc05</a>
</td>
<td>
GitHub
</td>
<td>
2024-06-19
</td>
<td>
Add files via upload
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="merfish-data-analysis" class="section level1">
<h1>MERFISH Data Analysis</h1>
</div>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>Spatial transcriptomics is a powerful technique for studying the
spatial organization of gene expression within tissues. Here, we apply
KODAMA to analyze the spatial transcriptomic data that measured the
mouse preoptic region of the hypothalamus using the MERFISH technology
from Moffitt et al., 2018. <a
href="https://www.science.org/doi/10.1126/science.aau5324">Link to
study</a>. We focus on the tissue sections Bregma -0.04, -0.09, -0.14,
-0.19, and -0.24 mm from a consecutive brain hypothalamic region of
animal 1. The original data can be downloaded from <a
href="https://datadryad.org/stash/dataset/doi:10.5061/dryad.8t8s248">Dryad</a>.</p>
</div>
<div id="loading-and-preprocessing-data" class="section level1">
<h1>Loading and Preprocessing Data</h1>
<pre class="r"><code># Load the necessary libraries
library(rgl)
library(irlba)
library(KODAMAextra)
library(scater)
library(SPARK)
library(ggplot2)
library(plotly)
library(mclust)
library(harmony)
library(bluster)
library(igraph)
n.cores=16</code></pre>
<pre class="r"><code># Load the MERFISH data
load(&quot;../MERFISH_Animal1.RData&quot;)

cols &lt;- c(&quot;#669bbc&quot;, &quot;#81b29a&quot;, &quot;#f2cc8f&quot;, &quot;#adc178&quot;,
          &quot;#dde5b6&quot;, &quot;#a8dadc&quot;, &quot;#e5989b&quot;, &quot;#e07a5f&quot;,
          &quot;#aae5b6&quot;, &quot;#a8aadc&quot;, &quot;#e59811&quot;, &quot;#aa7900&quot;)

# Define the slides to be analyzed
slides &lt;- c(&quot;-0.04&quot;, &quot;-0.09&quot;, &quot;-0.14&quot;, &quot;-0.19&quot;, &quot;-0.24&quot;)

# Initialize variables
xyz &lt;- NULL
tissue_segments &lt;- NULL
cell_type &lt;- NULL
RNA &lt;- NULL

# Extract spatial and expression data from each slide
for (i in slides) {
  x &lt;- info_mult[[i]]$x / 1000
  y &lt;- info_mult[[i]]$y / 1000
  z &lt;- as.numeric(i)
  slide_xyz &lt;- cbind(x - min(x), y - min(y), z)
  xyz &lt;- rbind(xyz, slide_xyz)
  tissue_segments &lt;- c(tissue_segments, info_mult[[i]]$z)
  cell_type &lt;- c(cell_type, info_mult[[i]]$Cell_class)
  RNA &lt;- rbind(RNA, t(cnts_mult[[i]]))
}

# Normalize RNA counts
RNA &lt;- t(normalizeCounts(t(RNA), log = TRUE))

# Convert tissue segments to factor with defined levels
tissue_segments &lt;- factor(tissue_segments, levels = c(&quot;V3&quot;, &quot;BST&quot;, &quot;fx&quot;, &quot;MPA&quot;, &quot;MPN&quot;, &quot;PV&quot;, &quot;PVH&quot;, &quot;PVT&quot;))

# Convert xyz to numeric matrix
xyz &lt;- matrix(as.numeric(as.matrix(xyz)), ncol = ncol(xyz))</code></pre>
</div>
<div id="identifying-differentially-expressed-genes"
class="section level1">
<h1>Identifying Differentially Expressed Genes</h1>
<pre class="r"><code># Initialize matrix to store p-values

top=multi_SPARKX(RNA,xyz[,-3],as.factor(xyz[,3]),n.cores = n.cores)</code></pre>
</div>
<div id="passing-message" class="section level1">
<h1>Passing message</h1>
<pre class="r"><code>passing.message = 
  function (data, spatial, number_knn = 15, quantile = 0.5, n.cores = 1) 
  {
    data=as.matrix(data)
    spatial=as.matrix(spatial)
    # Get dimensions of the input data
    nspots = nrow(data)
    nvariables = ncol(data)

    # Initialize result matrix
    data.2 &lt;- matrix(0, nrow = nspots, ncol = nvariables)
    knn=Rnanoflann::nn(spatial,spatial,number_knn)
    for(h in 1:nspots){       
      # Find nearest neighbors using spatial information
      
      
      # Initialize a temporary vector for computations
      temp = rep(0, nvariables)
      RNA.temp = data[knn$indices[h,], ]
      knn_gene = Rnanoflann::nn(RNA.temp, RNA.temp[1, , drop = FALSE], round(number_knn * quantile))$indices

      knn$distances=knn$distances/max(knn$distances)
      # Compute weighted sum for each variable
      for (i in 1:number_knn) {
        temp = temp + RNA.temp[i, ]/exp(knn$distances[h,i])
      }
      # Return the computed row
      data.2[h, ] &lt;-temp
    }

    rownames(data.2) = rownames(data)
    colnames(data.2) = colnames(data)
    data.2
  }

RNA.PM=passing.message(RNA[,top[1:100]],xyz)</code></pre>
<div id="dimensionality-reduction" class="section level2">
<h2>Dimensionality reduction</h2>
<pre class="r"><code>RNA.PM.scaled=scale(RNA.PM)
pca_results &lt;- irlba(A = RNA.PM.scaled, nv = 50)
pca.PM &lt;- pca_results$u %*% diag(pca_results$d)

pca.PM=RunHarmony(pca.PM,data.frame(z=xyz[,3]),&quot;z&quot;)


RNA.scaled=scale(RNA[,top[1:100]])
pca_results &lt;- irlba(A = RNA.scaled, nv = 50)
pca &lt;- pca_results$u %*% diag(pca_results$d)


par(mfrow = c(1, 2))
plot(pca, col = cols[tissue_segments], main = &quot;KODAMA tissue&quot;,pch=20)
plot(pca.PM, col = cols[tissue_segments], main = &quot;KODAMA Clusters&quot;,pch=20)</code></pre>
<p><img src="figure/MERFISH.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/432fc49eabe29b73dbed6a076d99a15956b907b6/docs/figure/MERFISH.Rmd/unnamed-chunk-5-1.png" target="_blank">432fc49</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2025-01-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/3374e66ce76ec3d26fcb912a9f6eba2f7341ed8f/docs/figure/MERFISH.Rmd/unnamed-chunk-5-1.png" target="_blank">3374e66</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-08-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/6f7daaccbcc7b968bc736664a45f9ebcaf73fba3/docs/figure/MERFISH.Rmd/unnamed-chunk-5-1.png" target="_blank">6f7daac</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-07-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/f8ca54aee4428df50d2cd0f41cbf2a0ceae01d7d/docs/figure/MERFISH.Rmd/unnamed-chunk-5-1.png" target="_blank">f8ca54a</a>
</td>
<td>
tkcaccia
</td>
<td>
2024-07-14
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/20a6dacd3051961f479a262b51ceb97224ff1d61/docs/figure/MERFISH.Rmd/unnamed-chunk-5-1.png" target="_blank">20a6dac</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-06-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="applying-kodama" class="section level1">
<h1>Applying KODAMA</h1>
<pre class="r"><code># Apply KODAMA to the PCA results
jj=KODAMA.matrix.parallel(pca.PM[,1:20],
                          spatial = xyz,
                          FUN= &quot;fastpls&quot; ,
                          landmarks = 100000,
                          splitting = 300,
                          ncomp = 50,
                          spatial.resolution = 0.3,
                          n.cores=n.cores,
                          seed = 543210)</code></pre>
<pre><code>Calculating Network

Calculating Network spatial</code></pre>
<pre><code>socket cluster with 16 nodes on host &#39;localhost&#39;
================================================================================
Finished parallel computation

[1] &quot;Calculation of dissimilarity matrix...&quot;
================================================================================</code></pre>
<pre class="r"><code>print(&quot;KODAMA finished&quot;)</code></pre>
<pre><code>[1] &quot;KODAMA finished&quot;</code></pre>
<pre class="r"><code>config=umap.defaults
config$n_neighbors=30
config$n_threads = n.cores
vis &lt;- KODAMA.visualization(jj, method = &quot;UMAP&quot;,config=config)

g &lt;- makeSNNGraph(as.matrix(vis), k = 100)
ncluster=8
res = 0.15
clu= cluster_louvain(g, resolution = res)$membership
lclu=length(unique(clu ))
while (lclu&lt;8) {
  res = res + 0.01
  clu=cluster_louvain(g, resolution = res)$membership
  lclu=length(unique(clu))
}</code></pre>
</div>
<div id="refinement-and-visualization" class="section level1">
<h1>Refinement and Visualization</h1>
<pre class="r"><code>ref=refine_SVM(xyz,clu,cost=1000)</code></pre>
<pre><code>[1] &quot;1&quot;</code></pre>
<pre class="r"><code>zlabels=ref

z=xyz[,3]
uz=unique(z)

result_KODAMA=list()
result_KODAMA$clusters=list()
result_KODAMA$feature_extraction=list()
result_KODAMA$tissue_segments=list()
result_KODAMA$xy=list()

ground_true=list()
ground_true$tissue_segments=list()
ground_true$clusters=list()
ground_true$xy=list()

for(i in 1:5){
  sel=z==uz[i]
  
    
  ground_true$tissue_segments[[i]]=tissue_segments[sel]
  ground_true$clusters[[i]]=tissue_segments[sel]
  ground_true$xy[[i]]=xyz[sel,-3]
  
  result_KODAMA$clusters[[i]]=zlabels[sel]
  result_KODAMA$tissue_segments[[i]]=tissue_segments[sel]
  result_KODAMA$xy[[i]]=xyz[sel,-3]
  result_KODAMA$feature_extraction[[i]]=vis[sel,]
}
result_KODAMA$xyz=xyz
result_KODAMA$clusters_allslide=zlabels

ground_true$xyz=xyz
ground_true$clusters_allslide=tissue_segments






save(ground_true,result_KODAMA,file=&quot;output/MERFISH-KODAMA-results.RData&quot;)
save(tissue_segments,vis,xyz,pca,pca.PM,RNA,RNA.PM,top,file=&quot;data/MERFISH-input.RData&quot;)

ARI=NULL
slides=unique(xyz[,3])
for(i in 1:length(slides)){
  sel=slides[i]==xyz[,3]
  ARI[i]=adjustedRandIndex(ref[sel], tissue_segments[sel])
}
print(ARI)</code></pre>
<pre><code>[1] 0.4758901 0.6160557 0.5421747 0.6638037 0.6626673</code></pre>
<pre class="r"><code># Plot the clusters
par(mfrow = c(1, 2))
plot(vis, col = cols[tissue_segments], main = &quot;KODAMA tissue&quot;,pch=20)
plot(vis, col = cols[clu], main = &quot;KODAMA Clusters&quot;,pch=20)</code></pre>
<p><img src="figure/MERFISH.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/2c6734f3f4c95e91bbd44f6be8cd3e45509ec5e0/docs/figure/MERFISH.Rmd/unnamed-chunk-7-1.png" target="_blank">2c6734f</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-09-05
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/3374e66ce76ec3d26fcb912a9f6eba2f7341ed8f/docs/figure/MERFISH.Rmd/unnamed-chunk-7-1.png" target="_blank">3374e66</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-08-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot_slide(xyz[,-3],as.factor(xyz[,3]),ref,nrow=1,col=cols)</code></pre>
<p><img src="figure/MERFISH.Rmd/unnamed-chunk-7-2.png" width="672" style="display: block; margin: auto;" />
# 3D Visualization for Slice -0.14</p>
<pre class="r"><code>library(rgl)

library(MASS)

library(misc3d)

#save(vis,xyz,ref,cols,file=&quot;MERFISH-test.RData&quot;)


#volume_rendering(xyz[!is.na(ref),],ref[!is.na(ref)],selection = c(&quot;V3&quot;,&quot;PVH&quot;,&quot;PV&quot;,&quot;fx&quot;),alpha = c(0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8) ,colors = cols,cells=c(20, 20, 20),level=exp(2.4))</code></pre>
<pre><code>[1] &quot;1&quot;
[1] 7
[1] 8
[1] 5
[1] 4
[1] 4
[1] 5
[1] 4
[1] 5
[1] 5
[1] 5
[1] 5
[1] 7
[1] 6
[1] 6
[1] 7
[1] 7
[1] 6
[1] 10
[1] 7
[1] 9
[1] 8
[1] &quot;1&quot;
[1] &quot;1&quot;</code></pre>
<pre><code>***************************************
  INPUT INFO:
    - Number of tissue sections: 1 
    - Number of cells/spots: 28317 
    - Number of genes: 155 
    - Potts interaction parameter estimation method: SW 
      - Estimate Potts interaction parameter with SW algorithm
  To list all hyper-parameters, Type listAllHyper(BASS_object)
***************************************
***** Exclude genes with 0 expression *****</code></pre>
<pre><code>Post-processing...
done</code></pre>
<pre><code>[1] 6</code></pre>
<pre><code>[1] 6</code></pre>
<pre><code>[1] 7</code></pre>
<pre><code>[1] 7</code></pre>
<pre><code>[1] 7</code></pre>
<pre><code>[1] 8</code></pre>
<pre><code>fitting ...
  |                                                                              |                                                                      |   0%  |                                                                              |===================================                                   |  50%  |                                                                              |======================================================================| 100%</code></pre>
<pre><code>variable initialize finish! 
predict Y and V! 
diff Energy = 258.460027 
Finish ICM step! 
iter = 2, loglik= -3313712.750000, dloglik=0.998457 
predict Y and V! 
diff Energy = 2.671663 
Finish ICM step! 
iter = 3, loglik= -3218905.500000, dloglik=0.028611 
predict Y and V! 
Finish ICM step! 
iter = 4, loglik= -3185529.250000, dloglik=0.010369 
predict Y and V! 
Finish ICM step! 
iter = 5, loglik= -3168624.750000, dloglik=0.005307 
predict Y and V! 
Finish ICM step! 
iter = 6, loglik= -3155711.750000, dloglik=0.004075 
predict Y and V! 
Finish ICM step! 
iter = 7, loglik= -3145455.250000, dloglik=0.003250 
predict Y and V! 
Finish ICM step! 
iter = 8, loglik= -3137200.500000, dloglik=0.002624 
predict Y and V! 
Finish ICM step! 
iter = 9, loglik= -3130864.000000, dloglik=0.002020 
predict Y and V! 
Finish ICM step! 
iter = 10, loglik= -3126362.000000, dloglik=0.001438 
predict Y and V! 
Finish ICM step! 
iter = 11, loglik= -3123223.000000, dloglik=0.001004 
predict Y and V! 
Finish ICM step! 
iter = 12, loglik= -3120983.250000, dloglik=0.000717 
predict Y and V! 
Finish ICM step! 
iter = 13, loglik= -3119290.000000, dloglik=0.000543 
predict Y and V! 
Finish ICM step! 
iter = 14, loglik= -3117982.250000, dloglik=0.000419 
predict Y and V! 
Finish ICM step! 
iter = 15, loglik= -3116926.750000, dloglik=0.000339 
predict Y and V! 
Finish ICM step! 
iter = 16, loglik= -3116056.750000, dloglik=0.000279 
predict Y and V! 
Finish ICM step! 
iter = 17, loglik= -3115299.500000, dloglik=0.000243 
predict Y and V! 
Finish ICM step! 
iter = 18, loglik= -3114635.500000, dloglik=0.000213 
predict Y and V! 
Finish ICM step! 
iter = 19, loglik= -3114024.750000, dloglik=0.000196 
predict Y and V! 
Finish ICM step! 
iter = 20, loglik= -3113462.750000, dloglik=0.000180 
predict Y and V! 
diff Energy = 0.501921 
Finish ICM step! 
iter = 21, loglik= -3112947.500000, dloglik=0.000165 
predict Y and V! 
Finish ICM step! 
iter = 22, loglik= -3112459.500000, dloglik=0.000157 
predict Y and V! 
diff Energy = 0.585441 
Finish ICM step! 
iter = 23, loglik= -3111979.000000, dloglik=0.000154 
predict Y and V! 
Finish ICM step! 
iter = 24, loglik= -3111522.000000, dloglik=0.000147 
predict Y and V! 
Finish ICM step! 
iter = 25, loglik= -3111093.750000, dloglik=0.000138 
predict Y and V! 
Finish ICM step! 
iter = 26, loglik= -3110639.250000, dloglik=0.000146 
predict Y and V! 
Finish ICM step! 
iter = 27, loglik= -3110208.000000, dloglik=0.000139 
predict Y and V! 
Finish ICM step! 
iter = 28, loglik= -3109780.250000, dloglik=0.000138 
predict Y and V! 
Finish ICM step! 
iter = 29, loglik= -3109331.750000, dloglik=0.000144 
predict Y and V! 
Finish ICM step! 
iter = 30, loglik= -3108864.750000, dloglik=0.000150 </code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.2 (2024-10-31)
Platform: x86_64-pc-linux-gnu
Running under: Ubuntu 20.04.6 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0 
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

time zone: Etc/UTC
tzcode source: system (glibc)

attached base packages:
[1] stats4    parallel  stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] BayesSpace_1.14.0           Seurat_5.1.0               
 [3] SeuratObject_5.0.2          sp_2.1-4                   
 [5] PRECAST_1.6.5               ProFAST_1.4                
 [7] gtools_3.9.5                SpatialExperiment_1.14.0   
 [9] Banksy_1.0.0                BASS_1.1.0.017             
[11] GIGrvg_0.8                  misc3d_0.9-1               
[13] MASS_7.3-61                 igraph_2.0.3               
[15] bluster_1.14.0              harmony_1.2.3              
[17] Rcpp_1.0.13-1               mclust_6.1.1               
[19] plotly_4.10.4               SPARK_1.1.1                
[21] scater_1.32.1               ggplot2_3.5.1              
[23] scuttle_1.14.0              SingleCellExperiment_1.26.0
[25] SummarizedExperiment_1.34.0 Biobase_2.64.0             
[27] GenomicRanges_1.56.2        GenomeInfoDb_1.40.1        
[29] IRanges_2.38.1              S4Vectors_0.42.1           
[31] BiocGenerics_0.50.0         MatrixGenerics_1.16.0      
[33] matrixStats_1.4.1           KODAMAextra_1.2            
[35] e1071_1.7-16                doParallel_1.0.17          
[37] iterators_1.0.14            foreach_1.5.2              
[39] KODAMA_3.0                  umap_0.2.10.0              
[41] Rtsne_0.17                  minerva_1.5.10             
[43] irlba_2.3.5.1               Matrix_1.7-1               
[45] rgl_1.3.16                  workflowr_1.7.1            

loaded via a namespace (and not attached):
  [1] progress_1.2.3            DirichletReg_0.7-1       
  [3] goftest_1.2-3             Biostrings_2.72.1        
  [5] vctrs_0.6.5               spatstat.random_3.3-2    
  [7] digest_0.6.37             png_0.1-8                
  [9] proxy_0.4-27              git2r_0.33.0             
 [11] ggrepel_0.9.6             org.Mm.eg.db_3.19.1      
 [13] deldir_2.0-4              parallelly_1.41.0        
 [15] combinat_0.0-8            Rnanoflann_0.0.3         
 [17] magick_2.8.4              reshape2_1.4.4           
 [19] GiRaF_1.0.1               httpuv_1.6.15            
 [21] withr_3.0.1               xfun_0.49                
 [23] ggpubr_0.6.0              survival_3.8-3           
 [25] memoise_2.0.1             ggbeeswarm_0.7.2         
 [27] zoo_1.8-12                pbapply_1.7-2            
 [29] Formula_1.2-5             prettyunits_1.2.0        
 [31] KEGGREST_1.44.1           promises_1.3.2           
 [33] httr_1.4.7                rstatix_0.7.2            
 [35] rhdf5filters_1.16.0       globals_0.16.3           
 [37] fitdistrplus_1.2-2        rhdf5_2.48.0             
 [39] ps_1.8.1                  rstudioapi_0.17.1        
 [41] UCSC.utils_1.0.0          miniUI_0.1.1.1           
 [43] generics_0.1.3            base64enc_0.1-3          
 [45] processx_3.8.4            curl_6.0.1               
 [47] zlibbioc_1.50.0           ScaledMatrix_1.12.0      
 [49] polyclip_1.10-7           doSNOW_1.0.20            
 [51] GenomeInfoDbData_1.2.12   SparseArray_1.4.8        
 [53] xtable_1.8-4              stringr_1.5.1            
 [55] pracma_2.4.4              evaluate_1.0.1           
 [57] S4Arrays_1.4.1            BiocFileCache_2.12.0     
 [59] hms_1.1.3                 colorspace_2.1-1         
 [61] filelock_1.0.3            ROCR_1.0-11              
 [63] reticulate_1.38.0         spatstat.data_3.1-4      
 [65] magrittr_2.0.3            lmtest_0.9-40            
 [67] later_1.4.0               viridis_0.6.5            
 [69] lattice_0.22-6            spatstat.geom_3.3-4      
 [71] future.apply_1.11.3       getPass_0.2-4            
 [73] scattermore_1.2           cowplot_1.1.3            
 [75] RcppAnnoy_0.0.22          class_7.3-23             
 [77] pillar_1.9.0              nlme_3.1-166             
 [79] compiler_4.4.2            beachmat_2.20.0          
 [81] RSpectra_0.16-2           stringi_1.8.4            
 [83] tensor_1.5                plyr_1.8.9               
 [85] crayon_1.5.3              abind_1.4-8              
 [87] locfit_1.5-9.10           org.Hs.eg.db_3.19.1      
 [89] bit_4.5.0.1               sandwich_3.1-1           
 [91] dplyr_1.1.4               whisker_0.4.1            
 [93] codetools_0.2-20          BiocSingular_1.20.0      
 [95] openssl_2.2.2             bslib_0.8.0              
 [97] DR.SC_3.4                 mime_0.12                
 [99] splines_4.4.2             fastDummies_1.7.4        
[101] dbplyr_2.5.0              sparseMatrixStats_1.16.0 
[103] maxLik_1.5-2.1            knitr_1.49               
[105] blob_1.2.4                utf8_1.2.4               
[107] fs_1.6.5                  listenv_0.9.1            
[109] label.switching_1.8       DelayedMatrixStats_1.26.0
[111] ggsignif_0.6.4            RcppHungarian_0.3        
[113] tibble_3.2.1              statmod_1.5.0            
[115] callr_3.7.6               lpSolve_5.6.23           
[117] pkgconfig_2.0.3           tools_4.4.2              
[119] cachem_1.1.0              aricode_1.0.3            
[121] RhpcBLASctl_0.23-42       RSQLite_2.3.9            
[123] viridisLite_0.4.2         DBI_1.2.3                
[125] fastmap_1.2.0             rmarkdown_2.29           
[127] scales_1.3.0              grid_4.4.2               
[129] ica_1.0-3                 broom_1.0.7              
[131] sass_0.4.9                coda_0.19-4.1            
[133] patchwork_1.3.0           dotCall64_1.2            
[135] carData_3.0-5             RANN_2.6.2               
[137] snow_0.4-4                farver_2.1.2             
[139] yaml_2.3.10               ggthemes_5.1.0           
[141] cli_3.6.3                 purrr_1.0.2              
[143] leiden_0.4.3.1            lifecycle_1.0.4          
[145] dbscan_1.2-0              askpass_1.2.1            
[147] uwot_0.2.2                backports_1.5.0          
[149] BiocParallel_1.38.0       gtable_0.3.5             
[151] rjson_0.2.23              ggridges_0.5.6           
[153] progressr_0.14.0          limma_3.60.3             
[155] edgeR_4.2.2               jsonlite_1.8.9           
[157] bitops_1.0-9              miscTools_0.6-28         
[159] RcppHNSW_0.6.0            xgboost_1.7.8.1          
[161] assertthat_0.2.1          bit64_4.5.2              
[163] spatstat.utils_3.1-2      BiocNeighbors_1.22.0     
[165] matlab_1.0.4.1            metapod_1.12.0           
[167] jquerylib_0.1.4           dqrng_0.4.1              
[169] spatstat.univar_3.1-1     lazyeval_0.2.2           
[171] shiny_1.10.0              htmltools_0.5.8.1        
[173] sctransform_0.4.1         rappdirs_0.3.3           
[175] glue_1.8.0                tcltk_4.4.2              
[177] spam_2.11-0               httr2_1.0.7              
[179] XVector_0.44.0            RCurl_1.98-1.16          
[181] rprojroot_2.0.4           scran_1.32.0             
[183] gridExtra_2.3             sccore_1.0.5             
[185] R6_2.5.1                  tidyr_1.3.1              
[187] CompQuadForm_1.4.3        labeling_0.4.3           
[189] cluster_2.1.8             Rhdf5lib_1.26.0          
[191] DelayedArray_0.30.1       tidyselect_1.2.1         
[193] vipor_0.4.7               xml2_1.3.6               
[195] car_3.1-3                 AnnotationDbi_1.66.0     
[197] future_1.34.0             leidenAlg_1.1.3          
[199] rsvd_1.0.5                munsell_0.5.1            
[201] KernSmooth_2.23-26        furrr_0.3.1              
[203] data.table_1.15.4         htmlwidgets_1.6.4        
[205] RColorBrewer_1.1-3        biomaRt_2.60.1           
[207] rlang_1.1.4               spatstat.sparse_3.1-0    
[209] spatstat.explore_3.3-3    fansi_1.0.6              
[211] beeswarm_0.4.0           </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
