<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>About</title>

<script src="site_libs/header-attrs-2.27/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">KODAMA-Analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="Installation.html">Installation</a>
</li>
<li>
  <a href="single-cell.html">Single-cell</a>
</li>
<li>
  <a href="simulation.html">Simulation</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Real data analyses
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="DLPFC-4.html">DLPFC (4 slides)</a>
    </li>
    <li>
      <a href="DLPFC-12.html">DLPFC (12 slides)</a>
    </li>
    <li>
      <a href="Prostate.html">Prostate</a>
    </li>
    <li>
      <a href="STARmap.html">STARmap (Wang et al., 2018)</a>
    </li>
    <li>
      <a href="MERFISH.html">MERFISH (Moffitt et al., 2018)</a>
    </li>
    <li>
      <a href="VisiumHD.html">Visium HD</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Wrapper
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Seurat.html">Seurat</a>
    </li>
    <li>
      <a href="Giotto.html">Giotto</a>
    </li>
  </ul>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/tkcaccia/KODAMA-Analysis">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">About</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-07-14
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>KODAMA-Analysis/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240618code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240618)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240618code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240618)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomtkcacciaKODAMAAnalysistree0051854b6093e26617fe534e1deab4ef02cb96d1targetblank0051854a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/tree/0051854b6093e26617fe534e1deab4ef02cb96d1" target="_blank">0051854</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomtkcacciaKODAMAAnalysistree0051854b6093e26617fe534e1deab4ef02cb96d1targetblank0051854a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/tkcaccia/KODAMA-Analysis/tree/0051854b6093e26617fe534e1deab4ef02cb96d1" target="_blank">0051854</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory

Untracked files:
    Untracked:  .h5seurat
    Untracked:  analysis/Single-cell.Rmd
    Untracked:  code/VisiumHD_CRC_download.sh
    Untracked:  data/Pathology.csv

Unstaged changes:
    Deleted:    _site.yml
    Deleted:    analysis/DLPFC.Rmd
    Deleted:    analysis/about.Rmd
    Deleted:    data/Moffitt_and_Bambah-Mukku_et_al_merfish_all_cells.csv

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/MERFISH.Rmd</code>) and HTML
(<code>docs/MERFISH.html</code>) files. If you’ve configured a remote
Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/0051854b6093e26617fe534e1deab4ef02cb96d1/analysis/MERFISH.Rmd" target="_blank">0051854</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-07-14
</td>
<td>
Start my new project
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/0638f9320233629c838d11f98cc4c9ef4b9137ad/analysis/MERFISH.Rmd" target="_blank">0638f93</a>
</td>
<td>
GitHub
</td>
<td>
2024-07-08
</td>
<td>
Update MERFISH.Rmd
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/d3de9e1bf63e1e35f3d967eaa80f3c8735ee32bf/docs/MERFISH.html" target="_blank">d3de9e1</a>
</td>
<td>
GitHub
</td>
<td>
2024-07-08
</td>
<td>
Update MERFISH.html
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/32f206ec6291cf1653fd45eb12008b97207ec96c/docs/MERFISH.html" target="_blank">32f206e</a>
</td>
<td>
GitHub
</td>
<td>
2024-07-08
</td>
<td>
Update MERFISH.html
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/940ec1385b5fac0e259842572da1962def1b36f1/docs/MERFISH.html" target="_blank">940ec13</a>
</td>
<td>
GitHub
</td>
<td>
2024-07-04
</td>
<td>
Update MERFISH.html
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/9ee53df126bf1758843f7581538a33d39bd97884/analysis/MERFISH.Rmd" target="_blank">9ee53df</a>
</td>
<td>
GitHub
</td>
<td>
2024-06-26
</td>
<td>
Update MERFISH.Rmd
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/35e8f25827e79ca29d8b44970ada451bee48482e/docs/MERFISH.html" target="_blank">35e8f25</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-06-26
</td>
<td>
Update docs with MERFISH.html
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/c3eee75f5d792665d77cce1691993c8025604892/analysis/MERFISH.Rmd" target="_blank">c3eee75</a>
</td>
<td>
GitHub
</td>
<td>
2024-06-26
</td>
<td>
Update MERFISH.Rmd
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/e7ddec66f16b8c7bfac6f4175eb87249e0fd2c28/analysis/MERFISH.Rmd" target="_blank">e7ddec6</a>
</td>
<td>
GitHub
</td>
<td>
2024-06-26
</td>
<td>
Update MERFISH.Rmd
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/eb29180b902b2329560807513555cbfa36849d1e/analysis/MERFISH.Rmd" target="_blank">eb29180</a>
</td>
<td>
GitHub
</td>
<td>
2024-06-26
</td>
<td>
Update MERFISH.Rmd
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/bd1988e34ddc2bb2ef6e4826bd15a1342e7d2f00/analysis/MERFISH.Rmd" target="_blank">bd1988e</a>
</td>
<td>
GitHub
</td>
<td>
2024-06-26
</td>
<td>
Update MERFISH.Rmd
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/911564c3294bcefc453a7097a348901b86544728/analysis/MERFISH.Rmd" target="_blank">911564c</a>
</td>
<td>
GitHub
</td>
<td>
2024-06-26
</td>
<td>
Update MERFISH.Rmd
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/246f0bdb380bae90145007978345f50c5020d4ee/analysis/MERFISH.Rmd" target="_blank">246f0bd</a>
</td>
<td>
GitHub
</td>
<td>
2024-06-25
</td>
<td>
Update MERFISH.Rmd
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/20a6dacd3051961f479a262b51ceb97224ff1d61/docs/MERFISH.html" target="_blank">20a6dac</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-06-25
</td>
<td>
Update docs with MERFISH.html
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/e56c4ff46a93e0e36aa1b2ca686bc55e06f1bdff/analysis/MERFISH.Rmd" target="_blank">e56c4ff</a>
</td>
<td>
GitHub
</td>
<td>
2024-06-20
</td>
<td>
Update MERFISH.Rmd
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/9df203d58ecad69cfd1e59c1dd8e1c571546628d/analysis/MERFISH.Rmd" target="_blank">9df203d</a>
</td>
<td>
GitHub
</td>
<td>
2024-06-20
</td>
<td>
Update MERFISH.Rmd
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/be3cdeb69c835b7428b656238860fc3a01504afc/analysis/MERFISH.Rmd" target="_blank">be3cdeb</a>
</td>
<td>
GitHub
</td>
<td>
2024-06-20
</td>
<td>
Update MERFISH.Rmd
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/2e7fa5ea49a52d62b390d184c49f12113aa2f189/analysis/MERFISH.Rmd" target="_blank">2e7fa5e</a>
</td>
<td>
GitHub
</td>
<td>
2024-06-20
</td>
<td>
Update MERFISH.Rmd
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/a21b9033dd2c224cd5dfe2f99aef2531cf55fd89/analysis/MERFISH.Rmd" target="_blank">a21b903</a>
</td>
<td>
GitHub
</td>
<td>
2024-06-20
</td>
<td>
Update MERFISH.Rmd
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/ee4ee17ff874a26eaf8c1e76e4a85ce3b9de867e/docs/MERFISH.html" target="_blank">ee4ee17</a>
</td>
<td>
GitHub
</td>
<td>
2024-06-19
</td>
<td>
Add files via upload
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/615fc05e05b60aed64683d1204afddc6b8c32f36/analysis/MERFISH.Rmd" target="_blank">615fc05</a>
</td>
<td>
GitHub
</td>
<td>
2024-06-19
</td>
<td>
Add files via upload
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Describe your project.</p>
<div id="merfish-data-analysis" class="section level1">
<h1>MERFISH Data Analysis</h1>
<p>The molecular imaging technology MERFISH (Multiplexed Error-Robust
Fluorescence In Situ Hybridization) has revolutionized RNA mapping at
the cellular level with unmatched spatial precision. This technique
enables the simultaneous visualization of numerous genes within
individual cells, offering unique insights into molecular processes in
biological tissues. MERFISH data provides detailed information on cell
structure and function, paving the way for new discoveries in cellular
and molecular biology.</p>
<p>The dataset analyzed, as presented in <a
href="https://datadryad.org/stash/dataset/doi:10.5061/dryad.8t8s248">this
article</a>, includes information on 36 mice from the hypothalamic
preoptic region of the brain, derived from samples taken from various
regions of the same animal. The “Bregma” column indicates the slice
location in Bregma coordinates, an anatomical reference point used in
neuroscience.</p>
<p>Our focus is on advanced analysis of MERFISH data from biological
tissues, introducing key advancements such as a 12 Extended Layer
Analysis method for detailed tissue structure understanding and a 3D
clustering feature for accurately representing the spatial relationships
and RNA interactions within cells.</p>
<p>Our main objective is to integrate multiple tissue sections for
unified analysis across different samples. We employ data preprocessing,
dimension reduction, and advanced clustering techniques to explore the
relationship between tissue structure and cellular behavior,
highlighting spatial variations and gene expression patterns associated
with specific biological functions.</p>
</div>
<div id="load-necessary-libraries" class="section level1">
<h1>Load necessary libraries</h1>
<pre class="r"><code>library(Seurat)
library(KODAMAextra)
library(ggplot2)
library(NMI)
library(mclust)
library(bluster)
library(igraph)</code></pre>
</div>
<div id="downloading-the-data" class="section level1">
<h1>Downloading the Data</h1>
<p>The data is downloaded from this <a
href="https://datadryad.org/stash/dataset/doi:10.5061/dryad.8t8s248">link</a>
for the continuation of the work.</p>
<pre class="r"><code># Load data
data &lt;- read.csv(&quot;../Merfish/hypothalamic_preoptic_region/Moffitt_and_Bambah-Mukku_et_al_merfish_all_cells.csv&quot;)
data &lt;- as.data.frame(data)</code></pre>
</div>
<div id="data-pre-processing" class="section level1">
<h1>Data Pre-processing</h1>
<p>In our data preprocessing process, we have undertaken a rigorous
approach to ensure the quality and relevance of the information we use
in our analysis. With a total of 170 genes initially included in our
data, our first step was to eliminate the “Blank” genes, which are
control or background genes that do not contribute to our main analysis.
This elimination allowed us to reduce our data set to 165 genes.
Subsequently, we identified and removed the “Fos” gene, suspected of
having an unwanted or disruptive impact on our data, leaving a total of
164 genes for our analysis. These initial cleaning steps are crucial to
ensure that the data we use is reliable and suitable for our analysis
objectives. By then filtering the data to retain only those
corresponding to Animal_ID 1 and specific Bregma values, we ensure that
we focus on a subset relevant to our study. This preprocessing process
is essential to prepare our data properly, by eliminating unnecessary
elements and focusing on the most relevant data for our subsequent
analysis.</p>
<pre class="r"><code># Data cleaning
colnames_data &lt;- colnames(data)
blankgenes &lt;- grep(&quot;Blank&quot;, colnames_data)
fos_index &lt;- which(colnames(data) == &quot;Fos&quot;)
df &lt;- data[, -c(fos_index, blankgenes)]
selected_bregma &lt;- unique(df$Bregma)

# Filter data for Animal_ID 1 and specific Bregma values
exp &lt;- subset(df, Animal_ID == 1 &amp; Bregma %in% selected_bregma)
rownames(exp) &lt;- exp$Cell_ID
data_list &lt;- split(exp, exp$Bregma)
data_list &lt;- data_list[match(selected_bregma, names(data_list))]


# Initialize lists 
xy &lt;- list()
pca &lt;- list()
v &lt;- list()
kk &lt;- list()
vis &lt;- list()
pred &lt;- list()
refine &lt;- list()
cons &lt;- list()
ARI &lt;- list()
NMI &lt;- list()


# Normalization and dimension reduction by PCA
for (i in names(data_list)) {
  print(i)
  x &lt;- data_list[[i]]$Centroid_X - min(data_list[[i]]$Centroid_X)
  y &lt;- data_list[[i]]$Centroid_Y - min(data_list[[i]]$Centroid_Y)
  xy[[i]] &lt;- cbind(x, y)
  rownames(xy[[i]]) &lt;- rownames(data_list[[i]])
  cons[[i]] &lt;- t(data_list[[i]][, 10:ncol(data_list[[i]])])
  colnames(cons[[i]]) &lt;- rownames(data_list[[i]])
  v[[i]] &lt;- t(LogNormalize(cons[[i]]))

#dimensionality reduction BY Principal Component Analysis (PCA)

  pca[[i]] &lt;- prcomp(v[[i]], scale. = TRUE)$x[, 1:50]

#  KODAMA clustering
    kk[[i]] &lt;- KODAMA.matrix.parallel(pca[[i]], spatial = xy[[i]], FUN = &quot;PLS&quot;, landmarks = 10000, n.cores = 4)
    vis[[i]] &lt;- KODAMA.visualization(kk[[i]], method = &quot;UMAP&quot;)
    names(vis[[i]]) &lt;- names(data_list[[i]])
    
    # Graph-based clustering
    g &lt;- bluster::makeSNNGraph(as.matrix(vis[[i]]), k = 100)
    g_walk &lt;- igraph::cluster_walktrap(g)
    pred[[i]] &lt;- as.character(igraph::cut_at(g_walk, no = 8))
    refine[[i]] &lt;- refinecluster(pred[[i]], xy[[i]], shape = &quot;hexagon&quot;)}</code></pre>
<pre><code>[1] &quot;0.26&quot;
socket cluster with 4 nodes on host &#39;localhost&#39;
================================================================================[1] &quot;Finished parallel computation&quot;

[1] &quot;Calculation of dissimilarity matrix...&quot;
================================================================================
[1] &quot;0.21&quot;
socket cluster with 4 nodes on host &#39;localhost&#39;
================================================================================[1] &quot;Finished parallel computation&quot;

[1] &quot;Calculation of dissimilarity matrix...&quot;
================================================================================
[1] &quot;0.16&quot;
socket cluster with 4 nodes on host &#39;localhost&#39;
================================================================================[1] &quot;Finished parallel computation&quot;

[1] &quot;Calculation of dissimilarity matrix...&quot;
================================================================================
[1] &quot;0.11&quot;
socket cluster with 4 nodes on host &#39;localhost&#39;
================================================================================[1] &quot;Finished parallel computation&quot;

[1] &quot;Calculation of dissimilarity matrix...&quot;
================================================================================
[1] &quot;0.06&quot;
socket cluster with 4 nodes on host &#39;localhost&#39;
================================================================================[1] &quot;Finished parallel computation&quot;

[1] &quot;Calculation of dissimilarity matrix...&quot;
================================================================================
[1] &quot;0.01&quot;
socket cluster with 4 nodes on host &#39;localhost&#39;
================================================================================[1] &quot;Finished parallel computation&quot;

[1] &quot;Calculation of dissimilarity matrix...&quot;
================================================================================
[1] &quot;-0.04&quot;
socket cluster with 4 nodes on host &#39;localhost&#39;
================================================================================[1] &quot;Finished parallel computation&quot;

[1] &quot;Calculation of dissimilarity matrix...&quot;
================================================================================
[1] &quot;-0.09&quot;
socket cluster with 4 nodes on host &#39;localhost&#39;
================================================================================[1] &quot;Finished parallel computation&quot;

[1] &quot;Calculation of dissimilarity matrix...&quot;
================================================================================
[1] &quot;-0.14&quot;
socket cluster with 4 nodes on host &#39;localhost&#39;
================================================================================[1] &quot;Finished parallel computation&quot;

[1] &quot;Calculation of dissimilarity matrix...&quot;
================================================================================
[1] &quot;-0.19&quot;
socket cluster with 4 nodes on host &#39;localhost&#39;
================================================================================[1] &quot;Finished parallel computation&quot;

[1] &quot;Calculation of dissimilarity matrix...&quot;
================================================================================
[1] &quot;-0.24&quot;
socket cluster with 4 nodes on host &#39;localhost&#39;
================================================================================[1] &quot;Finished parallel computation&quot;

[1] &quot;Calculation of dissimilarity matrix...&quot;
================================================================================
[1] &quot;-0.29&quot;
socket cluster with 4 nodes on host &#39;localhost&#39;
================================================================================[1] &quot;Finished parallel computation&quot;

[1] &quot;Calculation of dissimilarity matrix...&quot;
================================================================================</code></pre>
<p>#Visualization For better visualization of the results, it is
essential to import the <code>vis.R</code> file at this stage.</p>
<pre class="r"><code># Importing the vis.R code
source(&quot;data/vis.R&quot;)</code></pre>
<p>Visualizing the results of the Kodama clustering, a dimensionality
reduction method that enhances data understanding.</p>
<pre class="r"><code># Define colors for visualizations
cols &lt;- c(&quot;#669bbc&quot;, &quot;#81b29a&quot;, &quot;#f2cc8f&quot;, &quot;#adc178&quot;,
          &quot;#dde5b6&quot;, &quot;#a8dadc&quot;, &quot;#e5989b&quot;, &quot;#e07a5f&quot;)

# Visualize clusters
plotClustersFacet(xy, pred, selected_bregma, size = 0.2) +
  scale_color_manual(&quot;Domain&quot;, values = cols) +
  guides(color = guide_legend(nrow = 1, override.aes = list(size = 2)))</code></pre>
<p><img src="figure/MERFISH.Rmd/unnamed-chunk-5-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/20a6dacd3051961f479a262b51ceb97224ff1d61/docs/figure/MERFISH.Rmd/unnamed-chunk-5-1.png" target="_blank">20a6dac</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-06-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plotClustersFacet(xy, refine, selected_bregma, size = 1) +
  scale_color_manual(&quot;Domain&quot;, values = cols) +
  guides(color = guide_legend(nrow = 1, override.aes = list(size = 2)))</code></pre>
<p><img src="figure/MERFISH.Rmd/unnamed-chunk-5-2.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-2">
Past versions of unnamed-chunk-5-2.png
</button>
</p>
<div id="fig-unnamed-chunk-5-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/20a6dacd3051961f479a262b51ceb97224ff1d61/docs/figure/MERFISH.Rmd/unnamed-chunk-5-2.png" target="_blank">20a6dac</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-06-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>This script illustrates a standard approach to Merfish data analysis,
showcasing the use of commonly used bioinformatics tools. It provides a
comprehensive overview of the Merfish data analysis and visualization
process, crucial for understanding the spatial distribution of RNA in
cells.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.0 (2024-04-24)
Platform: x86_64-pc-linux-gnu
Running under: Ubuntu 20.04.6 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0 
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

time zone: Etc/UTC
tzcode source: system (glibc)

attached base packages:
[1] parallel  stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] igraph_2.0.3       bluster_1.14.0     mclust_6.1.1       NMI_2.0           
 [5] ggplot2_3.5.1      KODAMAextra_1.0    e1071_1.7-14       doParallel_1.0.17 
 [9] iterators_1.0.14   foreach_1.5.2      KODAMA_3.1         umap_0.2.10.0     
[13] Rtsne_0.17         minerva_1.5.10     Seurat_5.1.0       SeuratObject_5.0.2
[17] sp_2.1-4           workflowr_1.7.1   

loaded via a namespace (and not attached):
  [1] RColorBrewer_1.1-3     rstudioapi_0.16.0      jsonlite_1.8.8        
  [4] magrittr_2.0.3         spatstat.utils_3.0-5   farver_2.1.2          
  [7] rmarkdown_2.27         fs_1.6.4               vctrs_0.6.5           
 [10] ROCR_1.0-11            spatstat.explore_3.2-7 askpass_1.2.0         
 [13] htmltools_0.5.8.1      BiocNeighbors_1.22.0   sass_0.4.9            
 [16] sctransform_0.4.1      parallelly_1.37.1      KernSmooth_2.23-24    
 [19] bslib_0.7.0            htmlwidgets_1.6.4      ica_1.0-3             
 [22] plyr_1.8.9             plotly_4.10.4          zoo_1.8-12            
 [25] cachem_1.1.0           whisker_0.4.1          mime_0.12             
 [28] lifecycle_1.0.4        pkgconfig_2.0.3        Matrix_1.7-0          
 [31] R6_2.5.1               fastmap_1.2.0          fitdistrplus_1.1-11   
 [34] future_1.33.2          shiny_1.8.1.1          digest_0.6.36         
 [37] colorspace_2.1-0       S4Vectors_0.42.1       patchwork_1.2.0       
 [40] ps_1.7.7               rprojroot_2.0.4        tensor_1.5            
 [43] RSpectra_0.16-1        irlba_2.3.5.1          labeling_0.4.3        
 [46] progressr_0.14.0       fansi_1.0.6            spatstat.sparse_3.1-0 
 [49] httr_1.4.7             polyclip_1.10-6        abind_1.4-5           
 [52] compiler_4.4.0         proxy_0.4-27           withr_3.0.0           
 [55] BiocParallel_1.38.0    fastDummies_1.7.3      highr_0.11            
 [58] MASS_7.3-61            openssl_2.2.0          tools_4.4.0           
 [61] lmtest_0.9-40          httpuv_1.6.15          future.apply_1.11.2   
 [64] goftest_1.2-3          glue_1.7.0             callr_3.7.6           
 [67] nlme_3.1-165           promises_1.3.0         grid_4.4.0            
 [70] getPass_0.2-4          cluster_2.1.6          reshape2_1.4.4        
 [73] snow_0.4-4             generics_0.1.3         gtable_0.3.5          
 [76] spatstat.data_3.1-2    class_7.3-22           tidyr_1.3.1           
 [79] data.table_1.15.4      utf8_1.2.4             BiocGenerics_0.50.0   
 [82] spatstat.geom_3.2-9    RcppAnnoy_0.0.22       ggrepel_0.9.5         
 [85] RANN_2.6.1             pillar_1.9.0           stringr_1.5.1         
 [88] spam_2.10-0            RcppHNSW_0.6.0         later_1.3.2           
 [91] splines_4.4.0          dplyr_1.1.4            lattice_0.22-6        
 [94] survival_3.7-0         deldir_2.0-4           tidyselect_1.2.1      
 [97] miniUI_0.1.1.1         pbapply_1.7-2          knitr_1.48            
[100] git2r_0.33.0           gridExtra_2.3          scattermore_1.2       
[103] stats4_4.4.0           xfun_0.45              matrixStats_1.3.0     
[106] stringi_1.8.4          lazyeval_0.2.2         yaml_2.3.9            
[109] evaluate_0.24.0        codetools_0.2-20       tibble_3.2.1          
[112] cli_3.6.3              uwot_0.2.2             xtable_1.8-4          
[115] reticulate_1.38.0      munsell_0.5.1          processx_3.8.4        
[118] jquerylib_0.1.4        Rcpp_1.0.12            doSNOW_1.0.20         
[121] globals_0.16.3         spatstat.random_3.2-3  png_0.1-8             
[124] dotCall64_1.1-1        listenv_0.9.1          viridisLite_0.4.2     
[127] scales_1.3.0           ggridges_0.5.6         leiden_0.4.3.1        
[130] purrr_1.0.2            rlang_1.1.4            cowplot_1.1.3         </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
