<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>About</title>

<script src="site_libs/header-attrs-2.27/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">KODAMA-Analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="Installation.html">Installation</a>
</li>
<li>
  <a href="simulation.html">Simulation</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Real data analyses
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="DLPFC.html">DLPFC (Maynard et al., 2021)</a>
    </li>
    <li>
      <a href="Prostate.html">Prostate</a>
    </li>
    <li>
      <a href="STARmap.html">STARmap (Wang et al., 2018)</a>
    </li>
    <li>
      <a href="MERFISH.html">MERFISH (Moffitt et al., 2018)</a>
    </li>
    <li>
      <a href="VisiumHD.html">Visium HD</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Interoperability
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Seurat.html">Seurat</a>
    </li>
    <li>
      <a href="Giotto.html">Giotto</a>
    </li>
  </ul>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/tkcaccia/KODAMA-Analysis">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">About</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-07-12
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>KODAMA-Analysis/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, youâ€™ll want to first commit
it to the Git repo. If youâ€™re still working on the analysis, you can
ignore this warning. When youâ€™re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity itâ€™s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240618code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240618)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240618code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240618)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomtkcacciaKODAMAAnalysistree930428955e4ede1b1653c2567ca98f3f57edd33ctargetblank9304289a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/tree/930428955e4ede1b1653c2567ca98f3f57edd33c" target="_blank">9304289</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomtkcacciaKODAMAAnalysistree930428955e4ede1b1653c2567ca98f3f57edd33ctargetblank9304289a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/tkcaccia/KODAMA-Analysis/tree/930428955e4ede1b1653c2567ca98f3f57edd33c" target="_blank">9304289</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Untracked files:
    Untracked:  analysis/figure/

Unstaged changes:
    Modified:   analysis/MERFISH.Rmd
    Modified:   analysis/Prostate.Rmd

</code></pre>
<p>
Note that any generated files, e.g.Â HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/Prostate.Rmd</code>) and HTML
(<code>docs/Prostate.html</code>) files. If youâ€™ve configured a remote
Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/9b0bdad766b8342e6849ff967399777b230da94f/docs/Prostate.html" target="_blank">9b0bdad</a>
</td>
<td>
GitHub
</td>
<td>
2024-07-12
</td>
<td>
Update Prostate.html
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/4c71aaa3926d6cab26f2d762535adc7f23a782c5/docs/Prostate.html" target="_blank">4c71aaa</a>
</td>
<td>
GitHub
</td>
<td>
2024-07-12
</td>
<td>
Update Prostate.html
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/4a7b4cae32e4d1297c8ad2013dab64f0d7a78a6e/analysis/Prostate.Rmd" target="_blank">4a7b4ca</a>
</td>
<td>
GitHub
</td>
<td>
2024-07-12
</td>
<td>
Update Prostate.Rmd
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/d652bc631cf3de79434fe1b987c2ef65f2a80f38/docs/Prostate.html" target="_blank">d652bc6</a>
</td>
<td>
GitHub
</td>
<td>
2024-07-08
</td>
<td>
Update Prostate.html
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/cac859cee47ee0b941d957bd9ed75959e15bc24a/docs/Prostate.html" target="_blank">cac859c</a>
</td>
<td>
GitHub
</td>
<td>
2024-07-04
</td>
<td>
Update Prostate.html
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/ee4ee17ff874a26eaf8c1e76e4a85ce3b9de867e/docs/Prostate.html" target="_blank">ee4ee17</a>
</td>
<td>
GitHub
</td>
<td>
2024-06-19
</td>
<td>
Add files via upload
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/615fc05e05b60aed64683d1204afddc6b8c32f36/analysis/Prostate.Rmd" target="_blank">615fc05</a>
</td>
<td>
GitHub
</td>
<td>
2024-06-19
</td>
<td>
Add files via upload
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level3">
<h3>Introduction</h3>
<p>The data used in this analysis come from the Visium database, a
reference resource for spatial transcriptomics data. This database
provides detailed information on gene expression in various tissue
contexts, offering high-resolution spatial data.</p>
<p>For this tutorial, we focus on different types of prostate tissues,
including normal prostate, adenocarcinoma, acinar cell carcinoma, and
adjacent normal sections. These data are crucial for understanding the
variations in gene expression between healthy and cancerous tissues and
for identifying potential diagnostic and therapeutic markers.</p>
<p>The data can be downloaded using the following script: <a
href="https://github.com/tkcaccia/KODAMA-Analysis/blob/main/code/Prostate_download.sh">Prostate_download.sh</a>.
This script facilitates access to the raw data, which will then be
preprocessed and analyzed in the subsequent steps of our pipeline.</p>
</div>
<div id="preprocessing" class="section level3">
<h3>Preprocessing</h3>
<p>This section details the preprocessing of spatial transcriptomics
data, which is a crucial step for cleaning and preparing the data for
further analysis.</p>
<div id="loading-libraries-and-defining-tissue-types"
class="section level4">
<h4>Loading Libraries and Defining Tissue Types</h4>
<pre class="r"><code>library(SpatialExperiment)
library(scater)
library(nnSVG)
library(SPARK)
library(harmony)
library(scuttle)
library(BiocSingular)

tissues &lt;- c(&quot;Normal_prostate&quot;,
             &quot;Acinar_Cell_Carcinoma&quot;,
             &quot;Adjacent_normal_section&quot;,
             &quot;Adenocarcinoma&quot;)</code></pre>
<p>Begin by loading the necessary libraries for the analysis. Next,
define the different types of prostate tissues to be studied: normal
prostate, acinar cell carcinoma, adjacent normal sections, and
adenocarcinoma.</p>
</div>
<div id="reading-visium-data" class="section level4">
<h4>Reading Visium Data</h4>
<pre class="r"><code>dir &lt;- &quot;../Prostate/&quot;
address &lt;- file.path(dir, tissues, &quot;&quot;)

spe &lt;- read10xVisium(address, tissues,
                     type = &quot;sparse&quot;, data = &quot;raw&quot;,
                     images = &quot;lowres&quot;, load = FALSE)</code></pre>
<p>Specify the directory where the data is saved and load the Visium
data for the defined tissue types. The data is read in as sparse
matrices, with low-resolution images.</p>
</div>
<div id="saving-preprocessed-data" class="section level4">
<h4>Saving Preprocessed Data</h4>
<p>To facilitate access to the preprocessed data and avoid repeating the
preprocessing each time, save the <code>spe</code> object in an RData
file.</p>
<p>This allows loading the preprocessed data directly without having to
redo the preprocessing each time.</p>
</div>
<div id="loading-preprocessed-data" class="section level4">
<h4>Loading Preprocessed Data</h4>
<pre class="r"><code>metaData &lt;- SingleCellExperiment::colData(spe)
expr &lt;- SingleCellExperiment::counts(spe)
sample_names &lt;- unique(colData(spe)$sample_id)</code></pre>
<p>Load the preprocessed data and extract the metadata and gene
expression counts.</p>
</div>
<div id="filtering-tissue-spots-and-identifying-mitochondrial-genes"
class="section level4">
<h4>Filtering Tissue Spots and Identifying Mitochondrial Genes</h4>
<pre class="r"><code>spe &lt;- spe[, colData(spe)$in_tissue]

# Identify mitochondrial genes
is_mito &lt;- grepl(&quot;(^MT-)|(^mt-)&quot;, rowData(spe)$gene_name)
table(is_mito)</code></pre>
<pre><code>&lt; table of extent 0 &gt;</code></pre>
<p>Filter the spots located in the tissue and identify mitochondrial
genes, which are often used as quality indicators.</p>
</div>
<div id="calculating-quality-control-qc-metrics-per-spot"
class="section level4">
<h4>Calculating Quality Control (QC) Metrics per Spot</h4>
<pre class="r"><code># Calculate per-spot QC metrics
spe &lt;- addPerCellQC(spe, subsets = list(mito = is_mito))

# Select QC thresholds
qc_lib_size &lt;- colData(spe)$sum &lt; 500
qc_detected &lt;- colData(spe)$detected &lt; 250
qc_mito &lt;- colData(spe)$subsets_mito_percent &gt; 30
qc_cell_count &lt;- colData(spe)$cell_count &gt; 12

# Spots to discard
discard &lt;- qc_lib_size | qc_detected | qc_mito | qc_cell_count

if (length(discard) &gt; 0) {
  table(discard)
  colData(spe)$discard &lt;- discard
  
  # Filter low-quality spots
  spe &lt;- spe[, !colData(spe)$discard]
}
dim(spe)</code></pre>
<pre><code>[1] 36945 13417</code></pre>
<p>Calculate several QC metrics per spot, such as library size, number
of detected genes, percentage of mitochondrial genes, and cell count.
Define thresholds for these metrics and filter out low-quality
spots.</p>
</div>
<div id="filtering-genes" class="section level4">
<h4>Filtering Genes</h4>
<pre class="r"><code>colnames(rowData(spe)) &lt;- &quot;gene_name&quot;

spe &lt;- filter_genes(
  spe,
  filter_genes_ncounts = 2,   # Minimum counts
  filter_genes_pcspots = 0.5, # Minimum percentage of spots
  filter_mito = TRUE          # Filter mitochondrial genes
)

dim(spe)</code></pre>
<pre><code>[1] 12527 13417</code></pre>
<p>Filter genes based on the number of counts and the percentage of
spots in which they are present. Mitochondrial genes are also filtered
out.</p>
</div>
<div id="adjusting-spatial-coordinates" class="section level4">
<h4>Adjusting Spatial Coordinates</h4>
<pre class="r"><code>xy &lt;- spatialCoords(spe)
samples &lt;- unique(colData(spe)$sample_id)

for (j in 1:length(samples)) {
  sel &lt;- samples[j] == colData(spe)$sample_id
  xy[sel, 1] &lt;- spatialCoords(spe)[sel, 1] + 25000 * (j - 1)
}
spatialCoords(spe) &lt;- xy</code></pre>
<p>Adjust the spatial coordinates of each sample to separate them
visually, facilitating data analysis and visualization.</p>
</div>
<div id="normalizing-counts" class="section level4">
<h4>Normalizing Counts</h4>
<pre class="r"><code>spe &lt;- computeLibraryFactors(spe)
spe &lt;- logNormCounts(spe)</code></pre>
<p>normalize the counts using library size factors and apply a
logarithmic transformation to obtain data ready for more precise
analysis.</p>
<p>This preprocessing process cleans and normalizes the spatial
transcriptomics data, ensuring high-quality data ready for subsequent
analyses.</p>
</div>
</div>
<div id="feature-selection-with-spark" class="section level3">
<h3>Feature Selection with SPARK</h3>
<p>After preprocessing the data, the next step involves feature
selection using SPARK, which is crucial for identifying significant
genes across different tissue samples.</p>
<pre class="r"><code>pvalue_mat &lt;- matrix(NA, nrow = nrow(spe), ncol = length(sample_names))
rownames(pvalue_mat) &lt;- rownames(rowData(spe))

# Perform SPARK analysis for each sample
for (i in 1:length(sample_names)) {
  sel &lt;- colData(spe)$sample_id == sample_names[i]
  spe_sub &lt;- spe[, sel]
  
  sparkX &lt;- sparkx(logcounts(spe_sub), spatialCoords(spe_sub), numCores = 1, option = &quot;mixture&quot;,verbose=FALSE)
  
  pvalue_mat[rownames(sparkX$res_mtest), i] &lt;- sparkX$res_mtest$combinedPval
  print(sample_names[i])
}</code></pre>
<pre><code>## ===== SPARK-X INPUT INFORMATION ==== 
## number of total samples: 2543 
## number of total genes: 12519 
## Running with single core, may take some time 
## Testing With Projection Kernel
## Testing With Gaussian Kernel 1
## Testing With Gaussian Kernel 2
## Testing With Gaussian Kernel 3
## Testing With Gaussian Kernel 4
## Testing With Gaussian Kernel 5
## Testing With Cosine Kernel 1
## Testing With Cosine Kernel 2
## Testing With Cosine Kernel 3
## Testing With Cosine Kernel 4
## Testing With Cosine Kernel 5
[1] &quot;Normal_prostate&quot;
## ===== SPARK-X INPUT INFORMATION ==== 
## number of total samples: 3043 
## number of total genes: 12524 
## Running with single core, may take some time 
## Testing With Projection Kernel
## Testing With Gaussian Kernel 1
## Testing With Gaussian Kernel 2
## Testing With Gaussian Kernel 3
## Testing With Gaussian Kernel 4
## Testing With Gaussian Kernel 5
## Testing With Cosine Kernel 1
## Testing With Cosine Kernel 2
## Testing With Cosine Kernel 3
## Testing With Cosine Kernel 4
## Testing With Cosine Kernel 5
[1] &quot;Acinar_Cell_Carcinoma&quot;
## ===== SPARK-X INPUT INFORMATION ==== 
## number of total samples: 3460 
## number of total genes: 12521 
## Running with single core, may take some time 
## Testing With Projection Kernel
## Testing With Gaussian Kernel 1
## Testing With Gaussian Kernel 2
## Testing With Gaussian Kernel 3
## Testing With Gaussian Kernel 4
## Testing With Gaussian Kernel 5
## Testing With Cosine Kernel 1
## Testing With Cosine Kernel 2
## Testing With Cosine Kernel 3
## Testing With Cosine Kernel 4
## Testing With Cosine Kernel 5
[1] &quot;Adjacent_normal_section&quot;
## ===== SPARK-X INPUT INFORMATION ==== 
## number of total samples: 4371 
## number of total genes: 12525 
## Running with single core, may take some time 
## Testing With Projection Kernel
## Testing With Gaussian Kernel 1
## Testing With Gaussian Kernel 2
## Testing With Gaussian Kernel 3
## Testing With Gaussian Kernel 4
## Testing With Gaussian Kernel 5
## Testing With Cosine Kernel 1
## Testing With Cosine Kernel 2
## Testing With Cosine Kernel 3
## Testing With Cosine Kernel 4
## Testing With Cosine Kernel 5
[1] &quot;Adenocarcinoma&quot;</code></pre>
<pre class="r"><code>pvalue_mat=pvalue_mat[!is.na(rowSums(pvalue_mat)),]

oo=order(apply(pvalue_mat,1,function(x) mean(-log(x))),decreasing = TRUE)
top=rownames(pvalue_mat)[oo]</code></pre>
</div>
<div id="principal-component-analysis-pca" class="section level3">
<h3>Principal Component Analysis (PCA)</h3>
<p>After feature selection, principal component analysis (PCA) is
performed to explore the variance in the dataset and visualize sample
relationships.</p>
<p>sample_id=colData(spe)$sample_id</p>
<pre class="r"><code>library(ggplot2)

# Run PCA with top selected genes
spe &lt;- runPCA(spe, subset_row = top[1:3000], scale = TRUE)

# Run Harmony to adjust for batch effects
spe &lt;- RunHarmony(spe, group.by.vars = &quot;sample_id&quot;, lambda = NULL)

# Visualize PCA and Harmony results
plot(reducedDim(spe, type = &quot;PCA&quot;), col = as.factor(colData(spe)$sample_id), main = &quot;PCA&quot;)</code></pre>
<p><img src="figure/Prostate.Rmd/unnamed-chunk-10-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot(reducedDim(spe, type = &quot;HARMONY&quot;), col = as.factor(colData(spe)$sample_id), main = &quot;Harmony&quot;)</code></pre>
<p><img src="figure/Prostate.Rmd/unnamed-chunk-10-2.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pca=reducedDim(spe,type = &quot;HARMONY&quot;)[,1:50]
samples=as.factor(colData(spe)$sample_id)
xy=as.matrix(spatialCoords(spe))
data=t(logcounts(spe))</code></pre>
</div>
<div id="pathology-data-analysis" class="section level3">
<h3>Pathology Data Analysis</h3>
</div>
<div id="loading-pathology-data" class="section level3">
<h3>Loading Pathology Data</h3>
<p>To begin the pathology data analysis, load the corresponding
pathology data for adenocarcinoma samples. Ensure to replace the file
path with the correct location of your data.</p>
<pre class="r"><code>patho &lt;- read.csv(&quot;../Prostate/Adenocarcinoma/outs/Pathology.csv&quot;)</code></pre>
</div>
<div id="processing-pathology-data" class="section level3">
<h3>Processing Pathology Data</h3>
<p>The processing involves creating row names and associating pathology
information with the corresponding columns in the spe object.</p>
<pre class="r"><code>rownames(patho) &lt;- patho[,1]
pathology &lt;- rep(NA, ncol(spe))
sel &lt;- colData(spe)$sample_id == &quot;Adenocarcinoma&quot;
pathology[sel] &lt;- patho[rownames(colData(spe))[sel], &quot;Pathology&quot;]
pathology[pathology == &quot;&quot;] &lt;- NA
pathology &lt;- factor(pathology, levels = c(&quot;Invasive carcinoma&quot;,
                                          &quot;Blood vessel&quot;,
                                          &quot;Fibro-muscular tissue&quot;,
                                          &quot;Fibrous tissue&quot;,
                                          &quot;Immune Cells&quot;,
                                          &quot;Nerve&quot;,
                                          &quot;Normal gland&quot;))</code></pre>
</div>
<div id="visualization-of-pathology-data" class="section level3">
<h3>Visualization of Pathology Data</h3>
<p>Assign specific colors to each pathology category and visualize the
samples on a reduced dimension map (HARMONY), with each point colored
according to its pathology category.</p>
<pre class="r"><code>col_pathology &lt;- c(&quot;#0000ff&quot;, &quot;#e41a1c&quot;, &quot;#006400&quot;, &quot;#000000&quot;, &quot;#ffd700&quot;, &quot;#00ff00&quot;, &quot;#b2dfee&quot;)
plot(reducedDim(spe, type = &quot;HARMONY&quot;), pch = 20, col = col_pathology[pathology])</code></pre>
<p><img src="figure/Prostate.Rmd/unnamed-chunk-13-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="running-kodama-for-analysis" class="section level3">
<h3>Running KODAMA for Analysis</h3>
<p>The next step is running KODAMA, a method for dimensionality
reduction and visualization.</p>
<pre class="r"><code>library(KODAMAextra)

spe &lt;- RunKODAMAmatrix(spe,
                       reduction = &quot;HARMONY&quot;,
                       FUN = &quot;PLS&quot;,
                       landmarks = 100000,
                       splitting = 300,
                       f.par.pls = 50,
                       spatial.resolution = 0.4,
                       n.cores = 10)</code></pre>
<pre><code>socket cluster with 10 nodes on host &#39;localhost&#39;
================================================================================[1] &quot;Finished parallel computation&quot;

[1] &quot;Calculation of dissimilarity matrix...&quot;
================================================================================</code></pre>
<pre class="r"><code>config &lt;- umap.defaults
config$n_threads = 10
config$n_sgd_threads = &quot;auto&quot;

spe=RunKODAMAvisualization(spe,method=&quot;UMAP&quot;,config=config)

plot(reducedDim(spe,type = &quot;KODAMA&quot;),col=as.factor(colData(spe)$sample_id))</code></pre>
<p><img src="figure/Prostate.Rmd/unnamed-chunk-14-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-1">
Past versions of unnamed-chunk-14-1.png
</button>
</p>
<div id="fig-unnamed-chunk-14-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/fe5a2ba3ad736099b8b1fce2a4cd08711f6ef09c/docs/figure/Prostate.Rmd/unnamed-chunk-14-1.png" target="_blank">fe5a2ba</a>
</td>
<td>
GitHub
</td>
<td>
2024-07-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot(reducedDim(spe,type = &quot;KODAMA&quot;),col=pathology)</code></pre>
<p><img src="figure/Prostate.Rmd/unnamed-chunk-14-2.png" width="960" style="display: block; margin: auto;" /></p>
<p>This extended analysis includes principal component analysis (PCA),
pathology data analysis, and the application of KODAMA for
dimensionality reduction and visualization, enhancing the understanding
of spatial transcriptomics data in different prostate tissue types.</p>
</div>
<div id="gsva-enrichment-analysis-with-msigdb" class="section level3">
<h3>GSVA Enrichment Analysis with MSigDB</h3>
<p>To explore enriched biological processes in our spatial
transcriptomics data, we employ Gene Set Variation Analysis (GSVA) using
MSigDB gene sets as a reference. To download the necessary data, please
follow the steps provided at <a
href="https://www.gsea-msigdb.org/gsea/msigdb">this link</a> and create
an account if required.</p>
<div id="loading-packages-and-data" class="section level4">
<h4>Loading Packages and Data</h4>
<p>We start by loading the necessary packages and preparing our gene
data for analysis:</p>
<pre class="r"><code>library(&quot;GSVA&quot;)
library(&quot;GSA&quot;)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.0 (2024-04-24)
Platform: x86_64-pc-linux-gnu
Running under: Ubuntu 20.04.6 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0 
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

time zone: Etc/UTC
tzcode source: system (glibc)

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] GSA_1.03.3                  GSVA_1.52.3                
 [3] KODAMAextra_1.0             e1071_1.7-14               
 [5] doParallel_1.0.17           iterators_1.0.14           
 [7] foreach_1.5.2               KODAMA_3.1                 
 [9] umap_0.2.10.0               Rtsne_0.17                 
[11] minerva_1.5.10              BiocSingular_1.20.0        
[13] harmony_1.2.0               Rcpp_1.0.12                
[15] SPARK_1.1.1                 nnSVG_1.8.0                
[17] scater_1.32.0               ggplot2_3.5.1              
[19] scuttle_1.14.0              SpatialExperiment_1.14.0   
[21] SingleCellExperiment_1.26.0 SummarizedExperiment_1.34.0
[23] Biobase_2.64.0              GenomicRanges_1.56.1       
[25] GenomeInfoDb_1.40.1         IRanges_2.38.1             
[27] S4Vectors_0.42.1            BiocGenerics_0.50.0        
[29] MatrixGenerics_1.16.0       matrixStats_1.3.0          

loaded via a namespace (and not attached):
  [1] jsonlite_1.8.8            magrittr_2.0.3           
  [3] ggbeeswarm_0.7.2          magick_2.8.3             
  [5] rmarkdown_2.27            fs_1.6.4                 
  [7] zlibbioc_1.50.0           vctrs_0.6.5              
  [9] memoise_2.0.1             DelayedMatrixStats_1.26.0
 [11] askpass_1.2.0             CompQuadForm_1.4.3       
 [13] htmltools_0.5.8.1         S4Arrays_1.4.1           
 [15] BiocNeighbors_1.22.0      Rhdf5lib_1.26.0          
 [17] SparseArray_1.4.8         rhdf5_2.48.0             
 [19] sass_0.4.9                pracma_2.4.4             
 [21] bslib_0.7.0               cachem_1.1.0             
 [23] whisker_0.4.1             lifecycle_1.0.4          
 [25] pkgconfig_2.0.3           rsvd_1.0.5               
 [27] Matrix_1.7-0              R6_2.5.1                 
 [29] fastmap_1.2.0             GenomeInfoDbData_1.2.12  
 [31] digest_0.6.36             colorspace_2.1-0         
 [33] AnnotationDbi_1.66.0      rprojroot_2.0.4          
 [35] dqrng_0.4.1               RSpectra_0.16-1          
 [37] irlba_2.3.5.1             RSQLite_2.3.7            
 [39] beachmat_2.20.0           fansi_1.0.6              
 [41] httr_1.4.7                abind_1.4-5              
 [43] compiler_4.4.0            proxy_0.4-27             
 [45] BRISC_1.0.5               bit64_4.0.5              
 [47] withr_3.0.0               BiocParallel_1.38.0      
 [49] DBI_1.2.3                 viridis_0.6.5            
 [51] highr_0.11                HDF5Array_1.32.0         
 [53] R.utils_2.12.3            openssl_2.2.0            
 [55] DelayedArray_0.30.1       rjson_0.2.21             
 [57] rdist_0.0.5               tools_4.4.0              
 [59] vipor_0.4.7               beeswarm_0.4.0           
 [61] httpuv_1.6.15             R.oo_1.26.0              
 [63] glue_1.7.0                rhdf5filters_1.16.0      
 [65] promises_1.3.0            grid_4.4.0               
 [67] snow_0.4-4                generics_0.1.3           
 [69] gtable_0.3.5              class_7.3-22             
 [71] R.methodsS3_1.8.2         ScaledMatrix_1.12.0      
 [73] utf8_1.2.4                XVector_0.44.0           
 [75] ggrepel_0.9.5             RANN_2.6.1               
 [77] pillar_1.9.0              stringr_1.5.1            
 [79] limma_3.60.3              later_1.3.2              
 [81] dplyr_1.1.4               lattice_0.22-6           
 [83] bit_4.0.5                 annotate_1.82.0          
 [85] tidyselect_1.2.1          locfit_1.5-9.10          
 [87] Biostrings_2.72.1         pbapply_1.7-2            
 [89] knitr_1.48                git2r_0.33.0             
 [91] gridExtra_2.3             edgeR_4.2.0              
 [93] RhpcBLASctl_0.23-42       xfun_0.45                
 [95] statmod_1.5.0             DropletUtils_1.24.0      
 [97] stringi_1.8.4             UCSC.utils_1.0.0         
 [99] workflowr_1.7.1           yaml_2.3.9               
[101] evaluate_0.24.0           codetools_0.2-20         
[103] tibble_3.2.1              graph_1.82.0             
[105] cli_3.6.3                 matlab_1.0.4.1           
[107] xtable_1.8-4              reticulate_1.38.0        
[109] munsell_0.5.1             jquerylib_0.1.4          
[111] doSNOW_1.0.20             png_0.1-8                
[113] XML_3.99-0.17             blob_1.2.4               
[115] sparseMatrixStats_1.16.0  GSEABase_1.66.0          
[117] viridisLite_0.4.2         scales_1.3.0             
[119] crayon_1.5.3              rlang_1.1.4              
[121] KEGGREST_1.44.1           cowplot_1.1.3            </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
