<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Seurat pipeline</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">KODAMA-Analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="Installation.html">Installation</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Single-cell
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="singlecell_datamatrix.html">Data matrix</a>
    </li>
    <li>
      <a href="singlecell_seurat.html">Seurat</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Spatial
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="MERFISH.html">MERFISH (Moffitt et al., 2018)</a>
    </li>
    <li>
      <a href="DLPFC.html">Visium DLPFC (Maynard et al., 2021)</a>
    </li>
    <li>
      <a href="Prostate.html">Visium (10X database - Prostate)</a>
    </li>
    <li>
      <a href="VisiumHD.html">Visium HD (10X database - Colorectal cancer)</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Wrapper
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Seurat.html">Seurat</a>
    </li>
    <li>
      <a href="Giotto.html">Giotto</a>
    </li>
    <li>
      <a href="SpatialExperiment.html">SpatialExperiment</a>
    </li>
  </ul>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/tkcaccia/KODAMA-Analysis">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Seurat pipeline</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<<<<<<< HEAD
<strong>Last updated:</strong> 2025-04-13
=======
<strong>Last updated:</strong> 2024-09-01
>>>>>>> d89baaa7bfc3bffd762b0db8bac45eeb05443b97
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>KODAMA-Analysis/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240618code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240618)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240618code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240618)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<<<<<<< HEAD
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomtkcacciaKODAMAAnalysistree5f5ac634a8274a7b7c4cb315454f9c9aa7a12f86targetblank5f5ac63a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/tree/5f5ac634a8274a7b7c4cb315454f9c9aa7a12f86" target="_blank">5f5ac63</a>
=======
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomtkcacciaKODAMAAnalysistreeb1a5ea0a6f3df5fb99f8d37509de9c60563f7a14targetblankb1a5ea0a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/tree/b1a5ea0a6f3df5fb99f8d37509de9c60563f7a14" target="_blank">b1a5ea0</a>
>>>>>>> d89baaa7bfc3bffd762b0db8bac45eeb05443b97
</a>
</p>
</div>
<div
<<<<<<< HEAD
id="strongRepositoryversionstrongahrefhttpsgithubcomtkcacciaKODAMAAnalysistree5f5ac634a8274a7b7c4cb315454f9c9aa7a12f86targetblank5f5ac63a"
=======
id="strongRepositoryversionstrongahrefhttpsgithubcomtkcacciaKODAMAAnalysistreeb1a5ea0a6f3df5fb99f8d37509de9c60563f7a14targetblankb1a5ea0a"
>>>>>>> d89baaa7bfc3bffd762b0db8bac45eeb05443b97
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<<<<<<< HEAD
<a href="https://github.com/tkcaccia/KODAMA-Analysis/tree/5f5ac634a8274a7b7c4cb315454f9c9aa7a12f86" target="_blank">5f5ac63</a>.
=======
<a href="https://github.com/tkcaccia/KODAMA-Analysis/tree/b1a5ea0a6f3df5fb99f8d37509de9c60563f7a14" target="_blank">b1a5ea0</a>.
>>>>>>> d89baaa7bfc3bffd762b0db8bac45eeb05443b97
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
<<<<<<< HEAD
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  KODAMA.svg
    Untracked:  analysis/singlecell_datamatrix.Rmd
    Untracked:  analysis/singlecell_seurat.Rmd
    Untracked:  code/Acinar_Cell_Carcinoma.ipynb
    Untracked:  code/Adenocarcinoma.ipynb
    Untracked:  code/Adjacent_normal_section.ipynb
    Untracked:  code/DLFPC_preprocessing.R
    Untracked:  code/DLPFC - BANKSY.R
    Untracked:  code/DLPFC - BASS.R
    Untracked:  code/DLPFC - BAYESPACE.R
    Untracked:  code/DLPFC - Nonspatial.R
    Untracked:  code/DLPFC - PRECAST.R
    Untracked:  code/DLPFC_comparison.R
    Untracked:  code/DLPFC_results_analysis.R
    Untracked:  code/MERFISH - BANKSY.R
    Untracked:  code/MERFISH - BASS.R
    Untracked:  code/MERFISH - BAYESPACE.R
    Untracked:  code/MERFISH - Nonspatial.R
    Untracked:  code/MERFISH - PRECAST.R
    Untracked:  code/MERFISH_comparison.R
    Untracked:  code/MERFISH_results_analysis.R
    Untracked:  code/VisiumHD-CRC.ipynb
    Untracked:  code/VisiumHDassignment.py
    Untracked:  code/deep learning code DLPFC.R
    Untracked:  code/save tiles.py
    Untracked:  data/Adenocarcinoma.csv
    Untracked:  data/Annotations/
    Untracked:  data/DLFPC-Br5292-input.RData
    Untracked:  data/DLFPC-Br5595-input.RData
    Untracked:  data/DLFPC-Br8100-input.RData
    Untracked:  data/DLPFC-general.RData
    Untracked:  data/MERFISH-input.RData
    Untracked:  data/spots_classification_ALL.csv
    Untracked:  data/spots_classification_Acinar_Cell_Carcinoma.csv
    Untracked:  data/spots_classification_IF.csv
    Untracked:  data/spots_classification_Normal_prostate.csv
    Untracked:  data/trajectories.RData
    Untracked:  data/trajectories_VISIUMHD.RData
    Untracked:  output/BANSKY-results.RData
    Untracked:  output/BASS-results.RData
    Untracked:  output/BayesSpace-results.RData
    Untracked:  output/CRC-image.RData
    Untracked:  output/CRC-image2.RData
    Untracked:  output/CRC.png
    Untracked:  output/CRC2.png
    Untracked:  output/CRC7.png
    Untracked:  output/CRC8.png
    Untracked:  output/CRC_boxplot.png
    Untracked:  output/CRC_boxplot.svg
    Untracked:  output/CRC_boxplot2.svg
    Untracked:  output/CRC_linee.svg
    Untracked:  output/DL.RData
    Untracked:  output/DLFPC-All-2.RData
=======
    Ignored:    .Rhistory
    Ignored:    analysis/figure/

Untracked files:
    Untracked:  data/spots_classification_ALL.csv
    Untracked:  data/trajectories.RData
    Untracked:  output/DLFPC-All-2-3000-300-0.4.RData
>>>>>>> d89baaa7bfc3bffd762b0db8bac45eeb05443b97
    Untracked:  output/DLFPC-All.RData
    Untracked:  output/DLFPC-Br5292.RData
    Untracked:  output/DLFPC-Br5595.RData
    Untracked:  output/DLFPC-Br8100.RData
<<<<<<< HEAD
    Untracked:  output/DLFPC-variablesXdeeplearning.RData
    Untracked:  output/DLPFC-BANSKY-results.RData
    Untracked:  output/DLPFC-BASS-results.RData
    Untracked:  output/DLPFC-BayesSpace-results.RData
    Untracked:  output/DLPFC-Nonspatial-results.RData
    Untracked:  output/DLPFC-PRECAST-results.RData
    Untracked:  output/DLPFC_all_cluster.svg
    Untracked:  output/DLPFCpathway.RData
    Untracked:  output/Figure 1 - boxplot.pdf
    Untracked:  output/Figure 2 - DLPFC 10.pdf
    Untracked:  output/Figures/
    Untracked:  output/KODAMA-results.RData
    Untracked:  output/KODAMA_DLPFC_All_original.svg
    Untracked:  output/KODAMA_DLPFC_Br5595.svg
    Untracked:  output/KODAMA_DLPFC_Br5595_slide.svg
    Untracked:  output/Loupe.csv
    Untracked:  output/MERFISH-BANSKY-results.RData
    Untracked:  output/MERFISH-BASS-results.RData
    Untracked:  output/MERFISH-BayesSpace-results.RData
    Untracked:  output/MERFISH-KODAMA-results.RData
    Untracked:  output/MERFISH-Nonspatial-results.RData
    Untracked:  output/MERFISH-PRECAST-results.RData
    Untracked:  output/MERFISH.RData
    Untracked:  output/Nonspatial-results.RData
    Untracked:  output/Prostate.RData
    Untracked:  output/VisiumHD-RNA.RData
    Untracked:  output/VisiumHD-genes.pdf
    Untracked:  output/VisiumHD.RData
    Untracked:  output/boh.svg
    Untracked:  output/desmoplastic_distance_carcinoma.csv
    Untracked:  output/image.RData
    Untracked:  output/pp.RData
    Untracked:  output/pp2.RData
    Untracked:  output/pp3.RData
    Untracked:  output/pp4.RData
    Untracked:  output/pp5.RData
    Untracked:  output/prostate1.svg
    Untracked:  output/prostate2.svg
    Untracked:  output/prostate3.svg
    Untracked:  output/prostate4.svg
    Untracked:  output/prostate5.svg
    Untracked:  output/prostate6.svg
    Untracked:  output/prostate7.svg
    Untracked:  output/subclusters1.csv
    Untracked:  output/subclusters2.csv
    Untracked:  output/subclusters3.csv
    Untracked:  output/tight_boundary.geojson
    Untracked:  output/trajectory.csv
=======
    Untracked:  output/MERFISH.RData
    Untracked:  output/Prostate.RData
    Untracked:  output/VisiumHD.RData
    Untracked:  output/VisiumHD2.RData
    Untracked:  output/VisiumHD3.RData
    Untracked:  output/image.RData
>>>>>>> d89baaa7bfc3bffd762b0db8bac45eeb05443b97

Unstaged changes:
    Deleted:    analysis/D1.Rmd
    Deleted:    analysis/DLPFC-12.Rmd
    Deleted:    analysis/DLPFC-4.Rmd
    Modified:   analysis/DLPFC.Rmd
<<<<<<< HEAD
    Deleted:    analysis/DLPFC1.Rmd
    Deleted:    analysis/DLPFC10.Rmd
    Deleted:    analysis/DLPFC2.Rmd
    Deleted:    analysis/DLPFC3.Rmd
    Deleted:    analysis/DLPFC4.Rmd
    Deleted:    analysis/DLPFC5.Rmd
    Deleted:    analysis/DLPFC6.Rmd
    Deleted:    analysis/DLPFC7.Rmd
    Deleted:    analysis/DLPFC8.Rmd
    Deleted:    analysis/DLPFC9.Rmd
    Deleted:    analysis/Du1.Rmd
    Deleted:    analysis/Du10.Rmd
    Deleted:    analysis/Du11.Rmd
    Deleted:    analysis/Du12.Rmd
    Deleted:    analysis/Du13.Rmd
    Deleted:    analysis/Du14.Rmd
    Deleted:    analysis/Du15.Rmd
    Deleted:    analysis/Du16.Rmd
    Deleted:    analysis/Du17.Rmd
    Deleted:    analysis/Du18.Rmd
    Deleted:    analysis/Du19.Rmd
    Deleted:    analysis/Du2.Rmd
    Deleted:    analysis/Du20.Rmd
    Deleted:    analysis/Du3.Rmd
    Deleted:    analysis/Du4.Rmd
    Deleted:    analysis/Du5.Rmd
    Deleted:    analysis/Du6.Rmd
    Deleted:    analysis/Du7.Rmd
    Deleted:    analysis/Du8.Rmd
    Deleted:    analysis/Du9.Rmd
    Modified:   analysis/Giotto.Rmd
    Modified:   analysis/MERFISH.Rmd
    Deleted:    analysis/MERFISH1a (copy).Rmd
    Deleted:    analysis/MERFISH1a.Rmd
    Deleted:    analysis/MERFISH1b (copy).Rmd
    Deleted:    analysis/MERFISH1b.Rmd
    Deleted:    analysis/MERFISH2a (copy).Rmd
    Deleted:    analysis/MERFISH2a.Rmd
    Deleted:    analysis/MERFISH2b (copy).Rmd
    Deleted:    analysis/MERFISH2b.Rmd
    Deleted:    analysis/MERFISH3a (copy).Rmd
    Deleted:    analysis/MERFISH3a.Rmd
    Deleted:    analysis/MERFISH3b (copy).Rmd
    Deleted:    analysis/MERFISH3b.Rmd
    Deleted:    analysis/MERFISH4a (copy).Rmd
    Deleted:    analysis/MERFISH4a.Rmd
    Deleted:    analysis/MERFISH4b (copy).Rmd
    Deleted:    analysis/MERFISH4b.Rmd
    Modified:   analysis/Prostate.Rmd
    Deleted:    analysis/STARmap.Rmd
    Modified:   analysis/Seurat.Rmd
    Deleted:    analysis/Simulation.Rmd
    Deleted:    analysis/Single-cell.Rmd
    Modified:   analysis/SpatialExperiment.Rmd
    Modified:   analysis/VisiumHD.Rmd
    Modified:   code/VisiumHD_CRC_download.sh
    Deleted:    data/Pathology.csv
=======
    Modified:   analysis/Giotto.Rmd
    Modified:   analysis/MERFISH.Rmd
    Modified:   analysis/Prostate.Rmd
    Deleted:    analysis/STARmap.Rmd
    Modified:   analysis/VisiumHD.Rmd
>>>>>>> d89baaa7bfc3bffd762b0db8bac45eeb05443b97

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/Seurat.Rmd</code>) and HTML
(<code>docs/Seurat.html</code>) files. If you’ve configured a remote Git
repository (see <code>?wflow_git_remote</code>), click on the hyperlinks
in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/d1192e95a8ed7985fb6d9678ce0a6d850f882066/docs/Seurat.html" target="_blank">d1192e9</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-08-12
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/3374e66ce76ec3d26fcb912a9f6eba2f7341ed8f/docs/Seurat.html" target="_blank">3374e66</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-08-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/35ce733e4e0012d79f696f9c7dcdbf67e1623e80/docs/Seurat.html" target="_blank">35ce733</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-08-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/82fe167363197a5889eb59cb7e39511dfde2f22e/docs/Seurat.html" target="_blank">82fe167</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-07-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/6f7daaccbcc7b968bc736664a45f9ebcaf73fba3/docs/Seurat.html" target="_blank">6f7daac</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-07-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/3f7aad62bc5856b2bdec967ad08fe8dfcb44b2ad/analysis/Seurat.Rmd" target="_blank">3f7aad6</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-07-19
</td>
<td>
Start my new project
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/210dfdda3cc20b5c1af238a5ef698c98fee3d7a2/analysis/Seurat.Rmd" target="_blank">210dfdd</a>
</td>
<td>
GitHub
</td>
<td>
2024-07-16
</td>
<td>
Update Seurat.Rmd
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/7be8f59b0c4d82e877c5dfc8cfbfb7936ceceeb6/docs/Seurat.html" target="_blank">7be8f59</a>
</td>
<td>
tkcaccia
</td>
<td>
2024-07-15
</td>
<td>
updates
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/f8ca54aee4428df50d2cd0f41cbf2a0ceae01d7d/analysis/Seurat.Rmd" target="_blank">f8ca54a</a>
</td>
<td>
tkcaccia
</td>
<td>
2024-07-14
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/f8ca54aee4428df50d2cd0f41cbf2a0ceae01d7d/docs/Seurat.html" target="_blank">f8ca54a</a>
</td>
<td>
tkcaccia
</td>
<td>
2024-07-14
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/89a11c19aabfd67ffe5ad726376a5a2ee4b20539/analysis/Seurat.Rmd" target="_blank">89a11c1</a>
</td>
<td>
GitHub
</td>
<td>
2024-07-08
</td>
<td>
Add files via upload
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/2b5aad71a70086d1a4ae5ae75e37bcc58131fb7e/docs/Seurat.html" target="_blank">2b5aad7</a>
</td>
<td>
GitHub
</td>
<td>
2024-07-08
</td>
<td>
Add files via upload
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Seurat is an R toolkit widely used for single-cell genomics and
spatial transcriptomics, developed and maintained by the Satija Lab at
New York Genome Center. In this section, we will see the steps to
integrate KODAMA into the Seurat pipeline.</p>
<div id="loading-libraries" class="section level2">
<h2>Loading libraries</h2>
<p>First, we need to load the required libraries for this analysis. Make
sure you have the following packages installed.</p>
<pre class="r"><code>library(Seurat)
library(SeuratData)
library(KODAMA)
library(KODAMAextra)</code></pre>
</div>
<div id="dataset" class="section level2">
<h2>Dataset</h2>
<p>The data used are sagittal sections of the mouse brain generated
using Visium v1 technology saved in the Seurat library. For specific
details on loading and using the dataset, please take a look at the
documentation by typing ?stxBrain in your R environment. Install and
load the anterior region of the mouse brain dataset.</p>
<pre class="r"><code>#InstallData(&quot;stxBrain&quot;)
brain &lt;- LoadData(&quot;stxBrain&quot;, type = &quot;anterior1&quot;)</code></pre>
</div>
<div id="data-processing" class="section level2">
<h2>Data processing</h2>
<p>We will preprocess the data using Seurat’s SCTransform method and
perform PCA:</p>
<pre class="r"><code>brain &lt;- SCTransform(brain, assay = &quot;Spatial&quot;, verbose = FALSE)
brain &lt;- RunPCA(brain, assay = &quot;SCT&quot;, verbose = FALSE)</code></pre>
</div>
<div id="run-and-visualize-kodama" class="section level2">
<h2>Run and Visualize KODAMA</h2>
<p>Apply KODAMA to reduce dimensionality and visualize the results:</p>
<pre class="r"><code>brain &lt;- RunKODAMAmatrix(brain, reduction = &quot;pca&quot;)</code></pre>
<pre><code>Calculating Network

Calculating Network spatial
socket cluster with 1 nodes on host &#39;localhost&#39;
================================================================================
Finished parallel computation

[1] &quot;Calculation of dissimilarity matrix...&quot;
================================================================================</code></pre>
<pre class="r"><code>brain &lt;- RunKODAMAvisualization(brain)</code></pre>
</div>
<div id="clustering" class="section level2">
<h2>Clustering</h2>
<p>Apply clustering on the KODAMA plot.</p>
<pre class="r"><code>brain &lt;- FindNeighbors(brain, reduction = &quot;KODAMA&quot;, dims = 1:2)
brain &lt;- FindClusters(brain, verbose = FALSE, graph.name=&quot;RNA_snn&quot;,resolution = 0.2)</code></pre>
<p>Visualize the KODAMA results with DimPlot and SpatialDimPlot:</p>
<pre class="r"><code>p1 &lt;- DimPlot(brain, reduction = &quot;KODAMA&quot;, label = TRUE)
p2 &lt;- SpatialDimPlot(brain, label = TRUE, label.size = 3)
p1 + p2</code></pre>
<p><img src="figure/Seurat.Rmd/unnamed-chunk-6-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/6f7daaccbcc7b968bc736664a45f9ebcaf73fba3/docs/figure/Seurat.Rmd/unnamed-chunk-6-1.png" target="_blank">6f7daac</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-07-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.3 (2025-02-28)
Platform: x86_64-pc-linux-gnu
Running under: Ubuntu 20.04.6 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0 
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

time zone: Etc/UTC
tzcode source: system (glibc)

attached base packages:
[1] parallel  stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
<<<<<<< HEAD
 [1] KODAMAextra_1.2           e1071_1.7-16             
 [3] doParallel_1.0.17         iterators_1.0.14         
 [5] foreach_1.5.2             KODAMA_3.0               
 [7] Matrix_1.7-3              umap_0.2.10.0            
 [9] Rtsne_0.17                minerva_1.5.10           
[11] stxBrain.SeuratData_0.1.2 SeuratData_0.2.2.9002    
[13] Seurat_5.2.1              SeuratObject_5.0.2       
[15] sp_2.2-0                  workflowr_1.7.1          

loaded via a namespace (and not attached):
  [1] RcppAnnoy_0.0.22            splines_4.4.3              
  [3] later_1.4.1                 tibble_3.2.1               
  [5] polyclip_1.10-7             fastDummies_1.7.5          
  [7] lifecycle_1.0.4             tcltk_4.4.3                
  [9] rprojroot_2.0.4             Rnanoflann_0.0.3           
 [11] globals_0.16.3              processx_3.8.6             
 [13] lattice_0.22-7              MASS_7.3-65                
 [15] magrittr_2.0.3              plotly_4.10.4              
 [17] sass_0.4.9                  rmarkdown_2.29             
 [19] jquerylib_0.1.4             yaml_2.3.10                
 [21] httpuv_1.6.15               glmGamPoi_1.16.0           
 [23] sctransform_0.4.1           spam_2.11-1                
 [25] askpass_1.2.1               spatstat.sparse_3.1-0      
 [27] reticulate_1.42.0           cowplot_1.1.3              
 [29] pbapply_1.7-2               RColorBrewer_1.1-3         
 [31] abind_1.4-8                 zlibbioc_1.50.0            
 [33] GenomicRanges_1.56.2        purrr_1.0.4                
 [35] BiocGenerics_0.50.0         rappdirs_0.3.3             
 [37] misc3d_0.9-1                git2r_0.33.0               
 [39] GenomeInfoDbData_1.2.12     IRanges_2.38.1             
 [41] S4Vectors_0.42.1            ggrepel_0.9.6              
 [43] irlba_2.3.5.1               listenv_0.9.1              
 [45] spatstat.utils_3.1-3        goftest_1.2-3              
 [47] RSpectra_0.16-2             spatstat.random_3.3-3      
 [49] fitdistrplus_1.2-2          parallelly_1.43.0          
 [51] DelayedMatrixStats_1.26.0   codetools_0.2-20           
 [53] DelayedArray_0.30.1         tidyselect_1.2.1           
 [55] UCSC.utils_1.0.0            farver_2.1.2               
 [57] matrixStats_1.5.0           stats4_4.4.3               
 [59] spatstat.explore_3.4-2      jsonlite_2.0.0             
 [61] progressr_0.15.1            ggridges_0.5.6             
 [63] survival_3.8-3              tools_4.4.3                
 [65] snow_0.4-4                  ica_1.0-3                  
 [67] Rcpp_1.0.14                 glue_1.8.0                 
 [69] gridExtra_2.3               SparseArray_1.4.8          
 [71] xfun_0.51                   MatrixGenerics_1.16.0      
 [73] GenomeInfoDb_1.40.1         dplyr_1.1.4                
 [75] withr_3.0.2                 fastmap_1.2.0              
 [77] openssl_2.3.2               callr_3.7.6                
 [79] digest_0.6.37               R6_2.6.1                   
 [81] mime_0.13                   colorspace_2.1-1           
 [83] scattermore_1.2             tensor_1.5                 
 [85] spatstat.data_3.1-6         tidyr_1.3.1                
 [87] generics_0.1.3              data.table_1.17.0          
 [89] class_7.3-23                httr_1.4.7                 
 [91] htmlwidgets_1.6.4           S4Arrays_1.4.1             
 [93] whisker_0.4.1               uwot_0.2.3                 
 [95] pkgconfig_2.0.3             gtable_0.3.6               
 [97] lmtest_0.9-40               XVector_0.44.0             
 [99] htmltools_0.5.8.1           dotCall64_1.2              
[101] scales_1.3.0                Biobase_2.64.0             
[103] png_0.1-8                   doSNOW_1.0.20              
[105] spatstat.univar_3.1-2       knitr_1.50                 
[107] rstudioapi_0.17.1           reshape2_1.4.4             
[109] nlme_3.1-168                proxy_0.4-27               
[111] cachem_1.1.0                zoo_1.8-13                 
[113] stringr_1.5.1               KernSmooth_2.23-26         
[115] miniUI_0.1.1.1              pillar_1.10.1              
[117] grid_4.4.3                  vctrs_0.6.5                
[119] RANN_2.6.2                  promises_1.3.2             
[121] xtable_1.8-4                cluster_2.1.8.1            
[123] evaluate_1.0.3              cli_3.6.4                  
[125] compiler_4.4.3              rlang_1.1.5                
[127] crayon_1.5.3                future.apply_1.11.3        
[129] labeling_0.4.3              ps_1.9.0                   
[131] getPass_0.2-4               plyr_1.8.9                 
[133] fs_1.6.5                    stringi_1.8.7              
[135] viridisLite_0.4.2           deldir_2.0-4               
[137] munsell_0.5.1               lazyeval_0.2.2             
[139] spatstat.geom_3.3-6         RcppHNSW_0.6.0             
[141] patchwork_1.3.0             sparseMatrixStats_1.16.0   
[143] future_1.34.0               ggplot2_3.5.1              
[145] shiny_1.10.0                SummarizedExperiment_1.34.0
[147] ROCR_1.0-11                 igraph_2.1.4               
[149] bslib_0.9.0                </code></pre>
=======
 [1] KODAMAextra_1.0           bigmemory_4.6.4          
 [3] rgl_1.3.1                 misc3d_0.9-1             
 [5] e1071_1.7-14              doParallel_1.0.17        
 [7] iterators_1.0.14          foreach_1.5.2            
 [9] KODAMA_3.1                umap_0.2.10.0            
[11] Rtsne_0.17                minerva_1.5.10           
[13] stxBrain.SeuratData_0.1.2 SeuratData_0.2.2.9001    
[15] Seurat_5.1.0              SeuratObject_5.0.2       
[17] sp_2.1-4                  workflowr_1.7.1          

loaded via a namespace (and not attached):
  [1] RcppAnnoy_0.0.22            splines_4.4.1              
  [3] later_1.3.2                 tibble_3.2.1               
  [5] polyclip_1.10-7             fastDummies_1.7.4          
  [7] lifecycle_1.0.4             tcltk_4.4.1                
  [9] rprojroot_2.0.4             globals_0.16.3             
 [11] processx_3.8.4              lattice_0.22-6             
 [13] MASS_7.3-61                 magrittr_2.0.3             
 [15] plotly_4.10.4               sass_0.4.9                 
 [17] rmarkdown_2.27              jquerylib_0.1.4            
 [19] yaml_2.3.9                  httpuv_1.6.15              
 [21] glmGamPoi_1.16.0            sctransform_0.4.1          
 [23] spam_2.10-0                 askpass_1.2.0              
 [25] spatstat.sparse_3.1-0       reticulate_1.38.0          
 [27] cowplot_1.1.3               pbapply_1.7-2              
 [29] RColorBrewer_1.1-3          zlibbioc_1.50.0            
 [31] abind_1.4-5                 GenomicRanges_1.56.1       
 [33] purrr_1.0.2                 BiocGenerics_0.50.0        
 [35] rappdirs_0.3.3              git2r_0.33.0               
 [37] GenomeInfoDbData_1.2.12     IRanges_2.38.1             
 [39] S4Vectors_0.42.1            ggrepel_0.9.5              
 [41] irlba_2.3.5.1               listenv_0.9.1              
 [43] spatstat.utils_3.1-0        goftest_1.2-3              
 [45] RSpectra_0.16-1             spatstat.random_3.3-1      
 [47] fitdistrplus_1.2-1          parallelly_1.38.0          
 [49] DelayedMatrixStats_1.26.0   DelayedArray_0.30.1        
 [51] leiden_0.4.3.1              codetools_0.2-20           
 [53] tidyselect_1.2.1            farver_2.1.2               
 [55] UCSC.utils_1.0.0            matrixStats_1.3.0          
 [57] stats4_4.4.1                base64enc_0.1-3            
 [59] spatstat.explore_3.3-2      jsonlite_1.8.8             
 [61] progressr_0.14.0            ggridges_0.5.6             
 [63] survival_3.7-0              tools_4.4.1                
 [65] snow_0.4-4                  ica_1.0-3                  
 [67] Rcpp_1.0.12                 glue_1.7.0                 
 [69] SparseArray_1.4.8           gridExtra_2.3              
 [71] xfun_0.45                   MatrixGenerics_1.16.0      
 [73] GenomeInfoDb_1.40.1         dplyr_1.1.4                
 [75] withr_3.0.0                 fastmap_1.2.0              
 [77] fansi_1.0.6                 openssl_2.2.0              
 [79] callr_3.7.6                 digest_0.6.36              
 [81] R6_2.5.1                    mime_0.12                  
 [83] colorspace_2.1-0            scattermore_1.2            
 [85] tensor_1.5                  spatstat.data_3.1-2        
 [87] utf8_1.2.4                  tidyr_1.3.1                
 [89] generics_0.1.3              data.table_1.15.4          
 [91] class_7.3-22                S4Arrays_1.4.1             
 [93] httr_1.4.7                  htmlwidgets_1.6.4          
 [95] whisker_0.4.1               uwot_0.2.2                 
 [97] pkgconfig_2.0.3             gtable_0.3.5               
 [99] lmtest_0.9-40               XVector_0.44.0             
[101] htmltools_0.5.8.1           dotCall64_1.1-1            
[103] Biobase_2.64.0              scales_1.3.0               
[105] png_0.1-8                   doSNOW_1.0.20              
[107] spatstat.univar_3.0-0       bigmemory.sri_0.1.8        
[109] knitr_1.48                  rstudioapi_0.16.0          
[111] reshape2_1.4.4              uuid_1.2-1                 
[113] nlme_3.1-166                proxy_0.4-27               
[115] cachem_1.1.0                zoo_1.8-12                 
[117] stringr_1.5.1               KernSmooth_2.23-24         
[119] miniUI_0.1.1.1              pillar_1.9.0               
[121] grid_4.4.1                  vctrs_0.6.5                
[123] RANN_2.6.2                  promises_1.3.0             
[125] xtable_1.8-4                cluster_2.1.6              
[127] evaluate_0.24.0             cli_3.6.3                  
[129] compiler_4.4.1              rlang_1.1.4                
[131] crayon_1.5.3                future.apply_1.11.2        
[133] labeling_0.4.3              ps_1.7.7                   
[135] getPass_0.2-4               plyr_1.8.9                 
[137] fs_1.6.4                    stringi_1.8.4              
[139] viridisLite_0.4.2           deldir_2.0-4               
[141] munsell_0.5.1               lazyeval_0.2.2             
[143] spatstat.geom_3.3-2         Matrix_1.7-0               
[145] RcppHNSW_0.6.0              patchwork_1.2.0            
[147] sparseMatrixStats_1.16.0    future_1.34.0              
[149] ggplot2_3.5.1               shiny_1.9.1                
[151] highr_0.11                  SummarizedExperiment_1.34.0
[153] ROCR_1.0-11                 igraph_2.0.3               
[155] bslib_0.7.0                </code></pre>
>>>>>>> d89baaa7bfc3bffd762b0db8bac45eeb05443b97
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
