<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Visium HD: colonrectal cancer</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">KODAMA-Analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="Installation.html">Installation</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Single-cell
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="singlecell_datamatrix.html">Data matrix</a>
    </li>
    <li>
      <a href="singlecell_seurat.html">Seurat</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Spatial
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="MERFISH.html">MERFISH (Moffitt et al., 2018)</a>
    </li>
    <li>
      <a href="DLPFC.html">Visium DLPFC (Maynard et al., 2021)</a>
    </li>
    <li>
      <a href="Prostate.html">Visium (10X database - Prostate)</a>
    </li>
    <li>
      <a href="VisiumHD.html">Visium HD (10X database - Colorectal cancer)</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Wrapper
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Seurat.html">Seurat</a>
    </li>
    <li>
      <a href="Giotto.html">Giotto</a>
    </li>
    <li>
      <a href="SpatialExperiment.html">SpatialExperiment</a>
    </li>
  </ul>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/tkcaccia/KODAMA-Analysis">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Visium HD: colonrectal cancer</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<<<<<<< HEAD
<strong>Last updated:</strong> 2025-04-14
=======
<strong>Last updated:</strong> 2025-03-08
>>>>>>> d89baaa7bfc3bffd762b0db8bac45eeb05443b97
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>KODAMA-Analysis/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240618code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240618)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240618code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240618)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<<<<<<< HEAD
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomtkcacciaKODAMAAnalysistree5f5ac634a8274a7b7c4cb315454f9c9aa7a12f86targetblank5f5ac63a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/tree/5f5ac634a8274a7b7c4cb315454f9c9aa7a12f86" target="_blank">5f5ac63</a>
=======
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomtkcacciaKODAMAAnalysistree7de0c7444fdebf22d8827606a2b840417e344179targetblank7de0c74a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/tree/7de0c7444fdebf22d8827606a2b840417e344179" target="_blank">7de0c74</a>
>>>>>>> d89baaa7bfc3bffd762b0db8bac45eeb05443b97
</a>
</p>
</div>
<div
<<<<<<< HEAD
id="strongRepositoryversionstrongahrefhttpsgithubcomtkcacciaKODAMAAnalysistree5f5ac634a8274a7b7c4cb315454f9c9aa7a12f86targetblank5f5ac63a"
=======
id="strongRepositoryversionstrongahrefhttpsgithubcomtkcacciaKODAMAAnalysistree7de0c7444fdebf22d8827606a2b840417e344179targetblank7de0c74a"
>>>>>>> d89baaa7bfc3bffd762b0db8bac45eeb05443b97
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<<<<<<< HEAD
<a href="https://github.com/tkcaccia/KODAMA-Analysis/tree/5f5ac634a8274a7b7c4cb315454f9c9aa7a12f86" target="_blank">5f5ac63</a>.
=======
<a href="https://github.com/tkcaccia/KODAMA-Analysis/tree/7de0c7444fdebf22d8827606a2b840417e344179" target="_blank">7de0c74</a>.
>>>>>>> d89baaa7bfc3bffd762b0db8bac45eeb05443b97
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  KODAMA.svg
<<<<<<< HEAD
    Untracked:  analysis/singlecell_datamatrix.Rmd
    Untracked:  analysis/singlecell_seurat.Rmd
=======
    Untracked:  analysis/old_DLPFC.Rmd
>>>>>>> d89baaa7bfc3bffd762b0db8bac45eeb05443b97
    Untracked:  code/Acinar_Cell_Carcinoma.ipynb
    Untracked:  code/Adenocarcinoma.ipynb
    Untracked:  code/Adjacent_normal_section.ipynb
    Untracked:  code/DLFPC_preprocessing.R
    Untracked:  code/DLPFC - BANKSY.R
    Untracked:  code/DLPFC - BASS.R
    Untracked:  code/DLPFC - BAYESPACE.R
    Untracked:  code/DLPFC - Nonspatial.R
    Untracked:  code/DLPFC - PRECAST.R
    Untracked:  code/DLPFC_comparison.R
    Untracked:  code/DLPFC_results_analysis.R
    Untracked:  code/MERFISH - BANKSY.R
    Untracked:  code/MERFISH - BASS.R
    Untracked:  code/MERFISH - BAYESPACE.R
    Untracked:  code/MERFISH - Nonspatial.R
    Untracked:  code/MERFISH - PRECAST.R
    Untracked:  code/MERFISH_comparison.R
    Untracked:  code/MERFISH_results_analysis.R
    Untracked:  code/VisiumHD-CRC.ipynb
<<<<<<< HEAD
    Untracked:  code/VisiumHDassignment.py
    Untracked:  code/deep learning code DLPFC.R
    Untracked:  code/save tiles.py
=======
    Untracked:  code/deep learning code DLPFC.R
    Untracked:  code/save tiles.py
    Untracked:  data/Adenocarcinoma.csv
>>>>>>> d89baaa7bfc3bffd762b0db8bac45eeb05443b97
    Untracked:  data/Annotations/
    Untracked:  data/DLFPC-Br5292-input.RData
    Untracked:  data/DLFPC-Br5595-input.RData
    Untracked:  data/DLFPC-Br8100-input.RData
    Untracked:  data/DLPFC-general.RData
    Untracked:  data/MERFISH-input.RData
<<<<<<< HEAD
=======
    Untracked:  data/spots_classification_ALL.csv
    Untracked:  data/spots_classification_Acinar_Cell_Carcinoma.csv
    Untracked:  data/spots_classification_IF.csv
    Untracked:  data/spots_classification_Normal_prostate.csv
>>>>>>> d89baaa7bfc3bffd762b0db8bac45eeb05443b97
    Untracked:  data/trajectories.RData
    Untracked:  data/trajectories_VISIUMHD.RData
    Untracked:  output/BANSKY-results.RData
    Untracked:  output/BASS-results.RData
    Untracked:  output/BayesSpace-results.RData
    Untracked:  output/CRC-image.RData
    Untracked:  output/CRC-image2.RData
    Untracked:  output/CRC.png
    Untracked:  output/CRC2.png
    Untracked:  output/CRC7.png
    Untracked:  output/CRC8.png
    Untracked:  output/CRC_boxplot.png
    Untracked:  output/CRC_boxplot.svg
<<<<<<< HEAD
    Untracked:  output/CRC_boxplot2.svg
=======
>>>>>>> d89baaa7bfc3bffd762b0db8bac45eeb05443b97
    Untracked:  output/CRC_linee.svg
    Untracked:  output/DL.RData
    Untracked:  output/DLFPC-All-2.RData
    Untracked:  output/DLFPC-All.RData
    Untracked:  output/DLFPC-Br5292.RData
    Untracked:  output/DLFPC-Br5595.RData
    Untracked:  output/DLFPC-Br8100.RData
<<<<<<< HEAD
    Untracked:  output/DLFPC-variablesXdeeplearning.RData
=======
>>>>>>> d89baaa7bfc3bffd762b0db8bac45eeb05443b97
    Untracked:  output/DLPFC-BANSKY-results.RData
    Untracked:  output/DLPFC-BASS-results.RData
    Untracked:  output/DLPFC-BayesSpace-results.RData
    Untracked:  output/DLPFC-Nonspatial-results.RData
    Untracked:  output/DLPFC-PRECAST-results.RData
<<<<<<< HEAD
    Untracked:  output/DLPFC_all_cluster.svg
    Untracked:  output/DLPFCpathway.RData
=======
    Untracked:  output/DLPFC1.svg
    Untracked:  output/DLPFC_all_cluster.svg
>>>>>>> d89baaa7bfc3bffd762b0db8bac45eeb05443b97
    Untracked:  output/Figure 1 - boxplot.pdf
    Untracked:  output/Figure 2 - DLPFC 10.pdf
    Untracked:  output/Figures/
    Untracked:  output/KODAMA-results.RData
    Untracked:  output/KODAMA_DLPFC_All_original.svg
    Untracked:  output/KODAMA_DLPFC_Br5595.svg
    Untracked:  output/KODAMA_DLPFC_Br5595_slide.svg
<<<<<<< HEAD
    Untracked:  output/Loupe.csv
=======
>>>>>>> d89baaa7bfc3bffd762b0db8bac45eeb05443b97
    Untracked:  output/MERFISH-BANSKY-results.RData
    Untracked:  output/MERFISH-BASS-results.RData
    Untracked:  output/MERFISH-BayesSpace-results.RData
    Untracked:  output/MERFISH-KODAMA-results.RData
<<<<<<< HEAD
    Untracked:  output/MERFISH-Nonspatial-results.RData
    Untracked:  output/MERFISH-PRECAST-results.RData
    Untracked:  output/MERFISH.RData
    Untracked:  output/Nonspatial-results.RData
    Untracked:  output/Prostate-GSEA.csv
    Untracked:  output/Prostate-KODAMA.RData
    Untracked:  output/Prostate-trajectory.csv
    Untracked:  output/Prostate.RData
    Untracked:  output/VisiumHD-RNA.RData
    Untracked:  output/VisiumHD-genes.pdf
    Untracked:  output/VisiumHD.RData
    Untracked:  output/boh.svg
    Untracked:  output/desmoplastic_distance_carcinoma.csv
=======
    Untracked:  output/MERFISH-NONSPATIAL-results.RData
    Untracked:  output/MERFISH-PRECAST-results.RData
    Untracked:  output/MERFISH.RData
    Untracked:  output/Nonspatial-results.RData
    Untracked:  output/Prostate.RData
    Untracked:  output/RGB_DLPFC.png
    Untracked:  output/VisiumHD.RData
    Untracked:  output/VisiumHD3.RData
>>>>>>> d89baaa7bfc3bffd762b0db8bac45eeb05443b97
    Untracked:  output/image.RData
    Untracked:  output/pp.RData
    Untracked:  output/pp2.RData
    Untracked:  output/pp3.RData
    Untracked:  output/pp4.RData
    Untracked:  output/pp5.RData
    Untracked:  output/prostate1.svg
    Untracked:  output/prostate2.svg
    Untracked:  output/prostate3.svg
<<<<<<< HEAD
    Untracked:  output/subclusters1.csv
    Untracked:  output/subclusters2.csv
    Untracked:  output/subclusters3.csv
    Untracked:  output/tight_boundary.geojson
    Untracked:  output/trajectory.csv
=======
    Untracked:  output/prostate4.svg
    Untracked:  output/prostate5.svg
    Untracked:  output/prostate6.svg
    Untracked:  output/prostate7.svg
    Untracked:  output/tight_boundary.geojson
>>>>>>> d89baaa7bfc3bffd762b0db8bac45eeb05443b97

Unstaged changes:
    Deleted:    analysis/D1.Rmd
    Deleted:    analysis/DLPFC-12.Rmd
    Deleted:    analysis/DLPFC-4.Rmd
    Modified:   analysis/DLPFC.Rmd
    Deleted:    analysis/DLPFC1.Rmd
    Deleted:    analysis/DLPFC10.Rmd
    Deleted:    analysis/DLPFC2.Rmd
    Deleted:    analysis/DLPFC3.Rmd
    Deleted:    analysis/DLPFC4.Rmd
    Deleted:    analysis/DLPFC5.Rmd
    Deleted:    analysis/DLPFC6.Rmd
    Deleted:    analysis/DLPFC7.Rmd
    Deleted:    analysis/DLPFC8.Rmd
    Deleted:    analysis/DLPFC9.Rmd
    Deleted:    analysis/Du1.Rmd
    Deleted:    analysis/Du10.Rmd
    Deleted:    analysis/Du11.Rmd
    Deleted:    analysis/Du12.Rmd
    Deleted:    analysis/Du13.Rmd
    Deleted:    analysis/Du14.Rmd
    Deleted:    analysis/Du15.Rmd
    Deleted:    analysis/Du16.Rmd
    Deleted:    analysis/Du17.Rmd
    Deleted:    analysis/Du18.Rmd
    Deleted:    analysis/Du19.Rmd
    Deleted:    analysis/Du2.Rmd
    Deleted:    analysis/Du20.Rmd
    Deleted:    analysis/Du3.Rmd
    Deleted:    analysis/Du4.Rmd
    Deleted:    analysis/Du5.Rmd
    Deleted:    analysis/Du6.Rmd
    Deleted:    analysis/Du7.Rmd
    Deleted:    analysis/Du8.Rmd
    Deleted:    analysis/Du9.Rmd
    Modified:   analysis/Giotto.Rmd
    Modified:   analysis/MERFISH.Rmd
    Deleted:    analysis/MERFISH1a (copy).Rmd
    Deleted:    analysis/MERFISH1a.Rmd
    Deleted:    analysis/MERFISH1b (copy).Rmd
    Deleted:    analysis/MERFISH1b.Rmd
    Deleted:    analysis/MERFISH2a (copy).Rmd
    Deleted:    analysis/MERFISH2a.Rmd
    Deleted:    analysis/MERFISH2b (copy).Rmd
    Deleted:    analysis/MERFISH2b.Rmd
    Deleted:    analysis/MERFISH3a (copy).Rmd
    Deleted:    analysis/MERFISH3a.Rmd
    Deleted:    analysis/MERFISH3b (copy).Rmd
    Deleted:    analysis/MERFISH3b.Rmd
    Deleted:    analysis/MERFISH4a (copy).Rmd
    Deleted:    analysis/MERFISH4a.Rmd
    Deleted:    analysis/MERFISH4b (copy).Rmd
    Deleted:    analysis/MERFISH4b.Rmd
    Modified:   analysis/Prostate.Rmd
    Deleted:    analysis/STARmap.Rmd
<<<<<<< HEAD
    Modified:   analysis/Seurat.Rmd
    Deleted:    analysis/Simulation.Rmd
    Deleted:    analysis/Single-cell.Rmd
    Modified:   analysis/SpatialExperiment.Rmd
    Modified:   analysis/VisiumHD.Rmd
    Modified:   code/VisiumHD_CRC_download.sh
    Deleted:    data/Pathology.csv
    Deleted:    data/merfish.Rmd
    Deleted:    data/vis.R
=======
    Modified:   analysis/VisiumHD.Rmd
    Modified:   code/VisiumHD_CRC_download.sh
    Deleted:    data/Pathology.csv
>>>>>>> d89baaa7bfc3bffd762b0db8bac45eeb05443b97

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/VisiumHD.Rmd</code>) and HTML
(<code>docs/VisiumHD.html</code>) files. If you’ve configured a remote
Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/5b7dd635afa5a03cc9aa0f58e83d4ef96e087923/docs/VisiumHD.html" target="_blank">5b7dd63</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2025-01-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/86be707edff6acecf81558deefc732ce37df9149/analysis/VisiumHD.Rmd" target="_blank">86be707</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2025-01-10
</td>
<td>
Start my new project
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/7bba919f8c11c20ee17dfc45d94a93d6d9caa65e/analysis/VisiumHD.Rmd" target="_blank">7bba919</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2025-01-09
</td>
<td>
Start my new project
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/a423e5fa7c0b9ff8c2e5eb7d2b32a769723816fd/docs/VisiumHD.html" target="_blank">a423e5f</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-09-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/b0a97fe1976826a83414a553f195e6cacadcfd1f/analysis/VisiumHD.Rmd" target="_blank">b0a97fe</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-09-04
</td>
<td>
Start my new project
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/9bdaa70d970d3bd554bbfe0d7219a7def8f60643/docs/VisiumHD.html" target="_blank">9bdaa70</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-09-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/ca72951924783bded55a57e9a7b7af8f22dff40a/analysis/VisiumHD.Rmd" target="_blank">ca72951</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-09-04
</td>
<td>
Start my new project
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/098b08e3a2bdf95f877bd7e921d8123bac8d459d/docs/VisiumHD.html" target="_blank">098b08e</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-09-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/eb8066ed22e3d0e779e93ecfbaa7e86f19ff54d3/analysis/VisiumHD.Rmd" target="_blank">eb8066e</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-09-04
</td>
<td>
Start my new project
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/0010f3c6d5aa17bd4ac1b77564565534156f3a69/docs/VisiumHD.html" target="_blank">0010f3c</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-09-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/3f515c053e2a97d3ffdfa0b1aeb213fa11f40f65/analysis/VisiumHD.Rmd" target="_blank">3f515c0</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-09-04
</td>
<td>
Start my new project
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/51b04520c087f1b30229cf2939f79c567fedcd51/docs/VisiumHD.html" target="_blank">51b0452</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-09-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/c257b0ead34a12bbd4b0bf21e4ffe7e31acbded9/analysis/VisiumHD.Rmd" target="_blank">c257b0e</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-09-03
</td>
<td>
Start my new project
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/22e2ac6686694c36780c6e9f630148f45f38a65f/analysis/VisiumHD.Rmd" target="_blank">22e2ac6</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-08-26
</td>
<td>
Start my new project
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/d1192e95a8ed7985fb6d9678ce0a6d850f882066/docs/VisiumHD.html" target="_blank">d1192e9</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-08-12
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/5ef81484e2db340df5c626c8a7dd61d34f295f48/analysis/VisiumHD.Rmd" target="_blank">5ef8148</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-08-12
</td>
<td>
Start my new project
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/3374e66ce76ec3d26fcb912a9f6eba2f7341ed8f/docs/VisiumHD.html" target="_blank">3374e66</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-08-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/35ce733e4e0012d79f696f9c7dcdbf67e1623e80/docs/VisiumHD.html" target="_blank">35ce733</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-08-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/82fe167363197a5889eb59cb7e39511dfde2f22e/docs/VisiumHD.html" target="_blank">82fe167</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-07-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/b422e4318478a4de3e61332a0acf677640350efb/analysis/VisiumHD.Rmd" target="_blank">b422e43</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-07-24
</td>
<td>
Start my new project
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/6f7daaccbcc7b968bc736664a45f9ebcaf73fba3/docs/VisiumHD.html" target="_blank">6f7daac</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-07-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/5b97082045eb12ccb4f0314d2bed7f29f7e8e863/analysis/VisiumHD.Rmd" target="_blank">5b97082</a>
</td>
<td>
tkcaccia
</td>
<td>
2024-07-15
</td>
<td>
updates
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/7be8f59b0c4d82e877c5dfc8cfbfb7936ceceeb6/analysis/VisiumHD.Rmd" target="_blank">7be8f59</a>
</td>
<td>
tkcaccia
</td>
<td>
2024-07-15
</td>
<td>
updates
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/7be8f59b0c4d82e877c5dfc8cfbfb7936ceceeb6/docs/VisiumHD.html" target="_blank">7be8f59</a>
</td>
<td>
tkcaccia
</td>
<td>
2024-07-15
</td>
<td>
updates
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/79f73a27da5f8795085e9d164b276f9068ff4c0b/analysis/VisiumHD.Rmd" target="_blank">79f73a2</a>
</td>
<td>
GitHub
</td>
<td>
2024-07-14
</td>
<td>
Update VisiumHD.Rmd
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/f8ca54aee4428df50d2cd0f41cbf2a0ceae01d7d/docs/VisiumHD.html" target="_blank">f8ca54a</a>
</td>
<td>
tkcaccia
</td>
<td>
2024-07-14
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/d04c1e788b30e00c60a181bbd58284e3266f7ff4/docs/VisiumHD.html" target="_blank">d04c1e7</a>
</td>
<td>
GitHub
</td>
<td>
2024-07-08
</td>
<td>
Update VisiumHD.html
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/754c8bfca0fc33b0e0e24e8848501bc1dc601465/docs/VisiumHD.html" target="_blank">754c8bf</a>
</td>
<td>
GitHub
</td>
<td>
2024-07-04
</td>
<td>
Update VisiumHD.html
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tkcaccia/KODAMA-Analysis/ee4ee17ff874a26eaf8c1e76e4a85ce3b9de867e/docs/VisiumHD.html" target="_blank">ee4ee17</a>
</td>
<td>
GitHub
</td>
<td>
2024-06-19
</td>
<td>
Add files via upload
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/615fc05e05b60aed64683d1204afddc6b8c32f36/analysis/VisiumHD.Rmd" target="_blank">615fc05</a>
</td>
<td>
GitHub
</td>
<td>
2024-06-19
</td>
<td>
Add files via upload
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>The recently released VisiumHD platform by 10x Genomics significantly
improves spatial transcriptomics resolution by reducing the spot size
from 55 µm to an edge length of just 2 µm. This advancement eliminates
gaps between spots, enabling truly gap-free and bias-free single-cell
resolution. The high-density array allows for flexible data analysis at
multiple resolutions, enabling researchers to tailor spatial granularity
to specific biological questions. In the following analysis, we applied
KODAMA to data at an 8 µm resolution.</p>
</div>
<div id="loading-and-preprocessing-data" class="section level1">
<h1>Loading and Preprocessing Data</h1>
<p>The dataset can be downloaded using the following script: <a
href="https://github.com/tkcaccia/KODAMA-Analysis/blob/main/code/VisiumHD_CRC_download.sh">VisiumHD_CRC_download.sh</a>.
This script provides access to the raw data, which will be preprocessed
and analyzed in the subsequent steps of our pipeline.</p>
<p>The dataset will be then loaded in the R environment using the Seurat
pipeline.</p>
<pre class="r"><code>library(&quot;ggplot2&quot;)
library(&quot;patchwork&quot;)
library(&quot;dplyr&quot;)
library(&quot;Seurat&quot;)
library(&quot;KODAMA&quot;)
library(&quot;KODAMAextra&quot;)
library(&quot;bigmemory&quot;)

localdir=&quot;../Colorectal/outs/&quot;
object &lt;- Load10X_Spatial(data.dir = localdir, bin.size = c(8))</code></pre>
<p>We perform quality control on a spatial transcriptomics dataset by
removing low-quality spots with fewer than 100 UMIs, filtering out
mitochondrial genes, and retaining genes expressed with at least 1 count
in at least 0.5% of spots. The remaining high-quality genes are set as
variable features for downstream analysis.</p>
<pre class="r"><code>nCount_Spatial=colSums(object@assays$Spatial.008um$counts)

image=as.raster(object@images$slice1.008um@image)
save(image,file=&quot;output/CRC-image.RData&quot;)
#object@images$slice1.008um@scale.factors$hires

# plot(image,xlim=c(320,530),ylim=c(200,410))
# points(xy[,2]*0.007973422,nrow(image)-xy[,1]*0.007973422,pch=20)

#xy=as.matrix(GetTissueCoordinates(sp_obj)[,1:2])

#image=as.raster(imgData(object)$data[[1]])

#xy_sel=spatialCoords(spe_sub)
#xy_sel=xy_sel*scaleFactors(spe_sub)
#xy_sel[,2]=nrow(image)-xy_sel[,2]



sp_obj &lt;- subset(
  object,
  subset = nCount_Spatial.008um &gt; 100)

nCount_Spatial=colSums(sp_obj@assays$Spatial.008um$counts)



counts=sp_obj@assays$Spatial.008um$counts
is_mito &lt;- grepl(&quot;(^MT-)|(^mt-)&quot;, rownames(counts))
counts &lt;- counts[!is_mito,]

filter_genes_ncounts=1
filter_genes_pcspots=0.5
nspots &lt;- ceiling(filter_genes_pcspots/100 *  ncol(counts))
ix_remove &lt;- rowSums(counts &gt;= filter_genes_ncounts) &lt;   nspots
counts &lt;- counts[!ix_remove,]

QCgenes &lt;- rownames(counts)

VariableFeatures(sp_obj) = QCgenes

rm(counts)</code></pre>
<p>We prepare the filtered spatial transcriptomics data for
dimensionality reduction using principal component analysis (PCA). We
set the default assay, normalize the data, identify variable features,
and scale the data. We then extract the tissue coordinates and we
perform PCA using the filtered genes, storing the result as “pca.008um”.
Finally, PCA is displayed in a scatterplot.</p>
<pre class="r"><code>DefaultAssay(sp_obj) &lt;- &quot;Spatial.008um&quot;
sp_obj &lt;- NormalizeData(sp_obj)


sp_obj &lt;- FindVariableFeatures(sp_obj)
sp_obj &lt;- ScaleData(sp_obj)

xy=as.matrix(GetTissueCoordinates(sp_obj)[,1:2])

sp_obj &lt;- RunPCA(sp_obj, reduction.name = &quot;pca.008um&quot;)

dim(sp_obj)</code></pre>
<pre><code>[1]  18085 428381</code></pre>
<pre class="r"><code>plot(Seurat::Embeddings(sp_obj, reduction = &quot;pca.008um&quot;))</code></pre>
<<<<<<< HEAD
<p><img src="figure/VisiumHD.Rmd/unnamed-chunk-3-1.png" width="960" style="display: block; margin: auto;" /></p>
=======
<p><img src="figure/VisiumHD.Rmd/unnamed-chunk-1-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-1">
Past versions of unnamed-chunk-1-1.png
</button>
</p>
<div id="fig-unnamed-chunk-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/5b7dd635afa5a03cc9aa0f58e83d4ef96e087923/docs/figure/VisiumHD.Rmd/unnamed-chunk-1-1.png" target="_blank">5b7dd63</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2025-01-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/51b04520c087f1b30229cf2939f79c567fedcd51/docs/figure/VisiumHD.Rmd/unnamed-chunk-1-1.png" target="_blank">51b0452</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-09-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/d1192e95a8ed7985fb6d9678ce0a6d850f882066/docs/figure/VisiumHD.Rmd/unnamed-chunk-1-1.png" target="_blank">d1192e9</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-08-12
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/6f7daaccbcc7b968bc736664a45f9ebcaf73fba3/docs/figure/VisiumHD.Rmd/unnamed-chunk-1-1.png" target="_blank">6f7daac</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-07-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/7be8f59b0c4d82e877c5dfc8cfbfb7936ceceeb6/docs/figure/VisiumHD.Rmd/unnamed-chunk-1-1.png" target="_blank">7be8f59</a>
</td>
<td>
tkcaccia
</td>
<td>
2024-07-15
</td>
</tr>
</tbody>
</table>
>>>>>>> d89baaa7bfc3bffd762b0db8bac45eeb05443b97
</div>
<div id="kodama-analysis" class="section level1">
<h1>KODAMA analysis</h1>
<p>We performed the KODAMA analysis using as input the 50 principal
components of PCA and using 10000 landmarks.</p>
<pre class="r"><code>n.cores=8

sp_obj=RunKODAMAmatrix(sp_obj,
                       reduction = &quot;pca.008um&quot;,
                       landmarks = 10000,
                       n.cores=n.cores,
                       seed = 543210)

config &lt;- umap.defaults
config$n_threads = n.cores
config$n_sgd_threads = &quot;auto&quot;

sp_obj=RunKODAMAvisualization(sp_obj,method=&quot;UMAP&quot;,config=config)


kk_UMAP=Seurat::Embeddings(sp_obj, reduction = &quot;KODAMA&quot;)</code></pre>
</div>
<<<<<<< HEAD
<div id="tissue-annotation" class="section level1">
<h1>Tissue annotation</h1>
<p>The tissue was manually annotated using QuPath software and the
annotations were save in <a
href="https://github.com/tkcaccia/KODAMA-Analysis/blob/main/data/Annotations/Visium_HD_Human_Colon_Cancer_290325.geojson">Visium_HD_Human_Colon_Cancer_290325.geojson</a>.</p>
<p>Using the script <a
href="https://github.com/tkcaccia/KODAMA-Analysis/blob/main/code/VisiumHDassignment.py">VisiumHDassignment.py</a>
the annotations saved as *.geojson were assigned to the Visium spots and
saved in <a
href="https://github.com/tkcaccia/KODAMA-Analysis/blob/main/data/Annotations/spots_classification_VisiumHD.csv">spots_classification_VisiumHD.csv</a>.</p>
<pre class="r"><code>rr=read.csv(&quot;data/Annotations/spots_classification_VisiumHD.csv&quot;,sep=&quot;,&quot;)
=======
<pre class="r"><code>n.cores=8

sp_obj=RunKODAMAmatrix(sp_obj,
                    reduction = &quot;pca.008um&quot;,
                          FUN= &quot;fastpls&quot; ,
                          landmarks = 10000,
                          splitting = 300,
                          ncomp = 50,
                          spatial.resolution = 0.3,
                          n.cores=n.cores,
                          seed = 543210)</code></pre>
<pre><code>Calculating Network

Calculating Network spatial
socket cluster with 8 nodes on host &#39;localhost&#39;
================================================================================
Finished parallel computation

[1] &quot;Calculation of dissimilarity matrix...&quot;
================================================================================</code></pre>
<pre class="r"><code>config &lt;- umap.defaults
config$n_threads = n.cores
config$n_sgd_threads = &quot;auto&quot;

sp_obj=RunKODAMAvisualization(sp_obj,method=&quot;UMAP&quot;,config=config)



#sp_obj &lt;- RunKODAMAmatrix(sp_obj, reduction = &quot;pca.008um&quot;,
#                          FUN= &quot;PLS&quot; ,
#                          landmarks = 10000,
#                          splitting = 100,
#                          f.par.pls = 50,
#                          spatial.resolution = 0.4,
#                          n.cores=8)

#  print(&quot;KODAMA finished&quot;)
  
#     config=umap.defaults
#     config$n_threads = 8
#     config$n_sgd_threads = &quot;auto&quot;
#     sp_obj &lt;- RunKODAMAvisualization(sp_obj, method = &quot;UMAP&quot;,config=config)

     
kk_UMAP=Seurat::Embeddings(sp_obj, reduction = &quot;KODAMA&quot;)

save(kk_UMAP,xy,file=&quot;output/VisiumHD.RData&quot;)

load(&quot;output/VisiumHD.RData&quot;)   

rr=read.csv(&quot;data/spots_classification_ALL.csv&quot;,sep=&quot;,&quot;)
>>>>>>> d89baaa7bfc3bffd762b0db8bac45eeb05443b97
ss=strsplit(rr[,2],&quot;:&quot;)
ss=unlist(lapply(ss, function(x) x[2]))
ss=strsplit(ss,&quot;,&quot;)
ss=unlist(lapply(ss, function(x) x[1]))
ss=gsub(&quot;\&quot;&quot;,&quot;&quot;,ss)

rr[,2]=ss
n=ave(1:length(rr[,1]), rr[,1], FUN = seq_along)
rr=rr[n==1,]
rownames(rr)=rr[,1]
rr=rr[rownames(kk_UMAP),]

rr[,2]=substring(rr[,2],2)
<<<<<<< HEAD
=======
                     
    
  
>>>>>>> d89baaa7bfc3bffd762b0db8bac45eeb05443b97

table(rr[,&quot;classification&quot;])</code></pre>
<pre><code>
            blood vessel   desmoplastic submucosa                dysplasia 
<<<<<<< HEAD
                    1969                    55409                    89290 
dystrophic calcification            exocrine duct          external glands 
                     488                      158                     3108 
            immune cells   intramucosal carcinoma      intratumoral stroma 
                    2713                    69214                    13336 
      invasive carcinoma lamina propria dysplasia  lymphovascular channels 
                   37283                    26834                     1505 
       muscularis mucosa       muscularis propria             nerve fibers 
                    4785                    18023                      457 
            normal gland    normal lamina propria     oedematous submucosa 
                   30199                    16502                     5865 </code></pre>
=======
                    1969                    63892                    89367 
  dysplasia intermediate dystrophic calcification            exocrine duct 
                   69167                      488                      158 
         external glands     high-grade dysplasia             immune cells 
                    3108                      815                     2713 
      invasive carcinoma lamina propria dysplasia  lymphovascular channels 
                   37627                    10271                     1493 
       muscularis mucosa       muscularis propria             nerve fibers 
                    4769                    18023                      457 
            normal gland    normal lamina propria     oedematous submucosa 
                   30196                    16512                     5877 </code></pre>
>>>>>>> d89baaa7bfc3bffd762b0db8bac45eeb05443b97
<pre class="r"><code>library(ggplot2)


cols=sample(rainbow(15))
labels=as.factor(rr[,&quot;classification&quot;])
<<<<<<< HEAD

cols_tissue &lt;- c(&quot;#0000ff&quot;, &quot;#e41a1c&quot;, &quot;#006400&quot;, &quot;#00cc8f&quot; ,&quot;#0088dd&quot;,
                 &quot;#00ff00&quot;, &quot;#b2dfee&quot;,&quot;#669bbc&quot;, &quot;#81b29a&quot;, &quot;#ffd700&quot;,
                 &quot;#adc178&quot;, &quot;#aa1133&quot;, &quot;#1166dc&quot;, &quot;#e5989b&quot;, &quot;#e07a5f&quot;,
                 &quot;#cc00b6&quot;, &quot;#81ccff&quot;, &quot;#f2cc8f&quot;,&quot;#e0aa5f&quot;,&quot;#33b233&quot;, &quot;#aa228f&quot;,&quot;#aa7a6f&quot;)



df &lt;- data.frame(kk_UMAP[,1:2], tissue=labels)
plot1 = ggplot(df, aes(Dimensions_1, Dimensions_2, color = tissue)) +labs(title=&quot;KODAMA&quot;) +
  geom_point(size = 1) +
  theme_bw() + theme(legend.position = &quot;bottom&quot;)+
  scale_color_manual(&quot;Domain&quot;, values = cols_tissue) +
  guides(color = guide_legend(nrow = 4, 
                              override.aes = list(size = 4)))
plot1</code></pre>
<p><img src="figure/VisiumHD.Rmd/unnamed-chunk-6-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="trajectory-analysis" class="section level1">
<h1>Trajectory analysis</h1>
<p>The function new_trajectory allows us to draw manually a trajectory
into the KODAMA plot to identify the gradual changes in the gene
expression. The trajectory were previously drew and saved in the file <a
href="https://github.com/tkcaccia/KODAMA-Analysis/blob/main/data/trajectories_VISIUMHD.RData">trajectories_VISIUMHD.RData</a>.</p>
<pre class="r"><code>par(xpd = T, mar = par()$mar + c(0,0,0,7))



data=sp_obj@assays$Spatial.008um$data[rownames(sp_obj@assays$Spatial.008um$scale.data),]
data=as.matrix(data)</code></pre>
<pre><code>Warning in asMethod(object): sparse-&gt;dense coercion: allocating vector of size
6.4 GiB</code></pre>
<pre class="r"><code>data=t(data)
data=data[,-which(colMeans(data==0)&gt;0.99)]</code></pre>
<pre class="r"><code>load(&quot;data/trajectories_VISIUMHD.RData&quot;)

plot(kk_UMAP,cex=0.5,pch=20,col=cols_tissue[labels])
=======



cols_tissue &lt;- c(&quot;#0000ff&quot;, &quot;#e41a1c&quot;, &quot;#006400&quot;, &quot;#00cc8f&quot; ,&quot;#0088dd&quot;,
                 &quot;#00ff00&quot;, &quot;#b2dfee&quot;,&quot;#669bbc&quot;, &quot;#81b29a&quot;, &quot;#ffd700&quot;,
                 &quot;#adc178&quot;, &quot;#aa1133&quot;, &quot;#1166dc&quot;, &quot;#e5989b&quot;, &quot;#e07a5f&quot;,
                 &quot;#cc00b6&quot;, &quot;#81ccff&quot;, &quot;#f2cc8f&quot;,&quot;#e0aa5f&quot;,&quot;#33b233&quot;, &quot;#aa228f&quot;,&quot;#aa7a6f&quot;)



df &lt;- data.frame(kk_UMAP[,1:2], tissue=labels)
plot1 = ggplot(df, aes(Dimensions_1, Dimensions_2, color = tissue)) +labs(title=&quot;KODAMA&quot;) +
  geom_point(size = 1) +
  theme_bw() + theme(legend.position = &quot;bottom&quot;)+
  scale_color_manual(&quot;Domain&quot;, values = cols_tissue) +
  guides(color = guide_legend(nrow = 4, 
                              override.aes = list(size = 4)))
plot1</code></pre>
<p><img src="figure/VisiumHD.Rmd/unnamed-chunk-2-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-1">
Past versions of unnamed-chunk-2-1.png
</button>
</p>
<div id="fig-unnamed-chunk-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/5b7dd635afa5a03cc9aa0f58e83d4ef96e087923/docs/figure/VisiumHD.Rmd/unnamed-chunk-2-1.png" target="_blank">5b7dd63</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2025-01-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/098b08e3a2bdf95f877bd7e921d8123bac8d459d/docs/figure/VisiumHD.Rmd/unnamed-chunk-2-1.png" target="_blank">098b08e</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-09-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/0010f3c6d5aa17bd4ac1b77564565534156f3a69/docs/figure/VisiumHD.Rmd/unnamed-chunk-2-1.png" target="_blank">0010f3c</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-09-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/51b04520c087f1b30229cf2939f79c567fedcd51/docs/figure/VisiumHD.Rmd/unnamed-chunk-2-1.png" target="_blank">51b0452</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-09-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/d1192e95a8ed7985fb6d9678ce0a6d850f882066/docs/figure/VisiumHD.Rmd/unnamed-chunk-2-1.png" target="_blank">d1192e9</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-08-12
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/82fe167363197a5889eb59cb7e39511dfde2f22e/docs/figure/VisiumHD.Rmd/unnamed-chunk-2-1.png" target="_blank">82fe167</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2024-07-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>png(&quot;output/CRC.png&quot;,height = 2000,width = 2000)
plot1
dev.off()</code></pre>
<pre><code>png 
  2 </code></pre>
<pre class="r"><code>par(xpd = T, mar = par()$mar + c(0,0,0,7))

plot1=plot(kk_UMAP,cex=0.5,pch=20,col=cols_tissue[labels])
>>>>>>> d89baaa7bfc3bffd762b0db8bac45eeb05443b97
legend(max(kk_UMAP[,1])+0.05*dist(range(kk_UMAP[,1])), max(kk_UMAP[,2]),
       levels(labels),
       col = cols,
       cex = 0.8,
       pch=20)
<<<<<<< HEAD
mm1=new_trajectory (kk_UMAP,data = data,trace=tra1$xy)
mm2=new_trajectory (kk_UMAP,data = data,trace=tra2$xy)
mm3=new_trajectory (kk_UMAP,data = data,trace=tra3$xy)</code></pre>
<p><img src="figure/VisiumHD.Rmd/unnamed-chunk-10-1.png" width="960" style="display: block; margin: auto;" /></p>
=======

load(&quot;data/trajectories_VISIUMHD.RData&quot;)

data=sp_obj@assays$Spatial.008um$data[rownames(sp_obj@assays$Spatial.008um$scale.data),]
data=as.matrix(data)</code></pre>
<pre><code>Warning in asMethod(object): sparse-&gt;dense coercion: allocating vector of size
6.4 GiB</code></pre>
<pre class="r"><code>data=t(data)

mm1=new_trajectory (kk_UMAP,data = data,trace=tra1$xy)
mm2=new_trajectory (kk_UMAP,data = data,trace=tra2$xy)
mm3=new_trajectory (kk_UMAP,data = data,trace=tra3$xy)</code></pre>
<p><img src="figure/VisiumHD.Rmd/unnamed-chunk-2-2.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-2">
Past versions of unnamed-chunk-2-2.png
</button>
</p>
<div id="fig-unnamed-chunk-2-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/5b7dd635afa5a03cc9aa0f58e83d4ef96e087923/docs/figure/VisiumHD.Rmd/unnamed-chunk-2-2.png" target="_blank">5b7dd63</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2025-01-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
>>>>>>> d89baaa7bfc3bffd762b0db8bac45eeb05443b97
<pre class="r"><code>traj=rbind(mm1$trajectory,
           mm2$trajectory,
           mm3$trajectory)
y=rep(1:20,3)</code></pre>
<p>The genes were were correlated with the trajectory using the Spearman
correlation test.</p>
<pre class="r"><code>ma=multi_analysis(traj,y,FUN=&quot;correlation.test&quot;,method=&quot;spearman&quot;)
ma=ma[order(as.numeric(ma$`p-value`)),]
colnames(ma)=c(&quot;Feature   &quot;,&quot;rho   &quot;,&quot;p-value   &quot;,&quot;FDR   &quot;)
knitr::kable(ma[1:10,],row.names=FALSE)</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">Feature</th>
<th align="right">rho</th>
<th align="left">p-value</th>
<th align="left">FDR</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<<<<<<< HEAD
<td align="left">LCN2</td>
<td align="right">-0.88</td>
<td align="left">6.92e-21</td>
<td align="left">6.94e-18</td>
</tr>
<tr class="even">
<td align="left">SOD2</td>
<td align="right">-0.81</td>
<td align="left">3.83e-15</td>
<td align="left">1.92e-12</td>
</tr>
<tr class="odd">
<td align="left">CEBPD</td>
<td align="right">-0.81</td>
<td align="left">5.75e-15</td>
<td align="left">1.92e-12</td>
</tr>
<tr class="even">
<td align="left">CXCL3</td>
<td align="right">-0.77</td>
<td align="left">7.34e-13</td>
<td align="left">1.67e-10</td>
</tr>
<tr class="odd">
<td align="left">ID1</td>
<td align="right">-0.77</td>
<td align="left">8.35e-13</td>
<td align="left">1.67e-10</td>
</tr>
<tr class="even">
<td align="left">IL32</td>
<td align="right">-0.75</td>
<td align="left">4.17e-12</td>
<td align="left">6.97e-10</td>
</tr>
<tr class="odd">
<td align="left">PI3</td>
<td align="right">-0.74</td>
<td align="left">1.34e-11</td>
<td align="left">1.91e-09</td>
</tr>
<tr class="even">
<td align="left">CCL20</td>
<td align="right">-0.74</td>
<td align="left">1.73e-11</td>
<td align="left">2.17e-09</td>
</tr>
<tr class="odd">
<td align="left">CXCL1</td>
<td align="right">-0.74</td>
<td align="left">2.16e-11</td>
<td align="left">2.40e-09</td>
</tr>
<tr class="even">
<td align="left">TRIM31</td>
<td align="right">-0.73</td>
<td align="left">2.89e-11</td>
<td align="left">2.90e-09</td>
</tr>
</tbody>
</table>
<div id="validation-of-the-results-in-the-coad-tcga-cohort"
class="section level2">
<h2>Validation of the results in the COAD TCGA cohort</h2>
<p>The downregulation of CXCL3 across the progression of the carcinoma
was validated using the RNAseq data of the COAD TGCA cohort. Clinical
and gene expression data were downloaded from <a
href="http://firebrowse.org/">FireBrowse</a>.</p>
<pre class="r"><code># install.packages(&quot;readxl&quot;)
library(readxl)

# Read in Clinical Data:
coad=read.csv(&quot;../TCGA/COAD/COAD.clin.merged.picked.txt&quot;,sep=&quot;\t&quot;,check.names = FALSE, row.names = 1)
=======
<td align="left">TRIM31</td>
<td align="right">-0.83</td>
<td align="left">1.41e-16</td>
<td align="left">1.89e-13</td>
</tr>
<tr class="even">
<td align="left">CEMIP</td>
<td align="right">-0.82</td>
<td align="left">1.51e-15</td>
<td align="left">7.47e-13</td>
</tr>
<tr class="odd">
<td align="left">LYZ</td>
<td align="right">0.82</td>
<td align="left">1.67e-15</td>
<td align="left">7.47e-13</td>
</tr>
<tr class="even">
<td align="left">COL3A1</td>
<td align="right">0.80</td>
<td align="left">1.05e-14</td>
<td align="left">3.51e-12</td>
</tr>
<tr class="odd">
<td align="left">C1QC</td>
<td align="right">0.80</td>
<td align="left">1.8e-14</td>
<td align="left">4.45e-12</td>
</tr>
<tr class="even">
<td align="left">JCHAIN</td>
<td align="right">0.80</td>
<td align="left">1.99e-14</td>
<td align="left">4.45e-12</td>
</tr>
<tr class="odd">
<td align="left">DDIT4</td>
<td align="right">-0.80</td>
<td align="left">3.05e-14</td>
<td align="left">5.84e-12</td>
</tr>
<tr class="even">
<td align="left">PLAUR</td>
<td align="right">-0.79</td>
<td align="left">8.77e-14</td>
<td align="left">1.47e-11</td>
</tr>
<tr class="odd">
<td align="left">MMP2</td>
<td align="right">0.78</td>
<td align="left">1.17e-13</td>
<td align="left">1.75e-11</td>
</tr>
<tr class="even">
<td align="left">IGKC</td>
<td align="right">0.78</td>
<td align="left">1.32e-13</td>
<td align="left">1.77e-11</td>
</tr>
<tr class="odd">
<td align="left">A2M</td>
<td align="right">0.77</td>
<td align="left">8.76e-13</td>
<td align="left">1.07e-10</td>
</tr>
<tr class="even">
<td align="left">C1QA</td>
<td align="right">0.77</td>
<td align="left">1.03e-12</td>
<td align="left">1.11e-10</td>
</tr>
<tr class="odd">
<td align="left">C1QB</td>
<td align="right">0.77</td>
<td align="left">1.07e-12</td>
<td align="left">1.11e-10</td>
</tr>
<tr class="even">
<td align="left">ADM</td>
<td align="right">-0.76</td>
<td align="left">1.46e-12</td>
<td align="left">1.39e-10</td>
</tr>
<tr class="odd">
<td align="left">IGHG1</td>
<td align="right">0.76</td>
<td align="left">2.24e-12</td>
<td align="left">2.00e-10</td>
</tr>
<tr class="even">
<td align="left">CD74</td>
<td align="right">0.76</td>
<td align="left">2.58e-12</td>
<td align="left">2.16e-10</td>
</tr>
<tr class="odd">
<td align="left">PRSS22</td>
<td align="right">-0.76</td>
<td align="left">3.06e-12</td>
<td align="left">2.41e-10</td>
</tr>
<tr class="even">
<td align="left">RNF186</td>
<td align="right">-0.75</td>
<td align="left">5.15e-12</td>
<td align="left">3.83e-10</td>
</tr>
<tr class="odd">
<td align="left">EFNA1</td>
<td align="right">-0.75</td>
<td align="left">5.88e-12</td>
<td align="left">4.14e-10</td>
</tr>
<tr class="even">
<td align="left">HILPDA</td>
<td align="right">-0.75</td>
<td align="left">6.77e-12</td>
<td align="left">4.54e-10</td>
</tr>
<tr class="odd">
<td align="left">RHPN2</td>
<td align="right">-0.75</td>
<td align="left">7.6e-12</td>
<td align="left">4.85e-10</td>
</tr>
<tr class="even">
<td align="left">PFKFB4</td>
<td align="right">-0.74</td>
<td align="left">9.03e-12</td>
<td align="left">5.50e-10</td>
</tr>
<tr class="odd">
<td align="left">ITPKC</td>
<td align="right">-0.74</td>
<td align="left">9.73e-12</td>
<td align="left">5.67e-10</td>
</tr>
<tr class="even">
<td align="left">ITGA2</td>
<td align="right">-0.74</td>
<td align="left">1.3e-11</td>
<td align="left">7.26e-10</td>
</tr>
<tr class="odd">
<td align="left">ANKRD37</td>
<td align="right">-0.74</td>
<td align="left">1.69e-11</td>
<td align="left">9.05e-10</td>
</tr>
<tr class="even">
<td align="left">GGH</td>
<td align="right">-0.73</td>
<td align="left">2.48e-11</td>
<td align="left">1.28e-09</td>
</tr>
<tr class="odd">
<td align="left">COL6A2</td>
<td align="right">0.73</td>
<td align="left">3.87e-11</td>
<td align="left">1.92e-09</td>
</tr>
<tr class="even">
<td align="left">MS4A7</td>
<td align="right">0.73</td>
<td align="left">5.1e-11</td>
<td align="left">2.44e-09</td>
</tr>
<tr class="odd">
<td align="left">VEGFA</td>
<td align="right">-0.72</td>
<td align="left">1e-10</td>
<td align="left">4.62e-09</td>
</tr>
<tr class="even">
<td align="left">MS4A6A</td>
<td align="right">0.72</td>
<td align="left">1.23e-10</td>
<td align="left">5.49e-09</td>
</tr>
</tbody>
</table>
<div id="mirseq-analysis" class="section level1">
<h1>miRseq Analysis:</h1>
<p><strong>Analysing miRseq Gene Expression Data from a Colerectal
Adenocarcinoma Cohort:</strong></p>
<pre class="r"><code># install.packages(&quot;readxl&quot;)
library(readxl)</code></pre>
<p><strong><em>Prepare Clinical Data:</em></strong></p>
<pre class="r"><code># Read in Clinical Data:
coad=read.csv(&quot;../TCGA/COAD/COAD.clin.merged.picked.txt&quot;,sep=&quot;\t&quot;,check.names = FALSE, row.names = 1)

>>>>>>> d89baaa7bfc3bffd762b0db8bac45eeb05443b97
coad &lt;- as.data.frame(coad) 

# Clean column names: replace dots with dashes &amp; convert to uppercase
colnames(coad) = toupper(colnames(coad))

 # Transpose the dataframe so that rows become columns and vice versa
coad = t(coad) </code></pre>
<<<<<<< HEAD
<p><strong><em>Prepare RNA-seq expression data:</em></strong></p>
=======
<p><strong><em>Prepare miRNA-seq expression data:</em></strong></p>
>>>>>>> d89baaa7bfc3bffd762b0db8bac45eeb05443b97
<pre class="r"><code># Read RNA-seq expression data:
r = read.csv(&quot;../TCGA/COAD/COAD.rnaseqv2__illuminahiseq_rnaseqv2__unc_edu__Level_3__RSEM_genes_normalized__data.data.txt&quot;, sep = &quot;\t&quot;, check.names = FALSE, row.names = 1)
# Remove the first row:
r = r[-1,]
# Convert expression data to numeric matrix format
temp = matrix(as.numeric(as.matrix(r)), ncol=ncol(r))
colnames(temp) = colnames(r)  
rownames(temp) = rownames(r)  
RNA = temp  

# Transpose the matrix so that genes are rows and samples are columns
RNA = t(RNA)  </code></pre>
<p><strong><em>Extract patient and tissue information from column
names:</em></strong></p>
<pre class="r"><code>tcgaID = list()
 # Extract sample ID
tcgaID$sample.ID &lt;- substr(colnames(r), 1, 16)
# Extract patient ID
tcgaID$patient &lt;- substr(colnames(r), 1, 12)  
# Extract tissue type
tcgaID$tissue &lt;- substr(colnames(r), 14, 16)  

tcgaID = as.data.frame(tcgaID)  </code></pre>
<p><strong><em>Select Primary Solid Tumor tissue data
(“01A”):</em></strong></p>
<pre class="r"><code>sel=tcgaID$tissue == &quot;01A&quot;
tcgaID.sel = tcgaID[sel, ]

# Subset the RNA expression data to match selected samples
RNA.sel = RNA[sel, ]</code></pre>
<p><strong><em>Intersect patient IDs between clinical and RNA
data:</em></strong></p>
<pre class="r"><code>sel = intersect(tcgaID.sel$patient, rownames(coad))
# Subset the clinical data to include only selected patients:
coad.sel = coad[sel, ]
# Assign patient IDs as row names to the RNA data:
rownames(RNA.sel) = tcgaID.sel$patient
# Subset the RNA data to include only selected patients
RNA.sel = RNA.sel[sel, ]</code></pre>
<p><strong><em>Prepare labels for pathology stages:</em></strong></p>
<<<<<<< HEAD
<p>The tumor samples were classified based on their T stage: -
<code>t1</code>, <code>t2</code>, &amp; <code>t3</code> as “low” -
<code>t4</code>, <code>t4a</code>, &amp; <code>t4b</code> as “high” -
<code>tis</code> stages to <code>NA</code></p>
<pre class="r"><code>labelsTCGA = coad.sel[, &quot;pathology_T_stage&quot;]
labelsTCGA[labelsTCGA %in% c(&quot;t1&quot;, &quot;t2&quot;, &quot;t3&quot;, &quot;tis&quot;)] = &quot;low&quot;
labelsTCGA[labelsTCGA %in% c(&quot;t4&quot;, &quot;t4a&quot;, &quot;t4b&quot;)] = &quot;high&quot;
table(labelsTCGA)</code></pre>
<pre><code>labelsTCGA
high  low 
  38  242 </code></pre>
=======
<ul>
<li><p>Classify stages <code>t1</code>, <code>t2</code>, &amp;
<code>t3</code> as “low”</p></li>
<li><p>Classify stages <code>t4</code>, <code>t4a</code>, &amp;
<code>t4b</code> as “high”</p></li>
<li><p>Convert any <code>tis</code> stages to <code>NA</code></p></li>
</ul>
<pre class="r"><code>labelsTCGA = coad.sel[, &quot;pathology_T_stage&quot;]
labelsTCGA[labelsTCGA %in% c(&quot;t1&quot;, &quot;t2&quot;, &quot;t3&quot;, &quot;tis&quot;)] = &quot;low&quot;
labelsTCGA[labelsTCGA %in% c(&quot;t4&quot;, &quot;t4a&quot;, &quot;t4b&quot;)] = &quot;high&quot;</code></pre>
<p><strong><em>Log Transform the expression data for our selected gene
<code>CXCL2</code>:</em></strong></p>
<pre class="r"><code>CXCL2 &lt;- log(1 + RNA.sel[, &quot;CXCL2|2920&quot;])
LCN2 &lt;- log(1 + RNA.sel[,&quot;LCN2|3934&quot; ])</code></pre>
>>>>>>> d89baaa7bfc3bffd762b0db8bac45eeb05443b97
<p><strong><em>Boxplot to visualize the distribution of log transformed
gene expression by pathology stage:</em></strong></p>
<pre class="r"><code>colors=c(&quot;#0073c2bb&quot;,&quot;#efc000bb&quot;,&quot;#868686bb&quot;,&quot;#cd534cbb&quot;,&quot;#7aabdcbb&quot;,&quot;#003c67bb&quot;)

library(ggpubr)
<<<<<<< HEAD

gene.selected=&quot;CXCL3&quot;
gene.selected.RNA=colnames(RNA.sel)[pmatch(gene.selected,colnames(RNA.sel))]

CXCL3 &lt;- log(1 + RNA.sel[, gene.selected.RNA])
df=data.frame(variable=CXCL3,labels=labelsTCGA)
=======
df=data.frame(variable=CXCL2,labels=labelsTCGA)
>>>>>>> d89baaa7bfc3bffd762b0db8bac45eeb05443b97

my_comparisons=list()
my_comparisons[[1]]=c(1,2)

Nplot1=ggboxplot(df, x = &quot;labels&quot;, y = &quot;variable&quot;,fill=&quot;labels&quot;,
                 width = 0.8,
                 palette=colors,
                 add = &quot;jitter&quot;,            
                 add.params = list(size = 2, jitter = 0.2,fill=&quot;#ff0000aa&quot;, shape=21))+  
<<<<<<< HEAD
  ylab(&quot;CXCL3 gene expression (FPKM)&quot;)+ xlab(&quot;&quot;)+
  stat_compare_means(comparisons = my_comparisons,method=&quot;wilcox.test&quot;)

Nplot1</code></pre>
<p><img src="figure/VisiumHD.Rmd/unnamed-chunk-19-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>xy2=xy
xy2[,1]=xy[,2]
xy2[,2]=-xy[,1]

df &lt;- data.frame(xy2, tissue=labels)
plot2 = ggplot(df, aes(x, y, color = tissue)) +labs(title=&quot;KODAMA&quot;) +
  geom_point(size = 1) +
  theme_bw() + theme(legend.position = &quot;bottom&quot;)+
  scale_color_manual(&quot;Domain&quot;, values = cols_tissue) +
  guides(color = guide_legend(nrow = 4, 
                              override.aes = list(size = 4)))
plot2</code></pre>
<p><img src="figure/VisiumHD.Rmd/unnamed-chunk-21-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="proximity-analysis" class="section level1">
<h1>Proximity analysis</h1>
<p>The gene expression of the desmoplastic submucosa are analyzed. The
genes with gene expression correlates with the distance from the
invasive carcinoma are identified.</p>
<pre class="r"><code>sel_desmoplastic_submucosa=which(labels==&quot;desmoplastic submucosa&quot;)
xy_desmoplastic_submucosa=xy[sel_desmoplastic_submucosa,]
data_desmoplastic_submucosa=data[sel_desmoplastic_submucosa,]
data_desmoplastic_submucosa=data_desmoplastic_submucosa[,-which(colMeans(data_desmoplastic_submucosa==0)&gt;0.95)]
dim(data_desmoplastic_submucosa)</code></pre>
<pre><code>[1] 55409   201</code></pre>
<pre class="r"><code>sel_invasive_carcinoma=which(labels==&quot;invasive carcinoma&quot; | labels==&quot;intramucosal carcinoma&quot;)
xy_invasive_carcinoma=xy[sel_invasive_carcinoma,]

knn=Rnanoflann::nn(xy_invasive_carcinoma,xy_desmoplastic_submucosa,1)   

y=knn$distances[,1]</code></pre>
<pre class="r"><code># Define custom intervals
break_points &lt;-c(quantile(y,probs=c(seq(0,1,0.005))))

# Convert continuous data to intervals
distance_binned &lt;- cut(y, breaks = break_points)

gene_binned=apply(data_desmoplastic_submucosa,2,function(x)  tapply(x,distance_binned,mean))
break_points=break_points[-length(break_points)]


ma=multi_analysis(gene_binned,break_points,FUN=&quot;correlation.test&quot;,method=&quot;MINE&quot;)
ma=ma[order(as.numeric(ma$MIC),decreasing = TRUE),]
rownames(ma)=ma[,&quot;Feature&quot;]



knitr::kable(ma[1:20,],row.names=FALSE)</code></pre>
<table>
<thead>
=======
  ylab(&quot;CXCL2 gene expression (FPKM)&quot;)+ xlab(&quot;&quot;)+
  stat_compare_means(comparisons = my_comparisons,method=&quot;wilcox.test&quot;)

Nplot1</code></pre>
<p><img src="figure/VisiumHD.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-13-1">
Past versions of unnamed-chunk-13-1.png
</button>
</p>
<div id="fig-unnamed-chunk-13-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/5b7dd635afa5a03cc9aa0f58e83d4ef96e087923/docs/figure/VisiumHD.Rmd/unnamed-chunk-13-1.png" target="_blank">5b7dd63</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2025-01-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>svg(&quot;output/CRC_boxplot.svg&quot;)
Nplot1
dev.off()</code></pre>
<pre><code>png 
  2 </code></pre>
<pre class="r"><code>xy2=xy
xy2[,1]=xy[,2]
xy2[,2]=-xy[,1]
plot(xy2,col=cols_tissue[labels],pch=20,cex=0.5)</code></pre>
<p><img src="figure/VisiumHD.Rmd/unnamed-chunk-14-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-1">
Past versions of unnamed-chunk-14-1.png
</button>
</p>
<div id="fig-unnamed-chunk-14-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/5b7dd635afa5a03cc9aa0f58e83d4ef96e087923/docs/figure/VisiumHD.Rmd/unnamed-chunk-14-1.png" target="_blank">5b7dd63</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2025-01-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>df &lt;- data.frame(xy2, tissue=labels)
plot2 = ggplot(df, aes(x, y, color = tissue)) +labs(title=&quot;KODAMA&quot;) +
  geom_point(size = 1) +
  theme_bw() + theme(legend.position = &quot;bottom&quot;)+
  scale_color_manual(&quot;Domain&quot;, values = cols_tissue) +
  guides(color = guide_legend(nrow = 4, 
                              override.aes = list(size = 4)))
plot2</code></pre>
<p><img src="figure/VisiumHD.Rmd/unnamed-chunk-14-2.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-2">
Past versions of unnamed-chunk-14-2.png
</button>
</p>
<div id="fig-unnamed-chunk-14-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/5b7dd635afa5a03cc9aa0f58e83d4ef96e087923/docs/figure/VisiumHD.Rmd/unnamed-chunk-14-2.png" target="_blank">5b7dd63</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2025-01-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>png(&quot;output/CRC2.png&quot;,height = 2000,width = 2000)
plot2
dev.off()

sel_desmoplastic_submucosa=which(labels==&quot;desmoplastic submucosa&quot;)
kk_desmoplastic_submucosa=kk_UMAP[sel_desmoplastic_submucosa,]
xy_desmoplastic_submucosa=xy2[sel_desmoplastic_submucosa,]


g &lt;- bluster::makeSNNGraph(as.matrix(kk_desmoplastic_submucosa), k = 20)
g_walk &lt;- igraph::cluster_louvain(g,resolution = 0.005)
clu = g_walk$membership
names(clu)=rownames(kk_desmoplastic_submucosa)



df &lt;- data.frame(kk_desmoplastic_submucosa[,1:2], tissue=as.factor(clu))
plot3 = ggplot(df, aes(Dimensions_1, Dimensions_2, color = tissue)) +labs(title=&quot;KODAMA&quot;) +
  geom_point(size = 1) +
  theme_bw() + theme(legend.position = &quot;bottom&quot;)+
  scale_color_manual(&quot;Domain&quot;, values = cols_tissue) +
  guides(color = guide_legend(nrow = 4, 
                              override.aes = list(size = 4)))
plot3</code></pre>
<p><img src="figure/VisiumHD.Rmd/unnamed-chunk-14-3.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-3">
Past versions of unnamed-chunk-14-3.png
</button>
</p>
<div id="fig-unnamed-chunk-14-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/5b7dd635afa5a03cc9aa0f58e83d4ef96e087923/docs/figure/VisiumHD.Rmd/unnamed-chunk-14-3.png" target="_blank">5b7dd63</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2025-01-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>png(&quot;output/CRC7.png&quot;,height = 2000,width = 2000)
plot3
dev.off()


df &lt;- data.frame(xy_desmoplastic_submucosa, tissue=as.factor(clu))
plot4 = ggplot(df, aes(x, y, color = tissue)) +labs(title=&quot;KODAMA&quot;) +
  geom_point(size = 1) +
  theme_bw() + theme(legend.position = &quot;bottom&quot;)+
  scale_color_manual(&quot;Domain&quot;, values = cols_tissue) +
  guides(color = guide_legend(nrow = 4, 
                              override.aes = list(size = 4)))
plot4</code></pre>
<p><img src="figure/VisiumHD.Rmd/unnamed-chunk-14-4.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-4">
Past versions of unnamed-chunk-14-4.png
</button>
</p>
<div id="fig-unnamed-chunk-14-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/5b7dd635afa5a03cc9aa0f58e83d4ef96e087923/docs/figure/VisiumHD.Rmd/unnamed-chunk-14-4.png" target="_blank">5b7dd63</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2025-01-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>png(&quot;output/CRC8.png&quot;,height = 2000,width = 2000)
plot4
dev.off()



sel_desmoplastic_submucosa_selected=names(which(clu==names(which.max(table(clu)))))
kk_desmoplastic_submucosa_selected=kk_UMAP[sel_desmoplastic_submucosa_selected,]
xy_desmoplastic_submucosa_selected=xy2[sel_desmoplastic_submucosa_selected,]
data_desmoplastic_submucosa_selected=data[sel_desmoplastic_submucosa_selected,]
data_desmoplastic_submucosa_selected=data_desmoplastic_submucosa_selected[,-which(colMeans(data_desmoplastic_submucosa_selected==0)&gt;0.95)]

sel_invasive_carcinoma=which(labels==&quot;invasive carcinoma&quot; | labels==&quot;intermediate dysplasia&quot;)
kk_invasive_carcinoma=kk_UMAP[sel_invasive_carcinoma,]
xy_invasive_carcinoma=xy2[sel_invasive_carcinoma,]

knn=Rnanoflann::nn(xy_invasive_carcinoma,xy_desmoplastic_submucosa_selected,1)  

y=knn$distances[,1]



ma=multi_analysis(data_desmoplastic_submucosa_selected,y,FUN=&quot;correlation.test&quot;,method=&quot;spearman&quot;)
ma=ma[order(abs(as.numeric(ma$rho)),decreasing = TRUE),]
colnames(ma)=c(&quot;Feature   &quot;,&quot;rho   &quot;,&quot;p-value   &quot;,&quot;FDR   &quot;)



# 2) Define custom intervals
break_points &lt;-c(quantile(y,probs=c(seq(0,1,0.005))))

# 3) Convert continuous data to intervals
distance_binned &lt;- cut(y, breaks = break_points)

gene_binned=apply(data_desmoplastic_submucosa_selected,2,function(x)  tapply(x,distance_binned,mean))
break_points=break_points[-length(break_points)]

ma=multi_analysis(gene_binned,break_points,FUN=&quot;correlation.test&quot;,method=&quot;MINE&quot;)
ma=ma[order(as.numeric(ma$MIC),decreasing = TRUE),]
ma[1:10,]
rownames(ma)=ma[,&quot;Feature&quot;]
#plot(knn$distances,PMdata[,3])

zmax=NULL
for(i in 1:ncol(gene_binned)){
 df=data.frame(x=break_points,y=gene_binned[,i])
 ll=loess(y~x,data = df,span = 0.3)
 z=predict(ll,newdata = data.frame(x=break_points))
 zmax[i]=break_points[which.max(z)]
# points(break_points,z,type=&quot;l&quot;,col=2)
}
genes=colnames(gene_binned)
names(zmax)=genes


plot(log(1+zmax[genes]),ma[genes,]$MIC,cex=0.2)
text(log(1+zmax[genes]),ma[genes,]$MIC,labels=genes,cex=0.7)</code></pre>
<p><img src="figure/VisiumHD.Rmd/unnamed-chunk-14-5.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-5">
Past versions of unnamed-chunk-14-5.png
</button>
</p>
<div id="fig-unnamed-chunk-14-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/5b7dd635afa5a03cc9aa0f58e83d4ef96e087923/docs/figure/VisiumHD.Rmd/unnamed-chunk-14-5.png" target="_blank">5b7dd63</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2025-01-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot(log(1+break_points),gene_binned[,1],ylim=c(0,1),type=&quot;n&quot;)
 df=data.frame(x=break_points,
               HTRA3=gene_binned[,&quot;HTRA3&quot;],
               IGFBP5=gene_binned[,&quot;IGFBP5&quot;],
               CXCL14=gene_binned[,&quot;CXCL14&quot;],
               MMP11=gene_binned[,&quot;MMP11&quot;],
               TIMP3=gene_binned[,&quot;TIMP3&quot;],
               MGP=gene_binned[,&quot;MGP&quot;])
 
 ll=loess(IGFBP5~x,data = df,span = 0.3)
 IGFBP5=predict(ll,newdata = data.frame(x=break_points))
 
 ll=loess(CXCL14~x,data = df,span = 0.3)
 CXCL14=predict(ll,newdata = data.frame(x=break_points))
 
 ll=loess(MMP11~x,data = df,span = 0.3)
 MMP11=predict(ll,newdata = data.frame(x=break_points))
 
 ll=loess(TIMP3~x,data = df,span = 0.3)
 TIMP3=predict(ll,newdata = data.frame(x=break_points))
 
 ll=loess(HTRA3~x,data = df,span = 0.3)
 HTRA3=predict(ll,newdata = data.frame(x=break_points))
 
 ll=loess(MGP~x,data = df,span = 0.3)
 MGP=predict(ll,newdata = data.frame(x=break_points))
 points(log(1+break_points),HTRA3/max(HTRA3),type=&quot;l&quot;,col=2,lwd=3)
 points(log(1+break_points),IGFBP5/max(IGFBP5),type=&quot;l&quot;,col=3,lwd=3)
 points(log(1+break_points),CXCL14/max(CXCL14),type=&quot;l&quot;,col=4,lwd=3)
 points(log(1+break_points),MMP11/max(MMP11),type=&quot;l&quot;,col=5,lwd=3)
 points(log(1+break_points),TIMP3/max(TIMP3),type=&quot;l&quot;,col=6,lwd=3)
 points(log(1+break_points),MGP/max(MGP),type=&quot;l&quot;,col=7,lwd=3)</code></pre>
<p><img src="figure/VisiumHD.Rmd/unnamed-chunk-14-6.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-6">
Past versions of unnamed-chunk-14-6.png
</button>
</p>
<div id="fig-unnamed-chunk-14-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tkcaccia/KODAMA-Analysis/blob/5b7dd635afa5a03cc9aa0f58e83d4ef96e087923/docs/figure/VisiumHD.Rmd/unnamed-chunk-14-6.png" target="_blank">5b7dd63</a>
</td>
<td>
Stefano Cacciatore
</td>
<td>
2025-01-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>svg(&quot;output/CRC_linee.svg&quot;)
plot(log(1+break_points),gene_binned[,1],ylim=c(0,1),type=&quot;n&quot;)

 points(log(1+break_points),HTRA3/max(HTRA3),type=&quot;l&quot;,col=2,lwd=3)
 points(log(1+break_points),IGFBP5/max(IGFBP5),type=&quot;l&quot;,col=3,lwd=3)
 points(log(1+break_points),CXCL14/max(CXCL14),type=&quot;l&quot;,col=4,lwd=3)
 points(log(1+break_points),MMP11/max(MMP11),type=&quot;l&quot;,col=5,lwd=3)
 points(log(1+break_points),TIMP3/max(TIMP3),type=&quot;l&quot;,col=6,lwd=3)
 points(log(1+break_points),MGP/max(MGP),type=&quot;l&quot;,col=7,lwd=3)

dev.off()</code></pre>
<pre class="r"><code>knitr::kable(ma[1:30,],row.names=FALSE)</code></pre>
<table>
<thead>
>>>>>>> d89baaa7bfc3bffd762b0db8bac45eeb05443b97
<tr class="header">
<th align="left">Feature</th>
<th align="right">MIC</th>
<th align="left">p-value</th>
<th align="left">FDR</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<<<<<<< HEAD
<td align="left">IGFBP5</td>
<td align="right">1.00</td>
<td align="left">3.9e-239</td>
<td align="left">3.92e-237</td>
</tr>
<tr class="even">
<td align="left">HTRA3</td>
<td align="right">1.00</td>
<td align="left">9.1e-261</td>
<td align="left">1.83e-258</td>
</tr>
<tr class="odd">
<td align="left">MGP</td>
<td align="right">0.99</td>
<td align="left">7.32e-191</td>
<td align="left">3.68e-189</td>
</tr>
<tr class="even">
<td align="left">TIMP3</td>
<td align="right">0.92</td>
<td align="left">1.88e-171</td>
<td align="left">6.31e-170</td>
</tr>
<tr class="odd">
<td align="left">GREM1</td>
<td align="right">0.91</td>
<td align="left">4.3e-206</td>
<td align="left">2.88e-204</td>
</tr>
<tr class="even">
<td align="left">IGFBP3</td>
<td align="right">0.89</td>
<td align="left">4.32e-176</td>
<td align="left">1.74e-174</td>
</tr>
<tr class="odd">
<td align="left">SFRP4</td>
<td align="right">0.86</td>
<td align="left">1.84e-170</td>
<td align="left">5.29e-169</td>
</tr>
<tr class="even">
<td align="left">CXCL14</td>
<td align="right">0.84</td>
<td align="left">9.84e-148</td>
<td align="left">1.80e-146</td>
</tr>
<tr class="odd">
<td align="left">COL1A1</td>
<td align="right">0.83</td>
<td align="left">7.49e-152</td>
<td align="left">1.67e-150</td>
</tr>
<tr class="even">
<td align="left">MMP11</td>
<td align="right">0.80</td>
<td align="left">1.07e-141</td>
<td align="left">1.66e-140</td>
</tr>
<tr class="odd">
<td align="left">SPARC</td>
<td align="right">0.76</td>
<td align="left">1.99e-128</td>
<td align="left">2.50e-127</td>
</tr>
<tr class="even">
<td align="left">AEBP1</td>
<td align="right">0.76</td>
<td align="left">6.24e-99</td>
<td align="left">5.23e-98</td>
</tr>
<tr class="odd">
<td align="left">CCDC80</td>
<td align="right">0.75</td>
<td align="left">2.47e-129</td>
<td align="left">3.31e-128</td>
</tr>
<tr class="even">
<td align="left">SFRP2</td>
<td align="right">0.75</td>
<td align="left">3.47e-90</td>
<td align="left">2.49e-89</td>
</tr>
<tr class="odd">
<td align="left">ISLR</td>
<td align="right">0.75</td>
<td align="left">3.13e-161</td>
<td align="left">7.88e-160</td>
</tr>
<tr class="even">
<td align="left">COL1A2</td>
<td align="right">0.74</td>
<td align="left">7.6e-102</td>
<td align="left">7.27e-101</td>
</tr>
<tr class="odd">
<td align="left">DCN</td>
<td align="right">0.74</td>
<td align="left">5.3e-98</td>
<td align="left">4.26e-97</td>
</tr>
<tr class="even">
<td align="left">COL3A1</td>
<td align="right">0.73</td>
<td align="left">6.96e-132</td>
<td align="left">9.99e-131</td>
</tr>
<tr class="odd">
<td align="left">COL14A1</td>
<td align="right">0.73</td>
<td align="left">2.01e-100</td>
<td align="left">1.83e-99</td>
</tr>
<tr class="even">
<td align="left">A2M</td>
<td align="right">0.73</td>
<td align="left">2.73e-99</td>
<td align="left">2.38e-98</td>
</tr>
</tbody>
</table>
<pre class="r"><code> df=data.frame(x=break_points,
               HTRA3=gene_binned[,&quot;HTRA3&quot;],
               IGFBP5=gene_binned[,&quot;IGFBP5&quot;],
               CXCL14=gene_binned[,&quot;CXCL14&quot;],
               MMP11=gene_binned[,&quot;MMP11&quot;],
               TIMP3=gene_binned[,&quot;TIMP3&quot;],
               MGP=gene_binned[,&quot;MGP&quot;],
               GREM1=gene_binned[,&quot;GREM1&quot;],
               IGFBP3=gene_binned[,&quot;IGFBP3&quot;],
               SFRP4=gene_binned[,&quot;SFRP4&quot;])
 
 ll=loess(IGFBP5~x,data = df,span = 0.3)
 IGFBP5=predict(ll,newdata = data.frame(x=break_points))
 
 ll=loess(CXCL14~x,data = df,span = 0.3)
 CXCL14=predict(ll,newdata = data.frame(x=break_points))
 
 ll=loess(MMP11~x,data = df,span = 0.3)
 MMP11=predict(ll,newdata = data.frame(x=break_points))
 
 ll=loess(TIMP3~x,data = df,span = 0.3)
 TIMP3=predict(ll,newdata = data.frame(x=break_points))
 
 ll=loess(HTRA3~x,data = df,span = 0.3)
 HTRA3=predict(ll,newdata = data.frame(x=break_points))
 
 ll=loess(MGP~x,data = df,span = 0.3)
 MGP=predict(ll,newdata = data.frame(x=break_points))
 
 
 ll=loess(GREM1~x,data = df,span = 0.3)
 GREM1=predict(ll,newdata = data.frame(x=break_points))
 
 
 ll=loess(IGFBP3~x,data = df,span = 0.3)
 IGFBP3=predict(ll,newdata = data.frame(x=break_points))
 
 
 ll=loess(SFRP4~x,data = df,span = 0.3)
 SFRP4=predict(ll,newdata = data.frame(x=break_points))
 
 
 
cols_lines &lt;- c( &quot;#006400&quot;, &quot;#00cc8f&quot; ,&quot;#0088dd&quot;,
                  &quot;#b2dfee&quot;, &quot;#ffd700&quot;,   &quot;#adc178&quot;, &quot;#aa1133&quot;, &quot;#e07a5f&quot;,
                 &quot;#cc00b6&quot;, &quot;#f2cc8f&quot;, &quot;#aa228f&quot;,&quot;#aa7a6f&quot;)

plot(log(1+break_points),gene_binned[,1],ylim=c(0,1),type=&quot;n&quot;)

 
 
 points(log(1+break_points),HTRA3/max(HTRA3),type=&quot;l&quot;,col=cols_lines[1],lwd=3)
 points(log(1+break_points),IGFBP5/max(IGFBP5),type=&quot;l&quot;,col=cols_lines[2],lwd=3)
 points(log(1+break_points),CXCL14/max(CXCL14),type=&quot;l&quot;,col=cols_lines[3],lwd=3)
 points(log(1+break_points),MMP11/max(MMP11),type=&quot;l&quot;,col=cols_lines[4],lwd=3)
 points(log(1+break_points),TIMP3/max(TIMP3),type=&quot;l&quot;,col=cols_lines[5],lwd=3)
 points(log(1+break_points),MGP/max(MGP),type=&quot;l&quot;,col=cols_lines[6],lwd=3)
 points(log(1+break_points),GREM1/max(GREM1),type=&quot;l&quot;,col=cols_lines[7],lwd=3)
 points(log(1+break_points),IGFBP3/max(IGFBP3),type=&quot;l&quot;,col=cols_lines[8],lwd=3)
 points(log(1+break_points),SFRP4/max(SFRP4),type=&quot;l&quot;,col=cols_lines[9],lwd=3)</code></pre>
<p><img src="figure/VisiumHD.Rmd/unnamed-chunk-26-1.png" width="672" style="display: block; margin: auto;" /></p>
=======
<td align="left">SPARC</td>
<td align="right">0.97</td>
<td align="left">2.92e-216</td>
<td align="left">2.08e-214</td>
</tr>
<tr class="even">
<td align="left">IGFBP3</td>
<td align="right">0.97</td>
<td align="left">4.8e-242</td>
<td align="left">5.11e-240</td>
</tr>
<tr class="odd">
<td align="left">MGP</td>
<td align="right">0.97</td>
<td align="left">7.93e-189</td>
<td align="left">2.11e-187</td>
</tr>
<tr class="even">
<td align="left">COL11A1</td>
<td align="right">0.93</td>
<td align="left">9.14e-157</td>
<td align="left">9.74e-156</td>
</tr>
<tr class="odd">
<td align="left">COL1A1</td>
<td align="right">0.93</td>
<td align="left">9.27e-187</td>
<td align="left">2.19e-185</td>
</tr>
<tr class="even">
<td align="left">HTRA3</td>
<td align="right">0.91</td>
<td align="left">1.18e-215</td>
<td align="left">6.29e-214</td>
</tr>
<tr class="odd">
<td align="left">COMP</td>
<td align="right">0.91</td>
<td align="left">1.54e-159</td>
<td align="left">1.93e-158</td>
</tr>
<tr class="even">
<td align="left">COL3A1</td>
<td align="right">0.90</td>
<td align="left">2.05e-209</td>
<td align="left">8.75e-208</td>
</tr>
<tr class="odd">
<td align="left">SFRP4</td>
<td align="right">0.90</td>
<td align="left">4.64e-279</td>
<td align="left">9.88e-277</td>
</tr>
<tr class="even">
<td align="left">AEBP1</td>
<td align="right">0.90</td>
<td align="left">2.63e-159</td>
<td align="left">3.12e-158</td>
</tr>
<tr class="odd">
<td align="left">IGFBP5</td>
<td align="right">0.89</td>
<td align="left">2.17e-199</td>
<td align="left">7.71e-198</td>
</tr>
<tr class="even">
<td align="left">CTHRC1</td>
<td align="right">0.89</td>
<td align="left">1.1e-162</td>
<td align="left">1.47e-161</td>
</tr>
<tr class="odd">
<td align="left">GREM1</td>
<td align="right">0.88</td>
<td align="left">1.42e-172</td>
<td align="left">2.32e-171</td>
</tr>
<tr class="even">
<td align="left">TIMP3</td>
<td align="right">0.88</td>
<td align="left">3.73e-196</td>
<td align="left">1.14e-194</td>
</tr>
<tr class="odd">
<td align="left">CXCL14</td>
<td align="right">0.86</td>
<td align="left">7.92e-181</td>
<td align="left">1.62e-179</td>
</tr>
<tr class="even">
<td align="left">COL1A2</td>
<td align="right">0.86</td>
<td align="left">7.86e-167</td>
<td align="left">1.12e-165</td>
</tr>
<tr class="odd">
<td align="left">A2M</td>
<td align="right">0.86</td>
<td align="left">7.38e-168</td>
<td align="left">1.12e-166</td>
</tr>
<tr class="even">
<td align="left">LUM</td>
<td align="right">0.86</td>
<td align="left">1.37e-152</td>
<td align="left">1.27e-151</td>
</tr>
<tr class="odd">
<td align="left">ELN</td>
<td align="right">0.85</td>
<td align="left">5.31e-155</td>
<td align="left">5.39e-154</td>
</tr>
<tr class="even">
<td align="left">EPAS1</td>
<td align="right">0.82</td>
<td align="left">5.34e-147</td>
<td align="left">4.74e-146</td>
</tr>
<tr class="odd">
<td align="left">CD74</td>
<td align="right">0.82</td>
<td align="left">1.13e-145</td>
<td align="left">9.63e-145</td>
</tr>
<tr class="even">
<td align="left">LYZ</td>
<td align="right">0.81</td>
<td align="left">3.16e-144</td>
<td align="left">2.59e-143</td>
</tr>
<tr class="odd">
<td align="left">EFEMP1</td>
<td align="right">0.80</td>
<td align="left">6.87e-141</td>
<td align="left">5.42e-140</td>
</tr>
<tr class="even">
<td align="left">RGS5</td>
<td align="right">0.79</td>
<td align="left">8.38e-181</td>
<td align="left">1.62e-179</td>
</tr>
<tr class="odd">
<td align="left">THBS2</td>
<td align="right">0.79</td>
<td align="left">1.86e-105</td>
<td align="left">1.02e-104</td>
</tr>
<tr class="even">
<td align="left">ANGPTL2</td>
<td align="right">0.79</td>
<td align="left">9.27e-105</td>
<td align="left">4.94e-104</td>
</tr>
<tr class="odd">
<td align="left">ISLR</td>
<td align="right">0.79</td>
<td align="left">2.24e-95</td>
<td align="left">1.08e-94</td>
</tr>
<tr class="even">
<td align="left">MMP11</td>
<td align="right">0.79</td>
<td align="left">4.44e-159</td>
<td align="left">4.98e-158</td>
</tr>
<tr class="odd">
<td align="left">DCN</td>
<td align="right">0.78</td>
<td align="left">1.58e-174</td>
<td align="left">2.81e-173</td>
</tr>
<tr class="even">
<td align="left">MMP2</td>
<td align="right">0.77</td>
<td align="left">2.87e-124</td>
<td align="left">1.91e-123</td>
</tr>
</tbody>
</table>
>>>>>>> d89baaa7bfc3bffd762b0db8bac45eeb05443b97
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<<<<<<< HEAD
<pre><code>R version 4.4.3 (2025-02-28)
=======
<pre><code>R version 4.4.2 (2024-10-31)
>>>>>>> d89baaa7bfc3bffd762b0db8bac45eeb05443b97
Platform: x86_64-pc-linux-gnu
Running under: Ubuntu 20.04.6 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0 
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

time zone: Etc/UTC
tzcode source: system (glibc)

attached base packages:
[1] parallel  stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
<<<<<<< HEAD
 [1] ggpubr_0.6.0       readxl_1.4.5       bigmemory_4.6.4    KODAMAextra_1.2   
 [5] e1071_1.7-16       doParallel_1.0.17  iterators_1.0.14   foreach_1.5.2     
 [9] KODAMA_3.0         Matrix_1.7-3       umap_0.2.10.0      Rtsne_0.17        
[13] minerva_1.5.10     Seurat_5.2.1       SeuratObject_5.0.2 sp_2.2-0          
[17] dplyr_1.1.4        patchwork_1.3.0    ggplot2_3.5.1      workflowr_1.7.1   

loaded via a namespace (and not attached):
  [1] RcppAnnoy_0.0.22       splines_4.4.3          later_1.4.1           
  [4] tibble_3.2.1           cellranger_1.1.0       polyclip_1.10-7       
  [7] fastDummies_1.7.5      lifecycle_1.0.4        tcltk_4.4.3           
 [10] rstatix_0.7.2          rprojroot_2.0.4        globals_0.16.3        
 [13] processx_3.8.6         Rnanoflann_0.0.3       lattice_0.22-7        
 [16] hdf5r_1.3.12           MASS_7.3-65            backports_1.5.0       
 [19] magrittr_2.0.3         plotly_4.10.4          sass_0.4.9            
 [22] rmarkdown_2.29         jquerylib_0.1.4        yaml_2.3.10           
 [25] httpuv_1.6.15          sctransform_0.4.1      spam_2.11-1           
 [28] askpass_1.2.1          spatstat.sparse_3.1-0  reticulate_1.42.0     
 [31] cowplot_1.1.3          pbapply_1.7-2          RColorBrewer_1.1-3    
 [34] abind_1.4-8            purrr_1.0.4            misc3d_0.9-1          
 [37] git2r_0.33.0           ggrepel_0.9.6          irlba_2.3.5.1         
 [40] listenv_0.9.1          spatstat.utils_3.1-3   goftest_1.2-3         
 [43] RSpectra_0.16-2        spatstat.random_3.3-3  fitdistrplus_1.2-2    
 [46] parallelly_1.43.0      codetools_0.2-20       tidyselect_1.2.1      
 [49] farver_2.1.2           matrixStats_1.5.0      spatstat.explore_3.4-2
 [52] jsonlite_2.0.0         Formula_1.2-5          progressr_0.15.1      
 [55] ggridges_0.5.6         survival_3.8-3         tools_4.4.3           
 [58] ica_1.0-3              Rcpp_1.0.14            glue_1.8.0            
 [61] gridExtra_2.3          xfun_0.51              withr_3.0.2           
 [64] fastmap_1.2.0          openssl_2.3.2          callr_3.7.6           
 [67] digest_0.6.37          R6_2.6.1               mime_0.13             
 [70] colorspace_2.1-1       scattermore_1.2        tensor_1.5            
 [73] spatstat.data_3.1-6    tidyr_1.3.1            generics_0.1.3        
 [76] data.table_1.17.0      class_7.3-23           httr_1.4.7            
 [79] htmlwidgets_1.6.4      whisker_0.4.1          uwot_0.2.3            
 [82] pkgconfig_2.0.3        gtable_0.3.6           lmtest_0.9-40         
 [85] htmltools_0.5.8.1      carData_3.0-5          dotCall64_1.2         
 [88] scales_1.3.0           png_0.1-8              spatstat.univar_3.1-2 
 [91] bigmemory.sri_0.1.8    knitr_1.50             rstudioapi_0.17.1     
 [94] reshape2_1.4.4         uuid_1.2-1             nlme_3.1-168          
 [97] proxy_0.4-27           cachem_1.1.0           zoo_1.8-13            
[100] stringr_1.5.1          KernSmooth_2.23-26     miniUI_0.1.1.1        
[103] arrow_19.0.1           pillar_1.10.1          grid_4.4.3            
[106] vctrs_0.6.5            RANN_2.6.2             promises_1.3.2        
[109] car_3.1-3              xtable_1.8-4           cluster_2.1.8.1       
[112] evaluate_1.0.3         cli_3.6.4              compiler_4.4.3        
[115] rlang_1.1.5            future.apply_1.11.3    ggsignif_0.6.4        
[118] labeling_0.4.3         ps_1.9.0               getPass_0.2-4         
[121] plyr_1.8.9             fs_1.6.5               stringi_1.8.7         
[124] viridisLite_0.4.2      deldir_2.0-4           assertthat_0.2.1      
[127] munsell_0.5.1          lazyeval_0.2.2         spatstat.geom_3.3-6   
[130] RcppHNSW_0.6.0         bit64_4.6.0-1          future_1.34.0         
[133] shiny_1.10.0           ROCR_1.0-11            igraph_2.1.4          
[136] broom_1.0.8            bslib_0.9.0            bit_4.6.0             </code></pre>
=======
 [1] ggpubr_0.6.0       readxl_1.4.3       bigmemory_4.6.4    KODAMAextra_1.2   
 [5] e1071_1.7-16       doParallel_1.0.17  iterators_1.0.14   foreach_1.5.2     
 [9] KODAMA_3.0         Matrix_1.7-1       umap_0.2.10.0      Rtsne_0.17        
[13] minerva_1.5.10     Seurat_5.1.0       SeuratObject_5.0.2 sp_2.1-4          
[17] dplyr_1.1.4        patchwork_1.3.0    ggplot2_3.5.1      workflowr_1.7.1   

loaded via a namespace (and not attached):
  [1] RcppAnnoy_0.0.22       splines_4.4.2          later_1.4.0           
  [4] tibble_3.2.1           cellranger_1.1.0       polyclip_1.10-7       
  [7] fastDummies_1.7.4      lifecycle_1.0.4        rstatix_0.7.2         
 [10] tcltk_4.4.2            rprojroot_2.0.4        globals_0.16.3        
 [13] processx_3.8.5         Rnanoflann_0.0.3       lattice_0.22-6        
 [16] hdf5r_1.3.11           MASS_7.3-61            backports_1.5.0       
 [19] magrittr_2.0.3         plotly_4.10.4          sass_0.4.9            
 [22] rmarkdown_2.29         jquerylib_0.1.4        yaml_2.3.10           
 [25] httpuv_1.6.15          sctransform_0.4.1      spam_2.11-0           
 [28] askpass_1.2.1          spatstat.sparse_3.1-0  reticulate_1.38.0     
 [31] cowplot_1.1.3          pbapply_1.7-2          RColorBrewer_1.1-3    
 [34] abind_1.4-8            purrr_1.0.2            BiocGenerics_0.50.0   
 [37] misc3d_0.9-1           git2r_0.33.0           S4Vectors_0.42.1      
 [40] ggrepel_0.9.6          irlba_2.3.5.1          listenv_0.9.1         
 [43] spatstat.utils_3.1-2   goftest_1.2-3          RSpectra_0.16-2       
 [46] spatstat.random_3.3-2  fitdistrplus_1.2-2     parallelly_1.41.0     
 [49] leiden_0.4.3.1         codetools_0.2-20       tidyselect_1.2.1      
 [52] farver_2.1.2           stats4_4.4.2           matrixStats_1.4.1     
 [55] spatstat.explore_3.3-4 jsonlite_1.8.9         BiocNeighbors_1.22.0  
 [58] Formula_1.2-5          progressr_0.14.0       ggridges_0.5.6        
 [61] survival_3.8-3         tools_4.4.2            snow_0.4-4            
 [64] ica_1.0-3              Rcpp_1.0.13-1          glue_1.8.0            
 [67] gridExtra_2.3          xfun_0.49              withr_3.0.1           
 [70] fastmap_1.2.0          bluster_1.14.0         fansi_1.0.6           
 [73] openssl_2.2.2          callr_3.7.6            digest_0.6.37         
 [76] R6_2.5.1               mime_0.12              colorspace_2.1-1      
 [79] scattermore_1.2        tensor_1.5             spatstat.data_3.1-4   
 [82] utf8_1.2.4             tidyr_1.3.1            generics_0.1.3        
 [85] data.table_1.15.4      class_7.3-23           httr_1.4.7            
 [88] htmlwidgets_1.6.4      whisker_0.4.1          uwot_0.2.2            
 [91] pkgconfig_2.0.3        gtable_0.3.5           lmtest_0.9-40         
 [94] htmltools_0.5.8.1      carData_3.0-5          dotCall64_1.2         
 [97] scales_1.3.0           png_0.1-8              doSNOW_1.0.20         
[100] spatstat.univar_3.1-1  bigmemory.sri_0.1.8    knitr_1.49            
[103] rstudioapi_0.17.1      reshape2_1.4.4         uuid_1.2-1            
[106] nlme_3.1-166           proxy_0.4-27           cachem_1.1.0          
[109] zoo_1.8-12             stringr_1.5.1          KernSmooth_2.23-26    
[112] miniUI_0.1.1.1         vipor_0.4.7            arrow_16.1.0          
[115] ggrastr_1.0.2          pillar_1.9.0           grid_4.4.2            
[118] vctrs_0.6.5            RANN_2.6.2             promises_1.3.2        
[121] car_3.1-3              xtable_1.8-4           cluster_2.1.8         
[124] beeswarm_0.4.0         evaluate_1.0.1         cli_3.6.3             
[127] compiler_4.4.2         rlang_1.1.4            ggsignif_0.6.4        
[130] future.apply_1.11.3    labeling_0.4.3         ps_1.8.1              
[133] getPass_0.2-4          plyr_1.8.9             fs_1.6.5              
[136] ggbeeswarm_0.7.2       stringi_1.8.4          BiocParallel_1.38.0   
[139] viridisLite_0.4.2      deldir_2.0-4           assertthat_0.2.1      
[142] munsell_0.5.1          lazyeval_0.2.2         spatstat.geom_3.3-4   
[145] RcppHNSW_0.6.0         bit64_4.6.0-1          future_1.34.0         
[148] shiny_1.10.0           ROCR_1.0-11            broom_1.0.7           
[151] igraph_2.0.3           bslib_0.8.0            bit_4.5.0.1           </code></pre>
>>>>>>> d89baaa7bfc3bffd762b0db8bac45eeb05443b97
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
