---
title: "About"
output:
  workflowr::wflow_html:
    toc: true
editor_options:
  chunk_output_type: console
---

# Giotto Suite

## Introduction to Giotto 

Giotto Suite is a collection of open-source software tools, including data structures and methods, for the comprehensive analysis and visualization of spatial multi-omics data at multiple scales and resolutions. More information can be found [here](https://drieslab.github.io/Giotto_website/).

## Dataset Description

The data in this tutorial originates from a Visium Spatial Gene Expression slide of the adult mouse. This dataset is available on the 10X Genomics support site and can be downloaded using the following [code](code/Giotto_mouse_brain.sh).

## Library loading

```{r, fig.width=10, fig.height=4, warning=FALSE, message=FALSE}
library(KODAMA)
library(KODAMAextra)
library(Giotto)
```
## Dataset loading 
```{r, fig.width=10, fig.height=4, warning=FALSE, message=FALSE}

instrs = createGiottoInstructions(save_dir = '../Temporary',
                                  save_plot = FALSE,
                                  show_plot = TRUE,
                                  python_path = NULL)


## directly from visium folder
visium_brain = createGiottoVisiumObject(visium_dir = "../Giotto_Mouse_brain/",
                                        expr_data = 'raw',
                                        png_name = 'tissue_lowres_image.png',
                                        gene_column_index = 2,
                                        instructions = instrs,
                                        verbose = FALSE)


## check metadata
pDataDT(visium_brain)

## show plot
spatPlot2D(gobject = visium_brain, cell_color = 'in_tissue', point_size = 2,
           cell_color_code = c('0' = 'lightgrey', '1' = 'blue'))

```
###Loading and Preparing Data
Create Giotto Visium Object
```{r, fig.width=10, fig.height=4, warning=FALSE, message=FALSE}
# Provide path to Visium data folder
data_path <- '../Giotto_Mouse_brain/'

# Create Giotto Visium object
visium_brain <- createGiottoVisiumObject(visium_dir = data_path,
                                         expr_data = 'raw',
                                         png_name = 'tissue_lowres_image.png',
                                         gene_column_index = 2,
                                         instructions = instrs)
```
## Preprocessing
```{r, fig.width=10, fig.height=4, warning=FALSE, message=FALSE}
## subset on spots that were covered by tissue
metadata = pDataDT(visium_brain)
in_tissue_barcodes = metadata[in_tissue == 1]$cell_ID
visium_brain = subsetGiotto(visium_brain, cell_ids = in_tissue_barcodes)

## filter
visium_brain <- filterGiotto(gobject = visium_brain,
                             expression_threshold = 1,
                             feat_det_in_min_cells = 50,
                             min_det_feats_per_cell = 1000,
                             expression_values = c('raw'),
                             verbose = FALSE)
```
## Normalization and feature reduction
```{r, fig.width=10, fig.height=4, warning=FALSE, message=FALSE}

## normalize
visium_brain <- normalizeGiotto(gobject = visium_brain, scalefactor = 6000, verbose = FALSE)

## add gene & cell statistics
visium_brain <- addStatistics(gobject = visium_brain)


visium_brain <- calculateHVF(gobject = visium_brain)



gene_metadata = fDataDT(visium_brain)
featgenes = gene_metadata[hvf == 'yes' & perc_cells > 3 & mean_expr_det > 0.4]$feat_ID

## run PCA on expression values (default)
visium_brain <- runPCA(gobject = visium_brain, feats_to_use = featgenes)

```
## KODAMA analysis

```{r, fig.width=10, fig.height=4, warning=FALSE, message=FALSE}
visium_brain=RunKODAMAmatrix(visium_brain, f.par.pls = 50,FUN="PLS",n.cores=4)
visium_brain=RunKODAMAvisualization(visium_brain,method="UMAP")
```

## Clustering and visualization

```{r, fig.width=10, fig.height=4, warning=FALSE, message=FALSE}

visium_brain <- createNearestNetwork(gobject = visium_brain,dim_reduction_to_use = "KODAMA", dim_reduction_name="KODAMA",dimensions_to_use = 1:2, k = 15)

## Leiden clustering
visium_brain <- doLeidenCluster(gobject = visium_brain, resolution = 0.5, n_iterations = 1000,network_name = "sNN.KODAMA")

dimPlot2D(gobject = visium_brain, dim_reduction_to_use ="KODAMA", dim_reduction_name="KODAMA",cell_color = 'leiden_clus',point_size = 2)
spatPlot2D(gobject = visium_brain,cell_color = 'leiden_clus',point_size = 2.5)
```

