---
title: "About"
output:
  workflowr::wflow_html:
    toc: true
editor_options:
  chunk_output_type: console
---

# MERFISH Data Analysis

# Introduction

Spatial transcriptomics is a powerful technique for studying the spatial organization of gene expression within tissues.
Here, we apply KODAMA to analyze the spatial transcriptomic data that measured the mouse preoptic region of the hypothalamus
using the MERFISH technology from Moffitt et al., 2018. [Link to study](https://www.science.org/doi/10.1126/science.aau5324).
We focus on the tissue sections Bregma -0.04, -0.09, -0.14, -0.19, and -0.24 mm from a consecutive brain hypothalamic region of animal 1.
The original data can be downloaded from [Dryad](https://datadryad.org/stash/dataset/doi:10.5061/dryad.8t8s248).

# Loading and Preprocessing Data

```{r, message=FALSE, warning=FALSE}
# Load the necessary libraries
library(rgl)
library(irlba)
library(KODAMAextra)
library(scater)
library(SPARK)
library(ggplot2)
library(plotly)
library(mclust)

n.cores=16
```


```{r, message=FALSE, warning=FALSE}
# Load the MERFISH data
load("../MERFISH_Animal1.RData")

cols <- c("#669bbc", "#81b29a", "#f2cc8f", "#adc178",
          "#dde5b6", "#a8dadc", "#e5989b", "#e07a5f")

# Define the slides to be analyzed
slides <- c("-0.04", "-0.09", "-0.14", "-0.19", "-0.24")

# Initialize variables
xyz <- NULL
tissue_segments <- NULL
cell_type <- NULL
RNA <- NULL

# Extract spatial and expression data from each slide
for (i in slides) {
  x <- info_mult[[i]]$x / 1000
  y <- info_mult[[i]]$y / 1000
  z <- as.numeric(i)
  slide_xyz <- cbind(x - min(x), y - min(y), z)
  xyz <- rbind(xyz, slide_xyz)
  tissue_segments <- c(tissue_segments, info_mult[[i]]$z)
  cell_type <- c(cell_type, info_mult[[i]]$Cell_class)
  RNA <- rbind(RNA, t(cnts_mult[[i]]))
}

# Normalize RNA counts
RNA <- t(normalizeCounts(t(RNA), log = TRUE))

# Convert tissue segments to factor with defined levels
tissue_segments <- factor(tissue_segments, levels = c("V3", "BST", "fx", "MPA", "MPN", "PV", "PVH", "PVT"))

# Convert xyz to numeric matrix
xyz <- matrix(as.numeric(as.matrix(xyz)), ncol = ncol(xyz))
```

# Identifying Differentially Expressed Genes

```{r, fig.width=10, fig.height=4,results = "hide"}
# Initialize matrix to store p-values

top=multi_SPARKX(RNA,xyz[,-3],as.factor(xyz[,3]),n.cores = n.cores)
```

# Passing message


```{r, message=FALSE, warning=FALSE}


passing.message = 
  function (data, spatial, number_knn = 15, quantile = 0.5, n.cores = 1) 
  {
    data=as.matrix(data)
    spatial=as.matrix(spatial)
    # Get dimensions of the input data
    nspots = nrow(data)
    nvariables = ncol(data)

    # Initialize result matrix
    data.2 <- matrix(0, nrow = nspots, ncol = nvariables)
    knn=Rnanoflann::nn(spatial,spatial,number_knn)
    for(h in 1:nspots){       
      # Find nearest neighbors using spatial information
      
      
      # Initialize a temporary vector for computations
      temp = rep(0, nvariables)
      RNA.temp = data[knn$indices[h,], ]
      knn_gene = Rnanoflann::nn(RNA.temp, RNA.temp[1, , drop = FALSE], round(number_knn * quantile))$indices

      knn$distances=knn$distances/max(knn$distances)
      # Compute weighted sum for each variable
      for (i in 1:number_knn) {
        temp = temp + RNA.temp[i, ]/(1 + knn$distances[h,i]^2)
      }
      # Return the computed row
      data.2[h, ] <-temp
    }

    rownames(data.2) = rownames(data)
    colnames(data.2) = colnames(data)
    data.2
  }

RNA.PM=passing.message(RNA[,top[1:100]],xyz)

```

## Dimensionality reduction

```{r, message=FALSE, warning=FALSE}
RNA.PM.scaled=scale(RNA.PM)
pca_results <- irlba(A = RNA.PM.scaled, nv = 50)
pca.PM <- pca_results$u %*% diag(pca_results$d)

RNA.scaled=scale(RNA)
pca_results <- irlba(A = RNA.scaled, nv = 50)
pca <- pca_results$u %*% diag(pca_results$d)

plot(pca.PM,col=tissue_segments)

par(mfrow = c(1, 2))
plot(pca, col = cols[tissue_segments], main = "KODAMA tissue",pch=20)
plot(pca.PM, col = cols[tissue_segments], main = "KODAMA Clusters",pch=20)


uu=umap(pca[,1:30])$layout
uu.PM=umap(pca.PM[,1:10])$layout
par(mfrow = c(1, 2))
plot(uu, col = cols[tissue_segments], main = "KODAMA tissue",pch=20)
plot(uu.PM, col = cols[tissue_segments], main = "KODAMA Clusters",pch=20)


```

# Applying KODAMA

```{r, message=FALSE, warning=FALSE}
# Apply KODAMA to the PCA results
jj=KODAMA.matrix.parallel(pca[,1:10],
                          spatial = xyz,
                          FUN= "fastpls" ,
                          landmarks = 100000,
                          splitting = 300,
                          ncomp = 50,
                          spatial.resolution = 0.3,
                          n.cores=n.cores,
                          seed = 543210)

print("KODAMA finished")
           

config=umap.defaults
config$n_neighbors=30
config$n_threads = n.cores
vis <- KODAMA.visualization(jj, method = "UMAP",config=config)

g <- bluster::makeSNNGraph(as.matrix(vis), k = 100)
g_louvain <- igraph::cluster_louvain(g,resolution = 0.2)
clu = g_louvain$membership



```



# Refinement and Visualization

```{r, message=FALSE, warning=FALSE}
ref=refine_SVM(xyz,clu,cost=10000)
```

```{r, fig.width=10, fig.height=4, warning=FALSE, message=FALSE, echo=FALSE}
save(ref,tissue_segments,vis,xyz,pca,pca.PM,file="output/MERFISH.RData")
```

```{r, message=FALSE, warning=FALSE}
ARI=NULL
slides=unique(xyz[,3])
for(i in 1:length(slides)){
  sel=slides[i]==xyz[,3]
  ARI[i]=adjustedRandIndex(ref[sel], tissue_segments[sel])
}
print(ARI)

# Plot the clusters
par(mfrow = c(1, 2))
plot(vis, col = cols[tissue_segments], main = "KODAMA tissue",pch=20)
plot(vis, col = cols[clu], main = "KODAMA Clusters",pch=20)


plot_slide(xyz[,-3],as.factor(xyz[,3]),ref,nrow=1,col=cols)


```
# 3D Visualization for Slice -0.14

```{r, message=FALSE, warning=FALSE}


library(rgl)

library(MASS)

library(misc3d)

#save(vis,xyz,ref,cols,file="MERFISH-test.RData")


volume_rendering(xyz[!is.na(ref),],ref[!is.na(ref)],selection = c("V3","PVH","PV","fx"),alpha = c(0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8) ,colors = cols,cells=c(20, 20, 20),level=exp(2.4))


```
