---
title: "About"
output:
  workflowr::wflow_html:
    toc: true
editor_options:
  chunk_output_type: console
---

# MERFISH Data Analysis

# Introduction

Spatial transcriptomics is a powerful technique for studying the spatial organization of gene expression within tissues.
Here, we apply KODAMA to analyze the spatial transcriptomic data that measured the mouse preoptic region of the hypothalamus
using the MERFISH technology from Moffitt et al., 2018. [Link to study](https://www.science.org/doi/10.1126/science.aau5324).
We focus on the tissue sections Bregma -0.04, -0.09, -0.14, -0.19, and -0.24 mm from a consecutive brain hypothalamic region of animal 1.
The original data can be downloaded from [Dryad](https://datadryad.org/stash/dataset/doi:10.5061/dryad.8t8s248).

# Loading and Preprocessing Data

```{r load-data, message=FALSE, warning=FALSE}
# Load the necessary libraries
library(rgl)
library(irlba)
library(KODAMAextra)
library(scater)
library(SPARK)
library(ggplot2)
library(plotly)
library(mclust)

# Load the MERFISH data
load("../data/MERFISH_Animal1.RData")

# Define the slides to be analyzed
slides <- c("-0.04", "-0.09", "-0.14", "-0.19", "-0.24")

# Initialize variables
xyz <- NULL
tissue_segments <- NULL
cell_type <- NULL
RNA <- NULL

# Extract spatial and expression data from each slide
for (i in slides) {
  x <- info_mult[[i]]$x / 1000
  y <- info_mult[[i]]$y / 1000
  z <- as.numeric(i)
  slide_xyz <- cbind(x - min(x), y - min(y), z)
  xyz <- rbind(xyz, slide_xyz)
  tissue_segments <- c(tissue_segments, info_mult[[i]]$z)
  cell_type <- c(cell_type, info_mult[[i]]$Cell_class)
  RNA <- rbind(RNA, t(cnts_mult[[i]]))
}

# Normalize RNA counts
RNA <- t(normalizeCounts(t(RNA), log = TRUE))

# Convert tissue segments to factor with defined levels
tissue_segments <- factor(tissue_segments, levels = c("V3", "BST", "fx", "MPA", "MPN", "PV", "PVH", "PVT"))

# Convert xyz to numeric matrix
xyz <- matrix(as.numeric(as.matrix(xyz)), ncol = ncol(xyz))
```

# Identifying Differentially Expressed Genes

```{r diff-exp-genes, message=FALSE, warning=FALSE}
# Initialize matrix to store p-values
pvalue_mat <- matrix(nrow = ncol(RNA), ncol = length(slides))
rownames(pvalue_mat) <- colnames(RNA)

# Calculate p-values using SPARK
for (i in 1:length(slides)) {
  sel <- xyz[, 3] == slides[i]
  RNA_sub <- RNA[sel,]
  xyz_sup <- xyz[sel, -3]
  sparkX <- sparkx(t(RNA_sub), xyz_sup, numCores = 1, option = "mixture")
  pvalue_mat[, i] <- sparkX$res_mtest$combinedPval
  print(slides[i])
}

# Order genes by median -log(p-value)
oo <- order(apply(pvalue_mat, 1, function(x) median(-log(x))), decreasing = TRUE)
top <- colnames(RNA)[oo]
```

# Dimensionality Reduction and Clustering

```{r dim-reduce-cluster, message=FALSE, warning=FALSE}
# Select data for a specific slide
sel <- xyz[, 3] == -0.14
xy <- xyz[sel, 1:2]
tis <- tissue_segments[sel]
cel <- cell_type[sel]
RNA_sel <- RNA[sel,]

# Smoothing RNA data
ii <- knn_Armadillo(xy, xy, 10)
RNA_sel_2 <- RNA_sel[, top[1:100]]
RNA_sel_3 <- RNA_sel_2
for (h in 1:nrow(xy)) {
  temp <- RNA_sel_2[ii$nn_index[h,],, drop = FALSE]
  jj <- knn_Armadillo(temp, temp[1,, drop = FALSE], 8)
  RNA_sel_3[h,] <- colMeans(temp[jj$nn_index,])
}
RNA_sel_3 <- RNA_sel_3 / rowSums(ii$distances)
RNA_scaled <- scale(RNA_sel_3)

# Perform PCA
pca_results <- irlba(A = RNA_scaled, nv = 50)
pca <- pca_results$u %*% diag(pca_results$d)

# Perform UMAP
config <- umap.defaults
config$n_neighbors <- 50
config$n_threads <- 4
uu <- umap(pca[, 1:20], config = config)$layout

# Plot UMAP results
plot(uu, col = tis, main = "UMAP of PCA results")
```

# Applying KODAMA

```{r kodama, message=FALSE, warning=FALSE}
# Apply KODAMA to the PCA results
kk <- KODAMA.matrix.parallel(pca, spatial = xy, f.par.pls = 50, FUN = "PLS", landmarks = 100000, n.cores = 8, splitting = 50)

# Visualize results using UMAP
config <- umap.defaults
config$n_neighbors <- 100
v3 <- KODAMA.visualization(kk, method = "UMAP", config = config)

# Cluster the results
clu <- kmeans(v3, 8, nstart = 1000)$clust

# Plot the clusters
par(mfrow = c(1, 2))
plot(v3, col = tis, main = "KODAMA UMAP")
plot(v3, col = clu, main = "KODAMA Clusters")
```

# Refinement and Visualization

```{r refine-visualize, message=FALSE, warning=FALSE}
# Define tissue segment names
tissue_segment_names <- c("V3", "BST", "fx", "MPA", "MPN", "PV", "PVH", "PVT")

# Refine clusters using SVM
ref <- refine_SVM(xy, clu, rep(1, nrow(xy)), cost = 1000)

# Plot refined clusters
par(mfrow = c(1, 2))
plot(xy, col = tis, main = "Original Tissue Segments")
plot(xy, col = ref, main = "Refined Clusters")
adjustedRandIndex(ref, tis)

# Subset xyz to match the length of ref
xyz_subset <- xyz[sel, ]

# Create a data frame for ggplot
df <- data.frame(xyz_subset, tissue_segments = factor(ref, levels = 1:length(tissue_segment_names), labels = tissue_segment_names))
colnames(df) <- c("x", "y", "z", "tissue_segments")
df$z <- as.factor(df$z)
df$tissue_segments <- as.factor(df$tissue_segments)

# Define colors for the plot
cols <- c("#669bbc", "#81b29a", "#f2cc8f", "#adc178", "#dde5b6", "#a8dadc", "#e5989b", "#e07a5f")

# Plot the data using ggplot2
library(ggplot2)

ggplot(df, aes(x, y, color = tissue_segments)) +
  geom_point(size = 1) +
  facet_grid(~z) +
  theme_bw() +
  theme(
    legend.position = "bottom",
    axis.title = element_blank(),
    axis.text = element_blank(),
    axis.ticks = element_blank(),
    panel.grid = element_blank()
  ) +
  scale_color_manual("Domain", values = cols) +
  guides(color = guide_legend(nrow = 1, override.aes = list(size = 2)))


```
# 3D Visualization for Slice -0.14

```{r three-d-visualization, message=FALSE, warning=FALSE}
# Define colors for the plot
colors <- c("#669bbc", "#81b29a", "#f2cc8f", "#adc178", "#dde5b6", "#a8dadc", "#e5989b", "#e07a5f")

# Create a 3D scatter plot using plotly
fig <- plot_ly(df, 
                x = ~x, 
                y = ~y, 
                z = ~z, 
                color = ~tissue_segments, 
                colors = colors, 
                type = 'scatter3d', 
                mode = 'markers', 
                marker = list(size = 3)) %>%
  layout(scene = list(xaxis = list(title = 'X Coordinate'),
                      yaxis = list(title = 'Y Coordinate'),
                      zaxis = list(title = 'Z Coordinate')),
         title = "3D Scatter Plot of Tissue Segments")

# Show the plot
fig
```
