---
title: "MERFISH data analysis"
output:
  workflowr::wflow_html:
    toc: true
editor_options:
  chunk_output_type: console
---

# Introduction

Here, we apply KODAMA to analyze the spatial transcriptomic data that measured the mouse preoptic region of the hypothalamus using the MERFISH technology from Moffitt et al., 2018. [Link to study](https://www.science.org/doi/10.1126/science.aau5324).
We focus on the tissue sections Bregma -0.04, -0.09, -0.14, -0.19, and -0.24 mm from a consecutive brain hypothalamic region of animal 1. Data was retrieved from the [BASS GitHub repository](https://zhengli09.github.io/BASS-Analysis/MERFISH.html)
The original data can be downloaded from [Dryad](https://datadryad.org/stash/dataset/doi:10.5061/dryad.8t8s248).

# Loading and Preprocessing Data

The R library are loaded and the number of parallel computation is set.

```{r, message=FALSE, warning=FALSE}
# Load the necessary libraries
library(rgl)
library(irlba)
library(KODAMAextra)
library(scater)
library(SPARK)
library(ggplot2)
library(plotly)
library(mclust)
library(harmony)
library(bluster)
library(igraph)

n.cores=16
```


The spatial ttranscriptomic data and relative spatial coordinates are loaded in the R environment.

```{r, message=FALSE, warning=FALSE}
# Load the MERFISH data
load("../MERFISH_Animal1.RData")

cols <- c("#669bbc", "#81b29a", "#f2cc8f", "#adc178",
          "#dde5b6", "#a8dadc", "#e5989b", "#e07a5f",
          "#aae5b6", "#a8aadc", "#e59811", "#aa7900")

# Define the slides to be analyzed
slides <- c("-0.04", "-0.09", "-0.14", "-0.19", "-0.24")

# Initialize variables
xyz <- NULL
tissue_segments <- NULL
cell_type <- NULL
RNA <- NULL

# Extract spatial and expression data from each slide
for (i in slides) {
  x <- info_mult[[i]]$x / 1000
  y <- info_mult[[i]]$y / 1000
  z <- as.numeric(i)
  slide_xyz <- cbind(x - min(x), y - min(y), z)
  xyz <- rbind(xyz, slide_xyz)
  tissue_segments <- c(tissue_segments, info_mult[[i]]$z)
  cell_type <- c(cell_type, info_mult[[i]]$Cell_class)
  RNA <- rbind(RNA, t(cnts_mult[[i]]))
}

# Normalize RNA counts
RNA <- t(normalizeCounts(t(RNA), log = TRUE))

# Convert tissue segments to factor with defined levels
tissue_segments <- factor(tissue_segments, levels = c("V3", "BST", "fx", "MPA", "MPN", "PV", "PVH", "PVT"))

# Convert xyz to numeric matrix
xyz <- matrix(as.numeric(as.matrix(xyz)), ncol = ncol(xyz))
```

# Identifying Differentially Expressed Genes


Genes are ranked using the SPARK-X algorithm independently for each slide. The top gene are the spatially expressed genes.

```{r, fig.width=10, fig.height=4,results = "hide"}
top=multi_SPARKX(RNA,xyz[,-3],as.factor(xyz[,3]),n.cores = n.cores)
```

# Passing message
The gene expression value are processed using the passing-message function.

```{r, message=FALSE, warning=FALSE}
RNA.PM=passing.message(RNA[,top[1:100]],xyz)
```

## Dimensionality reduction

The dimensionality of the dataset is reduced using the principal component analysis (PCA) and batch-effect is removed using the harmony function.

```{r, message=FALSE, warning=FALSE}
RNA.PM.scaled=scale(RNA.PM)
pca_results <- irlba(A = RNA.PM.scaled, nv = 50)
pca.PM <- pca_results$u %*% diag(pca_results$d)

pca.PM=RunHarmony(pca.PM,data.frame(z=xyz[,3]),"z")


RNA.scaled=scale(RNA[,top[1:100]])
pca_results <- irlba(A = RNA.scaled, nv = 50)
pca <- pca_results$u %*% diag(pca_results$d)


par(mfrow = c(1, 2))
plot(pca, col = cols[tissue_segments], main = "PCA",pch=20,xlab="PC1",ylab="PC2")
plot(pca.PM, col = cols[tissue_segments], main = "PCA.PM",pch=20,xlab="PC1",ylab="PC2")


```

# Applying KODAMA

KODAMA is performed of the first 20 principal components of PCA.

```{r, message=FALSE, warning=FALSE}
# Apply KODAMA to the PCA results
jj=KODAMA.matrix.parallel(pca.PM[,1:20],
                          spatial = xyz,
                          landmarks = 100000,
                          n.cores=n.cores,
                          seed = 543210)

config=umap.defaults
config$n_neighbors=30
config$n_threads = n.cores
vis <- KODAMA.visualization(jj, method = "UMAP",config=config)
```


# Clustering and Refinement

A louvain clustering is performed.

```{r, message=FALSE, warning=FALSE}
g <- makeSNNGraph(as.matrix(vis), k = 100)
clu= louvain(g,ncluster = 8)$membership
```




```{r, message=FALSE, warning=FALSE}

ref=refine_SVM(xyz,clu,cost=1000)
```


```{r, fig.width=10, fig.height=4, warning=FALSE, message=FALSE, echo=FALSE}

zlabels=ref

z=xyz[,3]
uz=unique(z)

result_KODAMA=list()
result_KODAMA$clusters=list()
result_KODAMA$feature_extraction=list()
result_KODAMA$tissue_segments=list()
result_KODAMA$xy=list()

ground_true=list()
ground_true$tissue_segments=list()
ground_true$clusters=list()
ground_true$xy=list()


for(i in 1:5){
  sel=z==uz[i]
  
  ground_true$tissue_segments[[i]]=tissue_segments[sel]
  ground_true$clusters[[i]]=tissue_segments[sel]
  ground_true$xy[[i]]=xyz[sel,-3]
  
  result_KODAMA$clusters[[i]]=zlabels[sel]
  result_KODAMA$tissue_segments[[i]]=tissue_segments[sel]
  result_KODAMA$xy[[i]]=xyz[sel,-3]
  result_KODAMA$feature_extraction[[i]]=vis[sel,]
}
result_KODAMA$xyz=xyz
result_KODAMA$clusters_allslide=zlabels
result_KODAMA$feature_extraction_allslide=vis
result_KODAMA$tissue_segments_allslide=tissue_segments

ground_true$xyz=xyz
ground_true$clusters_allslide=tissue_segments

save(ground_true,result_KODAMA,file="output/MERFISH-KODAMA-results.RData")
save(tissue_segments,vis,xyz,pca,pca.PM,RNA,RNA.PM,top,file="data/MERFISH-input.RData")
```


```{r, fig.width=10, fig.height=4, warning=FALSE, message=FALSE, echo=FALSE}

ARI=NULL
slides=unique(xyz[,3])
for(i in 1:length(slides)){
  sel=slides[i]==xyz[,3]
  ARI[i]=adjustedRandIndex(ref[sel], tissue_segments[sel])
}
print(ARI)

# Plot the clusters
par(mfrow = c(1, 2))
plot(vis, col = cols[tissue_segments], main = "KODAMA tissue",pch=20)
plot(vis, col = cols[clu], main = "KODAMA Clusters",pch=20)


plot_slide(xyz[,-3],as.factor(xyz[,3]),tissue_segments,nrow=1,col=cols)
plot_slide(xyz[,-3],as.factor(xyz[,3]),ref,nrow=1,col=cols)


```


```{r, fig.width=10, fig.height=4, warning=FALSE, message=FALSE, echo=FALSE}

 cols <- c("#669bbc", "#81b29a", "#f2cc8f", "#adc178",
           "#dde5b6", "#a8dadc", "#e5989b", "#e07a5f",
           "#aae5b6", "#a8aadc", "#e59811", "#aa7900")
png("output/Figures/MERFISH/Fig 2b.png",width = 1500,height = 1500,bg = "transparent")
plot(vis, col = cols[tissue_segments],pch=20,axes=FALSE,xlab="",ylab="")
dev.off()
```


# 3D Visualization of Slice -0.14

```{r, message=FALSE, warning=FALSE}


library(rgl)
library(MASS)
library(misc3d)

volume_rendering(xyz[!is.na(ref),],ref[!is.na(ref)],selection = c("3","6","7","8"),alpha = c(0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8) ,colors = cols,cells=c(40, 40, 40),level=exp(2.7))
```


```{r, fig.width=10, fig.height=4, warning=FALSE, message=FALSE, echo=FALSE,results = "hide"}
source("code/MERFISH_comparison.R")
```
