---
title: "About"
output:
  workflowr::wflow_html:
    toc: true
editor_options:
  chunk_output_type: console
---
  
### Introduction
  
The data used in this analysis come from the Visium database, a reference resource for spatial transcriptomics data. This database provides detailed information on gene expression in various tissue contexts, offering high-resolution spatial data.

For this tutorial, we focus on different types of prostate tissues, including normal prostate, adenocarcinoma, acinar cell carcinoma, and adjacent normal sections. These data are crucial for understanding the variations in gene expression between healthy and cancerous tissues and for identifying potential diagnostic and therapeutic markers.

The data can be downloaded using the following script: [Prostate_download.sh](https://github.com/tkcaccia/KODAMA-Analysis/blob/main/code/Prostate_download.sh). This script facilitates access to the raw data, which will then be preprocessed and analyzed in the subsequent steps of our pipeline.




### Preprocessing

This section details the preprocessing of spatial transcriptomics data, which is a crucial step for cleaning and preparing the data for further analysis.

#### Loading Libraries and Defining Tissue Types

```{r, fig.width=10, fig.height=4, warning=FALSE, message=FALSE}
library(SpatialExperiment)
library(scater)
library(nnSVG)
library(SPARK)
library(harmony)
library(scuttle)
library(BiocSingular)
library(spatialLIBD)
library(KODAMAextra)
opar <- par()      # make a copy of current settings


tissues <- c("Normal_prostate",
             "Acinar_Cell_Carcinoma",
             "Adjacent_normal_section",
             "Adenocarcinoma")
n.cores=12

```

Begin by loading the necessary libraries for the analysis. Next, define the different types of prostate tissues to be studied: normal prostate, acinar cell carcinoma, adjacent normal sections, and adenocarcinoma.

#### Reading Visium Data

```{r, fig.width=10, fig.height=4, warning=FALSE, message=FALSE}
dir <- "../Prostate/"
address <- file.path(dir, tissues, "")

spe <- read10xVisium(address, tissues,
                     type = "sparse", data = "raw",
                     images = "lowres", load = FALSE)
rownames(colData(spe))=paste(gsub("-1","",rownames(colData(spe))),colData(spe)$sample_id,sep="-")

```

Visualization

```{r, fig.width=10, fig.height=4, warning=FALSE, message=FALSE}
par(mfrow = c(1, 4))

img=as.raster(getImg(spe, sample_id = "Normal_prostate" , image_id = NULL))
plot(img)
box(col="#77cc66",lwd=3)
mtext("Normal prostate")

img=as.raster(getImg(spe, sample_id = "Acinar_Cell_Carcinoma" , image_id = NULL))
plot(img)
box(col="#77cc66",lwd=3)
mtext("Acinar cell carcinoma")

img=as.raster(getImg(spe, sample_id = "Adenocarcinoma" , image_id = NULL))
plot(img)
box(col="#77cc66",lwd=3)
mtext("Adenocarcinoma")

img=as.raster(getImg(spe, sample_id = "Adjacent_normal_section" , image_id = NULL))
plot(img)
box(col="#77cc66",lwd=3)
mtext("Adjacent Normal section with IF")
```

### Loading Pathology Data

To begin the pathology data analysis, load the corresponding pathology data for adenocarcinoma samples. Ensure to replace the file path with the correct location of your data.

```{r, fig.width=10, fig.height=4, warning=FALSE, message=FALSE}

ss <- read.csv("data/Adenocarcinoma.csv")
annotation_Adenocarcinoma= ss[,2]
names(annotation_Adenocarcinoma)=ss[,1]
annotation_Adenocarcinoma=annotation_Adenocarcinoma[annotation_Adenocarcinoma!=""]
names(annotation_Adenocarcinoma)=paste(gsub("-1","",names(annotation_Adenocarcinoma)),"Adenocarcinoma",sep="-")
annotation_Acinar_Cell_Carcinoma=read_annotations("data/spots_classification_Acinar_Cell_Carcinoma.csv")
names(annotation_Acinar_Cell_Carcinoma)=paste(gsub("-1","",names(annotation_Acinar_Cell_Carcinoma)),"Acinar_Cell_Carcinoma",sep="-")
annotation_Adjacent_normal_section=read_annotations("data/spots_classification_IF.csv")
names(annotation_Adjacent_normal_section)=paste(gsub("-1","",names(annotation_Adjacent_normal_section)),"Adjacent_normal_section",sep="-")
annotation_Normal_Prostate=read_annotations("data/spots_classification_Normal_prostate.csv")
names(annotation_Normal_Prostate)=paste(gsub("-1","",names(annotation_Normal_Prostate)),"Normal_prostate",sep="-")

annotations = c(annotation_Normal_Prostate,
                annotation_Acinar_Cell_Carcinoma,
                annotation_Adjacent_normal_section,
                annotation_Adenocarcinoma)

```

#### Loading Preprocessed Data

```{r, fig.width=10, fig.height=4, warning=FALSE, message=FALSE}

metaData <- SingleCellExperiment::colData(spe)
expr <- SingleCellExperiment::counts(spe)
sample_names <- unique(colData(spe)$sample_id)
```

Load the preprocessed data and extract the metadata and gene expression counts.

#### Filtering Tissue Spots and Identifying Mitochondrial Genes

```{r, fig.width=10, fig.height=4, warning=FALSE, message=FALSE}
spe <- spe[, colData(spe)$in_tissue]

# Identify mitochondrial genes
is_mito <- grepl("(^MT-)|(^mt-)", rowData(spe)$gene_name)
```

Filter the spots located in the tissue and identify mitochondrial genes, which are often used as quality indicators.

#### Calculating Quality Control (QC) Metrics per Spot

```{r, fig.width=10, fig.height=4, warning=FALSE, message=FALSE}
# Calculate per-spot QC metrics
spe <- addPerCellQC(spe, subsets = list(mito = is_mito))

# Select QC thresholds
qc_lib_size <- colData(spe)$sum < 500
qc_detected <- colData(spe)$detected < 250
qc_mito <- colData(spe)$subsets_mito_percent > 30
qc_cell_count <- colData(spe)$cell_count > 12

# Spots to discard
discard <- qc_lib_size | qc_detected | qc_mito | qc_cell_count

if (length(discard) > 0) {
  table(discard)
  colData(spe)$discard <- discard
  
  # Filter low-quality spots
  spe <- spe[, !colData(spe)$discard]
}
dim(spe)
```

Calculate several QC metrics per spot, such as library size, number of detected genes, percentage of mitochondrial genes, and cell count. Define thresholds for these metrics and filter out low-quality spots.

#### Filtering Genes

```{r, fig.width=10, fig.height=4, warning=FALSE, message=FALSE}
colnames(rowData(spe)) <- "gene_name"

spe <- filter_genes(
  spe,
  filter_genes_ncounts = 2,   # Minimum counts
  filter_genes_pcspots = 0.5, # Minimum percentage of spots
  filter_mito = TRUE          # Filter mitochondrial genes
)

dim(spe)
```

Filter genes based on the number of counts and the percentage of spots in which they are present. Mitochondrial genes are also filtered out.


#### Normalizing Counts

```{r, fig.width=10, fig.height=4, warning=FALSE, message=FALSE}
spe <- computeLibraryFactors(spe)
spe <- logNormCounts(spe)
```

normalize the counts using library size factors and apply a logarithmic transformation to obtain data ready for more precise analysis.

This preprocessing process cleans and normalizes the spatial transcriptomics data, ensuring high-quality data ready for subsequent analyses.


### Feature Selection with SPARK

After preprocessing the data, the next step involves feature selection using SPARK, which is crucial for identifying significant genes across different tissue samples.

```{r, fig.width=10, fig.height=4, warning=FALSE, message=FALSE,results = "hide"}

top=multi_SPARKX(spe,n.cores=n.cores)

```
### Principal Component Analysis (PCA)

After feature selection, principal component analysis (PCA) is performed to explore the variance in the dataset and visualize sample relationships.



```{r, fig.width=10, fig.height=6, warning=FALSE, message=FALSE}

sample_id=colData(spe)$sample_id

library(ggplot2)
cols_tissue <- c("#0000b6cc", "#81b29acc", "#f2cc8fcc","#e07a5fcc")

# Run PCA with top selected genes
spe <- runPCA(spe, subset_row = top[1:3000], scale = TRUE)

# Run Harmony to adjust for batch effects
spe <- RunHarmony(spe, group.by.vars = "sample_id", lambda = NULL)

# Visualize PCA and Harmony results
df <- data.frame(reducedDim(spe,type = "PCA")[,1:2], tissue=as.factor(colData(spe)$sample_id))
plot1 = ggplot(df, aes(PC1, PC2, color = tissue)) +labs(title="PCA") +
  geom_point(size = 2) +
  theme_bw() + theme(legend.position = "bottom")+
  scale_color_manual("Domain", values = cols_tissue) +
  guides(color = guide_legend(nrow = 2, 
                              override.aes = list(size = 2)))

df <- data.frame(reducedDim(spe,type = "HARMONY")[,1:2], tissue=as.factor(colData(spe)$sample_id))
plot2 = ggplot(df, aes(HARMONY_1, HARMONY_2, color = tissue)) +labs(title="PCA") +
  geom_point(size = 2) +
  theme_bw() + theme(legend.position = "bottom")+
  scale_color_manual("Domain", values = cols_tissue) +
  guides(color = guide_legend(nrow = 2, 
                              override.aes = list(size = 2)))


pca=reducedDim(spe,type = "HARMONY")[,1:50]
samples=as.factor(colData(spe)$sample_id)
xy=as.matrix(spatialCoords(spe))
data=t(logcounts(spe))
plot1
plot2
svg("output/prostate1.svg",height = 3)
plot1
dev.off()
svg("output/prostate2.svg",height = 3)
plot2
dev.off()

```



### Processing Pathology Data

The processing involves creating row names and associating pathology information with the corresponding columns in the spe object.

```{r, fig.width=10, fig.height=6, warning=FALSE, message=FALSE}


annotations=annotations[rownames(colData(spe))]
annotations[annotations=="fibrous"]="fibromuscular"
names(annotations)=rownames(colData(spe))
```
### Visualization of Pathology Data

Assign specific colors to each pathology category and visualize the samples on a reduced dimension map (HARMONY), with each point colored according to its pathology category.


```{r, fig.width=10, fig.height=6, warning=FALSE, message=FALSE}

cols_pathology <- c("#0000ff", "#e41a1c", "#006400", "#000000", "#ffd700", 
                    "#00ff00", "#b2dfee","#669bbc", "#81b29a", "#f2cc8f",
                    "#adc178", "#aa1133", "#1166dc", "#e5989b", "#e07a5f")

df <- data.frame(pca[,1:2], tissue=annotations)
df=df[!is.na(annotations),]
plot3=ggplot(df, aes(HARMONY_1, HARMONY_2, color = tissue)) +labs(title="PCA") +
  geom_point(size = 2) +
  theme_bw() + theme(legend.position = "bottom")+
  scale_color_manual("Domain", values = cols_pathology) +
  guides(color = guide_legend(nrow = 2, 
                              override.aes = list(size = 2)))
svg("output/prostate3.svg",height = 3)
plot3
dev.off()

```

### Running KODAMA for Analysis

The next step is running KODAMA, a method for dimensionality reduction and visualization.

```{r, fig.width=10, fig.height=6, warning=FALSE, message=FALSE}

spe=RunKODAMAmatrix(spe,
                          reduction = "HARMONY",
                          FUN= "fastpls" ,
                          landmarks = 100000,
                          splitting = 300,
                          ncomp = 50,
                          spatial.resolution = 0.3,
                          n.cores=n.cores,
                          seed = 543210)



config <- umap.defaults
config$n_threads = n.cores
config$n_sgd_threads = "auto"

spe=RunKODAMAvisualization(spe,method="UMAP",config=config)


df <- data.frame(reducedDim(spe,type = "KODAMA")[,1:2], tissue=as.factor(colData(spe)$sample_id))
plot4=ggplot(df, aes(Dimension.1, Dimension.2, color = tissue)) +labs(title="KODAMA") +
  geom_point(size = 2) +
  theme_bw() + theme(legend.position = "bottom")+
  scale_color_manual("Domain", values = cols_tissue) +
  guides(color = guide_legend(nrow = 2, 
                              override.aes = list(size = 2)))




df <- data.frame(reducedDim(spe,type = "KODAMA")[,1:2], tissue=annotations)
plot5=ggplot(df, aes(Dimension.1, Dimension.2, color = tissue)) +labs(title="KODAMA") +
  geom_point(size = 2) +
  theme_bw() + theme(legend.position = "bottom")+
  scale_color_manual("Domain", values = cols_pathology) +
  guides(color = guide_legend(nrow = 3, 
                              override.aes = list(size = 2)))
plot4
plot5

svg("output/prostate4.svg",height = 4)
plot4
dev.off()
svg("output/prostate5.svg",height = 4)
plot5
dev.off()

```

```{r, fig.width=10, fig.height=4, warning=FALSE, message=FALSE, echo=FALSE}
save(spe,file="output/Prostate.RData")
```


### Clustering
```{r, fig.width=10, fig.height=4, warning=FALSE, message=FALSE, echo=FALSE}


ko=reducedDim(spe,type = "KODAMA")
g <- bluster::makeSNNGraph(as.matrix(ko), k = 50)
g_walk <- igraph::cluster_louvain(g,resolution = 0.2)
clu = g_walk$membership

table(clu,annotations)

cols_cluster <- c("#0000b6", "#81b29a", "#f2cc8f","#e07a5f",
          "#cc00b6", "#81ccff", "#00cc8f","#e0aa5f",
          "#0088dd", "#33b233", "#aa228f","#aa7a6f")


df <- data.frame(reducedDim(spe,type = "KODAMA")[,1:2], cluster=as.factor(clu))
df$cluster=as.factor(df$cluster)
plot6=ggplot(df, aes(Dimension.1, Dimension.2, color = cluster)) +
  geom_point(size = 2) +
  theme_bw() +
  theme(legend.position = "bottom")+
  
  scale_color_manual("Domain", values = cols_cluster) +
  guides(color = guide_legend(nrow = 2, 
                              override.aes = list(size = 2)))


plot7=plot_slide(xy,samples,as.factor(clu),nrow=1,col = cols_cluster)

plot6
plot7

svg("output/prostate6.svg",height = 4)
plot6
dev.off()
svg("output/prostate7.svg",height = 4)
plot7
dev.off()
```

### Trajectory

```{r, fig.width=10, fig.height=8, warning=FALSE, message=FALSE}

par(opar)
sel=colData(spe)$sample_id=="Acinar_Cell_Carcinoma"
spe_sub=spe[,sel]
image=as.raster(imgData(spe_sub)$data[[1]])
xy_sel=spatialCoords(spe_sub)
xy_sel=xy_sel*scaleFactors(spe_sub)
xy_sel[,2]=nrow(image)-xy_sel[,2]


plot(image)
points(xy_sel,cex=0.5,pch=20,col="#33333333")

data=as.matrix(t(logcounts(spe_sub)))



# nn1=new_trajectory (xy_sel,data = data)
# nn2=new_trajectory (xy_sel,data = data)
# nn3=new_trajectory (xy_sel,data = data)


load("data/trajectories.RData")

mm1=new_trajectory (xy_sel,data = data,trace=nn1$xy)
mm2=new_trajectory (xy_sel,data = data,trace=nn2$xy)
mm3=new_trajectory (xy_sel,data = data,trace=nn3$xy)

traj=rbind(mm1$trajectory,
           mm2$trajectory,
           mm3$trajectory)
traj=traj[,top[1:2000]]
y=rep(1:20,3)

ma=multi_analysis(traj,y,FUN="correlation.test",method="spearman")

ma=ma[order(as.numeric(ma$`p-value`)),]

ma[1:20,]
par(opar)
vis_gene(spe,"Acinar_Cell_Carcinoma","ENSG00000134339")
par(opar)
vis_gene(spe,"Acinar_Cell_Carcinoma","ENSG00000166741")


rowData(spe)[c("ENSG00000134339","ENSG00000166741"),]

samples=as.factor(colData(spe_sub)$sample_id)
xy=as.matrix(spatialCoords(spe_sub))
data=t(logcounts(spe_sub))[,top[1:2000]]
data=as.matrix(data)
PMdata=passing.message(data,xy/1000)

par(mfrow=c(1,2))

a=data[,"ENSG00000134339"]
b=data[,"ENSG00000166741"]
cc=cor.test(a,b)
txt=paste("r=",round(cc$estimate,digits=2))
plot(a,b,main="Pearson correlation",xlab="SAA2",ylab="NNMT")
mtext(txt)


a=PMdata[,"ENSG00000134339"]
b=PMdata[,"ENSG00000166741"]
cc=cor.test(a,b)
txt=paste("r=",round(cc$estimate,digits=2))
plot(a,b,main="APM correlation",xlab="SAA2",ylab="NNMT")
mtext(txt)




```

This extended analysis includes principal component analysis (PCA), pathology data analysis, and the application of KODAMA for dimensionality reduction and visualization, enhancing the understanding of spatial transcriptomics data in different prostate tissue types.

### GSVA Enrichment Analysis with MSigDB

To explore enriched biological processes in our spatial transcriptomics data, we employ Gene Set Variation Analysis (GSVA) using MSigDB gene sets as a reference. To download the necessary data, please follow the steps provided at [this link](https://www.gsea-msigdb.org/gsea/msigdb) and create an account if required.

#### Loading Packages and Data

We start by loading the necessary packages and preparing our gene data for analysis:



```{r, fig.width=10, fig.height=4, warning=FALSE, message=FALSE,results = "hide"}
 library("GSVA")
 library("GSA")
 library("VAM")
 geneset=GSA.read.gmt("../Genesets/msigdb_v2023.2.Hs_GMTs/h.all.v2023.2.Hs.symbols.gmt")
 names(geneset$genesets)=geneset$geneset.names
 genesets=geneset$genesets

countdata <- t(as.matrix(logcounts(spe)))

# library("gprofiler2")
# genes=gconvert(rownames(spe),organism="hsapiens",target="GENECARDS",filter_na = F)$target

genes=rowData(spe)[,"gene_name"]
spot_name=colnames(spe)
colnames(countdata)=genes
rownames(countdata)=paste(gsub("-1","",spot_name),spe$sample_id,sep="-")

save(genesets,countdata,file="Elemina.RData") 

li=lapply(genesets,function(x) which(genes %in% x))

VAM=vamForCollection(gene.expr=countdata, gene.set.collection=li)
pathway=VAM$distance.sq


ta=table(pathology,clu)
path_clust=levels(pathology)[apply(ta,2,which.max)]
clu2=rep(NA,length(clu))
clu2[(clu %in% which(path_clust=="Normal gland")) & pathology=="Normal gland"]="Normal-phenotype"
clu2[(clu %in% which(path_clust=="Invasive carcinoma")) & pathology=="Normal gland"]="Tumor-phenotype"
```
bla
```{r, fig.width=10, fig.height=8, warning=FALSE, message=FALSE}
ta


ma=multi_analysis(pathway[!is.na(clu2),],clu2[!is.na(clu2)])

ma=ma[order(as.numeric(ma$`p-value`)),]

ma[1:10,]









library(ggpubr)
library(gridExtra)
par(mai=c(3,3,3,3))
col_pathology <- c("#0000ff", "#e41a1c", "#006400", "#000000", "#ffd700", "#00ff00", "#b2dfee")


df=data.frame(variable=pathway[,"HALLMARK_MYC_TARGETS_V1"],labels=pathology)
df=df[!is.na(df$labels),]

my_comparisons=list(c(1,7))
Nplot1=ggboxplot(df, x = "labels", y = "variable", width = 0.8,palette = col_pathology,las=2,
                 fill="labels",ylim=c(200,1400),       
                 shape=21)+  
  ylab("HALLMARK_MYC_TARGETS_V1")+ 
  xlab("")+
  
  stat_compare_means(comparisons = my_comparisons,method="wilcox.test")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1),legend.position = "none",plot.margin = unit(c(2,1,1,1), "cm"))




cols=cols_cluster[c(1,2,3,6,11)]

df=data.frame(variable=pathway[,"HALLMARK_MYC_TARGETS_V1"],labels=clu)
df=df[!is.na(clu2),]
my_comparisons=list(c(4,3),c(4,2),c(4,1))
Nplot2=ggboxplot(df, x = "labels", y = "variable", width = 0.8,palette = cols,
                 fill="labels",add = "jitter", ylim=c(200,1400),                  
                 add.params = list(size = 0.5, jitter = 0.2,fill=2),
                 shape=21)+  
  ylab("HALLMARK_MYC_TARGETS_V1")+ 
  xlab("")+
  stat_compare_means(comparisons = my_comparisons,method="wilcox.test")+
  theme(legend.position = "none",plot.margin = unit(c(2,1,1,1), "cm"))

egg::ggarrange(Nplot1,Nplot2,widths = c(2,1.2),nrow=1,labels = c('A', 'B'))

```
