---
title: "About"
output:
  workflowr::wflow_html:
    toc: true
editor_options:
  chunk_output_type: console
---

Explore this comprehensive tutorial on spatial transcriptomics with Seurat and the KODAMA package.
This detailed guide takes you through all the essential steps to preprocess and effectively analyze
spatial transcriptomic data, focusing on the anterior region of the mouse brain. By the end of this journey,
you will master initial visualizations, precise data preprocessing, and the application of advanced dimensionality
reduction techniques using Seurat and KODAMA.

## Context and Importance
Spatial transcriptomics represents a significant advancement in studying gene expression
within tissue samples, providing a deep understanding of molecular and cellular interactions
within complex biological structures. This tutorial fully harnesses the potential of this innovative
approach using well-established tools within the bioinformatics community.

##Access to Data
The data used are sagittal sections of the mouse brain generated using Visium v1 technology.
You can access these data by downloading them directly here(https://support.10xgenomics.com/spatial-gene-expression/datasets). Once downloaded, they can be loaded
into Seurat using the Load10X_Spatial() function, which imports data generated by the spaceranger pipeline.
This creates a Seurat object containing both gene expression data at the spot level and the corresponding tissue slice image.

For specific details on the loading and usage procedure, refer to the documentation
by typing ?stxBrain in your R environment.


## Step 1: Load Necessary Libraries

First, we need to install and load the required libraries for this analysis. Make sure you have the following packages installed:

```{r,fig.width=10,fig.height=4,warning=FALSE,message=FALSE}
install.packages(c("Seurat", "SeuratData", "ggplot2", "patchwork", "dplyr", "KODAMA", "KODAMAextra"))
```

Load the libraries:

```{r,fig.width=10,fig.height=4,warning=FALSE,message=FALSE}
library(Seurat)
library(SeuratData)
library(ggplot2)
library(patchwork)
library(dplyr)
library(KODAMA)
library(KODAMAextra)
```

## Step 2: Install and Load Data

If you have not already installed the data, use the following command:

```{r,fig.width=10,fig.height=4,warning=FALSE,message=FALSE}
InstallData("stxBrain")
```

Load the anterior region of the mouse brain dataset:

```{r,fig.width=10,fig.height=4,warning=FALSE,message=FALSE}
brain <- LoadData("stxBrain", type = "anterior1")
```

## Step 3: Initial Visualization

Let's start with some initial visualizations to get a sense of our data:

```{r,fig.width=10,fig.height=4,warning=FALSE,message=FALSE}
plot1 <- VlnPlot(brain, features = "nCount_Spatial", pt.size = 0.1) + NoLegend()
plot2 <- SpatialFeaturePlot(brain, features = "nCount_Spatial") + theme(legend.position = "right")
wrap_plots(plot1, plot2)
```

## Step 4: Pre-process the Data

We will preprocess the data using Seurat's SCTransform method and perform PCA:

```{r,fig.width=10,fig.height=4,warning=FALSE,message=FALSE}
brain <- SCTransform(brain, assay = "Spatial", verbose = FALSE)
brain <- RunPCA(brain, assay = "SCT", verbose = FALSE)
brain <- FindNeighbors(brain, reduction = "pca", dims = 1:30)
brain <- FindClusters(brain, verbose = FALSE)
brain <- RunUMAP(brain, reduction = "pca", dims = 1:30)
```
## Step 5: Visualization of PCA and UMAP Results
Visualize the PCA and UMAP results to understand the clustering:

```{r,fig.width=10,fig.height=4,warning=FALSE,message=FALSE}
p1 <- DimPlot(brain, reduction = "umap", label = TRUE)
p2 <- SpatialDimPlot(brain, label = TRUE, label.size = 3)
p1 + p2
```

## Step 6: Visualization of Clusters

Highlight specific clusters using SpatialDimPlot:

```{r,fig.width=10,fig.height=4,warning=FALSE,message=FALSE}
SpatialDimPlot(brain, cells.highlight = CellsByIdentities(object = brain, idents = c(2, 1, 4, 3, 5, 8)), facet.highlight = TRUE, ncol = 3)
```

## Step 7: Run and Visualize KODAMA

Apply KODAMA to reduce dimensionality and visualize the results:

```{r,fig.width=10,fig.height=4,warning=FALSE,message=FALSE}
class(new_slide)
class(shift)
brain <- RunKODAMAmatrix(brain, reduction = "pca")
brain <- RunKODAMAvisualization(brain, method = "UMAP", config = config)
```
Visualize the KODAMA results with DimPlot and SpatialDimPlot:

```{r,fig.width=10,fig.height=4,warning=FALSE,message=FALSE}
p3 <- DimPlot(spe, reduction = "kodama", label = TRUE)
p4 <- SpatialDimPlot(spe, label = TRUE, label.size = 3)
p3 + p4
```

This tutorial provides a step-by-step guide to performing spatial transcriptomics analysis using Seurat and KODAMA. By following these steps, you will be able to preprocess your data, perform dimensionality reduction, and visualize the results effectively.
