---
title: "Seurat pipeline"
output:
  workflowr::wflow_html:
    toc: true
editor_options:
  chunk_output_type: console
---

Seurat is an R toolkit widely used for single-cell genomics and spatial transcriptomics, developed and maintained by the Satija Lab at New York Genome Center.
In this section, we will see the steps to integrate KODAMA into the Seurat pipeline.


## Loading libraries

First, we need to load the required libraries for this analysis. Make sure you have the following packages installed.


```{r,fig.width=10,fig.height=4,warning=FALSE,message=FALSE}
library(Seurat)
library(SeuratData)
library(KODAMA)
library(KODAMAextra)
```


## Dataset
The data used are sagittal sections of the mouse brain generated using Visium v1 technology saved in the Seurat library.
For specific details on loading and using the dataset, please take a look at the documentation by typing ?stxBrain in your R environment.
Install and load the anterior region of the mouse brain dataset.

```{r,fig.width=10,fig.height=4,warning=FALSE,message=FALSE}
InstallData("stxBrain")
brain <- LoadData("stxBrain", type = "anterior1")
```


## Data processing

We will preprocess the data using Seurat's SCTransform method and perform PCA:

```{r,fig.width=10,fig.height=4,warning=FALSE,message=FALSE}
brain <- SCTransform(brain, assay = "Spatial", verbose = FALSE)
brain <- RunPCA(brain, assay = "SCT", verbose = FALSE)
```


## Run and Visualize KODAMA

Apply KODAMA to reduce dimensionality and visualize the results:

```{r,fig.width=10,fig.height=4,warning=FALSE,message=FALSE}

brain <- RunKODAMAmatrix(brain, reduction = "pca")
brain <- RunKODAMAvisualization(brain, method = "UMAP")

```

## Clustering

Apply clustering on the KODAMA plot.

```{r,fig.width=10,fig.height=4,warning=FALSE,message=FALSE}
brain <- FindNeighbors(brain, reduction = "KODAMA", dims = 1:2)
brain <- FindClusters(brain, verbose = FALSE, graph.name="RNA_snn",resolution = 0.2)
```

Visualize the KODAMA results with DimPlot and SpatialDimPlot:

```{r,fig.width=10,fig.height=4,warning=FALSE,message=FALSE}
p1 <- DimPlot(brain, reduction = "KODAMA", label = TRUE)
p2 <- SpatialDimPlot(brain, label = TRUE, label.size = 3)
p1 + p2
```


