---
title: "About"
output:
  workflowr::wflow_html:
    toc: true
editor_options:
  chunk_output_type: console
---


Describe your project.
The data can be downloaded using the following script: [VisiumHD_CRC_download.sh](code/VisiumHD_CRC_download.sh). This script facilitates access to the raw data, which will then be preprocessed and analyzed in the subsequent steps of our pipeline.

The data can be downloaded using the following script: [VisiumHD_CRC_download.sh](https://github.com/tkcaccia/KODAMA-Analysis/blob/main/code/VisiumHD_CRC_download.sh). This script facilitates access to the raw data, which will then be preprocessed and analyzed in the subsequent steps of our pipeline.


```{r, fig.width=10, fig.height=4, warning=FALSE, message=FALSE}
library("ggplot2")
library("patchwork")
library("dplyr")
library("Seurat")
library("KODAMA")
library("KODAMAextra")
library("bigmemory")

localdir="../Colorectal/outs/"
object <- Load10X_Spatial(data.dir = localdir, bin.size = c(8))

image=as.raster(object@images$slice1.008um@image)
save(image,file="output/CRC-image.RData")
#object@images$slice1.008um@scale.factors$hires

# plot(image,xlim=c(320,530),ylim=c(200,410))
# points(xy[,2]*0.007973422,nrow(image)-xy[,1]*0.007973422,pch=20)

#xy=as.matrix(GetTissueCoordinates(sp_obj)[,1:2])

#image=as.raster(imgData(object)$data[[1]])

#xy_sel=spatialCoords(spe_sub)
#xy_sel=xy_sel*scaleFactors(spe_sub)
#xy_sel[,2]=nrow(image)-xy_sel[,2]


 vln.plot <- VlnPlot(object, features = "nCount_Spatial.008um", pt.size = 0) + NoLegend()
 count.plot <- SpatialFeaturePlot(object, features = "nCount_Spatial.008um", pt.size.factor = 1.2) +
   theme(legend.position = "right")

nCount_Spatial=colSums(object@assays$Spatial.008um$counts)
#w= which(nCount_Spatial >10)
#object@assays$Spatial.008um$counts= object@assays$Spatial.008um$counts[,w]
#object@meta.data=object@meta.data[w,]

sp_obj <- subset(
  object,
  subset = nCount_Spatial.008um > 100)



nCount_Spatial=colSums(sp_obj@assays$Spatial.008um$counts)


 counts=sp_obj@assays$Spatial.008um$counts
 is_mito <- grepl("(^MT-)|(^mt-)", rownames(counts))
 counts <- counts[!is_mito,]

 filter_genes_ncounts=1
 filter_genes_pcspots=0.5
 nspots <- ceiling(filter_genes_pcspots/100 *  ncol(counts))
 ix_remove <- rowSums(counts >= filter_genes_ncounts) <   nspots
 counts <- counts[!ix_remove,]

 QCgenes <- rownames(counts)

 VariableFeatures(sp_obj) = QCgenes

 rm(counts)


DefaultAssay(sp_obj) <- "Spatial.008um"
sp_obj <- NormalizeData(sp_obj)


sp_obj <- FindVariableFeatures(sp_obj)
sp_obj <- ScaleData(sp_obj)

xy=as.matrix(GetTissueCoordinates(sp_obj)[,1:2])

sp_obj <- RunPCA(sp_obj, reduction.name = "pca.008um")

dim(sp_obj)

plot(Seurat::Embeddings(sp_obj, reduction = "pca.008um"))
```

```{r, fig.width=10, fig.height=8}

#sp_obj <- RunKODAMAmatrix(sp_obj, reduction = "pca.008um",
#                          FUN= "PLS" ,
#                          landmarks = 10000,
#                          splitting = 100,
#                          f.par.pls = 50,
#                          spatial.resolution = 0.4,
#                          n.cores=8)

#  print("KODAMA finished")
  
#     config=umap.defaults
#     config$n_threads = 8
#     config$n_sgd_threads = "auto"
#     sp_obj <- RunKODAMAvisualization(sp_obj, method = "UMAP",config=config)

     
# kk_UMAP=Seurat::Embeddings(sp_obj, reduction = "KODAMA")

# save(kk_UMAP,xy,file="output/VisiumHD.RData")

load("output/VisiumHD3.RData")   

rr=read.csv("data/spots_classification_ALL.csv",sep=",")
ss=strsplit(rr[,2],":")
ss=unlist(lapply(ss, function(x) x[2]))
ss=strsplit(ss,",")
ss=unlist(lapply(ss, function(x) x[1]))
ss=gsub("\"","",ss)

rr[,2]=ss
n=ave(1:length(rr[,1]), rr[,1], FUN = seq_along)
rr=rr[n==1,]
rownames(rr)=rr[,1]
rr=rr[rownames(kk_UMAP),]

rr[,2]=substring(rr[,2],2)
                     
    
  

table(rr[,"classification"])



library(ggplot2)


cols=sample(rainbow(15))
labels=as.factor(rr[,"classification"])



cols_tissue <- c("#0000ff", "#e41a1c", "#006400",  "#ffd700","#0088dd",
                    "#00ff00", "#b2dfee","#669bbc", "#81b29a", "#f2cc8f",
                    "#adc178", "#aa1133", "#1166dc", "#e5989b", "#e07a5f",
                    "#cc00b6", "#81ccff", "#00cc8f","#e0aa5f","#33b233", "#aa228f","#aa7a6f")


df <- data.frame(kk_UMAP[,1:2], tissue=labels)
plot1 = ggplot(df, aes(Dimensions_1, Dimensions_2, color = tissue)) +labs(title="KODAMA") +
  geom_point(size = 1) +
  theme_bw() + theme(legend.position = "bottom")+
  scale_color_manual("Domain", values = cols_tissue) +
  guides(color = guide_legend(nrow = 4, 
                              override.aes = list(size = 4)))
plot1
png("output/CRC.png",height = 2000,width = 2000)
plot1
dev.off()


par(xpd = T, mar = par()$mar + c(0,0,0,7))

plot1=plot(kk_UMAP,cex=0.5,pch=20,col=cols_tissue[labels])
legend(max(kk_UMAP[,1])+0.05*dist(range(kk_UMAP[,1])), max(kk_UMAP[,2]),
       levels(labels),
       col = cols,
       cex = 0.8,
       pch=20)

load("data/trajectories_VISIUMHD.RData")

data=sp_obj@assays$Spatial.008um$data[rownames(sp_obj@assays$Spatial.008um$scale.data),]
data=as.matrix(data)
data=t(data)

mm1=new_trajectory (kk_UMAP,data = data,trace=tra1$xy)
mm2=new_trajectory (kk_UMAP,data = data,trace=tra2$xy)
mm3=new_trajectory (kk_UMAP,data = data,trace=tra3$xy)


traj=rbind(mm1$trajectory,
           mm2$trajectory,
           mm3$trajectory)
y=rep(1:20,3)

```

```{r, fig.width=10, fig.height=8, warning=FALSE, message=FALSE}
ma=multi_analysis(traj,y,FUN="correlation.test",method="spearman")
ma=ma[order(as.numeric(ma$`p-value`)),]
colnames(ma)=c("Feature   ","rho   ","p-value   ","FDR   ")
```

```{r, fig.width=10, fig.height=8, warning=FALSE, message=FALSE}
knitr::kable(ma[1:30,],row.names=FALSE)
```

# miRseq Analysis:

**Analysing miRseq Gene Expression Data from a Colerectal Adenocarcinoma Cohort:**

```{r, message=FALSE}
# install.packages("readxl")
library(readxl)

```

***Prepare Clinical Data:***

```{r}
# Read in Clinical Data:
coad=read.csv("../TCGA/COAD/COAD.clin.merged.picked.txt",sep="\t",check.names = FALSE, row.names = 1)

coad <- as.data.frame(coad) 

# Clean column names: replace dots with dashes & convert to uppercase
colnames(coad) = toupper(colnames(coad))

 # Transpose the dataframe so that rows become columns and vice versa
coad = t(coad) 
```

***Prepare miRNA-seq expression data:***

```{r}
# Read RNA-seq expression data:
r = read.csv("../TCGA/COAD/COAD.rnaseqv2__illuminahiseq_rnaseqv2__unc_edu__Level_3__RSEM_genes_normalized__data.data.txt", sep = "\t", check.names = FALSE, row.names = 1)
# Remove the first row:
r = r[-1,]
# Convert expression data to numeric matrix format
temp = matrix(as.numeric(as.matrix(r)), ncol=ncol(r))
colnames(temp) = colnames(r)  
rownames(temp) = rownames(r)  
RNA = temp  

# Transpose the matrix so that genes are rows and samples are columns
RNA = t(RNA)  

```

***Extract patient and tissue information from column names:***

```{r}

tcgaID = list()
 # Extract sample ID
tcgaID$sample.ID <- substr(colnames(r), 1, 16)
# Extract patient ID
tcgaID$patient <- substr(colnames(r), 1, 12)  
# Extract tissue type
tcgaID$tissue <- substr(colnames(r), 14, 16)  

tcgaID = as.data.frame(tcgaID)  
```

***Select Primary Solid Tumor tissue data ("01A"):***

```{r}
sel=tcgaID$tissue == "01A"
tcgaID.sel = tcgaID[sel, ]

# Subset the RNA expression data to match selected samples
RNA.sel = RNA[sel, ]
```

***Intersect patient IDs between clinical and RNA data:***

```{r}
sel = intersect(tcgaID.sel$patient, rownames(coad))
# Subset the clinical data to include only selected patients:
coad.sel = coad[sel, ]
# Assign patient IDs as row names to the RNA data:
rownames(RNA.sel) = tcgaID.sel$patient
# Subset the RNA data to include only selected patients
RNA.sel = RNA.sel[sel, ]
```

***Prepare labels for pathology stages:***

- Classify stages `t1`, `t2`, & `t3` as "low"

- Classify stages `t4`, `t4a`, & `t4b` as "high"

- Convert any `tis` stages to `NA`

```{r}

labelsTCGA = coad.sel[, "pathology_T_stage"]
labelsTCGA[labelsTCGA %in% c("t1", "t2", "t3", "tis")] = "low"
labelsTCGA[labelsTCGA %in% c("t4", "t4a", "t4b")] = "high"

```

***Log Transform the expression data for our selected gene `CXCL2`:***

```{r}
CXCL2 <- log(1 + RNA.sel[, "CXCL2|2920"])
LCN2 <- log(1 + RNA.sel[,"LCN2|3934" ])
```

***Boxplot to visualize the distribution of log transformed gene expression by pathology stage:***

```{r}
colors=c("#0073c2bb","#efc000bb","#868686bb","#cd534cbb","#7aabdcbb","#003c67bb")

library(ggpubr)
df=data.frame(variable=CXCL2,labels=labelsTCGA)

my_comparisons=list()
my_comparisons[[1]]=c(1,2)

Nplot1=ggboxplot(df, x = "labels", y = "variable",fill="labels",
                 width = 0.8,
                 palette=colors,
                 add = "jitter",            
                 add.params = list(size = 2, jitter = 0.2,fill="#ff0000aa", shape=21))+  
  ylab("CXCL2 gene expression (FPKM)")+ xlab("")+
  stat_compare_means(comparisons = my_comparisons,method="wilcox.test")

Nplot1

svg("output/CRC_boxplot.svg")
Nplot1
dev.off()


```


```{r, fig.width=10, fig.height=8, warning=FALSE, message=FALSE,results = "hide"}
xy2=xy
xy2[,1]=xy[,2]
xy2[,2]=-xy[,1]
plot(xy2,col=cols_tissue[labels],pch=20,cex=0.5)

df <- data.frame(xy2, tissue=labels)
plot2 = ggplot(df, aes(x, y, color = tissue)) +labs(title="KODAMA") +
  geom_point(size = 1) +
  theme_bw() + theme(legend.position = "bottom")+
  scale_color_manual("Domain", values = cols_tissue) +
  guides(color = guide_legend(nrow = 4, 
                              override.aes = list(size = 4)))
plot2

png("output/CRC2.png",height = 2000,width = 2000)
plot2
dev.off()

sel_desmoplastic_submucosa=which(labels=="desmoplastic submucosa")
kk_desmoplastic_submucosa=kk_UMAP[sel_desmoplastic_submucosa,]
xy_desmoplastic_submucosa=xy2[sel_desmoplastic_submucosa,]


g <- bluster::makeSNNGraph(as.matrix(kk_desmoplastic_submucosa), k = 20)
g_walk <- igraph::cluster_louvain(g,resolution = 0.005)
clu = g_walk$membership
names(clu)=rownames(kk_desmoplastic_submucosa)



df <- data.frame(kk_desmoplastic_submucosa[,1:2], tissue=as.factor(clu))
plot3 = ggplot(df, aes(Dimensions_1, Dimensions_2, color = tissue)) +labs(title="KODAMA") +
  geom_point(size = 1) +
  theme_bw() + theme(legend.position = "bottom")+
  scale_color_manual("Domain", values = cols_tissue) +
  guides(color = guide_legend(nrow = 4, 
                              override.aes = list(size = 4)))
plot3
png("output/CRC7.png",height = 2000,width = 2000)
plot3
dev.off()


df <- data.frame(xy_desmoplastic_submucosa, tissue=as.factor(clu))
plot4 = ggplot(df, aes(x, y, color = tissue)) +labs(title="KODAMA") +
  geom_point(size = 1) +
  theme_bw() + theme(legend.position = "bottom")+
  scale_color_manual("Domain", values = cols_tissue) +
  guides(color = guide_legend(nrow = 4, 
                              override.aes = list(size = 4)))
plot4

png("output/CRC8.png",height = 2000,width = 2000)
plot4
dev.off()



sel_desmoplastic_submucosa_selected=names(which(clu==names(which.max(table(clu)))))
kk_desmoplastic_submucosa_selected=kk_UMAP[sel_desmoplastic_submucosa_selected,]
xy_desmoplastic_submucosa_selected=xy2[sel_desmoplastic_submucosa_selected,]
data_desmoplastic_submucosa_selected=data[sel_desmoplastic_submucosa_selected,]
data_desmoplastic_submucosa_selected=data_desmoplastic_submucosa_selected[,-which(colMeans(data_desmoplastic_submucosa_selected==0)>0.95)]

sel_invasive_carcinoma=which(labels=="invasive carcinoma" | labels=="intermediate dysplasia")
kk_invasive_carcinoma=kk_UMAP[sel_invasive_carcinoma,]
xy_invasive_carcinoma=xy2[sel_invasive_carcinoma,]

knn=Rnanoflann::nn(xy_invasive_carcinoma,xy_desmoplastic_submucosa_selected,1)	

y=knn$distances[,1]



ma=multi_analysis(data_desmoplastic_submucosa_selected,y,FUN="correlation.test",method="spearman")
ma=ma[order(abs(as.numeric(ma$rho)),decreasing = TRUE),]
colnames(ma)=c("Feature   ","rho   ","p-value   ","FDR   ")



# 2) Define custom intervals
break_points <-c(quantile(y,probs=c(seq(0,1,0.005))))

# 3) Convert continuous data to intervals
distance_binned <- cut(y, breaks = break_points)

gene_binned=apply(data_desmoplastic_submucosa_selected,2,function(x)  tapply(x,distance_binned,mean))
break_points=break_points[-length(break_points)]

ma=multi_analysis(gene_binned,break_points,FUN="correlation.test",method="MINE")
ma=ma[order(as.numeric(ma$MIC),decreasing = TRUE),]
ma[1:10,]
rownames(ma)=ma[,"Feature"]
#plot(knn$distances,PMdata[,3])

zmax=NULL
for(i in 1:ncol(gene_binned)){
 df=data.frame(x=break_points,y=gene_binned[,i])
 ll=loess(y~x,data = df,span = 0.3)
 z=predict(ll,newdata = data.frame(x=break_points))
 zmax[i]=break_points[which.max(z)]
# points(break_points,z,type="l",col=2)
}
genes=colnames(gene_binned)
names(zmax)=genes


plot(log(1+zmax[genes]),ma[genes,]$MIC,cex=0.2)
text(log(1+zmax[genes]),ma[genes,]$MIC,labels=genes,cex=0.7)


plot(log(1+break_points),gene_binned[,1],ylim=c(0,1),type="n")
 df=data.frame(x=break_points,
               HTRA3=gene_binned[,"HTRA3"],
               IGFBP5=gene_binned[,"IGFBP5"],
               CXCL14=gene_binned[,"CXCL14"],
               MMP11=gene_binned[,"MMP11"],
               TIMP3=gene_binned[,"TIMP3"],
               MGP=gene_binned[,"MGP"])
 
 ll=loess(IGFBP5~x,data = df,span = 0.3)
 IGFBP5=predict(ll,newdata = data.frame(x=break_points))
 
 ll=loess(CXCL14~x,data = df,span = 0.3)
 CXCL14=predict(ll,newdata = data.frame(x=break_points))
 
 ll=loess(MMP11~x,data = df,span = 0.3)
 MMP11=predict(ll,newdata = data.frame(x=break_points))
 
 ll=loess(TIMP3~x,data = df,span = 0.3)
 TIMP3=predict(ll,newdata = data.frame(x=break_points))
 
 ll=loess(HTRA3~x,data = df,span = 0.3)
 HTRA3=predict(ll,newdata = data.frame(x=break_points))
 
 ll=loess(MGP~x,data = df,span = 0.3)
 MGP=predict(ll,newdata = data.frame(x=break_points))
 points(log(1+break_points),HTRA3/max(HTRA3),type="l",col=2,lwd=3)
 points(log(1+break_points),IGFBP5/max(IGFBP5),type="l",col=3,lwd=3)
 points(log(1+break_points),CXCL14/max(CXCL14),type="l",col=4,lwd=3)
 points(log(1+break_points),MMP11/max(MMP11),type="l",col=5,lwd=3)
 points(log(1+break_points),TIMP3/max(TIMP3),type="l",col=6,lwd=3)
 points(log(1+break_points),MGP/max(MGP),type="l",col=7,lwd=3)

 
svg("output/CRC_linee.svg")
plot(log(1+break_points),gene_binned[,1],ylim=c(0,1),type="n")

 points(log(1+break_points),HTRA3/max(HTRA3),type="l",col=2,lwd=3)
 points(log(1+break_points),IGFBP5/max(IGFBP5),type="l",col=3,lwd=3)
 points(log(1+break_points),CXCL14/max(CXCL14),type="l",col=4,lwd=3)
 points(log(1+break_points),MMP11/max(MMP11),type="l",col=5,lwd=3)
 points(log(1+break_points),TIMP3/max(TIMP3),type="l",col=6,lwd=3)
 points(log(1+break_points),MGP/max(MGP),type="l",col=7,lwd=3)

dev.off()

```


```{r, fig.width=10, fig.height=8, warning=FALSE, message=FALSE}
knitr::kable(ma[1:30,],row.names=FALSE)
```



